<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.555">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Figure7</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Figure7_files/libs/clipboard/clipboard.min.js"></script>
<script src="Figure7_files/libs/quarto-html/quarto.js"></script>
<script src="Figure7_files/libs/quarto-html/popper.min.js"></script>
<script src="Figure7_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Figure7_files/libs/quarto-html/anchor.min.js"></script>
<link href="Figure7_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Figure7_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Figure7_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Figure7_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Figure7_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Figure7</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="figure-7" class="level2">
<h2 class="anchored" data-anchor-id="figure-7">Figure 7</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.2     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::between()     masks data.table::between()
✖ dplyr::filter()      masks stats::filter()
✖ dplyr::first()       masks data.table::first()
✖ lubridate::hour()    masks data.table::hour()
✖ lubridate::isoweek() masks data.table::isoweek()
✖ dplyr::lag()         masks stats::lag()
✖ dplyr::last()        masks data.table::last()
✖ lubridate::mday()    masks data.table::mday()
✖ lubridate::minute()  masks data.table::minute()
✖ lubridate::month()   masks data.table::month()
✖ lubridate::quarter() masks data.table::quarter()
✖ lubridate::second()  masks data.table::second()
✖ purrr::transpose()   masks data.table::transpose()
✖ lubridate::wday()    masks data.table::wday()
✖ lubridate::week()    masks data.table::week()
✖ lubridate::yday()    masks data.table::yday()
✖ lubridate::year()    masks data.table::year()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>peptide_gene_mapping <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../../Databases/peptide_gene_mapping.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 731724 Columns: 6
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): seq_id, refseq_id, gene, sequence
dbl (2): index, gi

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>AA_colors <span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">"Pre&amp;Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span> <span class="ot">=</span> <span class="st">"#0ABAB5"</span>,</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>              <span class="st">"IC Cluster</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span> <span class="ot">=</span> <span class="st">"#e4eb6e"</span>,</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>              <span class="st">"Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span> <span class="ot">=</span> <span class="st">"#0A67BA"</span>,</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>              <span class="st">"Pre</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span> <span class="ot">=</span> <span class="st">"#B069EA"</span>,</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>              <span class="st">"Control</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span> <span class="ot">=</span> <span class="st">"#BA0A67"</span>,</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>              <span class="st">"Non</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span> <span class="ot">=</span> <span class="st">"darkgray"</span>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="do">## List from Zamecnik et al (paper and github link below)</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="do">## https://www.medrxiv.org/node/652357.external-links.html</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="do">## https://github.com/UCSF-Wilson-Lab/MS_DoD_and_ORIGINS_study_data</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>IC_cluster <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"SRSF4_342020</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="st">## B3G NT8_18711 ## Not in the library but reported in the manuscript?</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="st">SRSF4_342019</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="st">RIMS2_337979</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="st">CLASRP_13379</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="st">DENND4C_307092</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="st">SRSF7_101254</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="st">TRA2B_456817</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a><span class="st">RIMS2_627827</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a><span class="st">CLASRP_441187</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="st">RIMS2_306096</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a><span class="st">TRIO_395474</span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a><span class="st">SH3BP2_216602</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a><span class="st">SRSF4_682211</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a><span class="st">KRT75_287685</span></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a><span class="st">NKTR_139361</span></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a><span class="st">SRRM3_613856</span></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a><span class="st">KRT75_287686</span></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a><span class="st">SRRM3_558302</span></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a><span class="st">PPIG_130031</span></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a><span class="st">ZNF764_541112</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a><span class="st">SRSF7_331013</span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a><span class="st">USP31_222214</span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a><span class="st">RBMY1B_64485</span></span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a><span class="st">TRA2B_456818</span></span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a><span class="st">NKTR_139362</span></span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a><span class="st">MAP3K12_274436</span></span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a><span class="st">SRRM3_613857</span></span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a><span class="st">ZRANB2_281628</span></span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a><span class="st">CLASRP_13378</span></span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a><span class="st">USP31_222213</span></span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a><span class="st">NKTR_139346</span></span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a><span class="st">NKTR_139345</span></span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a><span class="st">EXO1_90756</span></span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a><span class="st">CHERP_554789</span></span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a><span class="st">LOC100652901_49861</span></span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a><span class="st">SRSF4_682210</span></span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a><span class="st">SRSF1_550431</span></span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a><span class="st">MAP3K12_274862</span></span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a><span class="st">MAP3K12_274863</span></span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a><span class="st">C5orf60_596251</span></span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a><span class="st">CHERP_554788</span></span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a><span class="st">MAP3K12_274437</span></span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a><span class="st">SRSF1_550429</span></span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a><span class="st">SRSF7_331012"</span>)</span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a>IC_cluster <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(IC_cluster,<span class="at">split  =</span> <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>))</span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a>out_dir <span class="ot">&lt;-</span> <span class="st">"../output/"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="preparing-ebv-specific-file-8mer" class="level3">
<h3 class="anchored" data-anchor-id="preparing-ebv-specific-file-8mer">Preparing EBV specific file (8mer)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../../epitope_alignment/Results/Phip_seq/Vasquez_8mer_3mis/"</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">"Epstein"</span>, files_to_read)]</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".tsv"</span>, files_to_read)]</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>files_completed <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"/stor/work/Ehrlich_COVID19/Cole/Molecular_Mimicry/Analysis/8mer_annotated_phipseq/"</span>)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="sc">!</span><span class="fu">gsub</span>(<span class="st">".tsv.gz"</span>, <span class="st">""</span>, files_to_read) <span class="sc">%in%</span> <span class="fu">gsub</span>(<span class="st">".csv.gz"</span>, <span class="st">""</span>, files_completed)]</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">order</span>(<span class="fu">file.info</span>(files_to_read)<span class="sc">$</span>size)]</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">nThread =</span> <span class="dv">12</span>)</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Skipping "</span>, active_file, <span class="st">" due to lack of data."</span>))</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a> active_data[, mismatches <span class="sc">:</span><span class="er">=</span> <span class="fu">mismatch_calling_fast</span>(query, target), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Data Merging "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Sys.time</span>()</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fwrite</span>(active_data, <span class="fu">paste0</span>(<span class="st">"../../../../Analysis/8mer_annotated_phipseq/"</span>, <span class="fu">gsub</span>(<span class="st">".tsv"</span>, <span class="st">".csv"</span>, active_file)), <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">compress =</span> <span class="st">"gzip"</span>, <span class="at">nThread =</span> <span class="dv">12</span>)</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(active_data, active_data_subset)</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gc</span>()</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>ebv_mimicry <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="fu">paste0</span>(<span class="st">"../../8mer_annotated_phipseq/Epstein-Barr_virus_UP000153037_8mer_3mis_hits.csv.gz"</span>), <span class="at">nThread =</span> <span class="dv">20</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>mimicry_model_matrix <span class="ot">&lt;-</span> ebv_mimicry <span class="sc">%&gt;%</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(target_seqname, target_start) <span class="sc">%&gt;%</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_min</span>(mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(target_seqname, target_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(target_seqname, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">factor</span>(mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">counts =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">target_seqname =</span> <span class="fu">gsub</span>(<span class="st">","</span>, <span class="st">"_"</span>, target_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">full_join</span>(peptide_gene_mapping <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"target_seqname"</span> <span class="ot">=</span> <span class="st">"seq_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">peptide_length =</span> <span class="fu">nchar</span>(sequence), <span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">freq =</span> counts <span class="sc">/</span> (peptide_length <span class="sc">-</span> <span class="dv">7</span>) <span class="sc">*</span><span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="fu">c</span>(target_seqname, peptide_length), <span class="at">names_from =</span> mismatches, <span class="at">values_from =</span> freq, <span class="at">names_expand =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(target_seqname,<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>)</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>  mimicry_model_matrix[<span class="fu">is.na</span>(mimicry_model_matrix)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(mimicry_model_matrix, <span class="st">"../../Key_Data/EBV_phipseq_8mers.csv.gz"</span>,</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>       <span class="at">nThread =</span> <span class="dv">16</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-phip-seq-data" class="level3">
<h3 class="anchored" data-anchor-id="load-phip-seq-data">Load Phip-seq Data</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>MS_phipseq <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"../../Databases/rpk_count_tables/dod_rpk_table.csv.gz"</span>, <span class="at">nThread =</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(sequence)) <span class="sc">%&gt;%</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MS070A_healthy_control_pre =</span> <span class="fu">as.numeric</span>(MS070A_healthy_control_pre )) <span class="sc">%&gt;%</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MS244A_healthy_control_post =</span> <span class="fu">as.numeric</span>(MS244A_healthy_control_post )) <span class="sc">%&gt;%</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MS117X_Case_post            =</span> <span class="fu">as.numeric</span>(MS117X_Case_post            )) <span class="sc">%&gt;%</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>MS_phipseq_meta <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colnames</span>(MS_phipseq)[<span class="sc">-</span><span class="dv">1</span><span class="sc">:-</span><span class="dv">4</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"sample_name"</span> <span class="ot">=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">timepoint =</span> <span class="fu">gsub</span>(<span class="st">".*_"</span>, <span class="st">""</span>, sample_name),</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">group =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"Case"</span>, sample_name), <span class="st">"MS"</span>,</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"healthy_control"</span>, sample_name), <span class="st">"Control"</span>, <span class="st">"Beads"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">timepoint =</span> <span class="fu">ifelse</span>(group <span class="sc">==</span> <span class="st">"Beads"</span>, <span class="cn">NA</span>, timepoint))</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>IC_cluster2 <span class="ot">&lt;-</span> MS_phipseq<span class="sc">$</span>peptide[<span class="fu">match</span>(IC_cluster, MS_phipseq<span class="sc">$</span>peptide_id)]</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(IC_cluster2) <span class="ot">&lt;-</span> IC_cluster</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="analysis" class="level3">
<h3 class="anchored" data-anchor-id="analysis">Analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Calcualte CTRL Means for each timepoint</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>control_pre <span class="ot">&lt;-</span> MS_phipseq <span class="sc">%&gt;%</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">contains</span>(<span class="fu">c</span>(<span class="st">"healthy_control_pre"</span>)))</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>control_pre_means <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(control_pre)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>control_post <span class="ot">&lt;-</span> MS_phipseq <span class="sc">%&gt;%</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">contains</span>(<span class="st">"healthy_control_post"</span>)) </span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>control_post_means <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(control_post)</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="do">### Pre MS Diagnoses</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>ms_pre <span class="ot">&lt;-</span> MS_phipseq <span class="sc">%&gt;%</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">contains</span>(<span class="st">"Case_pre"</span>))</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ms_pre) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(MS_phipseq)</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>ms_pre_FC <span class="ot">&lt;-</span> ms_pre <span class="sc">/</span> (control_pre_means <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>ms_pre_FC_mat <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(ms_pre_FC)</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>Auto_Abs_pre <span class="ot">&lt;-</span> (ms_pre_FC_mat<span class="sc">&gt;=</span><span class="dv">10</span>)</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(Auto_Abs_pre) <span class="ot">&lt;-</span> MS_phipseq<span class="sc">$</span>peptide</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>pre_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">rowSums</span>(Auto_Abs_pre))</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(pre_df) <span class="ot">&lt;-</span>  MS_phipseq<span class="sc">$</span>peptide</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a><span class="do">### Post MS Timepoint</span></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>ms_post <span class="ot">&lt;-</span> MS_phipseq <span class="sc">%&gt;%</span></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">contains</span>(<span class="st">"Case_post"</span>))</span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ms_post) <span class="ot">&lt;-</span> MS_phipseq<span class="sc">$</span>peptide</span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>ms_post_FC <span class="ot">&lt;-</span> ms_post <span class="sc">/</span> (control_post_means <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>ms_post_FC_mat <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(ms_post_FC)</span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ms_post_FC_mat)  <span class="ot">&lt;-</span> MS_phipseq<span class="sc">$</span>peptide</span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>Auto_Abs_post <span class="ot">&lt;-</span> (ms_post_FC_mat<span class="sc">&gt;=</span><span class="dv">10</span>) </span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(Auto_Abs_post) <span class="ot">&lt;-</span> MS_phipseq<span class="sc">$</span>peptide</span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>post_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">rowSums</span>(Auto_Abs_post))</span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(post_df) <span class="ot">&lt;-</span>  MS_phipseq<span class="sc">$</span>peptide</span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>post_df<span class="sc">$</span>peptide_id <span class="ot">&lt;-</span> MS_phipseq<span class="sc">$</span>peptide_id</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="control-calculations" class="level2">
<h2 class="anchored" data-anchor-id="control-calculations">Control Calculations</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="do">## To calculate the control FC, we calculate using a leave one out approach to</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="do">## actively exclude the evaluated sample from the control mean used for the calculation</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Pre Controls</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>mean_matrix_pre <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="at">nrow =</span> <span class="fu">nrow</span>(control_pre), <span class="at">ncol =</span> <span class="fu">ncol</span>(control_pre))</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(control_pre)){</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  mean_matrix_pre[,i] <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(control_pre[,<span class="sc">-</span>i])</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>ctl_pre_FC <span class="ot">&lt;-</span> control_pre <span class="sc">/</span> (mean_matrix_pre <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>ctl_pre_FC_b <span class="ot">&lt;-</span> (ctl_pre_FC<span class="sc">&gt;=</span><span class="dv">10</span>)</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>ctl_pre_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">rowSums</span>(ctl_pre_FC_b))</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ctl_pre_df) <span class="ot">&lt;-</span>  MS_phipseq<span class="sc">$</span>peptide</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Post Controls</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>mean_matrix_post <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="at">nrow =</span> <span class="fu">nrow</span>(control_post), <span class="at">ncol =</span> <span class="fu">ncol</span>(control_post))</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(control_post)){</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>  mean_matrix_post[,i] <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(control_post[,<span class="sc">-</span>i])</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>ctl_post_FC <span class="ot">&lt;-</span> control_post <span class="sc">/</span> (mean_matrix_post <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>ctl_post_FC_b <span class="ot">&lt;-</span> (ctl_post_FC<span class="sc">&gt;=</span><span class="dv">10</span>)</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ctl_post_FC_b) <span class="ot">&lt;-</span>  MS_phipseq<span class="sc">$</span>peptide</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>ctl_post_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">rowSums</span>(ctl_post_FC_b))</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ctl_post_df) <span class="ot">&lt;-</span>  MS_phipseq<span class="sc">$</span>peptide</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a>ctl_df <span class="ot">&lt;-</span> ctl_pre_df <span class="sc">%&gt;%</span></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(ctl_post_df <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"seq_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"Pre_CTL_Count"</span> <span class="ot">=</span> <span class="dv">2</span>, <span class="st">"Post_CTL_Count"</span> <span class="ot">=</span> <span class="dv">3</span>)</span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(ctl_df, <span class="fu">paste0</span>(<span class="st">"ctl_count_aab_leave_one_out.csv"</span>), <span class="at">row.names =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="panel-a" class="level2">
<h2 class="anchored" data-anchor-id="panel-a">Panel A</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>controls_df <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"ctl_count_aab_leave_one_out.csv.gz"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Pre_CTL_Count <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">|</span> Post_CTL_Count <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>mimicry_model_matrix <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"../../Key_Data/EBV_phipseq_8mers.csv.gz"</span>)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mimicry_model_matrix) <span class="ot">&lt;-</span> <span class="fu">as.character</span>(mimicry_model_matrix[<span class="dv">1</span>,])</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>mimicry_model_matrix <span class="ot">&lt;-</span> mimicry_model_matrix[<span class="sc">-</span><span class="dv">1</span>,]</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>mimicry_model_matrix[<span class="fu">is.na</span>(mimicry_model_matrix)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>pre_and_post <span class="ot">&lt;-</span> post_df <span class="sc">%&gt;%</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"post_MS_autoAbs"</span> <span class="ot">=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"target_seqname"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(pre_df <span class="sc">%&gt;%</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"pre_MS_autoAbs"</span> <span class="ot">=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"target_seqname"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(mimicry_model_matrix) <span class="sc">%&gt;%</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(peptide_gene_mapping <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(seq_id, gene, index), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"target_seqname"</span> <span class="ot">=</span> <span class="st">"seq_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">IC_cluster =</span> <span class="fu">ifelse</span>(target_seqname<span class="sc">%in%</span> IC_cluster2, <span class="st">"IC Cluster"</span>, <span class="st">"Non-IC Cluster"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(target_seqname)`
Joining with `by = join_by(target_seqname)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>pre_and_post<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">&lt;-</span> pre_and_post<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="sc">+</span> pre_and_post<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="sc">+</span> pre_and_post<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="sc">+</span> pre_and_post<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>pre_and_post<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">&lt;-</span> pre_and_post<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="sc">+</span> pre_and_post<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="sc">+</span> pre_and_post<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>pre_and_post<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">&lt;-</span> pre_and_post<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="sc">+</span> pre_and_post<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>concat_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(AutoAntibody_cutoff <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>){</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>tmp.data <span class="ot">&lt;-</span> pre_and_post <span class="sc">%&gt;%</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>,</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"2"</span>,<span class="st">"3"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(value <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"Mimic"</span>, <span class="st">"Non-mimic"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MS_autoAbs_call =</span> <span class="fu">ifelse</span>(pre_MS_autoAbs <span class="sc">&gt;=</span> AutoAntibody_cutoff, <span class="fu">ifelse</span>(post_MS_autoAbs <span class="sc">&gt;=</span>AutoAntibody_cutoff, <span class="st">"Pre&amp;Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>, <span class="st">"Pre</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>), <span class="fu">ifelse</span>(post_MS_autoAbs<span class="sc">&gt;=</span>AutoAntibody_cutoff, <span class="st">"Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>, <span class="st">"Non</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">MS_autoAbs_call =</span> <span class="fu">ifelse</span>(target_seqname <span class="sc">%in%</span> controls_df<span class="sc">$</span>seq_id, <span class="st">"Control</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>, MS_autoAbs_call)) <span class="sc">%&gt;%</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MS_autoAbs_call =</span> <span class="fu">ifelse</span>(target_seqname <span class="sc">%in%</span> pre_and_post<span class="sc">$</span>target_seqname[pre_and_post<span class="sc">$</span>IC_cluster <span class="sc">==</span> <span class="st">"IC Cluster"</span>],</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>                                   <span class="st">"IC Cluster</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>, MS_autoAbs_call)) <span class="sc">%&gt;%</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(MS_autoAbs_call, name, value) <span class="sc">%&gt;%</span></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> MS_autoAbs_call, <span class="at">values_from =</span> count)</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a><span class="do">## Add the pre and post antibodies back into the pre and post individual categories</span></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="st">"Pre</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(tmp.data) <span class="sc">&amp;</span>  <span class="st">"Pre&amp;Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(tmp.data) ){</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>  tmp.data<span class="sc">$</span><span class="st">`</span><span class="at">Pre</span><span class="sc">\n</span><span class="at">Auto-Antibody</span><span class="st">`</span> <span class="ot">&lt;-</span> tmp.data<span class="sc">$</span><span class="st">`</span><span class="at">Pre</span><span class="sc">\n</span><span class="at">Auto-Antibody</span><span class="st">`</span> <span class="sc">+</span> tmp.data<span class="sc">$</span><span class="st">`</span><span class="at">Pre&amp;Post</span><span class="sc">\n</span><span class="at">Auto-Antibody</span><span class="st">`</span></span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> <span class="cf">if</span> (<span class="st">"Pre&amp;Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(tmp.data)){</span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>  tmp.data<span class="sc">$</span><span class="st">`</span><span class="at">Pre</span><span class="sc">\n</span><span class="at">Auto-Antibody</span><span class="st">`</span> <span class="ot">&lt;-</span> tmp.data<span class="sc">$</span><span class="st">`</span><span class="at">Pre&amp;Post</span><span class="sc">\n</span><span class="at">Auto-Antibody</span><span class="st">`</span></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="st">"Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(tmp.data) <span class="sc">&amp;</span>  <span class="st">"Pre&amp;Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(tmp.data)){</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>  tmp.data<span class="sc">$</span><span class="st">`</span><span class="at">Post</span><span class="sc">\n</span><span class="at">Auto-Antibody</span><span class="st">`</span> <span class="ot">&lt;-</span> tmp.data<span class="sc">$</span><span class="st">`</span><span class="at">Post</span><span class="sc">\n</span><span class="at">Auto-Antibody</span><span class="st">`</span> <span class="sc">+</span> tmp.data<span class="sc">$</span><span class="st">`</span><span class="at">Pre&amp;Post</span><span class="sc">\n</span><span class="at">Auto-Antibody</span><span class="st">`</span></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> <span class="cf">if</span> (<span class="st">"Pre&amp;Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(tmp.data)){</span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>  tmp.data<span class="sc">$</span><span class="st">`</span><span class="at">Post</span><span class="sc">\n</span><span class="at">Auto-Antibody</span><span class="st">`</span> <span class="ot">&lt;-</span> tmp.data<span class="sc">$</span><span class="st">`</span><span class="at">Post&amp;Post</span><span class="sc">\n</span><span class="at">Auto-Antibody</span><span class="st">`</span></span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>tmp.data <span class="ot">&lt;-</span> tmp.data <span class="sc">%&gt;%</span></span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="sc">-</span>name, <span class="sc">-</span>value), <span class="at">names_to =</span> <span class="st">"MS_autoAbs_call"</span>, <span class="at">values_to =</span> <span class="st">"count"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(MS_autoAbs_call, name) <span class="sc">%&gt;%</span></span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(count), <span class="dv">0</span>, count)) <span class="sc">%&gt;%</span></span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">total =</span> <span class="fu">sum</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">freq =</span> count <span class="sc">/</span> <span class="fu">sum</span>(count) <span class="sc">*</span><span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(value <span class="sc">==</span> <span class="st">"Mimic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">ifelse</span>(name <span class="sc">==</span> <span class="dv">0</span>, <span class="fu">as.character</span>(name), <span class="fu">paste0</span>(<span class="st">"≤"</span>, name))) <span class="sc">%&gt;%</span></span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"≤1"</span>, <span class="st">"≤2"</span>, <span class="st">"≤3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MS_autoAbs_call =</span> <span class="fu">factor</span>(MS_autoAbs_call)) <span class="sc">%&gt;%</span> </span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">complete</span>(MS_autoAbs_call, name, value) <span class="sc">%&gt;%</span></span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(count), <span class="dv">0</span>, count),</span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a>         <span class="at">freq =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(freq), <span class="dv">0</span>, freq))</span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a>tmp.data<span class="sc">$</span>frequency <span class="ot">&lt;-</span> AutoAntibody_cutoff</span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a>concat_data <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(concat_data, tmp.data)</span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'MS_autoAbs_call', 'name'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'MS_autoAbs_call', 'name'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'MS_autoAbs_call', 'name'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'MS_autoAbs_call', 'name'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'MS_autoAbs_call', 'name'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'MS_autoAbs_call', 'name'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'MS_autoAbs_call', 'name'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'MS_autoAbs_call', 'name'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'MS_autoAbs_call', 'name'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'MS_autoAbs_call', 'name'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'MS_autoAbs_call', 'name'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'MS_autoAbs_call', 'name'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'MS_autoAbs_call', 'name'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'MS_autoAbs_call', 'name'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'MS_autoAbs_call', 'name'. You can override
using the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>aux_lines <span class="ot">&lt;-</span> concat_data <span class="sc">%&gt;%</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(frequency <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>MS_autoAbs_call <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Pre</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>, <span class="st">"Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>,<span class="st">"Pre&amp;Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"≤1"</span>, <span class="st">"≤2"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(MS_autoAbs_call, name, freq)</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(concat_data <span class="sc">%&gt;%</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(MS_autoAbs_call <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Pre</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>, <span class="st">"Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>,<span class="st">"Pre&amp;Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a> <span class="fu">filter</span>(name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"≤1"</span>, <span class="st">"≤2"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">freq_percent =</span> <span class="fu">ifelse</span>(MS_autoAbs_call <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>,<span class="st">"Pre&amp;Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>),</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>                               frequency<span class="sc">/</span><span class="dv">248</span><span class="sc">*</span><span class="dv">100</span>, frequency<span class="sc">/</span><span class="dv">250</span><span class="sc">*</span><span class="dv">100</span>)), <span class="fu">paste0</span>(out_dir, <span class="st">"Figure7a_datapoints.csv"</span>))</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>concat_data <span class="sc">%&gt;%</span></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(MS_autoAbs_call <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Pre</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>, <span class="st">"Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>,<span class="st">"Pre&amp;Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a> <span class="fu">filter</span>(name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"≤1"</span>, <span class="st">"≤2"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">freq_percent =</span> <span class="fu">ifelse</span>(MS_autoAbs_call <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>,<span class="st">"Pre&amp;Post</span><span class="sc">\n</span><span class="st">Auto-Antibody"</span>),</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>                               frequency<span class="sc">/</span><span class="dv">248</span><span class="sc">*</span><span class="dv">100</span>, frequency<span class="sc">/</span><span class="dv">250</span><span class="sc">*</span><span class="dv">100</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>freq_percent, <span class="at">y=</span> freq)) <span class="sc">+</span></span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">color =</span> MS_autoAbs_call)) <span class="sc">+</span></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> MS_autoAbs_call)) <span class="sc">+</span></span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">data =</span> aux_lines, <span class="fu">aes</span>(<span class="at">yintercept =</span> freq, <span class="at">color =</span> MS_autoAbs_call),</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>             <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text_repel(aes(label = count, color = MS_autoAbs_call)) +</span></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>name, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> AA_colors) <span class="sc">+</span></span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent of Auto-Antibodies with</span><span class="sc">\n</span><span class="st">an EBV matching 8mer (%)"</span>) <span class="sc">+</span></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Percent of MS patients positive for auto-antibody (%)"</span>) <span class="sc">+</span></span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">color=</span><span class="st">"black"</span>),</span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>)</span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure7_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"Figure7A.pdf"</span>), <span class="at">width =</span> <span class="dv">6</span>, <span class="at">height =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;e2&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;89&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;a4&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;e2&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;89&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;a4&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;e2&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;89&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;a4&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;e2&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;89&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;a4&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;e2&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;89&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;a4&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;e2&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;89&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;a4&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;e2&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;89&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;a4&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;e2&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;89&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;a4&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;e2&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;89&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;a4&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;e2&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;89&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤1' in 'mbcsToSbcs': dot substituted for &lt;a4&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;e2&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;89&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;a4&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;e2&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;89&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, :
conversion failure on '≤2' in 'mbcsToSbcs': dot substituted for &lt;a4&gt;</code></pre>
</div>
</div>
</section>
<section id="panel-b" class="level2">
<h2 class="anchored" data-anchor-id="panel-b">Panel B</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>pre_and_post <span class="ot">&lt;-</span> post_df <span class="sc">%&gt;%</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"post_MS_autoAbs"</span> <span class="ot">=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"target_seqname"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(pre_df <span class="sc">%&gt;%</span></span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"pre_MS_autoAbs"</span> <span class="ot">=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"target_seqname"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(!target_seqname %in% controls_df$seq_id) %&gt;%</span></span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(mimicry_model_matrix) <span class="sc">%&gt;%</span></span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(MS_phipseq <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(peptide, gene), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"target_seqname"</span> <span class="ot">=</span> <span class="st">"peptide"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">IC_cluster =</span> <span class="fu">ifelse</span>(target_seqname<span class="sc">%in%</span> IC_cluster2, <span class="st">"IC Cluster"</span>, <span class="st">"Non-IC Cluster"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(controls_df, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"target_seqname"</span> <span class="ot">=</span> <span class="st">"seq_id"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(target_seqname)`
Joining with `by = join_by(target_seqname)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>graph_data <span class="ot">&lt;-</span> pre_and_post <span class="sc">%&gt;%</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(IC_cluster <span class="sc">==</span> <span class="st">"Non-IC Cluster"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(Pre_CTL_Count) <span class="sc">&amp;</span> <span class="fu">is.na</span>(Post_CTL_Count) ) <span class="sc">%&gt;%</span></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(post_MS_autoAbs,<span class="at">n =</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span></span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(post_MS_autoAbs <span class="sc">&gt;=</span> <span class="dv">5</span> <span class="sc">|</span> pre_MS_autoAbs <span class="sc">&gt;=</span><span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">max =</span> <span class="fu">max</span>(post_MS_autoAbs, pre_MS_autoAbs)) <span class="sc">%&gt;%</span></span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(post_MS_autoAbs, pre_MS_autoAbs)) <span class="sc">%&gt;%</span></span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(name <span class="sc">==</span> <span class="st">"post_MS_autoAbs"</span>, value <span class="sc">/</span> <span class="dv">248</span><span class="sc">*</span><span class="dv">100</span>, value <span class="sc">/</span> <span class="dv">250</span><span class="sc">*</span><span class="dv">100</span>)) </span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-12"><a href="#cb55-12" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(graph_data, <span class="fu">paste0</span>(out_dir, <span class="st">"Figure7b_datapoints.csv"</span>))</span>
<span id="cb55-13"><a href="#cb55-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-14"><a href="#cb55-14" aria-hidden="true" tabindex="-1"></a>gg1 <span class="ot">&lt;-</span> graph_data <span class="sc">%&gt;%</span></span>
<span id="cb55-15"><a href="#cb55-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y=</span>value, <span class="at">x=</span><span class="fu">fct_reorder</span>(peptide_id, max), <span class="at">fill =</span> name)) <span class="sc">+</span></span>
<span id="cb55-16"><a href="#cb55-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb55-17"><a href="#cb55-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb55-18"><a href="#cb55-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent of participants (%)"</span>) <span class="sc">+</span></span>
<span id="cb55-19"><a href="#cb55-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Peptide"</span>) <span class="sc">+</span></span>
<span id="cb55-20"><a href="#cb55-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb55-21"><a href="#cb55-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#0A67BA"</span>, <span class="st">"#B069EA"</span>))<span class="sc">+</span></span>
<span id="cb55-22"><a href="#cb55-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb55-23"><a href="#cb55-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">color=</span><span class="st">"black"</span>),</span>
<span id="cb55-24"><a href="#cb55-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>)</span>
<span id="cb55-25"><a href="#cb55-25" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb55-26"><a href="#cb55-26" aria-hidden="true" tabindex="-1"></a>gg1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure7_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(showtext)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: sysfonts
Loading required package: showtextdb</code></pre>
</div>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">showtext_auto</span>()</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="fu">font_add_google</span>(<span class="st">"Roboto Mono"</span>, <span class="st">"Roboto Mono"</span>)</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>ebv_mimicry <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="fu">paste0</span>(<span class="st">"../../8mer_annotated_phipseq/Epstein-Barr_virus_UP000153037_8mer_3mis_hits.csv.gz"</span>), <span class="at">nThread =</span> <span class="dv">20</span>)</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>subsampled_mimicry <span class="ot">&lt;-</span> ebv_mimicry <span class="sc">%&gt;%</span></span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(target_seqname <span class="sc">%in%</span> MS_phipseq<span class="sc">$</span>peptide[<span class="fu">match</span>(graph_data<span class="sc">$</span>peptide_id, MS_phipseq<span class="sc">$</span>peptide_id)]) <span class="sc">%&gt;%</span></span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(mismatches <span class="sc">&lt;=</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(target_seqname, target_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(MS_phipseq <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(peptide, peptide_id), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"target_seqname"</span><span class="ot">=</span><span class="st">"peptide"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>full_position_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(subsampled_mimicry)){</span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>  active_protein <span class="ot">&lt;-</span> subsampled_mimicry[i, <span class="dv">4</span>]</span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a>  active_position <span class="ot">&lt;-</span> subsampled_mimicry[i, <span class="dv">2</span>]</span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">peptide_id =</span> <span class="fu">rep</span>(active_protein, <span class="dv">8</span>),</span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mismatches =</span> <span class="fu">rep</span>(subsampled_mimicry[i, <span class="dv">3</span>], <span class="dv">8</span>),</span>
<span id="cb58-18"><a href="#cb58-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">name =</span> active_position<span class="sc">:</span>(active_position<span class="sc">+</span><span class="dv">7</span>))</span>
<span id="cb58-19"><a href="#cb58-19" aria-hidden="true" tabindex="-1"></a>  full_position_df <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(full_position_df, tmp)</span>
<span id="cb58-20"><a href="#cb58-20" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb58-21"><a href="#cb58-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-22"><a href="#cb58-22" aria-hidden="true" tabindex="-1"></a>gg2 <span class="ot">&lt;-</span> MS_phipseq <span class="sc">%&gt;%</span></span>
<span id="cb58-23"><a href="#cb58-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(peptide_id <span class="sc">%in%</span> graph_data<span class="sc">$</span>peptide_id) <span class="sc">%&gt;%</span></span>
<span id="cb58-24"><a href="#cb58-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(gene, peptide_id, sequence) <span class="sc">%&gt;%</span></span>
<span id="cb58-25"><a href="#cb58-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(sequence,<span class="at">sep =</span> <span class="st">""</span>, <span class="fu">as.character</span>(<span class="fu">c</span>(<span class="st">"empty"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">49</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb58-26"><a href="#cb58-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>empty) <span class="sc">%&gt;%</span></span>
<span id="cb58-27"><a href="#cb58-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="sc">-</span>gene, <span class="sc">-</span>peptide_id)) <span class="sc">%&gt;%</span></span>
<span id="cb58-28"><a href="#cb58-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">as.numeric</span>(name)) <span class="sc">%&gt;%</span></span>
<span id="cb58-29"><a href="#cb58-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(full_position_df <span class="sc">%&gt;%</span></span>
<span id="cb58-30"><a href="#cb58-30" aria-hidden="true" tabindex="-1"></a>              <span class="fu">group_by</span>(peptide_id, name) <span class="sc">%&gt;%</span></span>
<span id="cb58-31"><a href="#cb58-31" aria-hidden="true" tabindex="-1"></a>              <span class="fu">slice_min</span>(mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb58-32"><a href="#cb58-32" aria-hidden="true" tabindex="-1"></a>              <span class="fu">distinct</span>(), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"name"</span>, <span class="st">"peptide_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb58-33"><a href="#cb58-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(mismatches), <span class="st">"&gt;2"</span>, mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb58-34"><a href="#cb58-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span> peptide_id)) <span class="sc">+</span></span>
<span id="cb58-35"><a href="#cb58-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb58-36"><a href="#cb58-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> value, <span class="at">color =</span> mismatches), <span class="at">size =</span> <span class="fl">2.5</span>, <span class="at">family =</span> <span class="st">"Roboto Mono"</span>) <span class="sc">+</span></span>
<span id="cb58-37"><a href="#cb58-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>() <span class="sc">+</span></span>
<span id="cb58-38"><a href="#cb58-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"1"</span> <span class="ot">=</span> <span class="st">"#FF00C3"</span>,</span>
<span id="cb58-39"><a href="#cb58-39" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"#FF0000"</span>, <span class="st">"&gt;2"</span> <span class="ot">=</span> <span class="st">"black"</span>))<span class="sc">+</span></span>
<span id="cb58-40"><a href="#cb58-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb58-41"><a href="#cb58-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">5</span>)</span>
<span id="cb58-42"><a href="#cb58-42" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb58-43"><a href="#cb58-43" aria-hidden="true" tabindex="-1"></a>gg2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure7_files/figure-html/unnamed-chunk-8-2.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(aplot)</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>gg1 <span class="sc">%&gt;%</span> <span class="fu">insert_left</span>(gg2, <span class="at">width =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure7_files/figure-html/unnamed-chunk-8-3.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"Figure7B.pdf"</span>), <span class="at">width =</span><span class="dv">8</span>, <span class="at">height =</span> <span class="dv">6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="supplemental-material" class="level2">
<h2 class="anchored" data-anchor-id="supplemental-material">Supplemental Material</h2>
<section id="figure-s8" class="level3">
<h3 class="anchored" data-anchor-id="figure-s8">Figure S8</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: grid</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
ComplexHeatmap version 2.16.0
Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
Github page: https://github.com/jokergoo/ComplexHeatmap
Documentation: http://jokergoo.github.io/ComplexHeatmap-reference

If you use it in published research, please cite either one:
- Gu, Z. Complex Heatmap Visualization. iMeta 2022.
- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
    genomic data. Bioinformatics 2016.


The new InteractiveComplexHeatmap package can directly export static 
complex heatmaps into an interactive Shiny app with zero effort. Have a try!

This message can be suppressed by:
  suppressPackageStartupMessages(library(ComplexHeatmap))
========================================</code></pre>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(circlize)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
circlize version 0.4.15
CRAN page: https://cran.r-project.org/package=circlize
Github page: https://github.com/jokergoo/circlize
Documentation: https://jokergoo.github.io/circlize_book/book/

If you use it in published research, please cite:
Gu, Z. circlize implements and enhances circular visualization
  in R. Bioinformatics 2014.

This message can be suppressed by:
  suppressPackageStartupMessages(library(circlize))
========================================</code></pre>
</div>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>col_fun <span class="ot">=</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">10</span>, <span class="dv">100</span>), <span class="fu">c</span>(<span class="st">"white"</span>, <span class="st">"orange"</span>, <span class="st">"red"</span>))</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>top_new_AutoAbs <span class="ot">&lt;-</span> graph_data<span class="sc">$</span>peptide_id</span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>ms_pre_FC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(ms_pre_FC)</span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ms_pre_FC) <span class="ot">&lt;-</span> MS_phipseq<span class="sc">$</span>peptide_id</span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>heat_df <span class="ot">&lt;-</span> <span class="fu">cbind.data.frame</span>(ms_pre_FC, ms_post_FC)</span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a>heat_df_pre <span class="ot">&lt;-</span> heat_df[<span class="fu">rownames</span>(heat_df) <span class="sc">%in%</span> <span class="fu">c</span>(top_new_AutoAbs, IC_cluster),]</span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(heat_df_pre, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS8_datapoints.csv"</span>))</span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-11"><a href="#cb66-11" aria-hidden="true" tabindex="-1"></a>hm <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>((<span class="fu">data.matrix</span>(heat_df_pre)), <span class="at">cluster_rows =</span> F, <span class="at">cluster_columns =</span> T,<span class="at">show_column_dend =</span> F, </span>
<span id="cb66-12"><a href="#cb66-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_column_names =</span> F, <span class="at">col =</span> col_fun, <span class="at">border =</span> T, <span class="at">name =</span> <span class="st">"FC over</span><span class="sc">\n</span><span class="st">Healthy Controls"</span>,</span>
<span id="cb66-13"><a href="#cb66-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">row_split =</span> <span class="fu">ifelse</span>(<span class="fu">rownames</span>(heat_df_pre) <span class="sc">%in%</span> IC_cluster, <span class="st">"IC Cluster"</span>, <span class="st">"Other MS Associated Auto-Antibodies"</span>),</span>
<span id="cb66-14"><a href="#cb66-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_split =</span> <span class="fu">ifelse</span>(<span class="fu">colnames</span>(heat_df_pre) <span class="sc">%in%</span> <span class="fu">colnames</span>(ms_pre), <span class="st">"Pre-Diagnosis"</span>, <span class="st">"Post-Diagnosis"</span>))</span>
<span id="cb66-15"><a href="#cb66-15" aria-hidden="true" tabindex="-1"></a>hm</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure7_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file=</span><span class="fu">paste0</span>(out_dir, <span class="st">"FigureS8.pdf"</span>), <span class="at">width =</span> <span class="dv">8</span>, <span class="at">height =</span> <span class="dv">11</span>)</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a> <span class="fu">draw</span>(hm)</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a> <span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>