<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Figure5_vSubFinal</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Figure5_vSubFinal2_files/libs/clipboard/clipboard.min.js"></script>
<script src="Figure5_vSubFinal2_files/libs/quarto-html/quarto.js"></script>
<script src="Figure5_vSubFinal2_files/libs/quarto-html/popper.min.js"></script>
<script src="Figure5_vSubFinal2_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Figure5_vSubFinal2_files/libs/quarto-html/anchor.min.js"></script>
<link href="Figure5_vSubFinal2_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Figure5_vSubFinal2_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Figure5_vSubFinal2_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Figure5_vSubFinal2_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Figure5_vSubFinal2_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Figure5_vSubFinal</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="figure-5" class="level1">
<h1>Figure 5</h1>
<section id="bootup" class="level2">
<h2 class="anchored" data-anchor-id="bootup">Bootup</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(seqinr)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringi)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(R.utils)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ReactomePA)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reticulate)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(treeio)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggtree)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(aplot)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(xlsx)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>out_dir <span class="ot">&lt;-</span> <span class="st">"../output/"</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../../Utilities.R"</span>)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>mart <span class="ot">&lt;-</span> <span class="fu">useEnsembl</span>(<span class="st">"ensembl"</span>, <span class="at">dataset =</span> <span class="st">"hsapiens_gene_ensembl"</span>) <span class="do">## , mirror = "useast"</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>mart_retrieved <span class="ot">&lt;-</span> <span class="fu">getBM</span>(</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">mart =</span> mart,</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">attributes =</span> <span class="fu">c</span>(</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>    <span class="st">'ensembl_gene_id'</span>,</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>    <span class="st">'entrezgene_id'</span>,</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>    <span class="st">'external_gene_name'</span>,</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>    <span class="st">'uniprot_gn_symbol'</span>,</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>    <span class="st">'uniprot_gn_id'</span>,</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chromosome_name'</span>,</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>    <span class="st">'description'</span>),</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">uniqueRows=</span><span class="cn">TRUE</span>)</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>meta_path <span class="ot">&lt;-</span> <span class="st">"../../Metadata/"</span></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"Final_Cohort_Metadata_v5.xlsx"</span>), <span class="at">sheet =</span> <span class="dv">1</span>)</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>duplicated_proteins <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"duplicated_proteins.csv"</span>))</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>human_duplicated_proteins <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"human_duplicated_or_removed_proteins.csv"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="dv">1</span>)</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>tissue_map <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"normal_tissue.tsv"</span>))</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a>organ_map <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"Tissue_Groupings.xlsx"</span>))</span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> <span class="fu">merge</span>(tissue_map, organ_map, <span class="at">by.x =</span> <span class="st">"Tissue"</span>, <span class="at">by.y =</span> <span class="st">"Var1"</span>, <span class="at">all.x =</span> T)</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(tissue_organ_map<span class="sc">$</span>Level)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Not detected"       "High"               "Low"               
[4] "Medium"             "Ascending"          "Descending"        
[7] "Not representative" "N/A"               </code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Level <span class="sc">!=</span> <span class="st">"Not detected"</span>,]</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Level <span class="sc">!=</span> <span class="st">"Not representative"</span>,]</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Level <span class="sc">!=</span> <span class="st">"N/A"</span>,]</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(tissue_organ_map<span class="sc">$</span>Reliability)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Uncertain" "Enhanced"  "Approved"  "Supported"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Reliability <span class="sc">!=</span> <span class="st">"Uncertain"</span>,]</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Reliability <span class="sc">!=</span> <span class="st">"NA"</span>,]</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>tissue_organ_map_condensed <span class="ot">&lt;-</span> tissue_organ_map <span class="sc">%&gt;%</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Tissue, Gene, <span class="st">`</span><span class="at">Cell type</span><span class="st">`</span>, <span class="st">`</span><span class="at">New Name</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Gene) <span class="sc">%&gt;%</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">organ_system =</span> <span class="fu">gsub</span>(<span class="st">"NA;"</span>, <span class="st">""</span>, <span class="fu">paste</span>(<span class="fu">unique</span>(<span class="st">`</span><span class="at">New Name</span><span class="st">`</span>), <span class="at">collapse =</span> <span class="st">";"</span>)),</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>                   <span class="at">tissue =</span> <span class="fu">gsub</span>(<span class="st">"NA;"</span>, <span class="st">""</span>, <span class="fu">paste</span>(<span class="fu">unique</span>(Tissue), <span class="at">collapse =</span> <span class="st">";"</span>)),</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>                   <span class="at">cell_type =</span> <span class="fu">gsub</span>(<span class="st">"NA;"</span>, <span class="st">""</span>, <span class="fu">paste</span>(<span class="fu">unique</span>(<span class="st">`</span><span class="at">Cell type</span><span class="st">`</span>), <span class="at">collapse =</span> <span class="st">";"</span>)))</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>protein_atlas <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"../../Databases/proteinatlas_2023-10-15.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="panel-b" class="level2">
<h2 class="anchored" data-anchor-id="panel-b">Panel B</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>ebv_mimicry <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"../../12mer_annotated/Epstein-Barr_virus_UP000153037_12mer_3mis_hits.csv.gz"</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>latent_ebv_proteins <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"P03211"</span>, <span class="st">"P03230"</span>, <span class="st">"P12978"</span>, <span class="st">"P13285"</span>, <span class="st">"Q8AZK7"</span>,</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"P03204"</span>, <span class="st">"P12977"</span>, <span class="st">"P03203"</span>)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>ebv_fasta <span class="ot">&lt;-</span> seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="st">"../../../epitope_alignment/Viral_Proteomes/Final_Cohort/Epstein-Barr_virus_UP000153037.fasta.gz"</span>, <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>, <span class="at">whole.header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>EBV_latency <span class="ot">&lt;-</span> <span class="fu">protein_compare</span>(<span class="at">virus_name =</span> <span class="st">"EBV"</span>,</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">mimicry_df =</span> ebv_mimicry,</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">fasta =</span> ebv_fasta,</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">protein_ids =</span> latent_ebv_proteins,</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">kmer_length =</span> <span class="dv">12</span>,</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">star_heights =</span> <span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">10</span>,<span class="dv">15</span>,<span class="dv">25</span>),</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">feature_name =</span> <span class="fu">c</span>(<span class="st">"Latent"</span>, <span class="st">"Lytic"</span>))</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>EBV_latency[[<span class="dv">1</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure5_vSubFinal2_files/figure-html/unnamed-chunk-1-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"Figure5B.pdf"</span>), <span class="at">width =</span> <span class="fl">4.5</span>, <span class="at">height =</span> <span class="dv">3</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(EBV_latency[[<span class="dv">2</span>]], <span class="fu">paste0</span>(out_dir, <span class="st">"TableS5.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"EBV_Latent_vs_Lytic"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="mtec-expression-test" class="level2">
<h2 class="anchored" data-anchor-id="mtec-expression-test">mTEC Expression Test</h2>
<section id="gabrielsen-i-et-al-2019-gene-expression" class="level3">
<h3 class="anchored" data-anchor-id="gabrielsen-i-et-al-2019-gene-expression">Gabrielsen I et al 2019 gene expression</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">"../../Databases/Gabrielsen_I_et_al_2019_TableS3.xlsx"</span>,</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">skip =</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"ensembl_gene_id"</span><span class="ot">=</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>(ensembl_gene_id, mTEC.MED, CD123.MED, CD141.MED, CD19.MED) <span class="sc">%&gt;%</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Any.MED =</span> <span class="fu">ifelse</span>(<span class="fu">sum</span>(<span class="fu">c</span>(mTEC.MED, CD123.MED, CD141.MED, CD19.MED)) <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                          <span class="dv">1</span>, <span class="dv">0</span>))</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="do">## gabrielsen_gene_set_binary</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>gabrielsen_gene_set_binary_both <span class="ot">&lt;-</span> all_genes <span class="sc">%&gt;%</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>ensembl_gene_id) <span class="sc">%&gt;%</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(ensembl_gene_id, name) <span class="sc">%&gt;%</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">paste0</span>(name,value)) <span class="sc">%&gt;%</span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>(name, ensembl_gene_id) <span class="sc">%&gt;%</span></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(mart_retrieved <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(ensembl_gene_id, uniprot_gn_id)) <span class="sc">%&gt;%</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>uniprot_gn_id <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">" "</span>, <span class="st">""</span>, <span class="cn">NA</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(uniprot_gn_id, name) <span class="sc">%&gt;%</span></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">uniprot =</span> <span class="dv">1</span>)</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a><span class="do">## The known frequencies of proteins</span></span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>protein_frequencies <span class="ot">&lt;-</span> gabrielsen_gene_set_binary_both <span class="sc">%&gt;%</span></span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cell_type =</span> <span class="fu">gsub</span>(<span class="st">".MED.*"</span>, <span class="st">""</span>, name),</span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">found =</span> <span class="fu">gsub</span>(<span class="st">".*MED"</span>, <span class="st">""</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> found, <span class="at">values_from =</span> count) <span class="sc">%&gt;%</span></span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">freq =</span> <span class="st">`</span><span class="at">1</span><span class="st">`</span><span class="sc">/</span>(<span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="sc">+</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="fisher-test-against-permuted" class="level3">
<h3 class="anchored" data-anchor-id="fisher-test-against-permuted">Fisher test against permuted</h3>
</section>
<section id="base-counts" class="level3">
<h3 class="anchored" data-anchor-id="base-counts">Base Counts</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../12mer_annotated/"</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".csv.gz"</span>, files_to_read)]</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>all_genes_enrichment <span class="ot">&lt;-</span> <span class="fu">unique</span>(all_genes<span class="sc">$</span>ensembl_gene_id)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file))</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">nThread =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(mismatches <span class="sc">&lt;=</span> <span class="dv">2</span>)</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>    zero_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">cell_type =</span> <span class="fu">c</span>(<span class="st">"CD123"</span>, <span class="st">"CD141"</span>, <span class="st">"CD19"</span>, <span class="st">"mTEC"</span>),</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>                    <span class="st">`</span><span class="at">0</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">rep</span>(<span class="dv">0</span>, <span class="dv">4</span>),</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>                    <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">rep</span>(<span class="dv">0</span>, <span class="dv">4</span>),</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">virus =</span> <span class="fu">rep</span>(<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer_3mis.*"</span>, <span class="st">""</span>, active_file)), <span class="dv">4</span>))</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(zero_df)[<span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>)</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>     out_final <span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rbind.fill</span>(out_final, zero_df)</span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>     <span class="cf">next</span></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>  counts_df <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> <span class="fu">c</span>(<span class="cn">NA</span>, <span class="st">""</span>, <span class="st">" "</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(gabrielsen_gene_set_binary_both) <span class="sc">%&gt;%</span></span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(name)) <span class="sc">%&gt;%</span></span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(query_start, target_start, name, uniprot, viral_id) <span class="sc">%&gt;%</span></span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">cell_type =</span> <span class="fu">gsub</span>(<span class="st">".MED.*"</span>, <span class="st">""</span>, name),</span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a>           <span class="at">expressed =</span> <span class="fu">gsub</span>(<span class="st">".*MED"</span>, <span class="st">""</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(viral_id, query_start, cell_type) <span class="sc">%&gt;%</span></span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_max</span>(expressed) <span class="sc">%&gt;%</span></span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(viral_id, query_start, cell_type, expressed) <span class="sc">%&gt;%</span></span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(cell_type, expressed) <span class="sc">%&gt;%</span></span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> expressed, <span class="at">values_from =</span> count) <span class="sc">%&gt;%</span></span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">virus =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer_3mis.*"</span>, <span class="st">""</span>, active_file)))</span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a>  out_final <span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rbind.fill</span>(out_final, counts_df)</span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Starting Adenoassociated_virus_UP000180764_12mer_3mis_hits.csv.gz"
[1] "Starting Aichi_virus_UP000119554_12mer_3mis_hits.csv.gz"
[1] "Starting Australian_bat_lyssavirus_UP000006934_12mer_3mis_hits.csv.gz"
[1] "Starting Banna_virus_UP000000832_12mer_3mis_hits.csv.gz"
[1] "Starting Barmah_forest_virus_UP000007609_12mer_3mis_hits.csv.gz"
[1] "Starting BK_polyomavirus_UP000145981_12mer_3mis_hits.csv.gz"
[1] "Starting Bunyamwera_virus_UP000002476_12mer_3mis_hits.csv.gz"
[1] "Starting Bunyavirus_La_Crosse_UP000008768_12mer_3mis_hits.csv.gz"
[1] "Starting Bunyavirus_snowshoe_hare_UP000150053_12mer_3mis_hits.csv.gz"
[1] "Starting Cercopithecine_alphaherpesvirus_UP000137988_12mer_3mis_hits.csv.gz"
[1] "Starting Chandipura_virus_UP000008448_12mer_3mis_hits.csv.gz"
[1] "Starting Chikungunya_virus_UP000000569_12mer_3mis_hits.csv.gz"
[1] "Starting Cosavirus_A_UP000029753_12mer_3mis_hits.csv.gz"
[1] "Starting Cowpox_virus_UP000152733_12mer_3mis_hits.csv.gz"
[1] "Starting Coxsackievirus_UP000007757_12mer_3mis_hits.csv.gz"
[1] "Starting Crimean-Congo_hemorrhagic_fever_virus_UP000008767_12mer_3mis_hits.csv.gz"
[1] "Starting Dengue_virus_UP000002500_12mer_3mis_hits.csv.gz"
[1] "Starting Dhori_virus_UP000181622_12mer_3mis_hits.csv.gz"
[1] "Starting Dugbe_virus_UP000000278_12mer_3mis_hits.csv.gz"
[1] "Starting Duvenhage_virus_UP000117552_12mer_3mis_hits.csv.gz"
[1] "Starting Eastern_equine_encephalitis_virus_UP000110644_12mer_3mis_hits.csv.gz"
[1] "Starting Ebolavirus_UP000007209_12mer_3mis_hits.csv.gz"
[1] "Starting Echovirus_UP000000670_12mer_3mis_hits.csv.gz"
[1] "Starting Encephalomyocarditis_virus_UP000002319_12mer_3mis_hits.csv.gz"
[1] "Starting Epstein-Barr_virus_UP000153037_12mer_3mis_hits.csv.gz"
[1] "Starting European_bat_lyssavirus_UP000008926_12mer_3mis_hits.csv.gz"
[1] "Starting Hantaan_virus_UP000008627_12mer_3mis_hits.csv.gz"
[1] "Starting Hendra_virus_UP000008771_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_A_virus_UP000006724_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_B_virus_UP000008591_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_C_virus_UP000000518_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_delta_virus_UP000006675_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_E_virus_UP000006705_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_G_virus_UP000007407_12mer_3mis_hits.csv.gz"
[1] "Starting Horsepox_virus_UP000111173_12mer_3mis_hits.csv.gz"
[1] "Starting Human_adenovirus_UP000008167_12mer_3mis_hits.csv.gz"
[1] "Starting Human_astrovirus_UP000001650_12mer_3mis_hits.csv.gz"
[1] "Starting Human_coronavirus_UP000006716_12mer_3mis_hits.csv.gz"
[1] "Starting Human_cytomegalovirus_merlin_UP000000938_12mer_3mis_hits.csv.gz"
[1] "Starting Human_enterovirus_68_UP000143570_12mer_3mis_hits.csv.gz"
[1] "Starting Human_enterovirus_70_UP000008683_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus_1_UP000009294_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus_2_UP000001874_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus_6_UP000009295_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus_7_UP000009246_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus_8_UP000000942_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus6b_UP000006930_12mer_3mis_hits.csv.gz"
[1] "Starting Human_immunodeficiency_virus_1_UP000002241_12mer_3mis_hits.csv.gz"
[1] "Starting Human_immunodeficiency_virus_2_UP000002242_12mer_3mis_hits.csv.gz"
[1] "Starting Human_papillomavirus_1_UP000006372_12mer_3mis_hits.csv.gz"
[1] "Starting Human_papillomavirus_2_UP000007710_12mer_3mis_hits.csv.gz"
[1] "Starting Human_papillomavirus_type_16_UP000009251_12mer_3mis_hits.csv.gz"
[1] "Starting Human_papillomavirus_type_18_UP000009109_12mer_3mis_hits.csv.gz"
[1] "Starting Human_parainfluenza_UP000109776_12mer_3mis_hits.csv.gz"
[1] "Starting Human_parvovirus_B19_UP000006624_12mer_3mis_hits.csv.gz"
[1] "Starting Human_respiratory_syncytial_virus_UP000002472_12mer_3mis_hits.csv.gz"
[1] "Starting Human_rhinovirus_UP000007070_12mer_3mis_hits.csv.gz"
[1] "Starting Human_SARS_coronavirus_UP000000354_12mer_3mis_hits.csv.gz"
[1] "Starting Human_spumaretrovirus_UP000138352_12mer_3mis_hits.csv.gz"
[1] "Starting Human_T-lymphotropic_virus_1_UP000001061_12mer_3mis_hits.csv.gz"
[1] "Starting Human_torovirus_UP000243571_12mer_3mis_hits.csv.gz"
[1] "Starting Influenza_A_virus_UP000009255_12mer_3mis_hits.csv.gz"
[1] "Starting Influenza_B_virus_UP000008158_12mer_3mis_hits.csv.gz"
[1] "Starting Influenza_C_virus_UP000008286_12mer_3mis_hits.csv.gz"
[1] "Starting Isfahan_virus_UP000204017_12mer_3mis_hits.csv.gz"
[1] "Starting Japanese_encephalitis_virus_UP000007214_12mer_3mis_hits.csv.gz"
[1] "Starting JC_polyomavirus_UP000008478_12mer_3mis_hits.csv.gz"
[1] "Starting Junin_mammarenavirus_UP000127886_12mer_3mis_hits.csv.gz"
[1] "Starting KI_Polyomavirus_UP000107189_12mer_3mis_hits.csv.gz"
[1] "Starting Kunjin_virus_UP000138183_12mer_3mis_hits.csv.gz"
[1] "Starting Lagos_bat_virus_UP000161306_12mer_3mis_hits.csv.gz"
[1] "Starting Lake_Victoria_marburgvirus_UP000180448_12mer_3mis_hits.csv.gz"
[1] "Starting Langat_virus_UP000180575_12mer_3mis_hits.csv.gz"
[1] "Starting Lassa_virus_UP000002473_12mer_3mis_hits.csv.gz"
[1] "Starting Lordsdale_virus_UP000007767_12mer_3mis_hits.csv.gz"
[1] "Starting Louping_ill_virus_UP000007644_12mer_3mis_hits.csv.gz"
[1] "Starting Lymphocytic_choriomeningitis_virus_UP000002474_12mer_3mis_hits.csv.gz"
[1] "Starting Machupo_virus_UP000233900_12mer_3mis_hits.csv.gz"
[1] "Starting Mayaro_virus_UP000007774_12mer_3mis_hits.csv.gz"
[1] "Starting Measles_virus_UP000008699_12mer_3mis_hits.csv.gz"
[1] "Starting Mengo_encephalomyocarditis_virus_UP000008663_12mer_3mis_hits.csv.gz"
[1] "Starting Merkel_cell_polyomavirus_UP000154903_12mer_3mis_hits.csv.gz"
[1] "Starting MERS_coronavirus_UP000147828_12mer_3mis_hits.csv.gz"
[1] "Starting MERS_UP000171868_12mer_3mis_hits.csv.gz"
[1] "Starting Mokola_virus_UP000006826_12mer_3mis_hits.csv.gz"
[1] "Starting Molluscum_contagiosum_virus_UP000000869_12mer_3mis_hits.csv.gz"
[1] "Starting Monkeypox_virus_UP000101269_12mer_3mis_hits.csv.gz"
[1] "Starting Mumps_virus_UP000002331_12mer_3mis_hits.csv.gz"
[1] "Starting Murray_valley_encephalitis_virus_UP000008863_12mer_3mis_hits.csv.gz"
[1] "Starting Nipah_virus_UP000170143_12mer_3mis_hits.csv.gz"
[1] "Starting Norwalk_virus_UP000000826_12mer_3mis_hits.csv.gz"
[1] "Starting Onyong_nyong_virus_UP000008868_12mer_3mis_hits.csv.gz"
[1] "Starting Orf_virus_UP000000870_12mer_3mis_hits.csv.gz"
[1] "Starting Oropouche_virus_UP000117676_12mer_3mis_hits.csv.gz"
[1] "Starting Pichinde_mammarenavirus_UP000139415_12mer_3mis_hits.csv.gz"
[1] "Starting Poliovirus_UP000000356_12mer_3mis_hits.csv.gz"
[1] "Starting Punta_toro_phlebovirus_UP000214480_12mer_3mis_hits.csv.gz"
[1] "Starting Puumala_virus_UP000110237_12mer_3mis_hits.csv.gz"
[1] "Starting Rabies_virus_UP000008649_12mer_3mis_hits.csv.gz"
[1] "Starting Rift_valley_fever_virus_UP000002477_12mer_3mis_hits.csv.gz"
[1] "Starting Rosavirus_A_UP000029743_12mer_3mis_hits.csv.gz"
[1] "Starting Ross_river_virus_UP000006579_12mer_3mis_hits.csv.gz"
[1] "Starting Rotavirus_A_UP000001119_12mer_3mis_hits.csv.gz"
[1] "Starting Rotavirus_B_UP000114294_12mer_3mis_hits.csv.gz"
[1] "Starting Rotavirus_C_UP000007664_12mer_3mis_hits.csv.gz"
[1] "Starting Rubella_virus_UP000000571_12mer_3mis_hits.csv.gz"
[1] "Starting Sagiyama_virus_UP000007138_12mer_3mis_hits.csv.gz"
[1] "Starting Salivirus_A_UP000029742_12mer_3mis_hits.csv.gz"
[1] "Starting Sandfly_fever_sicilian_virus_UP000207632_12mer_3mis_hits.csv.gz"
[1] "Starting Sapporo_virus_UP000000667_12mer_3mis_hits.csv.gz"
[1] "Starting SARS_coronavirus_2_UP000464024_12mer_3mis_hits.csv.gz"
[1] "Starting Semliki_forest_virus_UP000000570_12mer_3mis_hits.csv.gz"
[1] "Starting Seoul_virus_UP000207620_12mer_3mis_hits.csv.gz"
[1] "Starting Simian_foamy_virus_UP000001063_12mer_3mis_hits.csv.gz"
[1] "Starting Simian_virus_5_UP000007232_12mer_3mis_hits.csv.gz"
[1] "Starting Sindbis_virus_UP000006710_12mer_3mis_hits.csv.gz"
[1] "Starting Southampton_virus_UP000007226_12mer_3mis_hits.csv.gz"
[1] "Starting St._louis_encephalitis_virus_UP000171556_12mer_3mis_hits.csv.gz"
[1] "Starting Tick-borne_powassan_virus_UP000006848_12mer_3mis_hits.csv.gz"
[1] "Starting Torque_teno_virus_UP000001448_12mer_3mis_hits.csv.gz"
[1] "Starting Toscana_virus_UP000142444_12mer_3mis_hits.csv.gz"
[1] "Starting Uukuniemi_virus_UP000008595_12mer_3mis_hits.csv.gz"
[1] "Starting Vaccinia_virus_UP000000344_12mer_3mis_hits.csv.gz"
[1] "Starting Varicella-zoster_virus_UP000002602_12mer_3mis_hits.csv.gz"
[1] "Starting Variola_virus_UP000002060_12mer_3mis_hits.csv.gz"
[1] "Starting Venezuelan_equine_encephalitis_virus_UP000171441_12mer_3mis_hits.csv.gz"
[1] "Starting Vesicular_stomatitis_virus_UP000002327_12mer_3mis_hits.csv.gz"
[1] "Starting West_Nile_virus_UP000136458_12mer_3mis_hits.csv.gz"
[1] "Starting Western_equine_encephalitis_virus_UP000128494_12mer_3mis_hits.csv.gz"
[1] "Starting WU_polyomavirus_UP000118542_12mer_3mis_hits.csv.gz"
[1] "Starting Yaba_monkey_tumor_virus_UP000008596_12mer_3mis_hits.csv.gz"
[1] "Starting Yaba-like_disease_virus_UP000136581_12mer_3mis_hits.csv.gz"
[1] "Starting Yellow_fever_virus_UP000000360_12mer_3mis_hits.csv.gz"
[1] "Starting Zika_virus_UP000054557_12mer_3mis_hits.csv.gz"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_final, <span class="fu">paste0</span>(<span class="st">"../Figure5/output/"</span>, <span class="st">"mTEC_real.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="panel-c" class="level3">
<h3 class="anchored" data-anchor-id="panel-c">Panel C</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"mTEC_real.csv"</span>), <span class="at">row.names =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">virus =</span> <span class="fu">gsub</span>(<span class="st">"-.*"</span>, <span class="st">""</span>, virus)) <span class="sc">%&gt;%</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(protein_frequencies) <span class="sc">%&gt;%</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">X0</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(<span class="st">`</span><span class="at">X0</span><span class="st">`</span>), <span class="dv">0</span>, <span class="st">`</span><span class="at">X0</span><span class="st">`</span>),</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">X1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(<span class="st">`</span><span class="at">X1</span><span class="st">`</span>), <span class="dv">0</span>, <span class="st">`</span><span class="at">X1</span><span class="st">`</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">freq.viral =</span> <span class="st">`</span><span class="at">X1</span><span class="st">`</span> <span class="sc">/</span> (X1 <span class="sc">+</span>X0)) <span class="sc">%&gt;%</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.value =</span> <span class="cn">NA</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(cell_type)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(out_final)){</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  count_table <span class="ot">&lt;-</span> <span class="fu">mapply</span>(c,out_final[i, <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>)],out_final[i, <span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">6</span>)])</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  count_table[<span class="fu">is.na</span>(count_table)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> <span class="fu">try</span>(<span class="fu">fisher.test</span>(count_table), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>   <span class="cf">if</span>(<span class="fu">class</span>(tmp) <span class="sc">!=</span> <span class="st">"try-error"</span>){</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>      out_final<span class="sc">$</span>p.value[i] <span class="ot">&lt;-</span> tmp<span class="sc">$</span>p.value</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>   }</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>graph_data <span class="ot">&lt;-</span> out_final <span class="sc">%&gt;%</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"virus"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">virus</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">factor</span>(label)) <span class="sc">%&gt;%</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FC =</span> freq.viral <span class="sc">/</span> freq) <span class="sc">%&gt;%</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj =</span> <span class="fu">p.adjust</span>(p.value, <span class="st">"fdr"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">log10p =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adj)) <span class="sc">%&gt;%</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">direction =</span> <span class="fu">ifelse</span>(FC<span class="sc">&gt;</span><span class="dv">1</span>, <span class="st">"Up"</span>, <span class="st">"Down"</span>))</span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>graph_data_out <span class="ot">&lt;-</span> graph_data</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(graph_data_out)[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">6</span>)] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Hits_viral"</span>, <span class="st">"NoHit_viral"</span>, <span class="st">"Not_Expressed_Human"</span>, <span class="st">"Expressed_Human"</span>)</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(graph_data_out, <span class="fu">paste0</span>(out_dir, <span class="st">"TableS5.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"Thymic_Cell_Fisher_Test"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>gg5 <span class="ot">&lt;-</span> graph_data <span class="sc">%&gt;%</span></span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">difference =</span> freq.viral<span class="sc">*</span> <span class="dv">100</span> <span class="sc">-</span> freq <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>label, <span class="at">y=</span>cell_type)) <span class="sc">+</span></span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">size =</span>log10p, <span class="at">fill =</span> difference), <span class="at">shape =</span><span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_fill_manual(values = c("#6999EA", "#EA6983")) +</span></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>,<span class="dv">10</span>),</span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">colours=</span><span class="fu">c</span>(<span class="st">"#2213c2"</span>, <span class="st">"lightgray"</span>, <span class="st">"#b8180d"</span>),</span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">breaks=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="sc">-</span><span class="dv">5</span>, <span class="dv">0</span>, <span class="dv">5</span>, <span class="dv">10</span>),</span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>  <span class="at">guide =</span> <span class="fu">guide_colorbar</span>(<span class="at">frame.colour =</span> <span class="st">"black"</span>, <span class="at">ticks.colour =</span> <span class="st">"black"</span>),</span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">"Percent Difference"</span>, <span class="at">oob=</span>squish) <span class="sc">+</span></span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"00FFFFFF"</span>, <span class="st">"black"</span>)) <span class="sc">+</span> </span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">drop=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Thymus Cell Type"</span>) <span class="sc">+</span></span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(<span class="at">color=</span><span class="st">"grey"</span>, <span class="at">size=</span><span class="fl">0.1</span>), </span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">0</span>),</span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">fill=</span><span class="cn">NA</span>, <span class="at">size=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">position =</span> <span class="st">"top"</span>) <span class="sc">+</span></span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">size=</span><span class="fu">guide_legend</span>(<span class="st">"-log10 Transformed</span><span class="sc">\n</span><span class="st">Adjusted p-values"</span>),</span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a>         <span class="co">#fill=guide_legend("Percent Difference"),</span></span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a>         <span class="at">color=</span><span class="fu">guide_legend</span>(<span class="st">"Significant in all</span><span class="sc">\n</span><span class="st">three permutations"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The `size` argument of `element_rect()` is deprecated as of ggplot2 3.4.0.
ℹ Please use the `linewidth` argument instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.
ℹ Please use the `linewidth` argument instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Scale for x is already present.
Adding another scale for x, which will replace the existing scale.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>gg5</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure5_vSubFinal2_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>taxa <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Proteome.ID <span class="sc">%in%</span> graph_data<span class="sc">$</span><span class="st">`</span><span class="at">virus</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(family, order, class, phylum, kingdom) <span class="sc">%&gt;%</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Anelloviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Kolmioviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Kolmioviridae"</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">lapply</span>(taxa, as.factor), <span class="at">stringsAsFactors =</span><span class="cn">TRUE</span>)</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(check) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(taxa)</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> check[<span class="fu">complete.cases</span>(check),]</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">taxo2phylog</span>(<span class="fu">as.taxo</span>(check), <span class="at">root=</span><span class="st">"Root"</span>, <span class="at">abbrev=</span>F)</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>tax.phy <span class="ot">&lt;-</span> <span class="fu">as.phylo</span>(tmp)</span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tax.phy, <span class="at">branch.length=</span><span class="st">'none'</span>)</span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a>gg_tree</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure5_vSubFinal2_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>gg_tree[[<span class="st">"data"</span>]] <span class="ot">&lt;-</span> gg_tree[[<span class="st">"data"</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata <span class="sc">%&gt;%</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>               <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(Taxon.mnemonic, family), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>))</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span>  gg_tree <span class="sc">+</span> <span class="fu">geom_hilight</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">subset =</span> family <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2),<span class="at">fill =</span> family), <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>   <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> Family_palette_v2, <span class="at">name =</span> <span class="st">"Viral Family"</span>)</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>gg5 <span class="sc">%&gt;%</span></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">insert_bottom</span>(gg_tree <span class="sc">+</span> <span class="fu">coord_flip</span>(),<span class="at">height =</span> <span class="fl">0.2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure5_vSubFinal2_files/figure-html/unnamed-chunk-5-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"Figure5C.pdf"</span>), <span class="at">width =</span><span class="dv">15</span>, <span class="at">height =</span> <span class="dv">4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>