<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Figure2</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Figure2_vSubFinal_files/libs/clipboard/clipboard.min.js"></script>
<script src="Figure2_vSubFinal_files/libs/quarto-html/quarto.js"></script>
<script src="Figure2_vSubFinal_files/libs/quarto-html/popper.min.js"></script>
<script src="Figure2_vSubFinal_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Figure2_vSubFinal_files/libs/quarto-html/anchor.min.js"></script>
<link href="Figure2_vSubFinal_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Figure2_vSubFinal_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Figure2_vSubFinal_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Figure2_vSubFinal_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Figure2_vSubFinal_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Figure2</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="figure2" class="level2">
<h2 class="anchored" data-anchor-id="figure2">Figure2</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(seqinr)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringi)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(svMisc)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(xlsx)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(treeio)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggtree)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(aplot)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../../Utilities.R"</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>meta_path <span class="ot">&lt;-</span> <span class="st">"../../Metadata/"</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"Final_Cohort_Metadata_v5.xlsx"</span>), <span class="at">sheet =</span> <span class="dv">1</span>)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>duplicated_proteins <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"duplicated_proteins.csv"</span>))</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>human_duplicated_proteins <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"human_duplicated_or_removed_proteins.csv"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="dv">1</span>)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>final_cohort_naming <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../../Viral_Proteomes/Final_Cohort"</span>)</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>final_cohort_naming <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">gsub</span>(<span class="st">".fasta.gz"</span>, <span class="st">"_12mer_3mis_hits.tsv"</span>, final_cohort_naming))</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>out_dir <span class="ot">&lt;-</span> <span class="st">"../output/"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="random-scramble-processing" class="level2">
<h2 class="anchored" data-anchor-id="random-scramble-processing">Random Scramble Processing</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../../epitope_alignment/Results/Permutation_exp1"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".tsv"</span>, files_to_read)]</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>final_cohort_naming <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../../Viral_Proteomes/Final_Cohort"</span>)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>final_cohort_naming <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">gsub</span>(<span class="st">".fasta.gz"</span>, <span class="st">""</span>, final_cohort_naming))</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">gsub</span>(<span class="st">"-[1-9].*"</span>, <span class="st">""</span>, files_to_read) <span class="sc">%in%</span> <span class="fu">paste0</span>(<span class="st">"Full_Scramble_"</span>, final_cohort_naming)]</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"Proteome Length"</span>, <span class="st">"Number of Proteins"</span>))</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>out_protein_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" Pathogen number:"</span>,<span class="fu">match</span>(active_file, files_to_read), <span class="st">"/"</span>, <span class="fu">length</span>(files_to_read)))</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">nThread =</span> <span class="dv">10</span>)</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>    out_final[,<span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.tsv"</span>, <span class="st">""</span>, active_file)] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="cn">NA</span>,<span class="cn">NA</span>)</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>      data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../Viral_Proteomes/Final_Cohort/"</span>, <span class="fu">gsub</span>(<span class="st">"Full_Scramble_"</span>, <span class="st">""</span>, <span class="fu">gsub</span>(<span class="st">"-[1-9].*"</span>, <span class="st">""</span>, active_file)), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>       PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>      PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>uniprot <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(active_data<span class="sc">$</span>target_seqname, <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>]))</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>viral_id <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(active_data<span class="sc">$</span>query_seqname, <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>]))</span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Filter out any duplicated proteins, identified in </span></span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>  <span class="do">## ../../Scripts/Duplicate_Protein_Screen.R</span></span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>mismatches <span class="ot">&lt;-</span> <span class="fu">apply</span>(active_data, <span class="dv">1</span>, <span class="cf">function</span>(x){<span class="fu">DNA_Diff</span>(x[<span class="dv">1</span>], x[<span class="dv">6</span>])})</span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>mismatches <span class="ot">&lt;-</span> <span class="fu">factor</span>(active_data<span class="sc">$</span>mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))</span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(active_data)</span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(active_data<span class="sc">$</span>mismatches))</span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(out) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.tsv"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a>  <span class="do">############## Count Proteome Length ##############</span></span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../Viral_Proteomes/Final_Cohort/"</span>, <span class="fu">gsub</span>(<span class="st">"Full_Scramble_"</span>, <span class="st">""</span>, <span class="fu">gsub</span>(<span class="st">"-[1-9].*"</span>, <span class="st">""</span>, active_file)), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a>       PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a>      PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a>  out_final <span class="ot">&lt;-</span> <span class="fu">merge</span>(out_final, out, <span class="at">by =</span> <span class="st">"mismatches"</span>, <span class="at">all.x =</span> T)</span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-62"><a href="#cb2-62" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_final, <span class="st">"../../../Analysis/Figure2/output/12mer_perm1_random_scramble.csv"</span>, <span class="at">row.names =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="random-scramble-graphs" class="level2">
<h2 class="anchored" data-anchor-id="random-scramble-graphs">Random Scramble Graphs</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/12mer_perm1_random_scramble.csv"</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(out_final) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">."</span>, <span class="st">"-"</span>, <span class="fu">colnames</span>(out_final))</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>true_counts <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/12mer_proteome_everything.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() </span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(true_counts) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">."</span>, <span class="st">"-"</span>, <span class="fu">colnames</span>(true_counts))</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>true_counts[<span class="fu">is.na</span>(true_counts)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>random_scram_results <span class="ot">&lt;-</span> <span class="fu">t</span>(true_counts[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,]) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"Species"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Species <span class="sc">!=</span> <span class="st">"mismatches"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"0"</span><span class="ot">=</span><span class="st">"1"</span>,<span class="st">"1"</span><span class="ot">=</span><span class="st">"2"</span>,<span class="st">"2"</span><span class="ot">=</span><span class="st">"3"</span>,<span class="st">"3"</span><span class="ot">=</span><span class="st">"4"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>,<span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="st">`</span><span class="at">3</span><span class="st">`</span>),</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>),</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>Species) <span class="sc">%&gt;%</span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">t</span>(out_final[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,]) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rownames_to_column</span>(<span class="st">"Species"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(Species <span class="sc">!=</span> <span class="st">"mismatches"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"0"</span><span class="ot">=</span><span class="st">"1"</span>,<span class="st">"1"</span><span class="ot">=</span><span class="st">"2"</span>,<span class="st">"2"</span><span class="ot">=</span><span class="st">"3"</span>,<span class="st">"3"</span><span class="ot">=</span><span class="st">"4"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>,<span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="st">`</span><span class="at">3</span><span class="st">`</span>),</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>             <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>),</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>             <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>      <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>Species) <span class="sc">%&gt;%</span></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">value_rand =</span> value) <span class="sc">%&gt;%</span></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">Species =</span> <span class="fu">gsub</span>(<span class="st">"-[1-9].*"</span>, <span class="st">""</span>, Species)) <span class="sc">%&gt;%</span></span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(Species, name) <span class="sc">%&gt;%</span></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">value_rand =</span> <span class="fu">mean</span>(value_rand)) <span class="sc">%&gt;%</span></span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">Species =</span> <span class="fu">gsub</span>(<span class="st">"Full_Scramble_"</span>, <span class="st">""</span>, Species)), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Species"</span>, <span class="st">"name"</span>)</span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>  )  <span class="sc">%&gt;%</span></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"mismatches"</span> <span class="ot">=</span> <span class="st">"name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FC =</span> (value<span class="sc">+</span><span class="dv">10</span>) <span class="sc">/</span> (value_rand <span class="sc">+</span> <span class="dv">10</span>))</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>gg1 <span class="ot">&lt;-</span> random_scram_results <span class="sc">%&gt;%</span></span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, Species)) <span class="sc">%&gt;%</span></span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span><span class="fu">ifelse</span>(mismatches <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">"≤"</span>, mismatches), mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">factor</span>(mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"≤1"</span>, <span class="st">"≤2"</span>, <span class="st">"≤3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(mismatches, label, <span class="at">fill=</span>FC)) <span class="sc">+</span> <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Number of Mismatches"</span>) <span class="sc">+</span></span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_gradientn</span>(<span class="at">colours =</span> <span class="fu">c</span>(<span class="st">"#045375"</span>, <span class="st">"#089099"</span>, <span class="st">"#7CCBA2"</span>, <span class="st">"#FCDE9C"</span>, <span class="st">"#FAB27B"</span>, <span class="st">"#E23333"</span>),</span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>                       <span class="at">limits=</span><span class="fu">c</span>(<span class="fl">0.2</span>, <span class="dv">32</span>), <span class="at">oob =</span> scales<span class="sc">::</span>squish, <span class="at">trans =</span> <span class="st">"log"</span>,</span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>                       <span class="at">breaks =</span> <span class="fu">c</span>(<span class="fl">0.25</span>, <span class="fl">0.5</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">8</span>, <span class="dv">16</span>, <span class="dv">32</span>)) <span class="sc">+</span></span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">7</span>),</span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">9</span>),</span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">fill=</span><span class="cn">NA</span>, <span class="at">size=</span><span class="dv">1</span>)) </span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a>gg1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure2_vSubFinal_files/figure-html/perm1_graphing_1-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>taxa <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(family, order, class, phylum, kingdom) <span class="sc">%&gt;%</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Anelloviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Kolmioviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Kolmioviridae"</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">lapply</span>(taxa, as.factor), <span class="at">stringsAsFactors =</span><span class="cn">TRUE</span>)</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(check) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(taxa)</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> check[<span class="fu">complete.cases</span>(check),]</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">taxo2phylog</span>(<span class="fu">as.taxo</span>(check), <span class="at">root=</span><span class="st">"Root"</span>, <span class="at">abbrev=</span>F)</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>tax.phy <span class="ot">&lt;-</span> <span class="fu">as.phylo</span>(tmp)</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tax.phy, <span class="at">branch.length=</span><span class="st">'none'</span>)</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>gg_tree[[<span class="st">"data"</span>]] <span class="ot">&lt;-</span> gg_tree[[<span class="st">"data"</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata <span class="sc">%&gt;%</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>               <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(Taxon.mnemonic, family), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>))</span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span>  gg_tree <span class="sc">+</span> <span class="fu">geom_hilight</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">subset =</span> family <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2),<span class="at">fill =</span> family), <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>   <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> Family_palette_v2)</span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>gg1 <span class="sc">%&gt;%</span> <span class="fu">insert_left</span>(gg_tree,<span class="at">width =</span> <span class="fl">0.2</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure2_vSubFinal_files/figure-html/perm1_graphing_2-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(paste0(out_dir, "random_heatmap.pdf"), width = 4, height =12)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> random_scram_results <span class="sc">%&gt;%</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, Species)) <span class="sc">%&gt;%</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"Virus"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>))</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>wt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> FC <span class="sc">~</span> Chronic.or.Acute,</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(wt, <span class="fu">paste0</span>(out_dir, <span class="st">"figure2_statistics.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"Perm1_Chronic_vs_Acute_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">FALSE</span>)</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>gg3_perm1 <span class="ot">&lt;-</span> random_scram_results <span class="sc">%&gt;%</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, Species)) <span class="sc">%&gt;%</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">"Bacteria"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Chronic.or.Acute =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"Bacteria"</span>, Taxonomic.lineage), <span class="st">"Bacteria"</span>, Chronic.or.Acute)) <span class="sc">%&gt;%</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FC =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(FC), <span class="dv">0</span>, FC)) <span class="sc">%&gt;%</span></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(FC), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(FC)) <span class="sc">%&gt;%</span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span><span class="fu">ifelse</span>(mismatches <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">"≤"</span>, mismatches), mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">factor</span>(mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"≤1"</span>, <span class="st">"≤2"</span>, <span class="st">"≤3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>mismatches, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>)  <span class="sc">+</span></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>mismatches), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span><span class="fu">c</span>(<span class="st">"#1975EE"</span>, <span class="st">"#E78305"</span>, <span class="st">"black"</span>)) <span class="sc">+</span></span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Fold Change of</span><span class="sc">\n</span><span class="st">Actual over Permutation 1"</span>) <span class="sc">+</span></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log2_trans</span>(),</span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log2"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">2</span><span class="sc">^</span>x, <span class="dv">3</span>)))<span class="sc">+</span></span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">1</span>),</span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">6</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a>gg3_perm1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure2_vSubFinal_files/figure-html/perm1_graphing_3-1.png" class="img-fluid" width="384"></p>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(paste0(out_dir, "random_scram_chronicity.pdf"), width = 4, height = 3)</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> random_scram_results <span class="sc">%&gt;%</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, Species)) <span class="sc">%&gt;%</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Herpesviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Herpesviridae"</span>, <span class="st">"Herpesviridae"</span>, <span class="st">"Non-Herpesviridae"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Poxviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Poxviridae"</span>, <span class="st">"Poxviridae"</span>, <span class="st">"Non-Poxviridae"</span>))</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="do">## These results are used to manually add stars to the graph below in illustrator</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>kt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> FC <span class="sc">~</span> Family,</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>), <span class="at">method =</span> <span class="st">"kruskal.test"</span>)</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>wt.p <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> FC <span class="sc">~</span> Poxviridae,</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>wt.h <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> FC <span class="sc">~</span> Herpesviridae,</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>wt.hp <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Family <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Herpesviridae"</span>, <span class="st">"Poxviridae"</span>)), <span class="at">formula =</span> FC <span class="sc">~</span> Poxviridae,</span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>)))) <span class="sc">%&gt;%</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group1 =</span> <span class="st">"Herpesviridae"</span>)</span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(kt, <span class="fu">paste0</span>(out_dir, <span class="st">"figure2_statistics.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"Perm1_Viral_Family_Kruskal"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(wt.p, <span class="fu">paste0</span>(out_dir, <span class="st">"figure2_statistics.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"Perm1_Pox_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(wt.h, <span class="fu">paste0</span>(out_dir, <span class="st">"figure2_statistics.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"Perm1_Herpes_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(wt.hp, <span class="fu">paste0</span>(out_dir, <span class="st">"figure2_statistics.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"Perm1_Herpes_vs_Pox_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>gg4_perm1 <span class="ot">&lt;-</span> random_scram_results <span class="sc">%&gt;%</span></span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, Species)) <span class="sc">%&gt;%</span></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Family, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FC =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(FC), <span class="dv">0</span>, FC)) <span class="sc">%&gt;%</span></span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(FC), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(FC)) <span class="sc">%&gt;%</span></span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(count <span class="sc">&gt;</span> <span class="dv">4</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span><span class="fu">ifelse</span>(mismatches <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">"≤"</span>, mismatches), mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">factor</span>(mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"≤1"</span>, <span class="st">"≤2"</span>, <span class="st">"≤3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>mismatches, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>)  <span class="sc">+</span></span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>mismatches), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span>Family_palette_v2) <span class="sc">+</span></span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Fold Change of</span><span class="sc">\n</span><span class="st">Actual over Permutation 1"</span>)<span class="sc">+</span></span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log2_trans</span>(),</span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log2"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">2</span><span class="sc">^</span>x, <span class="dv">3</span>)))<span class="sc">+</span></span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">0.9</span>,<span class="fl">1.9</span>,<span class="fl">2.9</span>,<span class="fl">3.9</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="dv">14</span>, <span class="dv">21</span>, <span class="dv">25</span>, <span class="dv">3</span>),</span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.h<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">5</span>, <span class="at">lineheight =</span> <span class="fl">0.3</span>,</span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Herpesviridae"</span>]) <span class="sc">+</span></span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">1.1</span>,<span class="fl">2.1</span>,<span class="fl">3.1</span>,<span class="fl">4.1</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="dv">14</span>, <span class="dv">21</span>, <span class="dv">25</span>, <span class="dv">3</span>),</span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.p<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">5</span>, <span class="at">lineheight =</span> <span class="fl">0.3</span>,</span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Poxviridae"</span>]) <span class="sc">+</span></span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">inherit.aes =</span> <span class="cn">FALSE</span>, <span class="fu">aes</span>(<span class="at">x=</span><span class="fl">0.85</span>, <span class="at">ymin =</span> <span class="fl">3.666667</span>, <span class="at">ymax =</span> <span class="fl">10.3103821</span>),</span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> <span class="st">"gray"</span>, <span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">0.75</span>, <span class="fl">1.8</span>, <span class="fl">2.8</span>, <span class="fl">3.8</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="dv">6</span>),</span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.hp<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">5</span>, <span class="at">lineheight =</span> <span class="fl">0.3</span>,</span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> <span class="st">"gray"</span>)<span class="sc">+</span></span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a>gg4_perm1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure2_vSubFinal_files/figure-html/perm1_graphing_3-2.png" class="img-fluid" width="384"></p>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(paste0(out_dir, "random_scram_family.pdf"), width = 4, height = 3)</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggarrange</span>(gg3_perm1, gg4_perm1, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">1</span>, <span class="at">common.legend =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure2_vSubFinal_files/figure-html/perm1_graphing_3-3.png" class="img-fluid" width="384"></p>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(paste0(out_dir, "random_scram_combined.pdf"), width = 5, height = 3)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="reverse-processing" class="level2">
<h2 class="anchored" data-anchor-id="reverse-processing">Reverse Processing</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../../epitope_alignment/Results/Permutation_exp2"</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".tsv"</span>, files_to_read)]</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>final_cohort_naming <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../../Viral_Proteomes/Final_Cohort"</span>)</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>final_cohort_naming <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">gsub</span>(<span class="st">".fasta.gz"</span>, <span class="st">"_12mer_3mis_hits.tsv"</span>, final_cohort_naming))</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[files_to_read <span class="sc">%in%</span> <span class="fu">paste0</span>(<span class="st">"Reversed_"</span>, final_cohort_naming)]</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"Proteome Length"</span>, <span class="st">"Number of Proteins"</span>))</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>out_protein_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" Pathogen number:"</span>,<span class="fu">match</span>(active_file, files_to_read), <span class="st">"/"</span>, <span class="fu">length</span>(files_to_read)))</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file)</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>    out_final[,<span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.tsv"</span>, <span class="st">""</span>, active_file)] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="cn">NA</span>,<span class="cn">NA</span>)</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>      data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../Viral_Proteomes/Final_Cohort/"</span>, <span class="fu">gsub</span>(<span class="st">"Reversed_"</span>, <span class="st">""</span>, <span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.tsv"</span>, <span class="st">""</span>, active_file)), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>      PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>      PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a>  <span class="do">### This line is what makes the results unique!!</span></span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>uniprot <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(active_data<span class="sc">$</span>target_seqname, <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>]))</span>
<span id="cb10-34"><a href="#cb10-34" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>viral_id <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(active_data<span class="sc">$</span>query_seqname, <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>]))</span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Filter out any duplicated proteins, identified in </span></span>
<span id="cb10-36"><a href="#cb10-36" aria-hidden="true" tabindex="-1"></a>  <span class="do">## ../../Scripts/Duplicate_Protein_Screen.R</span></span>
<span id="cb10-37"><a href="#cb10-37" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb10-38"><a href="#cb10-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb10-39"><a href="#cb10-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb10-40"><a href="#cb10-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb10-41"><a href="#cb10-41" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>mismatches <span class="ot">&lt;-</span> <span class="fu">apply</span>(active_data, <span class="dv">1</span>, <span class="cf">function</span>(x){<span class="fu">DNA_Diff</span>(x[<span class="dv">1</span>], x[<span class="dv">6</span>])})</span>
<span id="cb10-42"><a href="#cb10-42" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>mismatches <span class="ot">&lt;-</span> <span class="fu">factor</span>(active_data<span class="sc">$</span>mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))</span>
<span id="cb10-43"><a href="#cb10-43" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(active_data)</span>
<span id="cb10-44"><a href="#cb10-44" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(active_data<span class="sc">$</span>mismatches))</span>
<span id="cb10-45"><a href="#cb10-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(out) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.tsv"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb10-46"><a href="#cb10-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-47"><a href="#cb10-47" aria-hidden="true" tabindex="-1"></a>  <span class="do">############## Count Proteome Length ##############</span></span>
<span id="cb10-48"><a href="#cb10-48" aria-hidden="true" tabindex="-1"></a>   <span class="co">#browser()</span></span>
<span id="cb10-49"><a href="#cb10-49" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../Viral_Proteomes/Final_Cohort/"</span>, <span class="fu">gsub</span>(<span class="st">"Reversed_"</span>, <span class="st">""</span>, <span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.tsv"</span>, <span class="st">""</span>, active_file)), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-50"><a href="#cb10-50" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb10-51"><a href="#cb10-51" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb10-52"><a href="#cb10-52" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb10-53"><a href="#cb10-53" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb10-54"><a href="#cb10-54" aria-hidden="true" tabindex="-1"></a>       PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb10-55"><a href="#cb10-55" aria-hidden="true" tabindex="-1"></a>       PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb10-56"><a href="#cb10-56" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb10-57"><a href="#cb10-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-58"><a href="#cb10-58" aria-hidden="true" tabindex="-1"></a>  out_final <span class="ot">&lt;-</span> <span class="fu">merge</span>(out_final, out, <span class="at">by =</span> <span class="st">"mismatches"</span>, <span class="at">all.x =</span> T)</span>
<span id="cb10-59"><a href="#cb10-59" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb10-60"><a href="#cb10-60" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb10-61"><a href="#cb10-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-62"><a href="#cb10-62" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-63"><a href="#cb10-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-64"><a href="#cb10-64" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_final, <span class="st">"../../../Analysis/Figure2/output/12mer_perm2_reverse.csv"</span>, <span class="at">row.names =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="reverse-proteomes-graphs" class="level2">
<h2 class="anchored" data-anchor-id="reverse-proteomes-graphs">Reverse Proteomes Graphs</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/12mer_perm2_reverse.csv"</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>true_counts <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/12mer_proteome_everything.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>true_counts[<span class="fu">is.na</span>(true_counts)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>reverse_results <span class="ot">&lt;-</span> <span class="fu">t</span>(true_counts[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,]) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"Species"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Species <span class="sc">!=</span> <span class="st">"mismatches"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"0"</span><span class="ot">=</span><span class="st">"1"</span>,<span class="st">"1"</span><span class="ot">=</span><span class="st">"2"</span>,<span class="st">"2"</span><span class="ot">=</span><span class="st">"3"</span>,<span class="st">"3"</span><span class="ot">=</span><span class="st">"4"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>,<span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="st">`</span><span class="at">3</span><span class="st">`</span>),</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>),</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>Species) <span class="sc">%&gt;%</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">t</span>(out_final[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,]) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rownames_to_column</span>(<span class="st">"Species"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(Species <span class="sc">!=</span> <span class="st">"mismatches"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"0"</span><span class="ot">=</span><span class="st">"1"</span>,<span class="st">"1"</span><span class="ot">=</span><span class="st">"2"</span>,<span class="st">"2"</span><span class="ot">=</span><span class="st">"3"</span>,<span class="st">"3"</span><span class="ot">=</span><span class="st">"4"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>,<span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="st">`</span><span class="at">3</span><span class="st">`</span>),</span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>             <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>),</span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>             <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>      <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>Species) <span class="sc">%&gt;%</span></span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">value_rev =</span> value) <span class="sc">%&gt;%</span></span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">Species =</span> <span class="fu">gsub</span>(<span class="st">"Reversed_"</span>, <span class="st">""</span>, Species)), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Species"</span>, <span class="st">"name"</span>)</span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>  )  <span class="sc">%&gt;%</span></span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"mismatches"</span> <span class="ot">=</span> <span class="st">"name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FC =</span> (value<span class="sc">+</span><span class="dv">10</span>) <span class="sc">/</span> (value_rev <span class="sc">+</span> <span class="dv">10</span>))</span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a>gg2 <span class="ot">&lt;-</span> reverse_results <span class="sc">%&gt;%</span></span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, Species)) <span class="sc">%&gt;%</span></span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span><span class="fu">ifelse</span>(mismatches <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">"≤"</span>, mismatches), mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">factor</span>(mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"≤1"</span>, <span class="st">"≤2"</span>, <span class="st">"≤3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(mismatches, label, <span class="at">fill=</span>FC)) <span class="sc">+</span> <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Number of Mismatches"</span>) <span class="sc">+</span></span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_gradientn</span>(<span class="at">colours =</span> <span class="fu">c</span>(<span class="st">"#045375"</span>, <span class="st">"#089099"</span>, <span class="st">"#7CCBA2"</span>, <span class="st">"#FCDE9C"</span>, <span class="st">"#FAB27B"</span>, <span class="st">"#E23333"</span>),</span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a>                       <span class="at">limits=</span><span class="fu">c</span>(<span class="fl">0.2</span>, <span class="dv">32</span>), <span class="at">oob =</span> scales<span class="sc">::</span>squish, <span class="at">trans =</span> <span class="st">"log"</span>,</span>
<span id="cb11-47"><a href="#cb11-47" aria-hidden="true" tabindex="-1"></a>                       <span class="at">breaks =</span> <span class="fu">c</span>(<span class="fl">0.25</span>, <span class="fl">0.5</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">8</span>, <span class="dv">16</span>, <span class="dv">32</span>)) <span class="sc">+</span></span>
<span id="cb11-48"><a href="#cb11-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-49"><a href="#cb11-49" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span><span class="fu">element_blank</span>(),<span class="co">#element_text(size=7),</span></span>
<span id="cb11-50"><a href="#cb11-50" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">9</span>),</span>
<span id="cb11-51"><a href="#cb11-51" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">fill=</span><span class="cn">NA</span>, <span class="at">size=</span><span class="dv">1</span>))</span>
<span id="cb11-52"><a href="#cb11-52" aria-hidden="true" tabindex="-1"></a>gg2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure2_vSubFinal_files/figure-html/perm2_graphing_1-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>taxa <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="do">##filter(Proteome.ID %in% out_protein_final$`Proteome ID`) %&gt;%</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(family, order, class, phylum, kingdom) <span class="sc">%&gt;%</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Anelloviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Kolmioviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Kolmioviridae"</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">lapply</span>(taxa, as.factor), <span class="at">stringsAsFactors =</span><span class="cn">TRUE</span>)</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(check) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(taxa)</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> check[<span class="fu">complete.cases</span>(check),]</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">taxo2phylog</span>(<span class="fu">as.taxo</span>(check), <span class="at">root=</span><span class="st">"Root"</span>, <span class="at">abbrev=</span>F)</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>tax.phy <span class="ot">&lt;-</span> <span class="fu">as.phylo</span>(tmp)</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tax.phy, <span class="at">branch.length=</span><span class="st">'none'</span>)</span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>gg_tree[[<span class="st">"data"</span>]] <span class="ot">&lt;-</span> gg_tree[[<span class="st">"data"</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata <span class="sc">%&gt;%</span></span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>               <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(Taxon.mnemonic, family), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>))</span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-31"><a href="#cb12-31" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span>  gg_tree <span class="sc">+</span> <span class="fu">geom_hilight</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">subset =</span> family <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2),<span class="at">fill =</span> family), <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb12-32"><a href="#cb12-32" aria-hidden="true" tabindex="-1"></a>   <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> Family_palette_v2)</span>
<span id="cb12-33"><a href="#cb12-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-34"><a href="#cb12-34" aria-hidden="true" tabindex="-1"></a>gg2 <span class="sc">%&gt;%</span> <span class="fu">insert_left</span>(gg_tree,<span class="at">width =</span> <span class="fl">0.2</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure2_vSubFinal_files/figure-html/perm2_graphing_2-1.png" class="img-fluid" width="480"></p>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(paste0(out_dir, "reversed_heatmap.pdf"), width = 5, height =10)</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="co">#gg2 %&gt;% insert_left(gg1) %&gt;% insert_left(gg_tree,width = 0.2)</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(paste0(out_dir, "perm1-2_heatmap.pdf"), width = 5, height =10)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> reverse_results <span class="sc">%&gt;%</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, Species)) <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"Virus"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>))</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>wt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> FC <span class="sc">~</span> Chronic.or.Acute,</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(wt, <span class="fu">paste0</span>(out_dir, <span class="st">"figure2_statistics.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"Perm2_Chronic_vs_Acute_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>gg3_perm2 <span class="ot">&lt;-</span> reverse_results <span class="sc">%&gt;%</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, Species)) <span class="sc">%&gt;%</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">"Bacteria"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Chronic.or.Acute =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"Bacteria"</span>, Taxonomic.lineage), <span class="st">"Bacteria"</span>, Chronic.or.Acute)) <span class="sc">%&gt;%</span></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FC =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(FC), <span class="dv">0</span>, FC)) <span class="sc">%&gt;%</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(FC), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(FC)) <span class="sc">%&gt;%</span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span><span class="fu">ifelse</span>(mismatches <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">"≤"</span>, mismatches), mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">factor</span>(mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"≤1"</span>, <span class="st">"≤2"</span>, <span class="st">"≤3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>mismatches, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>mismatches), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span><span class="fu">c</span>(<span class="st">"#1975EE"</span>, <span class="st">"#E78305"</span>, <span class="st">"black"</span>)) <span class="sc">+</span></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Fold Change of</span><span class="sc">\n</span><span class="st">Actual over Permutation 2"</span>)<span class="sc">+</span></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log2_trans</span>(),</span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log2"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">2</span><span class="sc">^</span>x, <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="fl">2.9</span>, <span class="fl">4.8</span>, <span class="fl">3.8</span>, <span class="fl">1.1</span>),</span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">6</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>gg3_perm2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure2_vSubFinal_files/figure-html/perm2_graphing_3-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(paste0(out_dir, "reverse_chronicity.pdf"), width = 4, height = 3)</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> reverse_results <span class="sc">%&gt;%</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, Species)) <span class="sc">%&gt;%</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Herpesviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Herpesviridae"</span>, <span class="st">"Herpesviridae"</span>, <span class="st">"Non-Herpesviridae"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Poxviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Poxviridae"</span>, <span class="st">"Poxviridae"</span>, <span class="st">"Non-Poxviridae"</span>))</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="do">## These results are used to manually add stars to the graph below in illustrator</span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>kt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> FC <span class="sc">~</span> Family,</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>), <span class="at">method =</span> <span class="st">"kruskal.test"</span>)</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>wt.p <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> FC <span class="sc">~</span> Poxviridae,</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>wt.h <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> FC <span class="sc">~</span> Herpesviridae,</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a>wt.hp <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Family <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Herpesviridae"</span>, <span class="st">"Poxviridae"</span>)), <span class="at">formula =</span> FC <span class="sc">~</span> Poxviridae,</span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group1 =</span> <span class="st">"Herpesviridae"</span>)</span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(kt, <span class="fu">paste0</span>(out_dir, <span class="st">"figure2_statistics.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"Perm2_Viral_Family_Kruskal"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(wt.p, <span class="fu">paste0</span>(out_dir, <span class="st">"figure2_statistics.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"Perm2_Pox_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(wt.h, <span class="fu">paste0</span>(out_dir, <span class="st">"figure2_statistics.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"Perm2_Herpes_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(wt.hp, <span class="fu">paste0</span>(out_dir, <span class="st">"figure2_statistics.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"Perm2_Herpes_vs_Pox_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>gg4_perm2 <span class="ot">&lt;-</span> reverse_results <span class="sc">%&gt;%</span></span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, Species)) <span class="sc">%&gt;%</span></span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Family, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FC =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(FC), <span class="dv">0</span>, FC)) <span class="sc">%&gt;%</span></span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(FC), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(FC)) <span class="sc">%&gt;%</span></span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(count <span class="sc">&gt;</span> <span class="dv">4</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span><span class="fu">ifelse</span>(mismatches <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">"≤"</span>, mismatches), mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">factor</span>(mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"≤1"</span>, <span class="st">"≤2"</span>, <span class="st">"≤3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>mismatches, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>))  <span class="sc">+</span></span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>mismatches), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb15-49"><a href="#cb15-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb15-50"><a href="#cb15-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span>Family_palette_v2) <span class="sc">+</span></span>
<span id="cb15-51"><a href="#cb15-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Fold Change of</span><span class="sc">\n</span><span class="st">Actual over Permutation 2"</span>)<span class="sc">+</span></span>
<span id="cb15-52"><a href="#cb15-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log2_trans</span>(),</span>
<span id="cb15-53"><a href="#cb15-53" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log2"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">2</span><span class="sc">^</span>x, <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb15-54"><a href="#cb15-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">0.9</span>,<span class="fl">1.9</span>,<span class="fl">2.9</span>,<span class="fl">3.9</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="dv">14</span>, <span class="dv">21</span>, <span class="dv">13</span>, <span class="dv">3</span>),</span>
<span id="cb15-55"><a href="#cb15-55" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.h<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">6</span>, <span class="at">lineheight =</span> <span class="fl">0.3</span>,</span>
<span id="cb15-56"><a href="#cb15-56" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Herpesviridae"</span>]) <span class="sc">+</span></span>
<span id="cb15-57"><a href="#cb15-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">1.1</span>,<span class="fl">2.1</span>,<span class="fl">3.1</span>,<span class="fl">4.1</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="dv">14</span>, <span class="dv">21</span>, <span class="dv">13</span>, <span class="dv">3</span>),</span>
<span id="cb15-58"><a href="#cb15-58" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.p<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">6</span>, <span class="at">lineheight =</span> <span class="fl">0.3</span>,</span>
<span id="cb15-59"><a href="#cb15-59" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Poxviridae"</span>])<span class="sc">+</span></span>
<span id="cb15-60"><a href="#cb15-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb15-61"><a href="#cb15-61" aria-hidden="true" tabindex="-1"></a>gg4_perm2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure2_vSubFinal_files/figure-html/perm2_graphing_3-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(paste0(out_dir, "reverse_family.pdf"), width = 4, height = 3)</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="co">#ggarrange(gg3_perm2, gg4_perm2, ncol = 2, nrow = 1, common.legend = TRUE)</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(paste0(out_dir, "reverse_combined.pdf"), width = 5, height = 3)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="class-scramble-processing" class="level2">
<h2 class="anchored" data-anchor-id="class-scramble-processing">Class Scramble Processing</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../../epitope_alignment/Results/Permutation_exp3"</span>)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".tsv"</span>, files_to_read)]</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Virus"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"Proteome Length"</span>, <span class="st">"Number of Proteins"</span>))</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>out_protein_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" Pathogen number:"</span>,<span class="fu">match</span>(active_file, files_to_read), <span class="st">"/"</span>, <span class="fu">length</span>(files_to_read)))</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">nThread =</span> <span class="dv">10</span>)</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>    out_final[,<span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.tsv"</span>, <span class="st">""</span>, active_file)] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="cn">NA</span>,<span class="cn">NA</span>)</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>      data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../Viral_Proteomes/Final_Cohort/"</span>, <span class="fu">gsub</span>(<span class="st">"AAClass_Scramble_"</span>, <span class="st">""</span>, <span class="fu">gsub</span>(<span class="st">"-[1-9].*"</span>, <span class="st">""</span>, active_file)), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>       PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>      PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a>  <span class="do">### This line is what makes the results unique!!</span></span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>uniprot <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(active_data<span class="sc">$</span>target_seqname, <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>]))</span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>viral_id <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(active_data<span class="sc">$</span>query_seqname, <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>]))</span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Filter out any duplicated proteins, identified in </span></span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a>  <span class="do">## ../../Scripts/Duplicate_Protein_Screen.R</span></span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>mismatches <span class="ot">&lt;-</span> <span class="fu">apply</span>(active_data, <span class="dv">1</span>, <span class="cf">function</span>(x){<span class="fu">DNA_Diff</span>(x[<span class="dv">1</span>], x[<span class="dv">6</span>])})</span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>mismatches <span class="ot">&lt;-</span> <span class="fu">factor</span>(active_data<span class="sc">$</span>mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))</span>
<span id="cb17-41"><a href="#cb17-41" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(active_data)</span>
<span id="cb17-42"><a href="#cb17-42" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(active_data<span class="sc">$</span>mismatches))</span>
<span id="cb17-43"><a href="#cb17-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(out) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.tsv"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb17-44"><a href="#cb17-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-45"><a href="#cb17-45" aria-hidden="true" tabindex="-1"></a>  <span class="do">############## Count Proteome Length ##############</span></span>
<span id="cb17-46"><a href="#cb17-46" aria-hidden="true" tabindex="-1"></a>   <span class="co">#browser()</span></span>
<span id="cb17-47"><a href="#cb17-47" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../Viral_Proteomes/Final_Cohort/"</span>, <span class="fu">gsub</span>(<span class="st">"AAClass_Scramble_"</span>, <span class="st">""</span>, <span class="fu">gsub</span>(<span class="st">"-[1-9].*"</span>, <span class="st">""</span>, active_file)), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-48"><a href="#cb17-48" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb17-49"><a href="#cb17-49" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb17-50"><a href="#cb17-50" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb17-51"><a href="#cb17-51" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb17-52"><a href="#cb17-52" aria-hidden="true" tabindex="-1"></a>      PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb17-53"><a href="#cb17-53" aria-hidden="true" tabindex="-1"></a>      PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb17-54"><a href="#cb17-54" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb17-55"><a href="#cb17-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-56"><a href="#cb17-56" aria-hidden="true" tabindex="-1"></a>  out_final <span class="ot">&lt;-</span> <span class="fu">merge</span>(out_final, out, <span class="at">by =</span> <span class="st">"mismatches"</span>, <span class="at">all.x =</span> T)</span>
<span id="cb17-57"><a href="#cb17-57" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb17-58"><a href="#cb17-58" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb17-59"><a href="#cb17-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-60"><a href="#cb17-60" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-61"><a href="#cb17-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-62"><a href="#cb17-62" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_final, <span class="st">"../../../Analysis/Figure2/output/12mer_perm3_class_scramble.csv"</span>, <span class="at">row.names =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="class-scramble-graphs" class="level2">
<h2 class="anchored" data-anchor-id="class-scramble-graphs">Class Scramble Graphs</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/12mer_perm3_class_scramble.csv"</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(out_final) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">."</span>,<span class="st">"-"</span>, <span class="fu">colnames</span>(out_final))</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>true_counts <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/12mer_proteome_everything.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>true_counts[<span class="fu">is.na</span>(true_counts)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(true_counts) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">."</span>,<span class="st">"-"</span>, <span class="fu">colnames</span>(true_counts))</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>class_scram_results <span class="ot">&lt;-</span> <span class="fu">t</span>(true_counts[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,]) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"Species"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Species <span class="sc">!=</span> <span class="st">"mismatches"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"0"</span><span class="ot">=</span><span class="st">"1"</span>,<span class="st">"1"</span><span class="ot">=</span><span class="st">"2"</span>,<span class="st">"2"</span><span class="ot">=</span><span class="st">"3"</span>,<span class="st">"3"</span><span class="ot">=</span><span class="st">"4"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>,<span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="st">`</span><span class="at">3</span><span class="st">`</span>),</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>),</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>Species) <span class="sc">%&gt;%</span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">t</span>(out_final[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,]) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rownames_to_column</span>(<span class="st">"Species"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(Species <span class="sc">!=</span> <span class="st">"mismatches"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"0"</span><span class="ot">=</span><span class="st">"1"</span>,<span class="st">"1"</span><span class="ot">=</span><span class="st">"2"</span>,<span class="st">"2"</span><span class="ot">=</span><span class="st">"3"</span>,<span class="st">"3"</span><span class="ot">=</span><span class="st">"4"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>,<span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="st">`</span><span class="at">3</span><span class="st">`</span>),</span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>             <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>),</span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>             <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>      <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>Species) <span class="sc">%&gt;%</span></span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">value_rand =</span> value) <span class="sc">%&gt;%</span></span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">Species =</span> <span class="fu">gsub</span>(<span class="st">"-[1-9].*"</span>, <span class="st">""</span>, Species)) <span class="sc">%&gt;%</span></span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(Species, name) <span class="sc">%&gt;%</span></span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">value_rand =</span> <span class="fu">mean</span>(value_rand)) <span class="sc">%&gt;%</span></span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">Species =</span> <span class="fu">gsub</span>(<span class="st">"AAClass_Scramble_"</span>, <span class="st">""</span>, Species)), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Species"</span>, <span class="st">"name"</span>)</span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a>  )  <span class="sc">%&gt;%</span></span>
<span id="cb18-37"><a href="#cb18-37" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"mismatches"</span> <span class="ot">=</span> <span class="st">"name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FC =</span> (value<span class="sc">+</span><span class="dv">10</span>) <span class="sc">/</span> (value_rand <span class="sc">+</span> <span class="dv">10</span>))</span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-40"><a href="#cb18-40" aria-hidden="true" tabindex="-1"></a>gg3 <span class="ot">&lt;-</span> class_scram_results <span class="sc">%&gt;%</span></span>
<span id="cb18-41"><a href="#cb18-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, Species)) <span class="sc">%&gt;%</span></span>
<span id="cb18-42"><a href="#cb18-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb18-43"><a href="#cb18-43" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb18-44"><a href="#cb18-44" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb18-45"><a href="#cb18-45" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-46"><a href="#cb18-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span><span class="fu">ifelse</span>(mismatches <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">"≤"</span>, mismatches), mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb18-47"><a href="#cb18-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">factor</span>(mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"≤1"</span>, <span class="st">"≤2"</span>, <span class="st">"≤3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb18-48"><a href="#cb18-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(mismatches, label, <span class="at">fill=</span>FC)) <span class="sc">+</span> <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb18-49"><a href="#cb18-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb18-50"><a href="#cb18-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Number of Mismatches"</span>) <span class="sc">+</span></span>
<span id="cb18-51"><a href="#cb18-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">colours =</span> <span class="fu">c</span>(<span class="st">"#045375"</span>, <span class="st">"#089099"</span>, <span class="st">"#7CCBA2"</span>, <span class="st">"#FCDE9C"</span>, <span class="st">"#FAB27B"</span>, <span class="st">"#E23333"</span>),</span>
<span id="cb18-52"><a href="#cb18-52" aria-hidden="true" tabindex="-1"></a>                       <span class="at">limits=</span><span class="fu">c</span>(<span class="fl">0.2</span>, <span class="dv">32</span>), <span class="at">oob =</span> scales<span class="sc">::</span>squish, <span class="at">trans =</span> <span class="st">"log"</span>,</span>
<span id="cb18-53"><a href="#cb18-53" aria-hidden="true" tabindex="-1"></a>                       <span class="at">breaks =</span> <span class="fu">c</span>(<span class="fl">0.25</span>, <span class="fl">0.5</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">8</span>, <span class="dv">16</span>, <span class="dv">32</span>)) <span class="sc">+</span></span>
<span id="cb18-54"><a href="#cb18-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-55"><a href="#cb18-55" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span><span class="fu">element_blank</span>(),</span>
<span id="cb18-56"><a href="#cb18-56" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">9</span>),</span>
<span id="cb18-57"><a href="#cb18-57" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">fill=</span><span class="cn">NA</span>, <span class="at">size=</span><span class="dv">1</span>))</span>
<span id="cb18-58"><a href="#cb18-58" aria-hidden="true" tabindex="-1"></a>gg3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure2_vSubFinal_files/figure-html/perm3_graphing_1-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>taxa <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="do">##filter(Proteome.ID %in% out_protein_final$`Proteome ID`) %&gt;%</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(family, order, class, phylum, kingdom) <span class="sc">%&gt;%</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Anelloviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Kolmioviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Kolmioviridae"</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">lapply</span>(taxa, as.factor), <span class="at">stringsAsFactors =</span><span class="cn">TRUE</span>)</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(check) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(taxa)</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> check[<span class="fu">complete.cases</span>(check),]</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">taxo2phylog</span>(<span class="fu">as.taxo</span>(check), <span class="at">root=</span><span class="st">"Root"</span>, <span class="at">abbrev=</span>F)</span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>tax.phy <span class="ot">&lt;-</span> <span class="fu">as.phylo</span>(tmp)</span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tax.phy, <span class="at">branch.length=</span><span class="st">'none'</span>)</span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a>gg_tree[[<span class="st">"data"</span>]] <span class="ot">&lt;-</span> gg_tree[[<span class="st">"data"</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata <span class="sc">%&gt;%</span></span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a>               <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(Taxon.mnemonic, family), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>))</span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span>  gg_tree <span class="sc">+</span> <span class="fu">geom_hilight</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">subset =</span> family <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2),<span class="at">fill =</span> family), <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a>   <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> Family_palette_v2)</span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a><span class="co">#gg3 %&gt;% insert_left(gg_tree,width = 0.2) </span></span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(paste0(out_dir, "class_scramble_heatmap.pdf"), width = 4, height =12)</span></span>
<span id="cb19-36"><a href="#cb19-36" aria-hidden="true" tabindex="-1"></a>gg3 <span class="sc">%&gt;%</span> <span class="fu">insert_left</span>(gg2) <span class="sc">%&gt;%</span> <span class="fu">insert_left</span>(gg1) <span class="sc">%&gt;%</span> <span class="fu">insert_left</span>(gg_tree,<span class="at">width =</span> <span class="fl">0.4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure2_vSubFinal_files/figure-html/perm3_graphing_2-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"Figure2B.pdf"</span>), <span class="at">width =</span> <span class="dv">5</span>, <span class="at">height =</span><span class="dv">13</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> class_scram_results <span class="sc">%&gt;%</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, Species)) <span class="sc">%&gt;%</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"Virus"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>))</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>wt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> FC <span class="sc">~</span> Chronic.or.Acute,</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(wt, <span class="fu">paste0</span>(out_dir, <span class="st">"figure2_statistics.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"Perm3_Chronic_vs_Acute_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>gg3_perm3 <span class="ot">&lt;-</span> class_scram_results <span class="sc">%&gt;%</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, Species)) <span class="sc">%&gt;%</span></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">"Bacteria"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Chronic.or.Acute =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"Bacteria"</span>, Taxonomic.lineage), <span class="st">"Bacteria"</span>, Chronic.or.Acute)) <span class="sc">%&gt;%</span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FC =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(FC), <span class="dv">0</span>, FC)) <span class="sc">%&gt;%</span></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(FC), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(FC)) <span class="sc">%&gt;%</span></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span><span class="fu">ifelse</span>(mismatches <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">"≤"</span>, mismatches), mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">factor</span>(mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"≤1"</span>, <span class="st">"≤2"</span>, <span class="st">"≤3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>mismatches, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>mismatches), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb21-30"><a href="#cb21-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb21-31"><a href="#cb21-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span><span class="fu">c</span>(<span class="st">"#1975EE"</span>, <span class="st">"#E78305"</span>, <span class="st">"black"</span>)) <span class="sc">+</span></span>
<span id="cb21-32"><a href="#cb21-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Fold Change of</span><span class="sc">\n</span><span class="st">Actual over Permutation 3"</span>)<span class="sc">+</span></span>
<span id="cb21-33"><a href="#cb21-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb21-34"><a href="#cb21-34" aria-hidden="true" tabindex="-1"></a>gg3_perm3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure2_vSubFinal_files/figure-html/perm3_graphing_3-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(paste0(out_dir, "class_scram_chronicity.pdf"), width = 4, height = 3)</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> class_scram_results <span class="sc">%&gt;%</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, Species)) <span class="sc">%&gt;%</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Herpesviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Herpesviridae"</span>, <span class="st">"Herpesviridae"</span>, <span class="st">"Non-Herpesviridae"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Poxviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Poxviridae"</span>, <span class="st">"Poxviridae"</span>, <span class="st">"Non-Poxviridae"</span>))</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="do">## These results are used to manually add stars to the graph below in illustrator</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>kt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> FC <span class="sc">~</span> Family,</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>), <span class="at">method =</span> <span class="st">"kruskal.test"</span>)</span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>wt.p <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> FC <span class="sc">~</span> Poxviridae,</span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>wt.h <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> FC <span class="sc">~</span> Herpesviridae,</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>wt.hp <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Family <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Herpesviridae"</span>, <span class="st">"Poxviridae"</span>)), <span class="at">formula =</span> FC <span class="sc">~</span> Poxviridae,</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group1 =</span> <span class="st">"Herpesviridae"</span>)</span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(kt, <span class="fu">paste0</span>(out_dir, <span class="st">"figure2_statistics.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"Perm3_Viral_Family_Kruskal"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(wt.p, <span class="fu">paste0</span>(out_dir, <span class="st">"figure2_statistics.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"Perm3_Pox_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(wt.h, <span class="fu">paste0</span>(out_dir, <span class="st">"figure2_statistics.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"Perm3_Herpes_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(wt.hp, <span class="fu">paste0</span>(out_dir, <span class="st">"figure2_statistics.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"Perm3_Herpes_vs_Pox_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a>gg4_perm3 <span class="ot">&lt;-</span> class_scram_results <span class="sc">%&gt;%</span></span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, Species)) <span class="sc">%&gt;%</span></span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Family, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb22-38"><a href="#cb22-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FC =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(FC), <span class="dv">0</span>, FC)) <span class="sc">%&gt;%</span></span>
<span id="cb22-39"><a href="#cb22-39" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(FC), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(FC)) <span class="sc">%&gt;%</span></span>
<span id="cb22-40"><a href="#cb22-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(count <span class="sc">&gt;</span> <span class="dv">4</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-41"><a href="#cb22-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb22-42"><a href="#cb22-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span><span class="fu">ifelse</span>(mismatches <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">"≤"</span>, mismatches), mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb22-43"><a href="#cb22-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">factor</span>(mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"≤1"</span>, <span class="st">"≤2"</span>, <span class="st">"≤3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb22-44"><a href="#cb22-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>mismatches, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb22-45"><a href="#cb22-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb22-46"><a href="#cb22-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb22-47"><a href="#cb22-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb22-48"><a href="#cb22-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb22-49"><a href="#cb22-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>mismatches), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb22-50"><a href="#cb22-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb22-51"><a href="#cb22-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span>Family_palette_v2) <span class="sc">+</span></span>
<span id="cb22-52"><a href="#cb22-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Fold Change of</span><span class="sc">\n</span><span class="st">Actual over Permutation 3"</span>) <span class="sc">+</span></span>
<span id="cb22-53"><a href="#cb22-53" aria-hidden="true" tabindex="-1"></a>   <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log2_trans</span>(),</span>
<span id="cb22-54"><a href="#cb22-54" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log2"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">2</span><span class="sc">^</span>x, <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb22-55"><a href="#cb22-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">0.9</span>,<span class="fl">1.9</span>,<span class="fl">2.9</span>,<span class="fl">3.9</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">3</span>),</span>
<span id="cb22-56"><a href="#cb22-56" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.h<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">6</span>, <span class="at">lineheight =</span> <span class="fl">0.3</span>,</span>
<span id="cb22-57"><a href="#cb22-57" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Herpesviridae"</span>]) <span class="sc">+</span></span>
<span id="cb22-58"><a href="#cb22-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">1.1</span>,<span class="fl">2.1</span>,<span class="fl">3.1</span>,<span class="fl">4.1</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">3</span>),</span>
<span id="cb22-59"><a href="#cb22-59" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.p<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">6</span>, <span class="at">lineheight =</span> <span class="fl">0.3</span>,</span>
<span id="cb22-60"><a href="#cb22-60" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Poxviridae"</span>])<span class="sc">+</span></span>
<span id="cb22-61"><a href="#cb22-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb22-62"><a href="#cb22-62" aria-hidden="true" tabindex="-1"></a>gg4_perm3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure2_vSubFinal_files/figure-html/perm3_graphing_3-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(paste0(out_dir, "class_scram_family.pdf"), width = 4, height = 3)</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="co">#ggarrange(gg3_perm3, gg4_perm3, ncol = 2, nrow = 1, common.legend = TRUE)</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(paste0(out_dir, "class_scram_combined.pdf"), width = 5, height = 3)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="panel-assembly" class="level2">
<h2 class="anchored" data-anchor-id="panel-assembly">Panel Assembly</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggarrange</span>(gg3_perm1, gg4_perm1,</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>          gg3_perm2, gg4_perm2,</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>          gg3_perm3, gg4_perm3,</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">3</span>, <span class="at">align =</span> <span class="st">"hv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Figure2_vSubFinal_files/figure-html/unnamed-chunk-1-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"Figure2C-E.pdf"</span>), <span class="at">width =</span> <span class="dv">5</span>, <span class="at">height =</span> <span class="dv">7</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>