<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.555">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Biological_Enrichment</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Figure4and5_files/libs/clipboard/clipboard.min.js"></script>
<script src="Figure4and5_files/libs/quarto-html/quarto.js"></script>
<script src="Figure4and5_files/libs/quarto-html/popper.min.js"></script>
<script src="Figure4and5_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Figure4and5_files/libs/quarto-html/anchor.min.js"></script>
<link href="Figure4and5_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Figure4and5_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Figure4and5_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Figure4and5_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Figure4and5_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Biological_Enrichment</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="figure-4-5---biological-enrichment" class="level2">
<h2 class="anchored" data-anchor-id="figure-4-5---biological-enrichment">Figure 4 &amp; 5 - Biological Enrichment</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dtplyr)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plyr)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggbeeswarm)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(aplot)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(treeio)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggtree)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(aplot)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggbeeswarm)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rstatix)</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lemon)</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>mart <span class="ot">&lt;-</span> <span class="fu">useEnsembl</span>(<span class="st">"ensembl"</span>, <span class="at">dataset =</span> <span class="st">"hsapiens_gene_ensembl"</span>) <span class="do">## , mirror = "useast"</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>mart_retrieved <span class="ot">&lt;-</span> <span class="fu">getBM</span>(</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">mart =</span> mart, <span class="at">useCache =</span> T, <span class="do">## Cached last: 2024-02-24</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">attributes =</span> <span class="fu">c</span>(</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>    <span class="st">'ensembl_gene_id'</span>,</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>    <span class="st">'entrezgene_id'</span>,</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>    <span class="st">'external_gene_name'</span>,</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>    <span class="st">'uniprot_gn_symbol'</span>,</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>    <span class="st">'uniprot_gn_id'</span>,</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chromosome_name'</span>,</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>    <span class="st">'description'</span>),</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">uniqueRows=</span><span class="cn">TRUE</span>)</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../../Utilities.R"</span>)</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>meta_path <span class="ot">&lt;-</span> <span class="st">"../../Metadata/"</span></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"Final_Cohort_Metadata_v5.xlsx"</span>), <span class="at">sheet =</span> <span class="dv">1</span>)</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>duplicated_proteins <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"duplicated_proteins.csv"</span>))</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>human_duplicated_proteins <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"human_duplicated_or_removed_proteins.csv"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="dv">1</span>)</span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>human_protein_metadata <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"uniprot_protein_metadata.xlsx"</span>))</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(human_protein_metadata) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">gsub</span>(<span class="st">" "</span>, <span class="st">"_"</span>, <span class="fu">tolower</span>(<span class="fu">colnames</span>(human_protein_metadata))), <span class="st">".human"</span>)</span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>out_dir <span class="ot">=</span> <span class="st">"../output/"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="kegg-enrichment-testing" class="level2">
<h2 class="anchored" data-anchor-id="kegg-enrichment-testing">KEGG Enrichment testing</h2>
<section id="actual-mimic-enrichment" class="level3">
<h3 class="anchored" data-anchor-id="actual-mimic-enrichment">Actual mimic enrichment</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../12mer_annotated/"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".csv.gz"</span>, files_to_read)]</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file))</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">nThread =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot)</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  Reactome <span class="ot">&lt;-</span> <span class="fu">enrichKEGG</span>(<span class="at">gene         =</span> <span class="fu">unique</span>(active_data<span class="sc">$</span>entrezgene_id)[<span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">unique</span>(active_data<span class="sc">$</span>entrezgene_id))],</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>                   <span class="at">organism     =</span> <span class="st">'hsa'</span>, </span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>                   <span class="at">use_internal_data =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>                   <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>,</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>                   <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>)</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>  out_final <span class="ot">&lt;-</span> <span class="fu">c</span>(out_final, Reactome)</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(out_final)[<span class="fu">length</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer.*"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(out_final, <span class="st">"../Figure4/output/KEGG_results_2024-02-24.rds"</span>)</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>df_out3 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(out_final)){</span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> out_final[[i]]<span class="sc">@</span>result</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">is.na</span>(active_data<span class="sc">$</span>Description[<span class="dv">1</span>])){</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>virus <span class="ot">&lt;-</span> <span class="fu">names</span>(out_final)[i]</span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>  df_out3 <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(df_out3, active_data)</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(df_out3, <span class="st">"..//Figure4/output/KEGG_results_2024-02-24.csv"</span>)</span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a><span class="do">## Manually add to xlsx doc</span></span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>plot.data <span class="ot">&lt;-</span> df_out3 <span class="sc">%&gt;%</span></span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adjust.manual =</span> <span class="fu">p.adjust</span>(pvalue, <span class="st">"fdr"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">virus =</span> <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, virus)) <span class="sc">%&gt;%</span></span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p.adjust.manual <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">log10.p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust.manual))</span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a>gg2 <span class="ot">&lt;-</span> plot.data <span class="sc">%&gt;%</span></span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Description <span class="sc">%in%</span> plot.data<span class="sc">$</span>Description[<span class="fu">duplicated</span>(plot.data<span class="sc">$</span>Description)]) <span class="sc">%&gt;%</span></span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Description, <span class="at">y=</span>virus, <span class="at">size =</span> log10.p.adjust, <span class="at">col =</span> log10.p.adjust)) <span class="sc">+</span></span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">size=</span><span class="fu">guide_legend</span>(<span class="st">"-log10 Transformed</span><span class="sc">\n</span><span class="st">Adjusted p-values"</span>),</span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a>         <span class="at">color=</span><span class="fu">guide_legend</span>(<span class="st">"-log10 Transformed</span><span class="sc">\n</span><span class="st">Adjusted p-values"</span>)) <span class="sc">+</span></span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradientn</span>(<span class="at">colors=</span><span class="fu">c</span>(<span class="st">"dodgerblue"</span>, <span class="st">"#bf5700"</span>)) <span class="sc">+</span></span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">limits=</span>rev) <span class="sc">+</span></span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(<span class="at">color=</span><span class="st">"grey"</span>, <span class="at">size=</span><span class="fl">0.1</span>), </span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">90</span>, <span class="at">hjust=</span><span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>,))</span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a>gg2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="motiff-enrichment" class="level3">
<h3 class="anchored" data-anchor-id="motiff-enrichment">Motiff Enrichment</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">## A subset analysis on the mimics with a greater frequency than 5</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>out_protein_final_count_12mer <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/12mer_proteome_protein_everything_count.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(count <span class="sc">&gt;=</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_id =</span> <span class="fu">id_split</span>(query_seqname))</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../12mer_annotated/"</span>)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".csv.gz"</span>, files_to_read)]</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file))</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">nThread =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> out_protein_final_count_12mer<span class="sc">$</span>viral_id)</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>  Reactome <span class="ot">&lt;-</span> <span class="fu">enrichKEGG</span>(<span class="at">gene         =</span> <span class="fu">unique</span>(active_data<span class="sc">$</span>entrezgene_id)[<span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">unique</span>(active_data<span class="sc">$</span>entrezgene_id))],</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>                   <span class="at">organism     =</span> <span class="st">'hsa'</span>, </span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>                   <span class="at">use_internal_data =</span> <span class="cn">FALSE</span>,</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>                   <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>,</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>                   <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>)</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>  out_final <span class="ot">&lt;-</span> <span class="fu">c</span>(out_final, Reactome)</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(out_final)[<span class="fu">length</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer.*"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(out_final, <span class="st">"../Figure4/output/KEGG_results_2024-02-24_&gt;=5gene_hits.rds"</span>)</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>df_out3 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(out_final)){</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> out_final[[i]]<span class="sc">@</span>result</span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">is.na</span>(active_data<span class="sc">$</span>Description[<span class="dv">1</span>])){</span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>virus <span class="ot">&lt;-</span> <span class="fu">names</span>(out_final)[i]</span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>  df_out3 <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(df_out3, active_data)</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(df_out3, <span class="st">"../Figure4/output/KEGG_results_2024-02-24_&gt;=5gene_hits.csv"</span>)</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a><span class="do">## Manually add to xlsx document</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="permutation-1" class="level3">
<h3 class="anchored" data-anchor-id="permutation-1">Permutation 1</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../../epitope_alignment/Results/Permutation_exp1/"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".tsv"</span>, files_to_read)]</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file))</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">nThread =</span> <span class="dv">10</span>)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>  active_data[, uniprot <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(target_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>  active_data[, viral_id <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(query_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>  active_data[, mismatches <span class="sc">:</span><span class="er">=</span> <span class="fu">mismatch_calling_fast</span>(query, target), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>     <span class="fu">lazy_dt</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(mart_retrieved <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(uniprot_gn_id, entrezgene_id), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="st">"uniprot_gn_id"</span>),</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>              <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>  Reactome <span class="ot">&lt;-</span> <span class="fu">enrichKEGG</span>(<span class="at">gene         =</span> <span class="fu">unique</span>(active_data<span class="sc">$</span>entrezgene_id)[<span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">unique</span>(active_data<span class="sc">$</span>entrezgene_id))],</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>                   <span class="at">organism     =</span> <span class="st">'hsa'</span>, </span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>                   <span class="at">use_internal_data =</span> <span class="cn">FALSE</span>,</span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>                   <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>,</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>                   <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>)</span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>  out_final <span class="ot">&lt;-</span> <span class="fu">c</span>(out_final, Reactome)</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(out_final)[<span class="fu">length</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer.*"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(out_final, <span class="st">"../../../Analysis/Figure4/output/KEGG_results_2024-02-24_perm1.rds"</span>)</span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a>df_out3 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(out_final)){</span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> out_final[[i]]<span class="sc">@</span>result</span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">is.na</span>(active_data<span class="sc">$</span>Description[<span class="dv">1</span>])){</span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>virus <span class="ot">&lt;-</span> <span class="fu">names</span>(out_final)[i]</span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a>  df_out3 <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(df_out3, active_data)</span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(df_out3, <span class="st">"../../../Analysis/Figure4/output/KEGG_results_2024-02-24_perm1.csv"</span>)</span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a>df_out3_perm1 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../../Analysis/Figure4/output/KEGG_results_2024-02-24_perm1.csv"</span>)</span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a>plot.data_perm1 <span class="ot">&lt;-</span> df_out3_perm1 <span class="sc">%&gt;%</span></span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adjust.manual =</span> <span class="fu">p.adjust</span>(pvalue, <span class="st">"fdr"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(grepl("virus", virus)) %&gt;%</span></span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">virus =</span> <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, virus)) <span class="sc">%&gt;%</span></span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p.adjust.manual <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">log10.p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust.manual)) <span class="co">#%&gt;%</span></span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a>gg2 <span class="ot">&lt;-</span> plot.data_perm1 <span class="sc">%&gt;%</span></span>
<span id="cb4-63"><a href="#cb4-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Description <span class="sc">%in%</span> plot.data_perm1<span class="sc">$</span>Description[<span class="fu">duplicated</span>(plot.data_perm1<span class="sc">$</span>Description)]) <span class="sc">%&gt;%</span></span>
<span id="cb4-64"><a href="#cb4-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Description, <span class="at">y=</span>virus, <span class="at">size =</span> log10.p.adjust, <span class="at">col =</span> log10.p.adjust)) <span class="sc">+</span></span>
<span id="cb4-65"><a href="#cb4-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb4-66"><a href="#cb4-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb4-67"><a href="#cb4-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">size=</span><span class="fu">guide_legend</span>(<span class="st">"-log10 Transformed</span><span class="sc">\n</span><span class="st">Adjusted p-values"</span>),</span>
<span id="cb4-68"><a href="#cb4-68" aria-hidden="true" tabindex="-1"></a>         <span class="at">color=</span><span class="fu">guide_legend</span>(<span class="st">"-log10 Transformed</span><span class="sc">\n</span><span class="st">Adjusted p-values"</span>)) <span class="sc">+</span></span>
<span id="cb4-69"><a href="#cb4-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradientn</span>(<span class="at">colors=</span><span class="fu">c</span>(<span class="st">"dodgerblue"</span>, <span class="st">"#bf5700"</span>)) <span class="sc">+</span></span>
<span id="cb4-70"><a href="#cb4-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">limits=</span>rev) <span class="sc">+</span></span>
<span id="cb4-71"><a href="#cb4-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(<span class="at">color=</span><span class="st">"grey"</span>, <span class="at">size=</span><span class="fl">0.1</span>), </span>
<span id="cb4-72"><a href="#cb4-72" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">90</span>, <span class="at">hjust=</span><span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>,))</span>
<span id="cb4-73"><a href="#cb4-73" aria-hidden="true" tabindex="-1"></a>gg2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="permutation-2" class="level3">
<h3 class="anchored" data-anchor-id="permutation-2">Permutation 2</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../../epitope_alignment/Results/Permutation_exp2/"</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".tsv"</span>, files_to_read)]</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="do">## For testing purposes use this subset</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file))</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">nThread =</span> <span class="dv">10</span>)</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>  active_data[, uniprot <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(target_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>  active_data[, viral_id <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(query_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>  active_data[, mismatches <span class="sc">:</span><span class="er">=</span> <span class="fu">mismatch_calling_fast</span>(query, target), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>     <span class="fu">lazy_dt</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(mart_retrieved <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(uniprot_gn_id, entrezgene_id), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="st">"uniprot_gn_id"</span>),</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>              <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>  Reactome <span class="ot">&lt;-</span> <span class="fu">enrichKEGG</span>(<span class="at">gene         =</span> <span class="fu">unique</span>(active_data<span class="sc">$</span>entrezgene_id)[<span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">unique</span>(active_data<span class="sc">$</span>entrezgene_id))],</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>                   <span class="at">organism     =</span> <span class="st">'hsa'</span>, </span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>                   <span class="at">use_internal_data =</span> <span class="cn">FALSE</span>,</span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>                   <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>,</span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>                   <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>)</span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a>  out_final <span class="ot">&lt;-</span> <span class="fu">c</span>(out_final, Reactome)</span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(out_final)[<span class="fu">length</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer.*"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(out_final, <span class="st">"../../../Analysis/Figure4/output/KEGG_results_2024-02-24_perm2.rds"</span>)</span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a>df_out3 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(out_final)){</span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> out_final[[i]]<span class="sc">@</span>result</span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">is.na</span>(active_data<span class="sc">$</span>Description[<span class="dv">1</span>])){</span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>virus <span class="ot">&lt;-</span> <span class="fu">names</span>(out_final)[i]</span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a>  df_out3 <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(df_out3, active_data)</span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(df_out3, <span class="st">"../../../Analysis/Figure4/output/KEGG_results_2024-02-24_perm2.csv"</span>)</span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a>df_out3_perm2 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../../Analysis/Figure4/output/KEGG_results_2024-02-24_perm2.csv"</span>)</span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a>plot.data_perm2 <span class="ot">&lt;-</span> df_out3_perm2 <span class="sc">%&gt;%</span></span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adjust.manual =</span> <span class="fu">p.adjust</span>(pvalue, <span class="st">"fdr"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(grepl("virus", virus)) %&gt;%</span></span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">virus =</span> <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, virus)) <span class="sc">%&gt;%</span></span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p.adjust.manual <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">log10.p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust.manual)) <span class="co">#%&gt;%</span></span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a>gg2 <span class="ot">&lt;-</span> plot.data_perm2 <span class="sc">%&gt;%</span></span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Description <span class="sc">%in%</span> plot.data_perm2<span class="sc">$</span>Description[<span class="fu">duplicated</span>(plot.data_perm2<span class="sc">$</span>Description)]) <span class="sc">%&gt;%</span></span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Description, <span class="at">y=</span>virus, <span class="at">size =</span> log10.p.adjust, <span class="at">col =</span> log10.p.adjust)) <span class="sc">+</span></span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">size=</span><span class="fu">guide_legend</span>(<span class="st">"-log10 Transformed</span><span class="sc">\n</span><span class="st">Adjusted p-values"</span>),</span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a>         <span class="at">color=</span><span class="fu">guide_legend</span>(<span class="st">"-log10 Transformed</span><span class="sc">\n</span><span class="st">Adjusted p-values"</span>)) <span class="sc">+</span></span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradientn</span>(<span class="at">colors=</span><span class="fu">c</span>(<span class="st">"dodgerblue"</span>, <span class="st">"#bf5700"</span>)) <span class="sc">+</span></span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">limits=</span>rev) <span class="sc">+</span></span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(<span class="at">color=</span><span class="st">"grey"</span>, <span class="at">size=</span><span class="fl">0.1</span>), </span>
<span id="cb5-69"><a href="#cb5-69" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">90</span>, <span class="at">hjust=</span><span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>,))</span>
<span id="cb5-70"><a href="#cb5-70" aria-hidden="true" tabindex="-1"></a>gg2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="permutation-3" class="level3">
<h3 class="anchored" data-anchor-id="permutation-3">Permutation 3</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../../epitope_alignment/Results/Permutation_exp3/"</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".tsv"</span>, files_to_read)]</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file))</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">nThread =</span> <span class="dv">10</span>)</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>  active_data[, uniprot <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(target_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>  active_data[, viral_id <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(query_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  active_data[, mismatches <span class="sc">:</span><span class="er">=</span> <span class="fu">mismatch_calling_fast</span>(query, target), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>     <span class="fu">lazy_dt</span>() <span class="sc">%&gt;%</span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(mart_retrieved <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(uniprot_gn_id, entrezgene_id), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="st">"uniprot_gn_id"</span>),</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>  Reactome <span class="ot">&lt;-</span> <span class="fu">enrichKEGG</span>(<span class="at">gene         =</span> <span class="fu">unique</span>(active_data<span class="sc">$</span>entrezgene_id)[<span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">unique</span>(active_data<span class="sc">$</span>entrezgene_id))],</span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>                   <span class="at">organism     =</span> <span class="st">'hsa'</span>, </span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>                   <span class="at">use_internal_data =</span> <span class="cn">FALSE</span>,</span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>                   <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>,</span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>                   <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>)</span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>  out_final <span class="ot">&lt;-</span> <span class="fu">c</span>(out_final, Reactome)</span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(out_final)[<span class="fu">length</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer.*"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(out_final, <span class="st">"../../../Analysis/Figure4/output/KEGG_results_2024-02-24_perm3.rds"</span>)</span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>df_out3 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(out_final)){</span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> out_final[[i]]<span class="sc">@</span>result</span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">is.na</span>(active_data<span class="sc">$</span>Description[<span class="dv">1</span>])){</span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>virus <span class="ot">&lt;-</span> <span class="fu">names</span>(out_final)[i]</span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a>  df_out3 <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(df_out3, active_data)</span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(df_out3, <span class="st">"../../../Figure4/output/KEGG_results_2024-02-24_perm3.csv"</span>)</span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a>df_out3_perm3 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../../Figure4/output/KEGG_results_2024-02-24_perm3.csv"</span>)</span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a>plot.data_perm2 <span class="ot">&lt;-</span> df_out3_perm3 <span class="sc">%&gt;%</span></span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adjust.manual =</span> <span class="fu">p.adjust</span>(pvalue, <span class="st">"fdr"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">virus =</span> <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, virus)) <span class="sc">%&gt;%</span></span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p.adjust.manual <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">log10.p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust.manual))</span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a>gg2 <span class="ot">&lt;-</span> plot.data_perm2 <span class="sc">%&gt;%</span></span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Description <span class="sc">%in%</span> plot.data_perm2<span class="sc">$</span>Description[<span class="fu">duplicated</span>(plot.data_perm2<span class="sc">$</span>Description)]) <span class="sc">%&gt;%</span></span>
<span id="cb6-59"><a href="#cb6-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Description, <span class="at">y=</span>virus, <span class="at">size =</span> log10.p.adjust, <span class="at">col =</span> log10.p.adjust)) <span class="sc">+</span></span>
<span id="cb6-60"><a href="#cb6-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb6-61"><a href="#cb6-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb6-62"><a href="#cb6-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">size=</span><span class="fu">guide_legend</span>(<span class="st">"-log10 Transformed</span><span class="sc">\n</span><span class="st">Adjusted p-values"</span>),</span>
<span id="cb6-63"><a href="#cb6-63" aria-hidden="true" tabindex="-1"></a>         <span class="at">color=</span><span class="fu">guide_legend</span>(<span class="st">"-log10 Transformed</span><span class="sc">\n</span><span class="st">Adjusted p-values"</span>)) <span class="sc">+</span></span>
<span id="cb6-64"><a href="#cb6-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradientn</span>(<span class="at">colors=</span><span class="fu">c</span>(<span class="st">"dodgerblue"</span>, <span class="st">"#bf5700"</span>)) <span class="sc">+</span></span>
<span id="cb6-65"><a href="#cb6-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">limits=</span>rev) <span class="sc">+</span></span>
<span id="cb6-66"><a href="#cb6-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(<span class="at">color=</span><span class="st">"grey"</span>, <span class="at">size=</span><span class="fl">0.1</span>), </span>
<span id="cb6-67"><a href="#cb6-67" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">90</span>, <span class="at">hjust=</span><span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>,))</span>
<span id="cb6-68"><a href="#cb6-68" aria-hidden="true" tabindex="-1"></a>gg2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="figure-4" class="level3">
<h3 class="anchored" data-anchor-id="figure-4">Figure 4</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>enrichment_res <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/KEGG_results_2024-02-24.csv"</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>perm2_enrichment_res <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/KEGG_results_2024-02-24_perm2.csv"</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>motiff_enrichment_res <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/KEGG_results_2024-02-24_&gt;=5gene_hits.csv"</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>plot.data_perm2 <span class="ot">&lt;-</span> perm2_enrichment_res <span class="sc">%&gt;%</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adjust.manual.perm2 =</span> <span class="fu">p.adjust</span>(pvalue, <span class="st">"fdr"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">virus =</span> <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, virus)) <span class="sc">%&gt;%</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p.adjust.manual.perm2 <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(virus, ID, Description, p.adjust.manual.perm2)</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="do">### If the pathway was not found in the reverse permutation, outline it in the original figure</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>plot.data <span class="ot">&lt;-</span> enrichment_res <span class="sc">%&gt;%</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(p.adjust.manual = p.adjust(pvalue, "fdr")) %&gt;%</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adjust.manual =</span> p.adjust) <span class="sc">%&gt;%</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">virus =</span> <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, virus)) <span class="sc">%&gt;%</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p.adjust.manual <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">log10.p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust.manual)) <span class="sc">%&gt;%</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">full_join</span>(plot.data_perm2) <span class="sc">%&gt;%</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">not_found_in_perm2 =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(p.adjust.manual.perm2), <span class="st">"Not Found"</span>, <span class="st">"Found"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(p.adjust))</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>cluster_order <span class="ot">&lt;-</span>plot.data <span class="sc">%&gt;%</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Description, virus, p.adjust) <span class="sc">%&gt;%</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust)) <span class="sc">%&gt;%</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> Description, <span class="at">names_from =</span> virus, <span class="at">values_from =</span> p.adjust) <span class="sc">%&gt;%</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"Description"</span>)</span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>cluster_order[<span class="fu">is.na</span>(cluster_order)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>heir_cluster <span class="ot">&lt;-</span> <span class="fu">hclust</span>(vegan<span class="sc">::</span><span class="fu">vegdist</span>(cluster_order,<span class="at">method=</span><span class="st">"jaccard"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>freq_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(plot.data<span class="sc">$</span>ID))</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>taxa <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Proteome.ID <span class="sc">%in%</span> plot.data<span class="sc">$</span>virus[plot.data<span class="sc">$</span>ID <span class="sc">%in%</span> freq_df<span class="sc">$</span>Var1[freq_df<span class="sc">$</span>Freq <span class="sc">&gt;=</span> <span class="dv">3</span>]]) <span class="sc">%&gt;%</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(family, order, class, phylum, kingdom) <span class="sc">%&gt;%</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Anelloviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Kolmioviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Kolmioviridae"</span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">lapply</span>(taxa, as.factor), <span class="at">stringsAsFactors =</span><span class="cn">TRUE</span>)</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(check) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(taxa)</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> check[<span class="fu">complete.cases</span>(check),]</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">taxo2phylog</span>(<span class="fu">as.taxo</span>(check), <span class="at">root=</span><span class="st">"Root"</span>, <span class="at">abbrev=</span>F)</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>tax.phy <span class="ot">&lt;-</span> <span class="fu">as.phylo</span>(tmp)</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tax.phy, <span class="at">branch.length=</span><span class="st">'none'</span>)</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a>gg_tree[[<span class="st">"data"</span>]] <span class="ot">&lt;-</span> gg_tree[[<span class="st">"data"</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata <span class="sc">%&gt;%</span></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a>               <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(Taxon.mnemonic, family), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>))</span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span>  gg_tree <span class="sc">+</span> <span class="fu">geom_hilight</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">subset =</span> family <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2),<span class="at">fill =</span> family), <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a>   <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> Family_palette_v2)</span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a>gg1 <span class="ot">&lt;-</span> plot.data <span class="sc">%&gt;%</span></span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(motiff_enrichment_res <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(virus, Description) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">Motiff_bound =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ID <span class="sc">%in%</span> freq_df<span class="sc">$</span>Var1[freq_df<span class="sc">$</span>Freq <span class="sc">&gt;=</span> <span class="dv">3</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Proteome.ID, Pathogen, <span class="st">`</span><span class="at">Taxon.mnemonic</span><span class="st">`</span>, <span class="st">`</span><span class="at">Organism</span><span class="st">`</span>, Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"virus"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb8-42"><a href="#cb8-42" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">virus</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb8-43"><a href="#cb8-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Description =</span> <span class="fu">factor</span>(Description, heir_cluster[[<span class="st">"labels"</span>]])) <span class="sc">%&gt;%</span></span>
<span id="cb8-44"><a href="#cb8-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Taxon.mnemonic, <span class="at">y=</span>Description, <span class="at">size =</span> log10.p.adjust, <span class="at">fill =</span> log10.p.adjust)) <span class="sc">+</span></span>
<span id="cb8-45"><a href="#cb8-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> not_found_in_perm2), <span class="at">shape =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb8-46"><a href="#cb8-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> . <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Motiff_bound), <span class="at">shape =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb8-47"><a href="#cb8-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb8-48"><a href="#cb8-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">size=</span><span class="fu">guide_legend</span>(<span class="st">"-log10 Transformed</span><span class="sc">\n</span><span class="st">Adjusted p-values"</span>),</span>
<span id="cb8-49"><a href="#cb8-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">color=</span><span class="fu">guide_legend</span>(<span class="st">"Not found in</span><span class="sc">\n</span><span class="st">permuted sets"</span>)) <span class="sc">+</span></span>
<span id="cb8-50"><a href="#cb8-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">colors=</span><span class="fu">c</span>(<span class="st">"dodgerblue"</span>, <span class="st">"red"</span>), <span class="at">name =</span> <span class="st">"-log10 Transformed</span><span class="sc">\n</span><span class="st">Adjusted p-values"</span>) <span class="sc">+</span></span>
<span id="cb8-51"><a href="#cb8-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>( <span class="st">"00FFFFFF"</span>, <span class="st">"black"</span>)) <span class="sc">+</span></span>
<span id="cb8-52"><a href="#cb8-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(<span class="at">color=</span><span class="st">"grey"</span>, <span class="at">size=</span><span class="fl">0.1</span>), </span>
<span id="cb8-53"><a href="#cb8-53" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb8-54"><a href="#cb8-54" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb8-55"><a href="#cb8-55" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">fill=</span><span class="cn">NA</span>, <span class="at">size=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb8-56"><a href="#cb8-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">" "</span>) <span class="sc">+</span> </span>
<span id="cb8-57"><a href="#cb8-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">position =</span> <span class="st">"right"</span>)</span>
<span id="cb8-58"><a href="#cb8-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-59"><a href="#cb8-59" aria-hidden="true" tabindex="-1"></a>gg1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure4and5_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>gg1 <span class="sc">%&gt;%</span> <span class="fu">insert_bottom</span>(gg_tree <span class="sc">+</span> <span class="fu">coord_flip</span>(), <span class="at">height =</span>  <span class="fu">c</span>(<span class="fl">0.07</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure4and5_files/figure-html/unnamed-chunk-8-2.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"Figure4A.pdf"</span>), <span class="at">width =</span> <span class="dv">12</span>, <span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(gg1<span class="sc">$</span>data, <span class="fu">paste0</span>(out_dir, <span class="st">"Figure4a_datapoints.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="panel-5a" class="level3">
<h3 class="anchored" data-anchor-id="panel-5a">Panel 5A</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>filter_nonsig_clusterProfiler <span class="ot">&lt;-</span> <span class="cf">function</span>(out_list, <span class="at">threshold =</span> <span class="fl">0.05</span>, pathway_list){</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(out_list)){</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>    out_list[[i]]<span class="sc">@</span>result <span class="ot">&lt;-</span> out_list[[i]]<span class="sc">@</span>result <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;=</span> threshold)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">nrow</span>(out_list[[i]]<span class="sc">@</span>result) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>       <span class="fu">names</span>(out_list)[i] <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">names</span>(out_list[i]), <span class="st">"Remove"</span>)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>      out_list[[i]]<span class="sc">@</span>geneSets <span class="ot">&lt;-</span> out_list[[i]]<span class="sc">@</span>geneSets[<span class="fu">names</span>(out_list[[i]]<span class="sc">@</span>geneSets) <span class="sc">%in%</span> out_list[[i]]<span class="sc">@</span>result<span class="sc">$</span>ID]</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">missing</span>(pathway_list)){</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(out_list)){</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>      out_list[[i]]<span class="sc">@</span>result <span class="ot">&lt;-</span> out_list[[i]]<span class="sc">@</span>result <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ID <span class="sc">%in%</span> pathway_list)</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">nrow</span>(out_list[[i]]<span class="sc">@</span>result) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>         <span class="fu">names</span>(out_list)[i] <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">names</span>(out_list[i]), <span class="st">"Remove"</span>)</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>        out_list[[i]]<span class="sc">@</span>geneSets <span class="ot">&lt;-</span> out_list[[i]]<span class="sc">@</span>geneSets[<span class="fu">names</span>(out_list[[i]]<span class="sc">@</span>geneSets) <span class="sc">%in%</span> out_list[[i]]<span class="sc">@</span>result<span class="sc">$</span>ID]</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>   }</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>  out_list <span class="ot">&lt;-</span> out_list[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"Remove"</span>, <span class="fu">names</span>(out_list) )]</span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../output/KEGG_results_2024-02-24.rds"</span>)</span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>pathways_selected <span class="ot">&lt;-</span> plot.data <span class="sc">%&gt;%</span></span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ID <span class="sc">%in%</span> freq_df<span class="sc">$</span>Var1[freq_df<span class="sc">$</span>Freq <span class="sc">&gt;=</span> <span class="dv">3</span>])</span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>out_final_filtered <span class="ot">&lt;-</span> <span class="fu">filter_nonsig_clusterProfiler</span>(out_final, <span class="at">threshold =</span> <span class="fl">0.001</span>, <span class="at">pathway_list =</span> pathways_selected<span class="sc">$</span>ID)</span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a><span class="do">## Because of the way we are running the pie charts, the viruses</span></span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a><span class="do">## need to be sorted by viral families</span></span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a>out_final_filtered <span class="ot">&lt;-</span> out_final_filtered[<span class="fu">order</span>(metadata<span class="sc">$</span>Family_Viral_Zone[ <span class="fu">match</span>(<span class="fu">names</span>(out_final_filtered), metadata<span class="sc">$</span><span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>)])]</span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a>total_enrichment <span class="ot">&lt;-</span> <span class="fu">merge_result</span>(out_final_filtered)</span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>color_vec <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(out_final_filtered))</span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(color_vec) <span class="ot">&lt;-</span> <span class="fu">levels</span>(total_enrichment<span class="sc">@</span>compareClusterResult<span class="sc">$</span>Cluster)</span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a>color_vec <span class="ot">&lt;-</span> metadata<span class="sc">$</span>Family_Viral_Zone[<span class="fu">match</span>(<span class="fu">names</span>(color_vec), metadata<span class="sc">$</span><span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>)]</span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a>color_vec_r <span class="ot">&lt;-</span> Family_palette_v2[<span class="fu">match</span>(color_vec, <span class="fu">names</span>(Family_palette_v2))]</span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a>color_vec_r[<span class="fu">is.na</span>(color_vec_r)] <span class="ot">&lt;-</span> <span class="st">"lightgray"</span></span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(color_vec_r) <span class="ot">&lt;-</span> <span class="fu">levels</span>(total_enrichment<span class="sc">@</span>compareClusterResult<span class="sc">$</span>Cluster)</span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a><span class="fu">cnetplot</span>(total_enrichment, <span class="at">node_label=</span><span class="st">"category"</span>,</span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a>         <span class="at">cex_category =</span> <span class="dv">10</span>,</span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a>         <span class="at">cex_line =</span> <span class="fl">0.25</span>, <span class="at">shadowtext =</span> <span class="st">"none"</span>,</span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a>         <span class="at">cex.params =</span> <span class="fu">list</span>(<span class="at">category_label =</span> <span class="fl">1.2</span>,<span class="at">edge =</span> <span class="cn">TRUE</span>), <span class="at">showCategory =</span><span class="dv">25</span>,</span>
<span id="cb11-47"><a href="#cb11-47" aria-hidden="true" tabindex="-1"></a>         <span class="at">layout =</span> <span class="st">"kk"</span>) <span class="sc">+</span></span>
<span id="cb11-48"><a href="#cb11-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> color_vec_r)<span class="sc">+</span></span>
<span id="cb11-49"><a href="#cb11-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure4and5_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"Figure5A.pdf"</span>), <span class="at">width =</span> <span class="dv">5</span>, <span class="at">height =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="autosome-x-and-y-chromosome-encoding-analysis" class="level2">
<h2 class="anchored" data-anchor-id="autosome-x-and-y-chromosome-encoding-analysis">Autosome, X, and Y Chromosome encoding analysis</h2>
<section id="actual-chromosome-mapping" class="level3">
<h3 class="anchored" data-anchor-id="actual-chromosome-mapping">Actual Chromosome Mapping</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../12mer_annotated/"</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".csv.gz"</span>, files_to_read)]</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="do">## For testing purposes use this subset</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>X_Y <span class="ot">=</span> <span class="cn">FALSE</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file))</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">nThread =</span> <span class="dv">10</span>)</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Filter out any duplicated proteins, identified in </span></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>  <span class="do">## ../../Scripts/Duplicate_Protein_Screen.R</span></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(human_protein_metadata <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(entry.human, proteomes.human) <span class="sc">%&gt;%</span></span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>                <span class="fu">mutate</span>(<span class="at">chromosome_name =</span> <span class="fu">gsub</span>(<span class="st">".* "</span>, <span class="st">""</span>, proteomes.human)) <span class="sc">%&gt;%</span></span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>                dplyr<span class="sc">::</span><span class="fu">select</span>(entry.human, chromosome_name), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="st">"entry.human"</span>))</span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(X_Y){</span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>    chrome_unique <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">chromosome_name =</span> <span class="fu">ifelse</span>(chromosome_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>), <span class="st">"Auto"</span>, chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>      <span class="fu">arrange</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(query, query_seqname, query_start) <span class="sc">%&gt;%</span></span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">chromosome_name =</span> <span class="fu">paste</span>(chromosome_name, <span class="at">collapse =</span> <span class="st">"&amp;"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">chromosome_name =</span> <span class="fu">gsub</span>(<span class="st">"&amp;Unplaced"</span>, <span class="st">""</span>, chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(chromosome_name <span class="sc">!=</span> <span class="st">"Unplaced"</span>)</span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>    chrome_unique <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">chromosome_name =</span> <span class="fu">ifelse</span>(chromosome_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>), <span class="st">"Auto"</span>, chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a>      <span class="fu">arrange</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(chromosome_name <span class="sc">!=</span> <span class="st">"Unplaced"</span>)</span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a>   <span class="do">## These lines basically say if a 12mer mimics multiple things and one thing is a mimic and the others are not, then ignore then call the 12mer a mTEC mimic</span></span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a>    active_data_chromo <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb13-48"><a href="#cb13-48" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(query_seqname, query_start) <span class="sc">%&gt;%</span></span>
<span id="cb13-49"><a href="#cb13-49" aria-hidden="true" tabindex="-1"></a>      <span class="fu">slice_min</span>(mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb13-50"><a href="#cb13-50" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb13-51"><a href="#cb13-51" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(chrome_unique) <span class="sc">%&gt;%</span></span>
<span id="cb13-52"><a href="#cb13-52" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb13-53"><a href="#cb13-53" aria-hidden="true" tabindex="-1"></a>     <span class="fu">group_by</span>(chromosome_name, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb13-54"><a href="#cb13-54" aria-hidden="true" tabindex="-1"></a>     dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb13-55"><a href="#cb13-55" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">pathogen =</span> active_file)</span>
<span id="cb13-56"><a href="#cb13-56" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb13-57"><a href="#cb13-57" aria-hidden="true" tabindex="-1"></a>  out_final <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_final, active_data_chromo)</span>
<span id="cb13-58"><a href="#cb13-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-59"><a href="#cb13-59" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-60"><a href="#cb13-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-61"><a href="#cb13-61" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_final, <span class="st">"../output/true_chromosome_mapping.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="permutation-evaluations" class="level3">
<h3 class="anchored" data-anchor-id="permutation-evaluations">Permutation Evaluations</h3>
<section id="permutation-1-1" class="level5">
<h5 class="anchored" data-anchor-id="permutation-1-1">Permutation 1</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Now do the rate count in the Permuted Proteomes</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../../epitope_alignment/Results/Permutation_exp1/"</span>)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".tsv"</span>, files_to_read)]</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="do">## For testing purposes use this subset</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>X_Y <span class="ot">=</span> <span class="cn">FALSE</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>out_final_perm1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" "</span>,</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>               <span class="fu">match</span>(active_file, files_to_read), <span class="st">"/"</span>, <span class="fu">length</span>(files_to_read)))</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(active_file)</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>mismatches <span class="ot">&lt;-</span> <span class="fu">apply</span>(active_data, <span class="dv">1</span>, <span class="cf">function</span>(x){<span class="fu">DNA_Diff</span>(x[<span class="dv">1</span>], x[<span class="dv">6</span>])})</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>mismatches <span class="ot">&lt;-</span> <span class="fu">factor</span>(active_data<span class="sc">$</span>mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>uniprot <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(active_data<span class="sc">$</span>target_seqname, <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>]))</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>viral_id <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(active_data<span class="sc">$</span>query_seqname, <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>]))</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>   <span class="do">## Filter out any duplicated proteins, identified in </span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>  <span class="do">## ../../Scripts/Duplicate_Protein_Screen.R</span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(human_protein_metadata <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(entry.human, proteomes.human) <span class="sc">%&gt;%</span></span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>                <span class="fu">mutate</span>(<span class="at">chromosome_name =</span> <span class="fu">gsub</span>(<span class="st">".* "</span>, <span class="st">""</span>, proteomes.human)) <span class="sc">%&gt;%</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>                dplyr<span class="sc">::</span><span class="fu">select</span>(entry.human, chromosome_name), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="st">"entry.human"</span>))</span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(X_Y){</span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>    chrome_unique <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">chromosome_name =</span> <span class="fu">ifelse</span>(chromosome_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>), <span class="st">"Auto"</span>, chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>      <span class="fu">arrange</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(query, query_seqname, query_start) <span class="sc">%&gt;%</span></span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">chromosome_name =</span> <span class="fu">paste</span>(chromosome_name, <span class="at">collapse =</span> <span class="st">"&amp;"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">chromosome_name =</span> <span class="fu">gsub</span>(<span class="st">"&amp;Unplaced"</span>, <span class="st">""</span>, chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(chromosome_name <span class="sc">!=</span> <span class="st">"Unplaced"</span>)</span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>    chrome_unique <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">chromosome_name =</span> <span class="fu">ifelse</span>(chromosome_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>), <span class="st">"Auto"</span>, chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a>      <span class="fu">arrange</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(chromosome_name <span class="sc">!=</span> <span class="st">"Unplaced"</span>)</span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a>   <span class="do">## These lines basically say if a 12mer mimics multiple things and one thing is a mimic and the others are not, then ignore then call the 12mer a mTEC mimic</span></span>
<span id="cb14-50"><a href="#cb14-50" aria-hidden="true" tabindex="-1"></a>    active_data_chromo <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb14-51"><a href="#cb14-51" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(query_seqname, query_start) <span class="sc">%&gt;%</span></span>
<span id="cb14-52"><a href="#cb14-52" aria-hidden="true" tabindex="-1"></a>      <span class="fu">slice_min</span>(mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb14-53"><a href="#cb14-53" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb14-54"><a href="#cb14-54" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(chrome_unique) <span class="sc">%&gt;%</span></span>
<span id="cb14-55"><a href="#cb14-55" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb14-56"><a href="#cb14-56" aria-hidden="true" tabindex="-1"></a>     <span class="fu">group_by</span>(chromosome_name, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb14-57"><a href="#cb14-57" aria-hidden="true" tabindex="-1"></a>     dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb14-58"><a href="#cb14-58" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">pathogen =</span> active_file)</span>
<span id="cb14-59"><a href="#cb14-59" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb14-60"><a href="#cb14-60" aria-hidden="true" tabindex="-1"></a>  out_final_perm1 <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_final_perm1, active_data_chromo)</span>
<span id="cb14-61"><a href="#cb14-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-62"><a href="#cb14-62" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-63"><a href="#cb14-63" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_final_perm1, <span class="st">"../output/perm1_random_chromosome.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="permutation-2-1" class="level5">
<h5 class="anchored" data-anchor-id="permutation-2-1">Permutation 2</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Now do the rate count in the Permuted Proteomes</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../../epitope_alignment/Results/Permutation_exp2/"</span>)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".tsv"</span>, files_to_read)]</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="do">## For testing purposes use this subset</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>X_Y <span class="ot">=</span> <span class="cn">FALSE</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>out_final_perm2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" "</span>,</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>               <span class="fu">match</span>(active_file, files_to_read), <span class="st">"/"</span>, <span class="fu">length</span>(files_to_read)))</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(active_file)</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>mismatches <span class="ot">&lt;-</span> <span class="fu">apply</span>(active_data, <span class="dv">1</span>, <span class="cf">function</span>(x){<span class="fu">DNA_Diff</span>(x[<span class="dv">1</span>], x[<span class="dv">6</span>])})</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>mismatches <span class="ot">&lt;-</span> <span class="fu">factor</span>(active_data<span class="sc">$</span>mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>uniprot <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(active_data<span class="sc">$</span>target_seqname, <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>]))</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>viral_id <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(active_data<span class="sc">$</span>query_seqname, <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>]))</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>   <span class="do">## Filter out any duplicated proteins, identified in </span></span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>  <span class="do">## ../../Scripts/Duplicate_Protein_Screen.R</span></span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(human_protein_metadata <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(entry.human, proteomes.human) <span class="sc">%&gt;%</span></span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>                <span class="fu">mutate</span>(<span class="at">chromosome_name =</span> <span class="fu">gsub</span>(<span class="st">".* "</span>, <span class="st">""</span>, proteomes.human)) <span class="sc">%&gt;%</span></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>                dplyr<span class="sc">::</span><span class="fu">select</span>(entry.human, chromosome_name), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="st">"entry.human"</span>))</span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(X_Y){</span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>    chrome_unique <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">chromosome_name =</span> <span class="fu">ifelse</span>(chromosome_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>), <span class="st">"Auto"</span>, chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a>      <span class="fu">arrange</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(query, query_seqname, query_start) <span class="sc">%&gt;%</span></span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">chromosome_name =</span> <span class="fu">paste</span>(chromosome_name, <span class="at">collapse =</span> <span class="st">"&amp;"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">chromosome_name =</span> <span class="fu">gsub</span>(<span class="st">"&amp;Unplaced"</span>, <span class="st">""</span>, chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(chromosome_name <span class="sc">!=</span> <span class="st">"Unplaced"</span>)</span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a>    chrome_unique <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">chromosome_name =</span> <span class="fu">ifelse</span>(chromosome_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>), <span class="st">"Auto"</span>, chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a>      <span class="fu">arrange</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(chromosome_name <span class="sc">!=</span> <span class="st">"Unplaced"</span>)</span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-49"><a href="#cb15-49" aria-hidden="true" tabindex="-1"></a>   <span class="do">## These lines basically say if a 12mer mimics multiple things and one thing is a mimic and the others are not, then ignore then call the 12mer a mTEC mimic</span></span>
<span id="cb15-50"><a href="#cb15-50" aria-hidden="true" tabindex="-1"></a>    active_data_chromo <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb15-51"><a href="#cb15-51" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(query_seqname, query_start) <span class="sc">%&gt;%</span></span>
<span id="cb15-52"><a href="#cb15-52" aria-hidden="true" tabindex="-1"></a>      <span class="fu">slice_min</span>(mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb15-53"><a href="#cb15-53" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb15-54"><a href="#cb15-54" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(chrome_unique) <span class="sc">%&gt;%</span></span>
<span id="cb15-55"><a href="#cb15-55" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb15-56"><a href="#cb15-56" aria-hidden="true" tabindex="-1"></a>     <span class="fu">group_by</span>(chromosome_name, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb15-57"><a href="#cb15-57" aria-hidden="true" tabindex="-1"></a>     dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb15-58"><a href="#cb15-58" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">pathogen =</span> active_file)</span>
<span id="cb15-59"><a href="#cb15-59" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb15-60"><a href="#cb15-60" aria-hidden="true" tabindex="-1"></a>  out_final_perm2 <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_final_perm2, active_data_chromo)</span>
<span id="cb15-61"><a href="#cb15-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-62"><a href="#cb15-62" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-63"><a href="#cb15-63" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_final_perm2, <span class="st">"../output/perm2_rev_chromosome.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="permutation-3-1" class="level5">
<h5 class="anchored" data-anchor-id="permutation-3-1">Permutation 3</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Now do the rate count in the Permuted Proteomes</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../../epitope_alignment/Results/Permutation_exp3/"</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".tsv"</span>, files_to_read)]</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="do">## For testing purposes use this subset</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>X_Y <span class="ot">=</span> <span class="cn">FALSE</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>out_final_perm3 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" "</span>,</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>               <span class="fu">match</span>(active_file, files_to_read), <span class="st">"/"</span>, <span class="fu">length</span>(files_to_read)))</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(active_file)</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>mismatches <span class="ot">&lt;-</span> <span class="fu">apply</span>(active_data, <span class="dv">1</span>, <span class="cf">function</span>(x){<span class="fu">DNA_Diff</span>(x[<span class="dv">1</span>], x[<span class="dv">6</span>])})</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>mismatches <span class="ot">&lt;-</span> <span class="fu">factor</span>(active_data<span class="sc">$</span>mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>uniprot <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(active_data<span class="sc">$</span>target_seqname, <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>]))</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>viral_id <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(active_data<span class="sc">$</span>query_seqname, <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>]))</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>   <span class="do">## Filter out any duplicated proteins, identified in </span></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>  <span class="do">## ../../Scripts/Duplicate_Protein_Screen.R</span></span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(human_protein_metadata <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(entry.human, proteomes.human) <span class="sc">%&gt;%</span></span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>                <span class="fu">mutate</span>(<span class="at">chromosome_name =</span> <span class="fu">gsub</span>(<span class="st">".* "</span>, <span class="st">""</span>, proteomes.human)) <span class="sc">%&gt;%</span></span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>                dplyr<span class="sc">::</span><span class="fu">select</span>(entry.human, chromosome_name), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="st">"entry.human"</span>))</span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(X_Y){</span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>    chrome_unique <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">chromosome_name =</span> <span class="fu">ifelse</span>(chromosome_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>), <span class="st">"Auto"</span>, chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a>      <span class="fu">arrange</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(query, query_seqname, query_start) <span class="sc">%&gt;%</span></span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">chromosome_name =</span> <span class="fu">paste</span>(chromosome_name, <span class="at">collapse =</span> <span class="st">"&amp;"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">chromosome_name =</span> <span class="fu">gsub</span>(<span class="st">"&amp;Unplaced"</span>, <span class="st">""</span>, chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(chromosome_name <span class="sc">!=</span> <span class="st">"Unplaced"</span>)</span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb16-41"><a href="#cb16-41" aria-hidden="true" tabindex="-1"></a>    chrome_unique <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb16-42"><a href="#cb16-42" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">chromosome_name =</span> <span class="fu">ifelse</span>(chromosome_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>), <span class="st">"Auto"</span>, chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb16-43"><a href="#cb16-43" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb16-44"><a href="#cb16-44" aria-hidden="true" tabindex="-1"></a>      <span class="fu">arrange</span>(query, query_seqname, query_start, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb16-45"><a href="#cb16-45" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb16-46"><a href="#cb16-46" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(chromosome_name <span class="sc">!=</span> <span class="st">"Unplaced"</span>)</span>
<span id="cb16-47"><a href="#cb16-47" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-48"><a href="#cb16-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-49"><a href="#cb16-49" aria-hidden="true" tabindex="-1"></a>   <span class="do">## These lines basically say if a 12mer mimics multiple things and one thing is a mimic and the others are not, then ignore then call the 12mer a mTEC mimic</span></span>
<span id="cb16-50"><a href="#cb16-50" aria-hidden="true" tabindex="-1"></a>    active_data_chromo <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb16-51"><a href="#cb16-51" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(query_seqname, query_start) <span class="sc">%&gt;%</span></span>
<span id="cb16-52"><a href="#cb16-52" aria-hidden="true" tabindex="-1"></a>      <span class="fu">slice_min</span>(mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb16-53"><a href="#cb16-53" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb16-54"><a href="#cb16-54" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(chrome_unique) <span class="sc">%&gt;%</span></span>
<span id="cb16-55"><a href="#cb16-55" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(chromosome_name)) <span class="sc">%&gt;%</span></span>
<span id="cb16-56"><a href="#cb16-56" aria-hidden="true" tabindex="-1"></a>     <span class="fu">group_by</span>(chromosome_name, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb16-57"><a href="#cb16-57" aria-hidden="true" tabindex="-1"></a>     dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb16-58"><a href="#cb16-58" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">pathogen =</span> active_file)</span>
<span id="cb16-59"><a href="#cb16-59" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb16-60"><a href="#cb16-60" aria-hidden="true" tabindex="-1"></a>  out_final_perm3 <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_final_perm3, active_data_chromo)</span>
<span id="cb16-61"><a href="#cb16-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-62"><a href="#cb16-62" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb16-63"><a href="#cb16-63" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_final_perm3, <span class="st">"../output/perm3_class_shuffle_chromosome.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="panel-5b" class="level3">
<h3 class="anchored" data-anchor-id="panel-5b">Panel 5b</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/true_chromosome_mapping.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>out_final<span class="sc">$</span>Proteome.ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer.*"</span>, <span class="st">""</span>, out_final<span class="sc">$</span>pathogen))</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>out_final_perm2 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/perm2_rev_chromosome.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>out_final_perm2<span class="sc">$</span>Proteome.ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer.*"</span>, <span class="st">""</span>, out_final_perm2<span class="sc">$</span>pathogen))</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>reversed_results <span class="ot">&lt;-</span> out_final <span class="sc">%&gt;%</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mismatches, <span class="at">values_from =</span> count) <span class="sc">%&gt;%</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), <span class="at">names_to=</span><span class="st">"mismatches"</span>, <span class="at">values_to=</span><span class="st">"count"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(mismatches), <span class="dv">0</span>, mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(out_final_perm2 <span class="sc">%&gt;%</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mismatches, <span class="at">values_from =</span> count) <span class="sc">%&gt;%</span></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>( <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>( <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), <span class="at">names_to=</span><span class="st">"mismatches"</span>, <span class="at">values_to=</span><span class="st">"count"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(mismatches), <span class="dv">0</span>, mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>pathogen), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"chromosome_name"</span>,<span class="st">"Proteome.ID"</span>, <span class="st">"mismatches"</span>), <span class="at">suffix =</span> <span class="fu">c</span>(<span class="st">".true"</span>, <span class="st">".permuted"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count.true =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(count.true), <span class="dv">0</span>, count.true))<span class="sc">%&gt;%</span></span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count.permuted =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(count.permuted), <span class="dv">0</span>, count.permuted)) <span class="sc">%&gt;%</span></span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FC =</span> (count.true <span class="sc">+</span> <span class="dv">3</span>)<span class="sc">/</span>(count.permuted <span class="sc">+</span> <span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a> <span class="co"># filter(chromosome_name %in% c(1:22, "Y", "X")) %&gt;%</span></span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(chromosome_name = ifelse(chromosome_name %in% c(1:22), "Auto", chromosome_name)) %&gt;%</span></span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(chromosome_name, mismatches, Proteome.ID) <span class="sc">%&gt;%</span></span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">median =</span> <span class="fu">median</span>(FC))</span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a>pd <span class="ot">&lt;-</span> <span class="fu">position_dodge</span>(<span class="fl">0.4</span>)</span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> out_final <span class="sc">%&gt;%</span></span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mismatches, <span class="at">values_from =</span> count) <span class="sc">%&gt;%</span></span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-41"><a href="#cb17-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb17-42"><a href="#cb17-42" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb17-43"><a href="#cb17-43" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb17-44"><a href="#cb17-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-45"><a href="#cb17-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), <span class="at">names_to=</span><span class="st">"mismatches"</span>, <span class="at">values_to=</span><span class="st">"count"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-46"><a href="#cb17-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(mismatches), <span class="dv">0</span>, mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb17-47"><a href="#cb17-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(out_final_perm2 <span class="sc">%&gt;%</span></span>
<span id="cb17-48"><a href="#cb17-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mismatches, <span class="at">values_from =</span> count) <span class="sc">%&gt;%</span></span>
<span id="cb17-49"><a href="#cb17-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>( <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb17-50"><a href="#cb17-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-51"><a href="#cb17-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb17-52"><a href="#cb17-52" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb17-53"><a href="#cb17-53" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb17-54"><a href="#cb17-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-55"><a href="#cb17-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>( <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), <span class="at">names_to=</span><span class="st">"mismatches"</span>, <span class="at">values_to=</span><span class="st">"count"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-56"><a href="#cb17-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(mismatches), <span class="dv">0</span>, mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb17-57"><a href="#cb17-57" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>pathogen), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"chromosome_name"</span>,<span class="st">"Proteome.ID"</span>, <span class="st">"mismatches"</span>), <span class="at">suffix =</span> <span class="fu">c</span>(<span class="st">".true"</span>, <span class="st">".permuted"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb17-58"><a href="#cb17-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count.true =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(count.true), <span class="dv">0</span>, count.true))<span class="sc">%&gt;%</span></span>
<span id="cb17-59"><a href="#cb17-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count.permuted =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(count.permuted), <span class="dv">0</span>, count.permuted)) <span class="sc">%&gt;%</span></span>
<span id="cb17-60"><a href="#cb17-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FC =</span> (count.true <span class="sc">+</span> <span class="dv">3</span>)<span class="sc">/</span>(count.permuted <span class="sc">+</span> <span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb17-61"><a href="#cb17-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb17-62"><a href="#cb17-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Family_Viral_Zone <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2)) <span class="sc">%&gt;%</span></span>
<span id="cb17-63"><a href="#cb17-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Proteome.ID, Family_Viral_Zone, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb17-64"><a href="#cb17-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(chromosome_name <span class="sc">!=</span> <span class="st">"Mitochondrion"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-65"><a href="#cb17-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">paste0</span>(<span class="st">"≤"</span>, mismatches, <span class="st">" mismatches"</span>)) </span>
<span id="cb17-66"><a href="#cb17-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-67"><a href="#cb17-67" aria-hidden="true" tabindex="-1"></a>supp_data <span class="ot">&lt;-</span> testing_data <span class="sc">%&gt;%</span></span>
<span id="cb17-68"><a href="#cb17-68" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(!(count.true == 0 &amp; count.permuted == 0)) %&gt;%</span></span>
<span id="cb17-69"><a href="#cb17-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(chromosome_name, FC, pathogen, Proteome.ID, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb17-70"><a href="#cb17-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> chromosome_name, <span class="at">values_from =</span> FC, <span class="at">id_cols =</span> <span class="fu">c</span>(pathogen, Proteome.ID, mismatches))</span>
<span id="cb17-71"><a href="#cb17-71" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> supp_data <span class="sc">%&gt;%</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(Auto, X, Y)), value <span class="sc">~</span> name, <span class="at">paired =</span> <span class="cn">TRUE</span>,</span>
<span id="cb17-72"><a href="#cb17-72" aria-hidden="true" tabindex="-1"></a>                       <span class="at">group.by =</span> <span class="st">"mismatches"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-73"><a href="#cb17-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">y.position =</span> <span class="fu">c</span>(<span class="fl">2.75</span>, <span class="fl">2.5</span>, <span class="fl">2.25</span>, <span class="fl">2.75</span>, <span class="fl">2.5</span>, <span class="fl">2.25</span>,<span class="fl">2.75</span>, <span class="fl">2.5</span>, <span class="fl">2.25</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb17-74"><a href="#cb17-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_significance</span>(<span class="st">"p.adj"</span>)</span>
<span id="cb17-75"><a href="#cb17-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-76"><a href="#cb17-76" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(supp_data, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData6_part1.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"Perm2_Chr_Data"</span>,  <span class="at">append =</span> <span class="cn">FALSE</span>)</span>
<span id="cb17-77"><a href="#cb17-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-78"><a href="#cb17-78" aria-hidden="true" tabindex="-1"></a><span class="do">## sort dataframe by group then by x</span></span>
<span id="cb17-79"><a href="#cb17-79" aria-hidden="true" tabindex="-1"></a>reversed_results <span class="sc">%&gt;%</span></span>
<span id="cb17-80"><a href="#cb17-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb17-81"><a href="#cb17-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Family_Viral_Zone <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2)) <span class="sc">%&gt;%</span></span>
<span id="cb17-82"><a href="#cb17-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Proteome.ID, Family_Viral_Zone, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb17-83"><a href="#cb17-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(chromosome_name <span class="sc">!=</span> <span class="st">"Mitochondrion"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-84"><a href="#cb17-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">paste0</span>(<span class="st">"≤"</span>, mismatches, <span class="st">" mismatches"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb17-85"><a href="#cb17-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>chromosome_name, <span class="at">y=</span>median, <span class="at">group =</span> Proteome.ID)) <span class="sc">+</span></span>
<span id="cb17-86"><a href="#cb17-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>,<span class="at">position =</span> pd) <span class="sc">+</span></span>
<span id="cb17-87"><a href="#cb17-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="fu">aes</span>(<span class="at">color =</span> Family_Viral_Zone), <span class="at">position =</span> pd) <span class="sc">+</span></span>
<span id="cb17-88"><a href="#cb17-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> Family_palette_v2) <span class="sc">+</span></span>
<span id="cb17-89"><a href="#cb17-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb17-90"><a href="#cb17-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>mismatches,<span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb17-91"><a href="#cb17-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log10_trans</span>(),</span>
<span id="cb17-92"><a href="#cb17-92" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log10"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">10</span><span class="sc">^</span>x, <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb17-93"><a href="#cb17-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb17-94"><a href="#cb17-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"FC of 12mers compared</span><span class="sc">\n</span><span class="st">to Perm 2"</span>) <span class="sc">+</span></span>
<span id="cb17-95"><a href="#cb17-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Chromosome"</span>) <span class="sc">+</span></span>
<span id="cb17-96"><a href="#cb17-96" aria-hidden="true" tabindex="-1"></a>  <span class="co">#stat_compare_means(comparisons = list(c("Auto", "Y"), c("X", "Y"))) +</span></span>
<span id="cb17-97"><a href="#cb17-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_pvalue_manual</span>(check,<span class="at">label =</span> <span class="st">"p.adj.signif"</span>) <span class="sc">+</span></span>
<span id="cb17-98"><a href="#cb17-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb17-99"><a href="#cb17-99" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.height=</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">'cm'</span>),</span>
<span id="cb17-100"><a href="#cb17-100" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.width=</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">'cm'</span>),</span>
<span id="cb17-101"><a href="#cb17-101" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">fill=</span><span class="cn">NA</span>, <span class="at">size=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb17-102"><a href="#cb17-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">0.316</span>,<span class="dv">800</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure4and5_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"Figure5B.pdf"</span>), <span class="at">width =</span><span class="fl">5.5</span>, <span class="at">height =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="supplemental-materials" class="level2">
<h2 class="anchored" data-anchor-id="supplemental-materials">Supplemental Materials</h2>
</section>
<section id="figure-s4" class="level2">
<h2 class="anchored" data-anchor-id="figure-s4">Figure S4</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>taxa <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Proteome.ID <span class="sc">%in%</span> plot.data<span class="sc">$</span>virus) <span class="sc">%&gt;%</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(family, order, class, phylum, kingdom) <span class="sc">%&gt;%</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Anelloviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Kolmioviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Kolmioviridae"</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">lapply</span>(taxa, as.factor), <span class="at">stringsAsFactors =</span><span class="cn">TRUE</span>)</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(check) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(taxa)</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> check[<span class="fu">complete.cases</span>(check),]</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">taxo2phylog</span>(<span class="fu">as.taxo</span>(check), <span class="at">root=</span><span class="st">"Root"</span>, <span class="at">abbrev=</span>F)</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>tax.phy <span class="ot">&lt;-</span> <span class="fu">as.phylo</span>(tmp)</span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tax.phy, <span class="at">branch.length=</span><span class="st">'none'</span>)</span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a>gg_tree[[<span class="st">"data"</span>]] <span class="ot">&lt;-</span> gg_tree[[<span class="st">"data"</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata <span class="sc">%&gt;%</span></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a>               <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(Taxon.mnemonic, family), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>))</span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span>  gg_tree <span class="sc">+</span> <span class="fu">geom_hilight</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">subset =</span> family <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2),<span class="at">fill =</span> family), <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a>   <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> Family_palette_v2)</span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a>ggsupp <span class="ot">&lt;-</span> plot.data <span class="sc">%&gt;%</span></span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(motiff_enrichment_res <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(virus, Description) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">Motiff_bound =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Proteome.ID, Pathogen, <span class="st">`</span><span class="at">Taxon.mnemonic</span><span class="st">`</span>, <span class="st">`</span><span class="at">Organism</span><span class="st">`</span>, Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"virus"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb19-36"><a href="#cb19-36" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb19-37"><a href="#cb19-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb19-38"><a href="#cb19-38" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">virus</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb19-39"><a href="#cb19-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb19-40"><a href="#cb19-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Description =</span> <span class="fu">factor</span>(Description, heir_cluster[[<span class="st">"labels"</span>]])) <span class="sc">%&gt;%</span></span>
<span id="cb19-41"><a href="#cb19-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Taxon.mnemonic, <span class="at">y=</span>Description, <span class="at">size =</span> log10.p.adjust, <span class="at">fill =</span> log10.p.adjust)) <span class="sc">+</span></span>
<span id="cb19-42"><a href="#cb19-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> not_found_in_perm2), <span class="at">shape =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb19-43"><a href="#cb19-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> . <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Motiff_bound), <span class="at">shape =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb19-44"><a href="#cb19-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb19-45"><a href="#cb19-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">size=</span><span class="fu">guide_legend</span>(<span class="st">"-log10 Transformed</span><span class="sc">\n</span><span class="st">Adjusted p-values"</span>),</span>
<span id="cb19-46"><a href="#cb19-46" aria-hidden="true" tabindex="-1"></a>         <span class="co">#fill=guide_legend("-log10 Transformed\nAdjusted p-values"),</span></span>
<span id="cb19-47"><a href="#cb19-47" aria-hidden="true" tabindex="-1"></a>         <span class="at">color=</span><span class="fu">guide_legend</span>(<span class="st">"Not found in</span><span class="sc">\n</span><span class="st">permuted sets"</span>)) <span class="sc">+</span></span>
<span id="cb19-48"><a href="#cb19-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">colors=</span><span class="fu">c</span>(<span class="st">"dodgerblue"</span>, <span class="st">"red"</span>), <span class="at">name =</span> <span class="st">"-log10 Transformed</span><span class="sc">\n</span><span class="st">Adjusted p-values"</span>) <span class="sc">+</span></span>
<span id="cb19-49"><a href="#cb19-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>( <span class="st">"00FFFFFF"</span>, <span class="st">"black"</span>)) <span class="sc">+</span></span>
<span id="cb19-50"><a href="#cb19-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">limits=</span>rev) <span class="sc">+</span></span>
<span id="cb19-51"><a href="#cb19-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(<span class="at">color=</span><span class="st">"grey"</span>, <span class="at">size=</span><span class="fl">0.1</span>), </span>
<span id="cb19-52"><a href="#cb19-52" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">90</span>, <span class="at">hjust=</span><span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb19-53"><a href="#cb19-53" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(Description, virus)`
Joining with `by = join_by(virus)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>ggsupp</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure4and5_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>ggsupp <span class="sc">%&gt;%</span> <span class="fu">insert_bottom</span>(gg_tree <span class="sc">+</span><span class="fu">coord_flip</span>(),<span class="at">height =</span>  <span class="fu">c</span>(<span class="fl">0.05</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure4and5_files/figure-html/unnamed-chunk-15-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"FigureS4.pdf"</span>), <span class="at">width =</span> <span class="dv">12</span>, <span class="at">height =</span> <span class="dv">24</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="figure-s5" class="level2">
<h2 class="anchored" data-anchor-id="figure-s5">Figure S5</h2>
<p>Heatmap of mimicry of human proteins by cell type, tissue, and organ system, using mapping from the human protein atlas. The code for this supplemental figure is in the other quato markdown document in this folder.</p>
</section>
<section id="figure-s6" class="level2">
<h2 class="anchored" data-anchor-id="figure-s6">Figure S6</h2>
<section id="permutation-1-2" class="level3">
<h3 class="anchored" data-anchor-id="permutation-1-2">Permutation 1</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/true_chromosome_mapping.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>out_final<span class="sc">$</span>Proteome.ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer.*"</span>, <span class="st">""</span>, out_final<span class="sc">$</span>pathogen))</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>out_final_perm1 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/perm1_random_chromosome.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>out_final_perm1<span class="sc">$</span>Proteome.ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer.*"</span>, <span class="st">""</span>, out_final_perm1<span class="sc">$</span>pathogen))</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>reversed_results <span class="ot">&lt;-</span> out_final <span class="sc">%&gt;%</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mismatches, <span class="at">values_from =</span> count) <span class="sc">%&gt;%</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), <span class="at">names_to=</span><span class="st">"mismatches"</span>, <span class="at">values_to=</span><span class="st">"count"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(mismatches), <span class="dv">0</span>, mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(out_final_perm1 <span class="sc">%&gt;%</span></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mismatches, <span class="at">values_from =</span> count) <span class="sc">%&gt;%</span></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>( <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>( <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), <span class="at">names_to=</span><span class="st">"mismatches"</span>, <span class="at">values_to=</span><span class="st">"count"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(mismatches), <span class="dv">0</span>, mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>pathogen), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"chromosome_name"</span>,<span class="st">"Proteome.ID"</span>, <span class="st">"mismatches"</span>), <span class="at">suffix =</span> <span class="fu">c</span>(<span class="st">".true"</span>, <span class="st">".permuted"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count.true =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(count.true), <span class="dv">0</span>, count.true))<span class="sc">%&gt;%</span></span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count.permuted =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(count.permuted), <span class="dv">0</span>, count.permuted)) <span class="sc">%&gt;%</span></span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FC =</span> (count.true <span class="sc">+</span> <span class="dv">3</span>)<span class="sc">/</span>(count.permuted <span class="sc">+</span> <span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(chromosome_name, mismatches, Proteome.ID) <span class="sc">%&gt;%</span></span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">median =</span> <span class="fu">median</span>(FC))</span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>pd <span class="ot">&lt;-</span> <span class="fu">position_dodge</span>(<span class="fl">0.4</span>)</span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> out_final <span class="sc">%&gt;%</span></span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mismatches, <span class="at">values_from =</span> count) <span class="sc">%&gt;%</span></span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), <span class="at">names_to=</span><span class="st">"mismatches"</span>, <span class="at">values_to=</span><span class="st">"count"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(mismatches), <span class="dv">0</span>, mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb24-45"><a href="#cb24-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(out_final_perm1 <span class="sc">%&gt;%</span></span>
<span id="cb24-46"><a href="#cb24-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mismatches, <span class="at">values_from =</span> count) <span class="sc">%&gt;%</span></span>
<span id="cb24-47"><a href="#cb24-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>( <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb24-48"><a href="#cb24-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-49"><a href="#cb24-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb24-50"><a href="#cb24-50" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb24-51"><a href="#cb24-51" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb24-52"><a href="#cb24-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-53"><a href="#cb24-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>( <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), <span class="at">names_to=</span><span class="st">"mismatches"</span>, <span class="at">values_to=</span><span class="st">"count"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-54"><a href="#cb24-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(mismatches), <span class="dv">0</span>, mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb24-55"><a href="#cb24-55" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>pathogen), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"chromosome_name"</span>,<span class="st">"Proteome.ID"</span>, <span class="st">"mismatches"</span>), <span class="at">suffix =</span> <span class="fu">c</span>(<span class="st">".true"</span>, <span class="st">".permuted"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-56"><a href="#cb24-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count.true =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(count.true), <span class="dv">0</span>, count.true))<span class="sc">%&gt;%</span></span>
<span id="cb24-57"><a href="#cb24-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count.permuted =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(count.permuted), <span class="dv">0</span>, count.permuted)) <span class="sc">%&gt;%</span></span>
<span id="cb24-58"><a href="#cb24-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FC =</span> (count.true <span class="sc">+</span> <span class="dv">3</span>)<span class="sc">/</span>(count.permuted <span class="sc">+</span> <span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-59"><a href="#cb24-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-60"><a href="#cb24-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Family_Viral_Zone <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2)) <span class="sc">%&gt;%</span></span>
<span id="cb24-61"><a href="#cb24-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Proteome.ID, Family_Viral_Zone, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb24-62"><a href="#cb24-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(chromosome_name <span class="sc">!=</span> <span class="st">"Mitochondrion"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-63"><a href="#cb24-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">paste0</span>(<span class="st">"≤"</span>, mismatches, <span class="st">" mismatches"</span>)) </span>
<span id="cb24-64"><a href="#cb24-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-65"><a href="#cb24-65" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(testing_data <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Proteome.ID, Pathogen, chromosome_name, mismatches, FC), <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS6a_datapoints.csv"</span>))</span>
<span id="cb24-66"><a href="#cb24-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-67"><a href="#cb24-67" aria-hidden="true" tabindex="-1"></a>supp_data <span class="ot">&lt;-</span> testing_data <span class="sc">%&gt;%</span></span>
<span id="cb24-68"><a href="#cb24-68" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(!(count.true == 0 &amp; count.permuted == 0)) %&gt;%</span></span>
<span id="cb24-69"><a href="#cb24-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(chromosome_name, FC, pathogen, Proteome.ID, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb24-70"><a href="#cb24-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> chromosome_name, <span class="at">values_from =</span> FC, <span class="at">id_cols =</span> <span class="fu">c</span>(pathogen, Proteome.ID, mismatches))</span>
<span id="cb24-71"><a href="#cb24-71" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> supp_data <span class="sc">%&gt;%</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(Auto, X, Y)), value <span class="sc">~</span> name, <span class="at">paired =</span> <span class="cn">TRUE</span>,</span>
<span id="cb24-72"><a href="#cb24-72" aria-hidden="true" tabindex="-1"></a>                       <span class="at">group.by =</span> <span class="st">"mismatches"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-73"><a href="#cb24-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">y.position =</span> <span class="fu">c</span>(<span class="fl">2.75</span>, <span class="fl">2.5</span>, <span class="fl">2.25</span>, <span class="fl">2.75</span>, <span class="fl">2.5</span>, <span class="fl">2.25</span>,<span class="fl">2.75</span>, <span class="fl">2.5</span>, <span class="fl">2.25</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-74"><a href="#cb24-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_significance</span>(<span class="st">"p.adj"</span>)</span>
<span id="cb24-75"><a href="#cb24-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-76"><a href="#cb24-76" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(supp_data, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData6_part1.xlsx"</span>),</span>
<span id="cb24-77"><a href="#cb24-77" aria-hidden="true" tabindex="-1"></a>                  <span class="at">sheetName =</span> <span class="st">"Perm1_Chr_Data"</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-78"><a href="#cb24-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-79"><a href="#cb24-79" aria-hidden="true" tabindex="-1"></a><span class="do">## sort dataframe by group then by x</span></span>
<span id="cb24-80"><a href="#cb24-80" aria-hidden="true" tabindex="-1"></a>ggS6A <span class="ot">&lt;-</span> reversed_results <span class="sc">%&gt;%</span></span>
<span id="cb24-81"><a href="#cb24-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-82"><a href="#cb24-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Family_Viral_Zone <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2)) <span class="sc">%&gt;%</span></span>
<span id="cb24-83"><a href="#cb24-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Proteome.ID, Family_Viral_Zone, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb24-84"><a href="#cb24-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(chromosome_name <span class="sc">!=</span> <span class="st">"Mitochondrion"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-85"><a href="#cb24-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">paste0</span>(<span class="st">"≤"</span>, mismatches, <span class="st">" mismatches"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-86"><a href="#cb24-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>chromosome_name, <span class="at">y=</span>median, <span class="at">group =</span> Proteome.ID)) <span class="sc">+</span></span>
<span id="cb24-87"><a href="#cb24-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>,<span class="at">position =</span> pd) <span class="sc">+</span></span>
<span id="cb24-88"><a href="#cb24-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="fu">aes</span>(<span class="at">color =</span> Family_Viral_Zone), <span class="at">position =</span> pd) <span class="sc">+</span></span>
<span id="cb24-89"><a href="#cb24-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> Family_palette_v2) <span class="sc">+</span></span>
<span id="cb24-90"><a href="#cb24-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb24-91"><a href="#cb24-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>mismatches,<span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb24-92"><a href="#cb24-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log10_trans</span>(),</span>
<span id="cb24-93"><a href="#cb24-93" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log10"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">10</span><span class="sc">^</span>x, <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb24-94"><a href="#cb24-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb24-95"><a href="#cb24-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"FC of 12mers compared</span><span class="sc">\n</span><span class="st">to Perm 1"</span>) <span class="sc">+</span></span>
<span id="cb24-96"><a href="#cb24-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Chromosome"</span>) <span class="sc">+</span></span>
<span id="cb24-97"><a href="#cb24-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_pvalue_manual</span>(check,<span class="at">label =</span> <span class="st">"p.adj.signif"</span>) <span class="sc">+</span></span>
<span id="cb24-98"><a href="#cb24-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb24-99"><a href="#cb24-99" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.height=</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">'cm'</span>),</span>
<span id="cb24-100"><a href="#cb24-100" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.width=</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">'cm'</span>),</span>
<span id="cb24-101"><a href="#cb24-101" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">fill=</span><span class="cn">NA</span>, <span class="at">size=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb24-102"><a href="#cb24-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">0.316</span>,<span class="dv">800</span>))</span>
<span id="cb24-103"><a href="#cb24-103" aria-hidden="true" tabindex="-1"></a>ggS6A</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure4and5_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"FigureS6A.pdf"</span>), <span class="at">width =</span><span class="fl">5.5</span>, <span class="at">height =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="permutation-3-2" class="level3">
<h3 class="anchored" data-anchor-id="permutation-3-2">Permutation 3</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/true_chromosome_mapping.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>out_final<span class="sc">$</span>Proteome.ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer.*"</span>, <span class="st">""</span>, out_final<span class="sc">$</span>pathogen))</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>out_final_perm3 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/perm3_class_shuffle_chromosome.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>out_final_perm3<span class="sc">$</span>Proteome.ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer.*"</span>, <span class="st">""</span>, out_final_perm3<span class="sc">$</span>pathogen))</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>reversed_results <span class="ot">&lt;-</span> out_final <span class="sc">%&gt;%</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mismatches, <span class="at">values_from =</span> count) <span class="sc">%&gt;%</span></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), <span class="at">names_to=</span><span class="st">"mismatches"</span>, <span class="at">values_to=</span><span class="st">"count"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(mismatches), <span class="dv">0</span>, mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(out_final_perm3 <span class="sc">%&gt;%</span></span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mismatches, <span class="at">values_from =</span> count) <span class="sc">%&gt;%</span></span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>( <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>( <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), <span class="at">names_to=</span><span class="st">"mismatches"</span>, <span class="at">values_to=</span><span class="st">"count"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(mismatches), <span class="dv">0</span>, mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>pathogen), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"chromosome_name"</span>,<span class="st">"Proteome.ID"</span>, <span class="st">"mismatches"</span>), <span class="at">suffix =</span> <span class="fu">c</span>(<span class="st">".true"</span>, <span class="st">".permuted"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count.true =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(count.true), <span class="dv">0</span>, count.true))<span class="sc">%&gt;%</span></span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count.permuted =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(count.permuted), <span class="dv">0</span>, count.permuted)) <span class="sc">%&gt;%</span></span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FC =</span> (count.true <span class="sc">+</span> <span class="dv">3</span>)<span class="sc">/</span>(count.permuted <span class="sc">+</span> <span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(chromosome_name, mismatches, Proteome.ID) <span class="sc">%&gt;%</span></span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">median =</span> <span class="fu">median</span>(FC))</span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a>pd <span class="ot">&lt;-</span> <span class="fu">position_dodge</span>(<span class="fl">0.4</span>)</span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> out_final <span class="sc">%&gt;%</span></span>
<span id="cb26-36"><a href="#cb26-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mismatches, <span class="at">values_from =</span> count) <span class="sc">%&gt;%</span></span>
<span id="cb26-37"><a href="#cb26-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb26-38"><a href="#cb26-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-39"><a href="#cb26-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb26-40"><a href="#cb26-40" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb26-41"><a href="#cb26-41" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb26-42"><a href="#cb26-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-43"><a href="#cb26-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), <span class="at">names_to=</span><span class="st">"mismatches"</span>, <span class="at">values_to=</span><span class="st">"count"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-44"><a href="#cb26-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(mismatches), <span class="dv">0</span>, mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb26-45"><a href="#cb26-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(out_final_perm3 <span class="sc">%&gt;%</span></span>
<span id="cb26-46"><a href="#cb26-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mismatches, <span class="at">values_from =</span> count) <span class="sc">%&gt;%</span></span>
<span id="cb26-47"><a href="#cb26-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>( <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb26-48"><a href="#cb26-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-49"><a href="#cb26-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb26-50"><a href="#cb26-50" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb26-51"><a href="#cb26-51" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>( <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb26-52"><a href="#cb26-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-53"><a href="#cb26-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>( <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), <span class="at">names_to=</span><span class="st">"mismatches"</span>, <span class="at">values_to=</span><span class="st">"count"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-54"><a href="#cb26-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(mismatches), <span class="dv">0</span>, mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb26-55"><a href="#cb26-55" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>pathogen), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"chromosome_name"</span>,<span class="st">"Proteome.ID"</span>, <span class="st">"mismatches"</span>), <span class="at">suffix =</span> <span class="fu">c</span>(<span class="st">".true"</span>, <span class="st">".permuted"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb26-56"><a href="#cb26-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count.true =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(count.true), <span class="dv">0</span>, count.true))<span class="sc">%&gt;%</span></span>
<span id="cb26-57"><a href="#cb26-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">count.permuted =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(count.permuted), <span class="dv">0</span>, count.permuted)) <span class="sc">%&gt;%</span></span>
<span id="cb26-58"><a href="#cb26-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FC =</span> (count.true <span class="sc">+</span> <span class="dv">3</span>)<span class="sc">/</span>(count.permuted <span class="sc">+</span> <span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb26-59"><a href="#cb26-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb26-60"><a href="#cb26-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Family_Viral_Zone <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2)) <span class="sc">%&gt;%</span></span>
<span id="cb26-61"><a href="#cb26-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Proteome.ID, Family_Viral_Zone, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb26-62"><a href="#cb26-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(chromosome_name <span class="sc">!=</span> <span class="st">"Mitochondrion"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-63"><a href="#cb26-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">paste0</span>(<span class="st">"≤"</span>, mismatches, <span class="st">" mismatches"</span>)) </span>
<span id="cb26-64"><a href="#cb26-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-65"><a href="#cb26-65" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(testing_data <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Proteome.ID, Pathogen, chromosome_name, mismatches, FC), <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS6b_datapoints.csv"</span>))</span>
<span id="cb26-66"><a href="#cb26-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-67"><a href="#cb26-67" aria-hidden="true" tabindex="-1"></a>supp_data <span class="ot">&lt;-</span> testing_data <span class="sc">%&gt;%</span></span>
<span id="cb26-68"><a href="#cb26-68" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(!(count.true == 0 &amp; count.permuted == 0)) %&gt;%</span></span>
<span id="cb26-69"><a href="#cb26-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(chromosome_name, FC, pathogen, Proteome.ID, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb26-70"><a href="#cb26-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> chromosome_name, <span class="at">values_from =</span> FC, <span class="at">id_cols =</span> <span class="fu">c</span>(pathogen, Proteome.ID, mismatches))</span>
<span id="cb26-71"><a href="#cb26-71" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> supp_data <span class="sc">%&gt;%</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(Auto, X, Y)), value <span class="sc">~</span> name, <span class="at">paired =</span> <span class="cn">TRUE</span>,</span>
<span id="cb26-72"><a href="#cb26-72" aria-hidden="true" tabindex="-1"></a>                       <span class="at">group.by =</span> <span class="st">"mismatches"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-73"><a href="#cb26-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">y.position =</span> <span class="fu">c</span>(<span class="fl">2.75</span>, <span class="fl">2.5</span>, <span class="fl">2.25</span>, <span class="fl">2.75</span>, <span class="fl">2.5</span>, <span class="fl">2.25</span>,<span class="fl">2.75</span>, <span class="fl">2.5</span>, <span class="fl">2.25</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb26-74"><a href="#cb26-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_significance</span>(<span class="st">"p.adj"</span>)</span>
<span id="cb26-75"><a href="#cb26-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-76"><a href="#cb26-76" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(supp_data, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData6_part1.xlsx"</span>),</span>
<span id="cb26-77"><a href="#cb26-77" aria-hidden="true" tabindex="-1"></a>                 <span class="at">sheetName =</span> <span class="st">"Perm3_Chr_Data"</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb26-78"><a href="#cb26-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-79"><a href="#cb26-79" aria-hidden="true" tabindex="-1"></a><span class="do">## sort dataframe by group then by x</span></span>
<span id="cb26-80"><a href="#cb26-80" aria-hidden="true" tabindex="-1"></a>ggS6B <span class="ot">&lt;-</span> reversed_results <span class="sc">%&gt;%</span></span>
<span id="cb26-81"><a href="#cb26-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb26-82"><a href="#cb26-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Family_Viral_Zone <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2)) <span class="sc">%&gt;%</span></span>
<span id="cb26-83"><a href="#cb26-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Proteome.ID, Family_Viral_Zone, chromosome_name) <span class="sc">%&gt;%</span></span>
<span id="cb26-84"><a href="#cb26-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(chromosome_name <span class="sc">!=</span> <span class="st">"Mitochondrion"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-85"><a href="#cb26-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">paste0</span>(<span class="st">"≤"</span>, mismatches, <span class="st">" mismatches"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb26-86"><a href="#cb26-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>chromosome_name, <span class="at">y=</span>median, <span class="at">group =</span> Proteome.ID)) <span class="sc">+</span></span>
<span id="cb26-87"><a href="#cb26-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>,<span class="at">position =</span> pd) <span class="sc">+</span></span>
<span id="cb26-88"><a href="#cb26-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="fu">aes</span>(<span class="at">color =</span> Family_Viral_Zone), <span class="at">position =</span> pd) <span class="sc">+</span></span>
<span id="cb26-89"><a href="#cb26-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> Family_palette_v2) <span class="sc">+</span></span>
<span id="cb26-90"><a href="#cb26-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb26-91"><a href="#cb26-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>mismatches,<span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb26-92"><a href="#cb26-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log10_trans</span>(),</span>
<span id="cb26-93"><a href="#cb26-93" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log10"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">10</span><span class="sc">^</span>x, <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb26-94"><a href="#cb26-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb26-95"><a href="#cb26-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"FC of 12mers compared</span><span class="sc">\n</span><span class="st">to Perm 3"</span>) <span class="sc">+</span></span>
<span id="cb26-96"><a href="#cb26-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Chromosome"</span>) <span class="sc">+</span></span>
<span id="cb26-97"><a href="#cb26-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_pvalue_manual</span>(check,<span class="at">label =</span> <span class="st">"p.adj.signif"</span>) <span class="sc">+</span></span>
<span id="cb26-98"><a href="#cb26-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb26-99"><a href="#cb26-99" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.height=</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">'cm'</span>),</span>
<span id="cb26-100"><a href="#cb26-100" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.width=</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">'cm'</span>),</span>
<span id="cb26-101"><a href="#cb26-101" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">fill=</span><span class="cn">NA</span>, <span class="at">size=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb26-102"><a href="#cb26-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">0.316</span>,<span class="dv">800</span>))</span>
<span id="cb26-103"><a href="#cb26-103" aria-hidden="true" tabindex="-1"></a>ggS6B</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure4and5_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"FigureS6B.pdf"</span>), <span class="at">width =</span><span class="fl">5.5</span>, <span class="at">height =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggarrange</span>(ggS6A, ggS6B, <span class="at">align =</span> <span class="st">"hv"</span>, <span class="at">nrow =</span> <span class="dv">1</span>, <span class="at">ncol =</span><span class="dv">2</span>, <span class="at">common.legend =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure4and5_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"FigureS6.pdf"</span>), <span class="at">width =</span><span class="dv">8</span>, <span class="at">height =</span> <span class="dv">6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>