<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>FigureS5</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="FigureS5_files/libs/clipboard/clipboard.min.js"></script>
<script src="FigureS5_files/libs/quarto-html/quarto.js"></script>
<script src="FigureS5_files/libs/quarto-html/popper.min.js"></script>
<script src="FigureS5_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="FigureS5_files/libs/quarto-html/anchor.min.js"></script>
<link href="FigureS5_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="FigureS5_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="FigureS5_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="FigureS5_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="FigureS5_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">FigureS5</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="figure-s5" class="level1">
<h1>Figure S5</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(seqinr)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringi)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(R.utils)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(treeio)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggtree)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(aplot)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>out_dir <span class="ot">&lt;-</span> <span class="st">"../output/"</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../../Utilities.R"</span>)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>meta_path <span class="ot">&lt;-</span> <span class="st">"../../Metadata/"</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"Final_Cohort_Metadata_v5.xlsx"</span>), <span class="at">sheet =</span> <span class="dv">1</span>)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>duplicated_proteins <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"duplicated_proteins.csv"</span>))</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>human_duplicated_proteins <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"human_duplicated_or_removed_proteins.csv"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="dv">1</span>)</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>tissue_map <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"normal_tissue.tsv"</span>))</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>organ_map <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"Tissue_Groupings.xlsx"</span>))</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> <span class="fu">merge</span>(tissue_map, organ_map, <span class="at">by.x =</span> <span class="st">"Tissue"</span>, <span class="at">by.y =</span> <span class="st">"Var1"</span>, <span class="at">all.x =</span> T)</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(tissue_organ_map<span class="sc">$</span>Level)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Not detected"       "High"               "Low"               
[4] "Medium"             "Ascending"          "Descending"        
[7] "Not representative" "N/A"               </code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Level <span class="sc">!=</span> <span class="st">"Not detected"</span>,]</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Level <span class="sc">!=</span> <span class="st">"Not representative"</span>,]</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Level <span class="sc">!=</span> <span class="st">"N/A"</span>,]</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Reliability <span class="sc">!=</span> <span class="st">"Uncertain"</span>,]</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Reliability <span class="sc">!=</span> <span class="st">"NA"</span>,]</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>tissue_organ_map_condensed <span class="ot">&lt;-</span> tissue_organ_map <span class="sc">%&gt;%</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Tissue, Gene, <span class="st">`</span><span class="at">Cell type</span><span class="st">`</span>, <span class="st">`</span><span class="at">New Name</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Gene) <span class="sc">%&gt;%</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">organ_system =</span> <span class="fu">gsub</span>(<span class="st">"NA;"</span>, <span class="st">""</span>, <span class="fu">paste</span>(<span class="fu">unique</span>(<span class="st">`</span><span class="at">New Name</span><span class="st">`</span>), <span class="at">collapse =</span> <span class="st">";"</span>)),</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>                   <span class="at">tissue =</span> <span class="fu">gsub</span>(<span class="st">"NA;"</span>, <span class="st">""</span>, <span class="fu">paste</span>(<span class="fu">unique</span>(Tissue), <span class="at">collapse =</span> <span class="st">";"</span>)),</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>                   <span class="at">cell_type =</span> <span class="fu">gsub</span>(<span class="st">"NA;"</span>, <span class="st">""</span>, <span class="fu">paste</span>(<span class="fu">unique</span>(<span class="st">`</span><span class="at">Cell type</span><span class="st">`</span>), <span class="at">collapse =</span> <span class="st">";"</span>)))</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>protein_atlas <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"../../Databases/proteinatlas_2023-10-15.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="cell-type-enrichment" class="level2">
<h2 class="anchored" data-anchor-id="cell-type-enrichment">Cell Type Enrichment</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../12mer_annotated/"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".csv.gz"</span>, files_to_read)]</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="do">## For testing purposes use this subset</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>protein_atlas_celltypes <span class="ot">&lt;-</span> protein_atlas <span class="sc">%&gt;%</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Uniprot, <span class="st">`</span><span class="at">RNA single cell type specific nTPM</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_longer_delim</span>(<span class="st">`</span><span class="at">RNA single cell type specific nTPM</span><span class="st">`</span>, <span class="st">";"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">RNA single cell type specific nTPM</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">":.*"</span>, <span class="st">""</span>, <span class="st">`</span><span class="at">RNA single cell type specific nTPM</span><span class="st">`</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"cell_type"</span> <span class="ot">=</span> <span class="st">`</span><span class="at">RNA single cell type specific nTPM</span><span class="st">`</span>,</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>                <span class="st">"uniprot"</span> <span class="ot">=</span> <span class="st">"Uniprot"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(cell_type))</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>output_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">cell_type =</span> <span class="fu">unique</span>(protein_atlas_celltypes<span class="sc">$</span>cell_type))</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file))</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">nThread =</span> <span class="dv">16</span>) </span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(protein_atlas_celltypes, <span class="at">by =</span> <span class="st">"uniprot"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>     <span class="fu">distinct</span>(query, query_seqname, query_start, cell_type) </span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(active_data<span class="sc">$</span>cell_type)) <span class="sc">==</span> <span class="fu">nrow</span>(active_data)){</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>  name_replacement <span class="ot">&lt;-</span> <span class="st">"Freq"</span>; <span class="fu">names</span>(name_replacement) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_12mer.*"</span>, <span class="st">""</span>, active_file)</span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a> output_df <span class="ot">&lt;-</span> output_df <span class="sc">%&gt;%</span></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>   <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(<span class="fu">table</span>(active_data<span class="sc">$</span>cell_type)) <span class="sc">%&gt;%</span></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>   dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"cell_type"</span> <span class="ot">=</span> <span class="st">"Var1"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>     dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="sc">!!!</span> name_replacement))</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Finished with"</span>, active_file))</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Starting Adenoassociated_virus_UP000180764_12mer_3mis_hits.csv.gz"
[1] "Finished withAdenoassociated_virus_UP000180764_12mer_3mis_hits.csv.gz"
[1] "Starting Aichi_virus_UP000119554_12mer_3mis_hits.csv.gz"
[1] "Finished withAichi_virus_UP000119554_12mer_3mis_hits.csv.gz"
[1] "Starting Australian_bat_lyssavirus_UP000006934_12mer_3mis_hits.csv.gz"
[1] "Finished withAustralian_bat_lyssavirus_UP000006934_12mer_3mis_hits.csv.gz"
[1] "Starting Banna_virus_UP000000832_12mer_3mis_hits.csv.gz"
[1] "Finished withBanna_virus_UP000000832_12mer_3mis_hits.csv.gz"
[1] "Starting Barmah_forest_virus_UP000007609_12mer_3mis_hits.csv.gz"
[1] "Finished withBarmah_forest_virus_UP000007609_12mer_3mis_hits.csv.gz"
[1] "Starting BK_polyomavirus_UP000145981_12mer_3mis_hits.csv.gz"
[1] "Finished withBK_polyomavirus_UP000145981_12mer_3mis_hits.csv.gz"
[1] "Starting Bunyamwera_virus_UP000002476_12mer_3mis_hits.csv.gz"
[1] "Finished withBunyamwera_virus_UP000002476_12mer_3mis_hits.csv.gz"
[1] "Starting Bunyavirus_La_Crosse_UP000008768_12mer_3mis_hits.csv.gz"
[1] "Finished withBunyavirus_La_Crosse_UP000008768_12mer_3mis_hits.csv.gz"
[1] "Starting Bunyavirus_snowshoe_hare_UP000150053_12mer_3mis_hits.csv.gz"
[1] "Finished withBunyavirus_snowshoe_hare_UP000150053_12mer_3mis_hits.csv.gz"
[1] "Starting Cercopithecine_alphaherpesvirus_UP000137988_12mer_3mis_hits.csv.gz"
[1] "Finished withCercopithecine_alphaherpesvirus_UP000137988_12mer_3mis_hits.csv.gz"
[1] "Starting Chandipura_virus_UP000008448_12mer_3mis_hits.csv.gz"
[1] "Finished withChandipura_virus_UP000008448_12mer_3mis_hits.csv.gz"
[1] "Starting Chikungunya_virus_UP000000569_12mer_3mis_hits.csv.gz"
[1] "Finished withChikungunya_virus_UP000000569_12mer_3mis_hits.csv.gz"
[1] "Starting Cosavirus_A_UP000029753_12mer_3mis_hits.csv.gz"
[1] "Finished withCosavirus_A_UP000029753_12mer_3mis_hits.csv.gz"
[1] "Starting Cowpox_virus_UP000152733_12mer_3mis_hits.csv.gz"
[1] "Finished withCowpox_virus_UP000152733_12mer_3mis_hits.csv.gz"
[1] "Starting Coxsackievirus_UP000007757_12mer_3mis_hits.csv.gz"
[1] "Finished withCoxsackievirus_UP000007757_12mer_3mis_hits.csv.gz"
[1] "Starting Crimean-Congo_hemorrhagic_fever_virus_UP000008767_12mer_3mis_hits.csv.gz"
[1] "Finished withCrimean-Congo_hemorrhagic_fever_virus_UP000008767_12mer_3mis_hits.csv.gz"
[1] "Starting Dengue_virus_UP000002500_12mer_3mis_hits.csv.gz"
[1] "Finished withDengue_virus_UP000002500_12mer_3mis_hits.csv.gz"
[1] "Starting Dhori_virus_UP000181622_12mer_3mis_hits.csv.gz"
[1] "Finished withDhori_virus_UP000181622_12mer_3mis_hits.csv.gz"
[1] "Starting Dugbe_virus_UP000000278_12mer_3mis_hits.csv.gz"
[1] "Finished withDugbe_virus_UP000000278_12mer_3mis_hits.csv.gz"
[1] "Starting Duvenhage_virus_UP000117552_12mer_3mis_hits.csv.gz"
[1] "Finished withDuvenhage_virus_UP000117552_12mer_3mis_hits.csv.gz"
[1] "Starting Eastern_equine_encephalitis_virus_UP000110644_12mer_3mis_hits.csv.gz"
[1] "Finished withEastern_equine_encephalitis_virus_UP000110644_12mer_3mis_hits.csv.gz"
[1] "Starting Ebolavirus_UP000007209_12mer_3mis_hits.csv.gz"
[1] "Finished withEbolavirus_UP000007209_12mer_3mis_hits.csv.gz"
[1] "Starting Echovirus_UP000000670_12mer_3mis_hits.csv.gz"
[1] "Finished withEchovirus_UP000000670_12mer_3mis_hits.csv.gz"
[1] "Starting Encephalomyocarditis_virus_UP000002319_12mer_3mis_hits.csv.gz"
[1] "Finished withEncephalomyocarditis_virus_UP000002319_12mer_3mis_hits.csv.gz"
[1] "Starting Epstein-Barr_virus_UP000153037_12mer_3mis_hits.csv.gz"
[1] "Finished withEpstein-Barr_virus_UP000153037_12mer_3mis_hits.csv.gz"
[1] "Starting European_bat_lyssavirus_UP000008926_12mer_3mis_hits.csv.gz"
[1] "Finished withEuropean_bat_lyssavirus_UP000008926_12mer_3mis_hits.csv.gz"
[1] "Starting Hantaan_virus_UP000008627_12mer_3mis_hits.csv.gz"
[1] "Finished withHantaan_virus_UP000008627_12mer_3mis_hits.csv.gz"
[1] "Starting Hendra_virus_UP000008771_12mer_3mis_hits.csv.gz"
[1] "Finished withHendra_virus_UP000008771_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_A_virus_UP000006724_12mer_3mis_hits.csv.gz"
[1] "Finished withHepatitis_A_virus_UP000006724_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_B_virus_UP000008591_12mer_3mis_hits.csv.gz"
[1] "Finished withHepatitis_B_virus_UP000008591_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_C_virus_UP000000518_12mer_3mis_hits.csv.gz"
[1] "Finished withHepatitis_C_virus_UP000000518_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_delta_virus_UP000006675_12mer_3mis_hits.csv.gz"
[1] "Finished withHepatitis_delta_virus_UP000006675_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_E_virus_UP000006705_12mer_3mis_hits.csv.gz"
[1] "Finished withHepatitis_E_virus_UP000006705_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_G_virus_UP000007407_12mer_3mis_hits.csv.gz"
[1] "Finished withHepatitis_G_virus_UP000007407_12mer_3mis_hits.csv.gz"
[1] "Starting Horsepox_virus_UP000111173_12mer_3mis_hits.csv.gz"
[1] "Finished withHorsepox_virus_UP000111173_12mer_3mis_hits.csv.gz"
[1] "Starting Human_adenovirus_UP000008167_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_adenovirus_UP000008167_12mer_3mis_hits.csv.gz"
[1] "Starting Human_astrovirus_UP000001650_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_astrovirus_UP000001650_12mer_3mis_hits.csv.gz"
[1] "Starting Human_coronavirus_UP000006716_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_coronavirus_UP000006716_12mer_3mis_hits.csv.gz"
[1] "Starting Human_cytomegalovirus_merlin_UP000000938_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_cytomegalovirus_merlin_UP000000938_12mer_3mis_hits.csv.gz"
[1] "Starting Human_enterovirus_68_UP000143570_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_enterovirus_68_UP000143570_12mer_3mis_hits.csv.gz"
[1] "Starting Human_enterovirus_70_UP000008683_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_enterovirus_70_UP000008683_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus_1_UP000009294_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_herpesvirus_1_UP000009294_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus_2_UP000001874_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_herpesvirus_2_UP000001874_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus_6_UP000009295_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_herpesvirus_6_UP000009295_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus_7_UP000009246_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_herpesvirus_7_UP000009246_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus_8_UP000000942_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_herpesvirus_8_UP000000942_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus6b_UP000006930_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_herpesvirus6b_UP000006930_12mer_3mis_hits.csv.gz"
[1] "Starting Human_immunodeficiency_virus_1_UP000002241_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_immunodeficiency_virus_1_UP000002241_12mer_3mis_hits.csv.gz"
[1] "Starting Human_immunodeficiency_virus_2_UP000002242_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_immunodeficiency_virus_2_UP000002242_12mer_3mis_hits.csv.gz"
[1] "Starting Human_papillomavirus_1_UP000006372_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_papillomavirus_1_UP000006372_12mer_3mis_hits.csv.gz"
[1] "Starting Human_papillomavirus_2_UP000007710_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_papillomavirus_2_UP000007710_12mer_3mis_hits.csv.gz"
[1] "Starting Human_papillomavirus_type_16_UP000009251_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_papillomavirus_type_16_UP000009251_12mer_3mis_hits.csv.gz"
[1] "Starting Human_papillomavirus_type_18_UP000009109_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_papillomavirus_type_18_UP000009109_12mer_3mis_hits.csv.gz"
[1] "Starting Human_parainfluenza_UP000109776_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_parainfluenza_UP000109776_12mer_3mis_hits.csv.gz"
[1] "Starting Human_parvovirus_B19_UP000006624_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_parvovirus_B19_UP000006624_12mer_3mis_hits.csv.gz"
[1] "Starting Human_respiratory_syncytial_virus_UP000002472_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_respiratory_syncytial_virus_UP000002472_12mer_3mis_hits.csv.gz"
[1] "Starting Human_rhinovirus_UP000007070_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_rhinovirus_UP000007070_12mer_3mis_hits.csv.gz"
[1] "Starting Human_SARS_coronavirus_UP000000354_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_SARS_coronavirus_UP000000354_12mer_3mis_hits.csv.gz"
[1] "Starting Human_spumaretrovirus_UP000138352_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_spumaretrovirus_UP000138352_12mer_3mis_hits.csv.gz"
[1] "Starting Human_T-lymphotropic_virus_1_UP000001061_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_T-lymphotropic_virus_1_UP000001061_12mer_3mis_hits.csv.gz"
[1] "Starting Human_torovirus_UP000243571_12mer_3mis_hits.csv.gz"
[1] "Starting Influenza_A_virus_UP000009255_12mer_3mis_hits.csv.gz"
[1] "Finished withInfluenza_A_virus_UP000009255_12mer_3mis_hits.csv.gz"
[1] "Starting Influenza_B_virus_UP000008158_12mer_3mis_hits.csv.gz"
[1] "Finished withInfluenza_B_virus_UP000008158_12mer_3mis_hits.csv.gz"
[1] "Starting Influenza_C_virus_UP000008286_12mer_3mis_hits.csv.gz"
[1] "Finished withInfluenza_C_virus_UP000008286_12mer_3mis_hits.csv.gz"
[1] "Starting Isfahan_virus_UP000204017_12mer_3mis_hits.csv.gz"
[1] "Finished withIsfahan_virus_UP000204017_12mer_3mis_hits.csv.gz"
[1] "Starting Japanese_encephalitis_virus_UP000007214_12mer_3mis_hits.csv.gz"
[1] "Finished withJapanese_encephalitis_virus_UP000007214_12mer_3mis_hits.csv.gz"
[1] "Starting JC_polyomavirus_UP000008478_12mer_3mis_hits.csv.gz"
[1] "Finished withJC_polyomavirus_UP000008478_12mer_3mis_hits.csv.gz"
[1] "Starting Junin_mammarenavirus_UP000127886_12mer_3mis_hits.csv.gz"
[1] "Finished withJunin_mammarenavirus_UP000127886_12mer_3mis_hits.csv.gz"
[1] "Starting KI_Polyomavirus_UP000107189_12mer_3mis_hits.csv.gz"
[1] "Finished withKI_Polyomavirus_UP000107189_12mer_3mis_hits.csv.gz"
[1] "Starting Kunjin_virus_UP000138183_12mer_3mis_hits.csv.gz"
[1] "Finished withKunjin_virus_UP000138183_12mer_3mis_hits.csv.gz"
[1] "Starting Lagos_bat_virus_UP000161306_12mer_3mis_hits.csv.gz"
[1] "Finished withLagos_bat_virus_UP000161306_12mer_3mis_hits.csv.gz"
[1] "Starting Lake_Victoria_marburgvirus_UP000180448_12mer_3mis_hits.csv.gz"
[1] "Finished withLake_Victoria_marburgvirus_UP000180448_12mer_3mis_hits.csv.gz"
[1] "Starting Langat_virus_UP000180575_12mer_3mis_hits.csv.gz"
[1] "Finished withLangat_virus_UP000180575_12mer_3mis_hits.csv.gz"
[1] "Starting Lassa_virus_UP000002473_12mer_3mis_hits.csv.gz"
[1] "Finished withLassa_virus_UP000002473_12mer_3mis_hits.csv.gz"
[1] "Starting Lordsdale_virus_UP000007767_12mer_3mis_hits.csv.gz"
[1] "Finished withLordsdale_virus_UP000007767_12mer_3mis_hits.csv.gz"
[1] "Starting Louping_ill_virus_UP000007644_12mer_3mis_hits.csv.gz"
[1] "Finished withLouping_ill_virus_UP000007644_12mer_3mis_hits.csv.gz"
[1] "Starting Lymphocytic_choriomeningitis_virus_UP000002474_12mer_3mis_hits.csv.gz"
[1] "Finished withLymphocytic_choriomeningitis_virus_UP000002474_12mer_3mis_hits.csv.gz"
[1] "Starting Machupo_virus_UP000233900_12mer_3mis_hits.csv.gz"
[1] "Finished withMachupo_virus_UP000233900_12mer_3mis_hits.csv.gz"
[1] "Starting Mayaro_virus_UP000007774_12mer_3mis_hits.csv.gz"
[1] "Finished withMayaro_virus_UP000007774_12mer_3mis_hits.csv.gz"
[1] "Starting Measles_virus_UP000008699_12mer_3mis_hits.csv.gz"
[1] "Finished withMeasles_virus_UP000008699_12mer_3mis_hits.csv.gz"
[1] "Starting Mengo_encephalomyocarditis_virus_UP000008663_12mer_3mis_hits.csv.gz"
[1] "Finished withMengo_encephalomyocarditis_virus_UP000008663_12mer_3mis_hits.csv.gz"
[1] "Starting Merkel_cell_polyomavirus_UP000154903_12mer_3mis_hits.csv.gz"
[1] "Finished withMerkel_cell_polyomavirus_UP000154903_12mer_3mis_hits.csv.gz"
[1] "Starting MERS_coronavirus_UP000147828_12mer_3mis_hits.csv.gz"
[1] "Finished withMERS_coronavirus_UP000147828_12mer_3mis_hits.csv.gz"
[1] "Starting MERS_UP000171868_12mer_3mis_hits.csv.gz"
[1] "Finished withMERS_UP000171868_12mer_3mis_hits.csv.gz"
[1] "Starting Mokola_virus_UP000006826_12mer_3mis_hits.csv.gz"
[1] "Finished withMokola_virus_UP000006826_12mer_3mis_hits.csv.gz"
[1] "Starting Molluscum_contagiosum_virus_UP000000869_12mer_3mis_hits.csv.gz"
[1] "Finished withMolluscum_contagiosum_virus_UP000000869_12mer_3mis_hits.csv.gz"
[1] "Starting Monkeypox_virus_UP000101269_12mer_3mis_hits.csv.gz"
[1] "Finished withMonkeypox_virus_UP000101269_12mer_3mis_hits.csv.gz"
[1] "Starting Mumps_virus_UP000002331_12mer_3mis_hits.csv.gz"
[1] "Finished withMumps_virus_UP000002331_12mer_3mis_hits.csv.gz"
[1] "Starting Murray_valley_encephalitis_virus_UP000008863_12mer_3mis_hits.csv.gz"
[1] "Finished withMurray_valley_encephalitis_virus_UP000008863_12mer_3mis_hits.csv.gz"
[1] "Starting Nipah_virus_UP000170143_12mer_3mis_hits.csv.gz"
[1] "Finished withNipah_virus_UP000170143_12mer_3mis_hits.csv.gz"
[1] "Starting Norwalk_virus_UP000000826_12mer_3mis_hits.csv.gz"
[1] "Finished withNorwalk_virus_UP000000826_12mer_3mis_hits.csv.gz"
[1] "Starting Onyong_nyong_virus_UP000008868_12mer_3mis_hits.csv.gz"
[1] "Finished withOnyong_nyong_virus_UP000008868_12mer_3mis_hits.csv.gz"
[1] "Starting Orf_virus_UP000000870_12mer_3mis_hits.csv.gz"
[1] "Finished withOrf_virus_UP000000870_12mer_3mis_hits.csv.gz"
[1] "Starting Oropouche_virus_UP000117676_12mer_3mis_hits.csv.gz"
[1] "Finished withOropouche_virus_UP000117676_12mer_3mis_hits.csv.gz"
[1] "Starting Pichinde_mammarenavirus_UP000139415_12mer_3mis_hits.csv.gz"
[1] "Finished withPichinde_mammarenavirus_UP000139415_12mer_3mis_hits.csv.gz"
[1] "Starting Poliovirus_UP000000356_12mer_3mis_hits.csv.gz"
[1] "Finished withPoliovirus_UP000000356_12mer_3mis_hits.csv.gz"
[1] "Starting Punta_toro_phlebovirus_UP000214480_12mer_3mis_hits.csv.gz"
[1] "Finished withPunta_toro_phlebovirus_UP000214480_12mer_3mis_hits.csv.gz"
[1] "Starting Puumala_virus_UP000110237_12mer_3mis_hits.csv.gz"
[1] "Finished withPuumala_virus_UP000110237_12mer_3mis_hits.csv.gz"
[1] "Starting Rabies_virus_UP000008649_12mer_3mis_hits.csv.gz"
[1] "Finished withRabies_virus_UP000008649_12mer_3mis_hits.csv.gz"
[1] "Starting Rift_valley_fever_virus_UP000002477_12mer_3mis_hits.csv.gz"
[1] "Finished withRift_valley_fever_virus_UP000002477_12mer_3mis_hits.csv.gz"
[1] "Starting Rosavirus_A_UP000029743_12mer_3mis_hits.csv.gz"
[1] "Finished withRosavirus_A_UP000029743_12mer_3mis_hits.csv.gz"
[1] "Starting Ross_river_virus_UP000006579_12mer_3mis_hits.csv.gz"
[1] "Finished withRoss_river_virus_UP000006579_12mer_3mis_hits.csv.gz"
[1] "Starting Rotavirus_A_UP000001119_12mer_3mis_hits.csv.gz"
[1] "Finished withRotavirus_A_UP000001119_12mer_3mis_hits.csv.gz"
[1] "Starting Rotavirus_B_UP000114294_12mer_3mis_hits.csv.gz"
[1] "Finished withRotavirus_B_UP000114294_12mer_3mis_hits.csv.gz"
[1] "Starting Rotavirus_C_UP000007664_12mer_3mis_hits.csv.gz"
[1] "Finished withRotavirus_C_UP000007664_12mer_3mis_hits.csv.gz"
[1] "Starting Rubella_virus_UP000000571_12mer_3mis_hits.csv.gz"
[1] "Finished withRubella_virus_UP000000571_12mer_3mis_hits.csv.gz"
[1] "Starting Sagiyama_virus_UP000007138_12mer_3mis_hits.csv.gz"
[1] "Finished withSagiyama_virus_UP000007138_12mer_3mis_hits.csv.gz"
[1] "Starting Salivirus_A_UP000029742_12mer_3mis_hits.csv.gz"
[1] "Finished withSalivirus_A_UP000029742_12mer_3mis_hits.csv.gz"
[1] "Starting Sandfly_fever_sicilian_virus_UP000207632_12mer_3mis_hits.csv.gz"
[1] "Finished withSandfly_fever_sicilian_virus_UP000207632_12mer_3mis_hits.csv.gz"
[1] "Starting Sapporo_virus_UP000000667_12mer_3mis_hits.csv.gz"
[1] "Finished withSapporo_virus_UP000000667_12mer_3mis_hits.csv.gz"
[1] "Starting SARS_coronavirus_2_UP000464024_12mer_3mis_hits.csv.gz"
[1] "Finished withSARS_coronavirus_2_UP000464024_12mer_3mis_hits.csv.gz"
[1] "Starting Semliki_forest_virus_UP000000570_12mer_3mis_hits.csv.gz"
[1] "Finished withSemliki_forest_virus_UP000000570_12mer_3mis_hits.csv.gz"
[1] "Starting Seoul_virus_UP000207620_12mer_3mis_hits.csv.gz"
[1] "Finished withSeoul_virus_UP000207620_12mer_3mis_hits.csv.gz"
[1] "Starting Simian_foamy_virus_UP000001063_12mer_3mis_hits.csv.gz"
[1] "Finished withSimian_foamy_virus_UP000001063_12mer_3mis_hits.csv.gz"
[1] "Starting Simian_virus_5_UP000007232_12mer_3mis_hits.csv.gz"
[1] "Finished withSimian_virus_5_UP000007232_12mer_3mis_hits.csv.gz"
[1] "Starting Sindbis_virus_UP000006710_12mer_3mis_hits.csv.gz"
[1] "Finished withSindbis_virus_UP000006710_12mer_3mis_hits.csv.gz"
[1] "Starting Southampton_virus_UP000007226_12mer_3mis_hits.csv.gz"
[1] "Finished withSouthampton_virus_UP000007226_12mer_3mis_hits.csv.gz"
[1] "Starting St._louis_encephalitis_virus_UP000171556_12mer_3mis_hits.csv.gz"
[1] "Finished withSt._louis_encephalitis_virus_UP000171556_12mer_3mis_hits.csv.gz"
[1] "Starting Tick-borne_powassan_virus_UP000006848_12mer_3mis_hits.csv.gz"
[1] "Finished withTick-borne_powassan_virus_UP000006848_12mer_3mis_hits.csv.gz"
[1] "Starting Torque_teno_virus_UP000001448_12mer_3mis_hits.csv.gz"
[1] "Finished withTorque_teno_virus_UP000001448_12mer_3mis_hits.csv.gz"
[1] "Starting Toscana_virus_UP000142444_12mer_3mis_hits.csv.gz"
[1] "Finished withToscana_virus_UP000142444_12mer_3mis_hits.csv.gz"
[1] "Starting Uukuniemi_virus_UP000008595_12mer_3mis_hits.csv.gz"
[1] "Finished withUukuniemi_virus_UP000008595_12mer_3mis_hits.csv.gz"
[1] "Starting Vaccinia_virus_UP000000344_12mer_3mis_hits.csv.gz"
[1] "Finished withVaccinia_virus_UP000000344_12mer_3mis_hits.csv.gz"
[1] "Starting Varicella-zoster_virus_UP000002602_12mer_3mis_hits.csv.gz"
[1] "Finished withVaricella-zoster_virus_UP000002602_12mer_3mis_hits.csv.gz"
[1] "Starting Variola_virus_UP000002060_12mer_3mis_hits.csv.gz"
[1] "Finished withVariola_virus_UP000002060_12mer_3mis_hits.csv.gz"
[1] "Starting Venezuelan_equine_encephalitis_virus_UP000171441_12mer_3mis_hits.csv.gz"
[1] "Finished withVenezuelan_equine_encephalitis_virus_UP000171441_12mer_3mis_hits.csv.gz"
[1] "Starting Vesicular_stomatitis_virus_UP000002327_12mer_3mis_hits.csv.gz"
[1] "Finished withVesicular_stomatitis_virus_UP000002327_12mer_3mis_hits.csv.gz"
[1] "Starting West_Nile_virus_UP000136458_12mer_3mis_hits.csv.gz"
[1] "Finished withWest_Nile_virus_UP000136458_12mer_3mis_hits.csv.gz"
[1] "Starting Western_equine_encephalitis_virus_UP000128494_12mer_3mis_hits.csv.gz"
[1] "Finished withWestern_equine_encephalitis_virus_UP000128494_12mer_3mis_hits.csv.gz"
[1] "Starting WU_polyomavirus_UP000118542_12mer_3mis_hits.csv.gz"
[1] "Finished withWU_polyomavirus_UP000118542_12mer_3mis_hits.csv.gz"
[1] "Starting Yaba_monkey_tumor_virus_UP000008596_12mer_3mis_hits.csv.gz"
[1] "Finished withYaba_monkey_tumor_virus_UP000008596_12mer_3mis_hits.csv.gz"
[1] "Starting Yaba-like_disease_virus_UP000136581_12mer_3mis_hits.csv.gz"
[1] "Finished withYaba-like_disease_virus_UP000136581_12mer_3mis_hits.csv.gz"
[1] "Starting Yellow_fever_virus_UP000000360_12mer_3mis_hits.csv.gz"
[1] "Finished withYellow_fever_virus_UP000000360_12mer_3mis_hits.csv.gz"
[1] "Starting Zika_virus_UP000054557_12mer_3mis_hits.csv.gz"
[1] "Finished withZika_virus_UP000054557_12mer_3mis_hits.csv.gz"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>ggc <span class="ot">&lt;-</span> output_df <span class="sc">%&gt;%</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>cell_type) <span class="sc">%&gt;%</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">as.numeric</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(value), <span class="dv">0</span>, value)) <span class="sc">%&gt;%</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, cell_type) <span class="sc">%&gt;%</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">sum</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">total =</span> <span class="fu">sum</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">freq =</span> count <span class="sc">/</span> total <span class="sc">*</span><span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer.*"</span>, <span class="st">""</span>, name))) <span class="sc">%&gt;%</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>cell_type, <span class="at">y=</span>label)) <span class="sc">+</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">fill =</span> freq)) <span class="sc">+</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">colours =</span> <span class="fu">c</span>(<span class="st">"#045375"</span>, <span class="st">"#089099"</span>, <span class="st">"#7CCBA2"</span>, <span class="st">"#FCDE9C"</span>, <span class="st">"#FAB27B"</span>, <span class="st">"#E23333"</span>),</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>                       <span class="at">name =</span> <span class="st">"Cell Type (%)"</span>) <span class="sc">+</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Cell Type"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'name'. You can override using the
`.groups` argument.
Joining with `by = join_by(Proteome.ID)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>ggc</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FigureS5_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="organ-and-tissue-enrichment" class="level2">
<h2 class="anchored" data-anchor-id="organ-and-tissue-enrichment">Organ and Tissue Enrichment</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../12mer_annotated/"</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".csv.gz"</span>, files_to_read)]</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="do">## For testing purposes use this subset</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span><span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>Organ_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="st">"organ"</span> <span class="ot">=</span> <span class="fu">unique</span>(tissue_organ_map<span class="sc">$</span><span class="st">`</span><span class="at">New Name</span><span class="st">`</span>))</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>Tissue_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="st">"tissue"</span> <span class="ot">=</span> <span class="fu">unique</span>(tissue_organ_map<span class="sc">$</span>Tissue))</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file))</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file) </span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>    <span class="co">#dplyr::select(-organ_system, -cell_type, -tissue) %&gt;%</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(tissue_organ_map <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"ensembl_gene_id"</span> <span class="ot">=</span> <span class="st">"Gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>     <span class="co">#filter(mismatches == 3) %&gt;%</span></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>     <span class="fu">filter</span>(Level <span class="sc">!=</span> <span class="st">"Low"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>     <span class="fu">filter</span>(Level <span class="sc">!=</span> <span class="st">"NA"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>     <span class="fu">distinct</span>(query, query_seqname, query_start, Tissue, <span class="st">`</span><span class="at">New Name</span><span class="st">`</span>) </span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>  final_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(query, query_seqname, query_start, <span class="st">`</span><span class="at">New Name</span><span class="st">`</span>)</span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>  final_data2 <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(query, query_seqname, query_start, <span class="st">`</span><span class="at">Tissue</span><span class="st">`</span>)</span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a>  Organ_df <span class="ot">&lt;-</span> <span class="fu">merge</span>(Organ_df, <span class="fu">as.data.frame</span>(<span class="fu">table</span>(final_data<span class="sc">$</span><span class="st">`</span><span class="at">New Name</span><span class="st">`</span>)), <span class="at">by.x =</span> <span class="st">"organ"</span>, <span class="at">by.y =</span> <span class="st">"Var1"</span> )</span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>     Tissue_df <span class="ot">&lt;-</span> <span class="fu">merge</span>(Tissue_df, <span class="fu">as.data.frame</span>(<span class="fu">table</span>(final_data2<span class="sc">$</span>Tissue)), <span class="at">by.x =</span> <span class="st">"tissue"</span>, <span class="at">by.y =</span> <span class="st">"Var1"</span> )</span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>     <span class="fu">colnames</span>(Organ_df)[<span class="fu">ncol</span>(Organ_df)] <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.tsv"</span>, <span class="st">""</span>, active_file)</span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>     <span class="fu">colnames</span>(Tissue_df)[<span class="fu">ncol</span>(Tissue_df)] <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.tsv"</span>, <span class="st">""</span>, active_file)</span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>     <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Finished with"</span>, active_file))</span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Starting Adenoassociated_virus_UP000180764_12mer_3mis_hits.csv.gz"
[1] "Finished withAdenoassociated_virus_UP000180764_12mer_3mis_hits.csv.gz"
[1] "Starting Aichi_virus_UP000119554_12mer_3mis_hits.csv.gz"
[1] "Finished withAichi_virus_UP000119554_12mer_3mis_hits.csv.gz"
[1] "Starting Australian_bat_lyssavirus_UP000006934_12mer_3mis_hits.csv.gz"
[1] "Finished withAustralian_bat_lyssavirus_UP000006934_12mer_3mis_hits.csv.gz"
[1] "Starting Banna_virus_UP000000832_12mer_3mis_hits.csv.gz"
[1] "Finished withBanna_virus_UP000000832_12mer_3mis_hits.csv.gz"
[1] "Starting Barmah_forest_virus_UP000007609_12mer_3mis_hits.csv.gz"
[1] "Finished withBarmah_forest_virus_UP000007609_12mer_3mis_hits.csv.gz"
[1] "Starting BK_polyomavirus_UP000145981_12mer_3mis_hits.csv.gz"
[1] "Finished withBK_polyomavirus_UP000145981_12mer_3mis_hits.csv.gz"
[1] "Starting Bunyamwera_virus_UP000002476_12mer_3mis_hits.csv.gz"
[1] "Finished withBunyamwera_virus_UP000002476_12mer_3mis_hits.csv.gz"
[1] "Starting Bunyavirus_La_Crosse_UP000008768_12mer_3mis_hits.csv.gz"
[1] "Finished withBunyavirus_La_Crosse_UP000008768_12mer_3mis_hits.csv.gz"
[1] "Starting Bunyavirus_snowshoe_hare_UP000150053_12mer_3mis_hits.csv.gz"
[1] "Finished withBunyavirus_snowshoe_hare_UP000150053_12mer_3mis_hits.csv.gz"
[1] "Starting Cercopithecine_alphaherpesvirus_UP000137988_12mer_3mis_hits.csv.gz"
[1] "Finished withCercopithecine_alphaherpesvirus_UP000137988_12mer_3mis_hits.csv.gz"
[1] "Starting Chandipura_virus_UP000008448_12mer_3mis_hits.csv.gz"
[1] "Finished withChandipura_virus_UP000008448_12mer_3mis_hits.csv.gz"
[1] "Starting Chikungunya_virus_UP000000569_12mer_3mis_hits.csv.gz"
[1] "Finished withChikungunya_virus_UP000000569_12mer_3mis_hits.csv.gz"
[1] "Starting Cosavirus_A_UP000029753_12mer_3mis_hits.csv.gz"
[1] "Finished withCosavirus_A_UP000029753_12mer_3mis_hits.csv.gz"
[1] "Starting Cowpox_virus_UP000152733_12mer_3mis_hits.csv.gz"
[1] "Finished withCowpox_virus_UP000152733_12mer_3mis_hits.csv.gz"
[1] "Starting Coxsackievirus_UP000007757_12mer_3mis_hits.csv.gz"
[1] "Finished withCoxsackievirus_UP000007757_12mer_3mis_hits.csv.gz"
[1] "Starting Crimean-Congo_hemorrhagic_fever_virus_UP000008767_12mer_3mis_hits.csv.gz"
[1] "Finished withCrimean-Congo_hemorrhagic_fever_virus_UP000008767_12mer_3mis_hits.csv.gz"
[1] "Starting Dengue_virus_UP000002500_12mer_3mis_hits.csv.gz"
[1] "Finished withDengue_virus_UP000002500_12mer_3mis_hits.csv.gz"
[1] "Starting Dhori_virus_UP000181622_12mer_3mis_hits.csv.gz"
[1] "Finished withDhori_virus_UP000181622_12mer_3mis_hits.csv.gz"
[1] "Starting Dugbe_virus_UP000000278_12mer_3mis_hits.csv.gz"
[1] "Finished withDugbe_virus_UP000000278_12mer_3mis_hits.csv.gz"
[1] "Starting Duvenhage_virus_UP000117552_12mer_3mis_hits.csv.gz"
[1] "Finished withDuvenhage_virus_UP000117552_12mer_3mis_hits.csv.gz"
[1] "Starting Eastern_equine_encephalitis_virus_UP000110644_12mer_3mis_hits.csv.gz"
[1] "Finished withEastern_equine_encephalitis_virus_UP000110644_12mer_3mis_hits.csv.gz"
[1] "Starting Ebolavirus_UP000007209_12mer_3mis_hits.csv.gz"
[1] "Finished withEbolavirus_UP000007209_12mer_3mis_hits.csv.gz"
[1] "Starting Echovirus_UP000000670_12mer_3mis_hits.csv.gz"
[1] "Finished withEchovirus_UP000000670_12mer_3mis_hits.csv.gz"
[1] "Starting Encephalomyocarditis_virus_UP000002319_12mer_3mis_hits.csv.gz"
[1] "Finished withEncephalomyocarditis_virus_UP000002319_12mer_3mis_hits.csv.gz"
[1] "Starting Epstein-Barr_virus_UP000153037_12mer_3mis_hits.csv.gz"
[1] "Finished withEpstein-Barr_virus_UP000153037_12mer_3mis_hits.csv.gz"
[1] "Starting European_bat_lyssavirus_UP000008926_12mer_3mis_hits.csv.gz"
[1] "Finished withEuropean_bat_lyssavirus_UP000008926_12mer_3mis_hits.csv.gz"
[1] "Starting Hantaan_virus_UP000008627_12mer_3mis_hits.csv.gz"
[1] "Finished withHantaan_virus_UP000008627_12mer_3mis_hits.csv.gz"
[1] "Starting Hendra_virus_UP000008771_12mer_3mis_hits.csv.gz"
[1] "Finished withHendra_virus_UP000008771_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_A_virus_UP000006724_12mer_3mis_hits.csv.gz"
[1] "Finished withHepatitis_A_virus_UP000006724_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_B_virus_UP000008591_12mer_3mis_hits.csv.gz"
[1] "Finished withHepatitis_B_virus_UP000008591_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_C_virus_UP000000518_12mer_3mis_hits.csv.gz"
[1] "Finished withHepatitis_C_virus_UP000000518_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_delta_virus_UP000006675_12mer_3mis_hits.csv.gz"
[1] "Finished withHepatitis_delta_virus_UP000006675_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_E_virus_UP000006705_12mer_3mis_hits.csv.gz"
[1] "Finished withHepatitis_E_virus_UP000006705_12mer_3mis_hits.csv.gz"
[1] "Starting Hepatitis_G_virus_UP000007407_12mer_3mis_hits.csv.gz"
[1] "Finished withHepatitis_G_virus_UP000007407_12mer_3mis_hits.csv.gz"
[1] "Starting Horsepox_virus_UP000111173_12mer_3mis_hits.csv.gz"
[1] "Finished withHorsepox_virus_UP000111173_12mer_3mis_hits.csv.gz"
[1] "Starting Human_adenovirus_UP000008167_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_adenovirus_UP000008167_12mer_3mis_hits.csv.gz"
[1] "Starting Human_astrovirus_UP000001650_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_astrovirus_UP000001650_12mer_3mis_hits.csv.gz"
[1] "Starting Human_coronavirus_UP000006716_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_coronavirus_UP000006716_12mer_3mis_hits.csv.gz"
[1] "Starting Human_cytomegalovirus_merlin_UP000000938_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_cytomegalovirus_merlin_UP000000938_12mer_3mis_hits.csv.gz"
[1] "Starting Human_enterovirus_68_UP000143570_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_enterovirus_68_UP000143570_12mer_3mis_hits.csv.gz"
[1] "Starting Human_enterovirus_70_UP000008683_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_enterovirus_70_UP000008683_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus_1_UP000009294_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_herpesvirus_1_UP000009294_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus_2_UP000001874_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_herpesvirus_2_UP000001874_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus_6_UP000009295_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_herpesvirus_6_UP000009295_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus_7_UP000009246_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_herpesvirus_7_UP000009246_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus_8_UP000000942_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_herpesvirus_8_UP000000942_12mer_3mis_hits.csv.gz"
[1] "Starting Human_herpesvirus6b_UP000006930_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_herpesvirus6b_UP000006930_12mer_3mis_hits.csv.gz"
[1] "Starting Human_immunodeficiency_virus_1_UP000002241_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_immunodeficiency_virus_1_UP000002241_12mer_3mis_hits.csv.gz"
[1] "Starting Human_immunodeficiency_virus_2_UP000002242_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_immunodeficiency_virus_2_UP000002242_12mer_3mis_hits.csv.gz"
[1] "Starting Human_papillomavirus_1_UP000006372_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_papillomavirus_1_UP000006372_12mer_3mis_hits.csv.gz"
[1] "Starting Human_papillomavirus_2_UP000007710_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_papillomavirus_2_UP000007710_12mer_3mis_hits.csv.gz"
[1] "Starting Human_papillomavirus_type_16_UP000009251_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_papillomavirus_type_16_UP000009251_12mer_3mis_hits.csv.gz"
[1] "Starting Human_papillomavirus_type_18_UP000009109_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_papillomavirus_type_18_UP000009109_12mer_3mis_hits.csv.gz"
[1] "Starting Human_parainfluenza_UP000109776_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_parainfluenza_UP000109776_12mer_3mis_hits.csv.gz"
[1] "Starting Human_parvovirus_B19_UP000006624_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_parvovirus_B19_UP000006624_12mer_3mis_hits.csv.gz"
[1] "Starting Human_respiratory_syncytial_virus_UP000002472_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_respiratory_syncytial_virus_UP000002472_12mer_3mis_hits.csv.gz"
[1] "Starting Human_rhinovirus_UP000007070_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_rhinovirus_UP000007070_12mer_3mis_hits.csv.gz"
[1] "Starting Human_SARS_coronavirus_UP000000354_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_SARS_coronavirus_UP000000354_12mer_3mis_hits.csv.gz"
[1] "Starting Human_spumaretrovirus_UP000138352_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_spumaretrovirus_UP000138352_12mer_3mis_hits.csv.gz"
[1] "Starting Human_T-lymphotropic_virus_1_UP000001061_12mer_3mis_hits.csv.gz"
[1] "Finished withHuman_T-lymphotropic_virus_1_UP000001061_12mer_3mis_hits.csv.gz"
[1] "Starting Human_torovirus_UP000243571_12mer_3mis_hits.csv.gz"
[1] "Starting Influenza_A_virus_UP000009255_12mer_3mis_hits.csv.gz"
[1] "Finished withInfluenza_A_virus_UP000009255_12mer_3mis_hits.csv.gz"
[1] "Starting Influenza_B_virus_UP000008158_12mer_3mis_hits.csv.gz"
[1] "Finished withInfluenza_B_virus_UP000008158_12mer_3mis_hits.csv.gz"
[1] "Starting Influenza_C_virus_UP000008286_12mer_3mis_hits.csv.gz"
[1] "Finished withInfluenza_C_virus_UP000008286_12mer_3mis_hits.csv.gz"
[1] "Starting Isfahan_virus_UP000204017_12mer_3mis_hits.csv.gz"
[1] "Finished withIsfahan_virus_UP000204017_12mer_3mis_hits.csv.gz"
[1] "Starting Japanese_encephalitis_virus_UP000007214_12mer_3mis_hits.csv.gz"
[1] "Finished withJapanese_encephalitis_virus_UP000007214_12mer_3mis_hits.csv.gz"
[1] "Starting JC_polyomavirus_UP000008478_12mer_3mis_hits.csv.gz"
[1] "Finished withJC_polyomavirus_UP000008478_12mer_3mis_hits.csv.gz"
[1] "Starting Junin_mammarenavirus_UP000127886_12mer_3mis_hits.csv.gz"
[1] "Finished withJunin_mammarenavirus_UP000127886_12mer_3mis_hits.csv.gz"
[1] "Starting KI_Polyomavirus_UP000107189_12mer_3mis_hits.csv.gz"
[1] "Finished withKI_Polyomavirus_UP000107189_12mer_3mis_hits.csv.gz"
[1] "Starting Kunjin_virus_UP000138183_12mer_3mis_hits.csv.gz"
[1] "Finished withKunjin_virus_UP000138183_12mer_3mis_hits.csv.gz"
[1] "Starting Lagos_bat_virus_UP000161306_12mer_3mis_hits.csv.gz"
[1] "Finished withLagos_bat_virus_UP000161306_12mer_3mis_hits.csv.gz"
[1] "Starting Lake_Victoria_marburgvirus_UP000180448_12mer_3mis_hits.csv.gz"
[1] "Finished withLake_Victoria_marburgvirus_UP000180448_12mer_3mis_hits.csv.gz"
[1] "Starting Langat_virus_UP000180575_12mer_3mis_hits.csv.gz"
[1] "Finished withLangat_virus_UP000180575_12mer_3mis_hits.csv.gz"
[1] "Starting Lassa_virus_UP000002473_12mer_3mis_hits.csv.gz"
[1] "Finished withLassa_virus_UP000002473_12mer_3mis_hits.csv.gz"
[1] "Starting Lordsdale_virus_UP000007767_12mer_3mis_hits.csv.gz"
[1] "Finished withLordsdale_virus_UP000007767_12mer_3mis_hits.csv.gz"
[1] "Starting Louping_ill_virus_UP000007644_12mer_3mis_hits.csv.gz"
[1] "Finished withLouping_ill_virus_UP000007644_12mer_3mis_hits.csv.gz"
[1] "Starting Lymphocytic_choriomeningitis_virus_UP000002474_12mer_3mis_hits.csv.gz"
[1] "Finished withLymphocytic_choriomeningitis_virus_UP000002474_12mer_3mis_hits.csv.gz"
[1] "Starting Machupo_virus_UP000233900_12mer_3mis_hits.csv.gz"
[1] "Finished withMachupo_virus_UP000233900_12mer_3mis_hits.csv.gz"
[1] "Starting Mayaro_virus_UP000007774_12mer_3mis_hits.csv.gz"
[1] "Finished withMayaro_virus_UP000007774_12mer_3mis_hits.csv.gz"
[1] "Starting Measles_virus_UP000008699_12mer_3mis_hits.csv.gz"
[1] "Finished withMeasles_virus_UP000008699_12mer_3mis_hits.csv.gz"
[1] "Starting Mengo_encephalomyocarditis_virus_UP000008663_12mer_3mis_hits.csv.gz"
[1] "Finished withMengo_encephalomyocarditis_virus_UP000008663_12mer_3mis_hits.csv.gz"
[1] "Starting Merkel_cell_polyomavirus_UP000154903_12mer_3mis_hits.csv.gz"
[1] "Finished withMerkel_cell_polyomavirus_UP000154903_12mer_3mis_hits.csv.gz"
[1] "Starting MERS_coronavirus_UP000147828_12mer_3mis_hits.csv.gz"
[1] "Finished withMERS_coronavirus_UP000147828_12mer_3mis_hits.csv.gz"
[1] "Starting MERS_UP000171868_12mer_3mis_hits.csv.gz"
[1] "Finished withMERS_UP000171868_12mer_3mis_hits.csv.gz"
[1] "Starting Mokola_virus_UP000006826_12mer_3mis_hits.csv.gz"
[1] "Finished withMokola_virus_UP000006826_12mer_3mis_hits.csv.gz"
[1] "Starting Molluscum_contagiosum_virus_UP000000869_12mer_3mis_hits.csv.gz"
[1] "Finished withMolluscum_contagiosum_virus_UP000000869_12mer_3mis_hits.csv.gz"
[1] "Starting Monkeypox_virus_UP000101269_12mer_3mis_hits.csv.gz"
[1] "Finished withMonkeypox_virus_UP000101269_12mer_3mis_hits.csv.gz"
[1] "Starting Mumps_virus_UP000002331_12mer_3mis_hits.csv.gz"
[1] "Finished withMumps_virus_UP000002331_12mer_3mis_hits.csv.gz"
[1] "Starting Murray_valley_encephalitis_virus_UP000008863_12mer_3mis_hits.csv.gz"
[1] "Finished withMurray_valley_encephalitis_virus_UP000008863_12mer_3mis_hits.csv.gz"
[1] "Starting Nipah_virus_UP000170143_12mer_3mis_hits.csv.gz"
[1] "Finished withNipah_virus_UP000170143_12mer_3mis_hits.csv.gz"
[1] "Starting Norwalk_virus_UP000000826_12mer_3mis_hits.csv.gz"
[1] "Finished withNorwalk_virus_UP000000826_12mer_3mis_hits.csv.gz"
[1] "Starting Onyong_nyong_virus_UP000008868_12mer_3mis_hits.csv.gz"
[1] "Finished withOnyong_nyong_virus_UP000008868_12mer_3mis_hits.csv.gz"
[1] "Starting Orf_virus_UP000000870_12mer_3mis_hits.csv.gz"
[1] "Finished withOrf_virus_UP000000870_12mer_3mis_hits.csv.gz"
[1] "Starting Oropouche_virus_UP000117676_12mer_3mis_hits.csv.gz"
[1] "Finished withOropouche_virus_UP000117676_12mer_3mis_hits.csv.gz"
[1] "Starting Pichinde_mammarenavirus_UP000139415_12mer_3mis_hits.csv.gz"
[1] "Finished withPichinde_mammarenavirus_UP000139415_12mer_3mis_hits.csv.gz"
[1] "Starting Poliovirus_UP000000356_12mer_3mis_hits.csv.gz"
[1] "Finished withPoliovirus_UP000000356_12mer_3mis_hits.csv.gz"
[1] "Starting Punta_toro_phlebovirus_UP000214480_12mer_3mis_hits.csv.gz"
[1] "Finished withPunta_toro_phlebovirus_UP000214480_12mer_3mis_hits.csv.gz"
[1] "Starting Puumala_virus_UP000110237_12mer_3mis_hits.csv.gz"
[1] "Finished withPuumala_virus_UP000110237_12mer_3mis_hits.csv.gz"
[1] "Starting Rabies_virus_UP000008649_12mer_3mis_hits.csv.gz"
[1] "Finished withRabies_virus_UP000008649_12mer_3mis_hits.csv.gz"
[1] "Starting Rift_valley_fever_virus_UP000002477_12mer_3mis_hits.csv.gz"
[1] "Finished withRift_valley_fever_virus_UP000002477_12mer_3mis_hits.csv.gz"
[1] "Starting Rosavirus_A_UP000029743_12mer_3mis_hits.csv.gz"
[1] "Finished withRosavirus_A_UP000029743_12mer_3mis_hits.csv.gz"
[1] "Starting Ross_river_virus_UP000006579_12mer_3mis_hits.csv.gz"
[1] "Finished withRoss_river_virus_UP000006579_12mer_3mis_hits.csv.gz"
[1] "Starting Rotavirus_A_UP000001119_12mer_3mis_hits.csv.gz"
[1] "Finished withRotavirus_A_UP000001119_12mer_3mis_hits.csv.gz"
[1] "Starting Rotavirus_B_UP000114294_12mer_3mis_hits.csv.gz"
[1] "Finished withRotavirus_B_UP000114294_12mer_3mis_hits.csv.gz"
[1] "Starting Rotavirus_C_UP000007664_12mer_3mis_hits.csv.gz"
[1] "Finished withRotavirus_C_UP000007664_12mer_3mis_hits.csv.gz"
[1] "Starting Rubella_virus_UP000000571_12mer_3mis_hits.csv.gz"
[1] "Finished withRubella_virus_UP000000571_12mer_3mis_hits.csv.gz"
[1] "Starting Sagiyama_virus_UP000007138_12mer_3mis_hits.csv.gz"
[1] "Finished withSagiyama_virus_UP000007138_12mer_3mis_hits.csv.gz"
[1] "Starting Salivirus_A_UP000029742_12mer_3mis_hits.csv.gz"
[1] "Finished withSalivirus_A_UP000029742_12mer_3mis_hits.csv.gz"
[1] "Starting Sandfly_fever_sicilian_virus_UP000207632_12mer_3mis_hits.csv.gz"
[1] "Finished withSandfly_fever_sicilian_virus_UP000207632_12mer_3mis_hits.csv.gz"
[1] "Starting Sapporo_virus_UP000000667_12mer_3mis_hits.csv.gz"
[1] "Finished withSapporo_virus_UP000000667_12mer_3mis_hits.csv.gz"
[1] "Starting SARS_coronavirus_2_UP000464024_12mer_3mis_hits.csv.gz"
[1] "Finished withSARS_coronavirus_2_UP000464024_12mer_3mis_hits.csv.gz"
[1] "Starting Semliki_forest_virus_UP000000570_12mer_3mis_hits.csv.gz"
[1] "Finished withSemliki_forest_virus_UP000000570_12mer_3mis_hits.csv.gz"
[1] "Starting Seoul_virus_UP000207620_12mer_3mis_hits.csv.gz"
[1] "Finished withSeoul_virus_UP000207620_12mer_3mis_hits.csv.gz"
[1] "Starting Simian_foamy_virus_UP000001063_12mer_3mis_hits.csv.gz"
[1] "Finished withSimian_foamy_virus_UP000001063_12mer_3mis_hits.csv.gz"
[1] "Starting Simian_virus_5_UP000007232_12mer_3mis_hits.csv.gz"
[1] "Finished withSimian_virus_5_UP000007232_12mer_3mis_hits.csv.gz"
[1] "Starting Sindbis_virus_UP000006710_12mer_3mis_hits.csv.gz"
[1] "Finished withSindbis_virus_UP000006710_12mer_3mis_hits.csv.gz"
[1] "Starting Southampton_virus_UP000007226_12mer_3mis_hits.csv.gz"
[1] "Finished withSouthampton_virus_UP000007226_12mer_3mis_hits.csv.gz"
[1] "Starting St._louis_encephalitis_virus_UP000171556_12mer_3mis_hits.csv.gz"
[1] "Finished withSt._louis_encephalitis_virus_UP000171556_12mer_3mis_hits.csv.gz"
[1] "Starting Tick-borne_powassan_virus_UP000006848_12mer_3mis_hits.csv.gz"
[1] "Finished withTick-borne_powassan_virus_UP000006848_12mer_3mis_hits.csv.gz"
[1] "Starting Torque_teno_virus_UP000001448_12mer_3mis_hits.csv.gz"
[1] "Finished withTorque_teno_virus_UP000001448_12mer_3mis_hits.csv.gz"
[1] "Starting Toscana_virus_UP000142444_12mer_3mis_hits.csv.gz"
[1] "Finished withToscana_virus_UP000142444_12mer_3mis_hits.csv.gz"
[1] "Starting Uukuniemi_virus_UP000008595_12mer_3mis_hits.csv.gz"
[1] "Finished withUukuniemi_virus_UP000008595_12mer_3mis_hits.csv.gz"
[1] "Starting Vaccinia_virus_UP000000344_12mer_3mis_hits.csv.gz"
[1] "Finished withVaccinia_virus_UP000000344_12mer_3mis_hits.csv.gz"
[1] "Starting Varicella-zoster_virus_UP000002602_12mer_3mis_hits.csv.gz"
[1] "Finished withVaricella-zoster_virus_UP000002602_12mer_3mis_hits.csv.gz"
[1] "Starting Variola_virus_UP000002060_12mer_3mis_hits.csv.gz"
[1] "Finished withVariola_virus_UP000002060_12mer_3mis_hits.csv.gz"
[1] "Starting Venezuelan_equine_encephalitis_virus_UP000171441_12mer_3mis_hits.csv.gz"
[1] "Finished withVenezuelan_equine_encephalitis_virus_UP000171441_12mer_3mis_hits.csv.gz"
[1] "Starting Vesicular_stomatitis_virus_UP000002327_12mer_3mis_hits.csv.gz"
[1] "Finished withVesicular_stomatitis_virus_UP000002327_12mer_3mis_hits.csv.gz"
[1] "Starting West_Nile_virus_UP000136458_12mer_3mis_hits.csv.gz"
[1] "Finished withWest_Nile_virus_UP000136458_12mer_3mis_hits.csv.gz"
[1] "Starting Western_equine_encephalitis_virus_UP000128494_12mer_3mis_hits.csv.gz"
[1] "Finished withWestern_equine_encephalitis_virus_UP000128494_12mer_3mis_hits.csv.gz"
[1] "Starting WU_polyomavirus_UP000118542_12mer_3mis_hits.csv.gz"
[1] "Finished withWU_polyomavirus_UP000118542_12mer_3mis_hits.csv.gz"
[1] "Starting Yaba_monkey_tumor_virus_UP000008596_12mer_3mis_hits.csv.gz"
[1] "Finished withYaba_monkey_tumor_virus_UP000008596_12mer_3mis_hits.csv.gz"
[1] "Starting Yaba-like_disease_virus_UP000136581_12mer_3mis_hits.csv.gz"
[1] "Finished withYaba-like_disease_virus_UP000136581_12mer_3mis_hits.csv.gz"
[1] "Starting Yellow_fever_virus_UP000000360_12mer_3mis_hits.csv.gz"
[1] "Finished withYellow_fever_virus_UP000000360_12mer_3mis_hits.csv.gz"
[1] "Starting Zika_virus_UP000054557_12mer_3mis_hits.csv.gz"
[1] "Finished withZika_virus_UP000054557_12mer_3mis_hits.csv.gz"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>ggt <span class="ot">&lt;-</span> Tissue_df <span class="sc">%&gt;%</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>tissue) <span class="sc">%&gt;%</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">as.numeric</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, tissue) <span class="sc">%&gt;%</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">sum</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">total =</span> <span class="fu">sum</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">freq =</span> count <span class="sc">/</span> total <span class="sc">*</span><span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer.*"</span>, <span class="st">""</span>, name))) <span class="sc">%&gt;%</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>tissue, <span class="at">y=</span>label)) <span class="sc">+</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">fill =</span> freq)) <span class="sc">+</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">colours =</span> <span class="fu">c</span>(<span class="st">"#045375"</span>, <span class="st">"#089099"</span>, <span class="st">"#7CCBA2"</span>, <span class="st">"#FCDE9C"</span>, <span class="st">"#FAB27B"</span>, <span class="st">"#E23333"</span>),</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>                       <span class="at">name =</span> <span class="st">"Tissue (%)"</span>) <span class="sc">+</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Tissue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'name'. You can override using the
`.groups` argument.
Joining with `by = join_by(Proteome.ID)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>ggt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FigureS5_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>ggo <span class="ot">&lt;-</span> Organ_df <span class="sc">%&gt;%</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>organ) <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">as.numeric</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, organ) <span class="sc">%&gt;%</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">sum</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">total =</span> <span class="fu">sum</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">freq =</span> count <span class="sc">/</span> total <span class="sc">*</span><span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer.*"</span>, <span class="st">""</span>, name))) <span class="sc">%&gt;%</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>organ, <span class="at">y=</span>label)) <span class="sc">+</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">fill =</span> freq)) <span class="sc">+</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">colours =</span> <span class="fu">c</span>(<span class="st">"#045375"</span>, <span class="st">"#089099"</span>, <span class="st">"#7CCBA2"</span>, <span class="st">"#FCDE9C"</span>, <span class="st">"#FAB27B"</span>, <span class="st">"#E23333"</span>),</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>                       <span class="at">name =</span> <span class="st">"Organ System (%)"</span>) <span class="sc">+</span></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Organ System"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'name'. You can override using the
`.groups` argument.
Joining with `by = join_by(Proteome.ID)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>ggo</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FigureS5_files/figure-html/unnamed-chunk-5-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>viruses <span class="ot">&lt;-</span> Organ_df <span class="sc">%&gt;%</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>organ) <span class="sc">%&gt;%</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">as.numeric</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, organ) <span class="sc">%&gt;%</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">sum</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">total =</span> <span class="fu">sum</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">freq =</span> count <span class="sc">/</span> total <span class="sc">*</span><span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer.*"</span>, <span class="st">""</span>, name)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'name'. You can override using the
`.groups` argument.</code></pre>
</div>
</div>
<section id="panel-assembly" class="level3">
<h3 class="anchored" data-anchor-id="panel-assembly">Panel assembly</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>taxa <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Proteome.ID <span class="sc">%in%</span> viruses<span class="sc">$</span><span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(family, order, class, phylum, kingdom) <span class="sc">%&gt;%</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Anelloviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Kolmioviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Kolmioviridae"</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">lapply</span>(taxa, as.factor), <span class="at">stringsAsFactors =</span><span class="cn">TRUE</span>)</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(check) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(taxa)</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> check[<span class="fu">complete.cases</span>(check),]</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">taxo2phylog</span>(<span class="fu">as.taxo</span>(check), <span class="at">root=</span><span class="st">"Root"</span>, <span class="at">abbrev=</span>F)</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>tax.phy <span class="ot">&lt;-</span> <span class="fu">as.phylo</span>(tmp)</span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tax.phy, <span class="at">branch.length=</span><span class="st">'none'</span>)</span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a>gg_tree</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FigureS5_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>gg_tree[[<span class="st">"data"</span>]] <span class="ot">&lt;-</span> gg_tree[[<span class="st">"data"</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata <span class="sc">%&gt;%</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>               <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(Taxon.mnemonic, family), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>))</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span>  gg_tree <span class="sc">+</span> <span class="fu">geom_hilight</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">subset =</span> family <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2),<span class="at">fill =</span> family), <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>   <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> Family_palette_v2, <span class="at">name =</span> <span class="st">"Viral Family"</span>)</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>ggc <span class="sc">%&gt;%</span></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">insert_right</span>(ggt, <span class="at">width =</span> <span class="fl">0.5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">insert_right</span>(ggo, <span class="at">width =</span> <span class="fl">0.15</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">insert_left</span>(gg_tree, <span class="at">width =</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FigureS5_files/figure-html/unnamed-chunk-6-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"FigureS5.pdf"</span>), <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">17</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>