<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.555">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Figure1</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Figure1_files/libs/clipboard/clipboard.min.js"></script>
<script src="Figure1_files/libs/quarto-html/quarto.js"></script>
<script src="Figure1_files/libs/quarto-html/popper.min.js"></script>
<script src="Figure1_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Figure1_files/libs/quarto-html/anchor.min.js"></script>
<link href="Figure1_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Figure1_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Figure1_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Figure1_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Figure1_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Figure1</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="figure-1" class="level2">
<h2 class="anchored" data-anchor-id="figure-1">Figure 1</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/stor/work/Melamed_COVID/Cole/git_clones/MolecularMimicry/Analysis/Figure1/src/"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dtplyr)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plyr)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggbeeswarm)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(wesanderson)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(aplot)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(treeio)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggtree)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(aplot)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggbeeswarm)</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="do">### mart &lt;- useMart('ENSEMBL_MART_ENSEMBL')</span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="do">### mart &lt;- useDataset('hsapiens_gene_ensembl', mart)</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>mart <span class="ot">&lt;-</span> <span class="fu">useEnsembl</span>(<span class="st">"ensembl"</span>, <span class="at">dataset =</span> <span class="st">"hsapiens_gene_ensembl"</span>) <span class="do">## , mirror = "useast"</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>mart_retrieved <span class="ot">&lt;-</span> <span class="fu">getBM</span>(</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">mart =</span> mart, <span class="at">useCache =</span> T,</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">attributes =</span> <span class="fu">c</span>(</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>    <span class="st">'ensembl_gene_id'</span>,</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>    <span class="st">'entrezgene_id'</span>,</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>    <span class="st">'external_gene_name'</span>,</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>    <span class="st">'uniprot_gn_symbol'</span>,</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>    <span class="st">'uniprot_gn_id'</span>,</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chromosome_name'</span>,</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>    <span class="st">'description'</span>),</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">uniqueRows=</span><span class="cn">TRUE</span>)</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>elm_classes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"../../Databases/elm_classes.tsv"</span>, <span class="at">skip =</span> <span class="dv">5</span>)</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>elm_classes_human <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"../../Databases/elm_instances_human.tsv"</span>, <span class="at">skip =</span> <span class="dv">5</span>)</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../../Utilities.R"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>meta_path <span class="ot">&lt;-</span> <span class="st">"../../Metadata/"</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"Final_Cohort_Metadata_v5.xlsx"</span>), <span class="at">sheet =</span> <span class="dv">1</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>duplicated_proteins <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"duplicated_proteins.csv"</span>))</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>human_duplicated_proteins <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"human_duplicated_or_removed_proteins.csv"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="dv">1</span>)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>proteome_metadata <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"uniprot_proteome_metadata.tsv"</span>))</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>viral_protein_metadata <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"uniprot_viral_protein_metadata.xlsx"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in Z1741 / R1741C26: got 'NON_STD 64;
/note="Selenocysteine"; /evidence="ECO:0000250"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U1978 / R1978C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA1978 / R1978C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AIC64083.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in W1994 / R1994C23: got 'MASS SPECTROMETRY:
[Non-structural protein 8]: Mass=21871; Method=Electrospray;
Evidence={ECO:0000269|PubMed:32083638}; MASS SPECTROMETRY: [Non-structural
protein 9]: Mass=12403; Method=Electrospray;
Evidence={ECO:0000269|PubMed:32083638}; MASS SPECTROMETRY: [Non-structural
protein 10]: Mass=14974; Method=Electrospray;
Evidence={ECO:0000269|PubMed:32083638};'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in W1995 / R1995C23: got 'MASS SPECTROMETRY:
[Non-structural protein 8]: Mass=21871; Method=Electrospray;
Evidence={ECO:0000269|PubMed:32083638}; MASS SPECTROMETRY: [Non-structural
protein 9]: Mass=12403; Method=Electrospray;
Evidence={ECO:0000269|PubMed:32083638}; MASS SPECTROMETRY: [Non-structural
protein 10]: Mass=14974; Method=Electrospray;
Evidence={ECO:0000269|PubMed:32083638};'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2169 / R2169C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2169 / R2169C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54757.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2170 / R2170C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2170 / R2170C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54658.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2177 / R2177C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2177 / R2177C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54660.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2178 / R2178C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2178 / R2178C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54759.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2181 / R2181C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2181 / R2181C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54755.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2207 / R2207C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2207 / R2207C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54739.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2217 / R2217C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2217 / R2217C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54742.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2229 / R2229C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2229 / R2229C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54758.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2230 / R2230C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2230 / R2230C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54659.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2244 / R2244C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2244 / R2244C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54749.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2246 / R2246C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2246 / R2246C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54741.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3051 / R3051C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3051 / R3051C27: got 'NON_TER 723;
/evidence="ECO:0000313|EMBL:CAD53434.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3061 / R3061C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3061 / R3061C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53393.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3062 / R3062C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3062 / R3062C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53389.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3063 / R3063C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3063 / R3063C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53394.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3064 / R3064C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3064 / R3064C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53391.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3065 / R3065C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3065 / R3065C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53390.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3066 / R3066C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3066 / R3066C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53386.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3067 / R3067C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3067 / R3067C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53392.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3068 / R3068C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3068 / R3068C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53388.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3116 / R3116C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3116 / R3116C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53466.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3442 / R3442C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3442 / R3442C27: got 'NON_TER 49;
/evidence="ECO:0000313|EMBL:AAD25087.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in Z3605 / R3605C26: got 'NON_STD 181;
/note="Selenocysteine"; /evidence="ECO:0000313|EMBL:CAL35617.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in Z3677 / R3677C26: got 'NON_STD 17;
/note="Selenocysteine"; /evidence="ECO:0000255"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U4860 / R4860C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA4860 / R4860C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAL35657.1"'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(viral_protein_metadata) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">gsub</span>(<span class="st">" "</span>, <span class="st">"_"</span>, <span class="fu">tolower</span>(<span class="fu">colnames</span>(viral_protein_metadata))), <span class="st">".viral"</span>)</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>human_protein_metadata <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"uniprot_protein_metadata.xlsx"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L1053 / R1053C12: got 'NON_STD 96;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in R2045 / R2045C18: got 'BIOPHYSICOCHEMICAL
PROPERTIES: Absorption: Abs(max)=420 nm;'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in R2046 / R2046C18: got 'BIOPHYSICOCHEMICAL
PROPERTIES: Absorption: Abs(max)=560 nm;'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in R2047 / R2047C18: got 'BIOPHYSICOCHEMICAL
PROPERTIES: Absorption: Abs(max)=530 nm {ECO:0000305|PubMed:2937147};'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L2235 / R2235C12: got 'NON_STD 49;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L2996 / R2996C12: got 'NON_STD 40;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L3216 / R3216C12: got 'NON_STD 73;
/note="Selenocysteine"; /evidence="ECO:0000269|PubMed:2229017"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L3884 / R3884C12: got 'NON_STD 73;
/note="Selenocysteine"; /evidence="ECO:0000305|PubMed:8039723,
ECO:0000305|PubMed:9705830"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L4382 / R4382C12: got 'NON_STD 126;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L4386 / R4386C12: got 'NON_STD 59;
/note="Selenocysteine"; NON_STD 300; /note="Selenocysteine"; NON_STD 318;
/note="Selenocysteine"; NON_STD 330; /note="Selenocysteine"; NON_STD 345;
/note="Selenocysteine"; NON_STD 352; /note="Selenocysteine"; NON_STD 367;
/note="Selenocysteine"; NON_STD 369; /note="Selenocysteine"; NON_STD 376;
/note="Selenocysteine"; NON_STD 378; /note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L4724 / R4724C12: got 'NON_STD 170;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L5044 / R5044C12: got 'NON_STD 49;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L6708 / R6708C12: got 'NON_STD 648;
/note="Selenocysteine"; /evidence="ECO:0000269|PubMed:8650234"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L8342 / R8342C12: got 'NON_STD 642;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L9715 / R9715C12: got 'NON_STD 48;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L9929 / R9929C12: got 'NON_STD 133;
/note="Selenocysteine"; NON_STD 266; /note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L10958 / R10958C12: got 'NON_STD 60;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L11117 / R11117C12: got 'NON_STD 188;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L11330 / R11330C12: got 'NON_STD 667;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L11638 / R11638C12: got 'NON_STD 387;
/note="Selenocysteine"; /evidence="ECO:0000269|PubMed:12775843"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L12326 / R12326C12: got 'NON_STD 523;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L12976 / R12976C12: got 'NON_STD 127;
/note="Selenocysteine"; NON_STD 462; /note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L12977 / R12977C12: got 'NON_STD 95;
/note="Selenocysteine"; /evidence="ECO:0000305|PubMed:10608886"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L14255 / R14255C12: got 'NON_STD 92;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L14927 / R14927C12: got 'NON_STD 73;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L14928 / R14928C12: got 'NON_STD 273;
/note="Selenocysteine"; /evidence="ECO:0000305|PubMed:12775843"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L18715 / R18715C12: got 'NON_STD 13;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in M20179 / R20179C13: got 'NON_CONS 336..337;
/evidence="ECO:0000305"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L33085 / R33085C12: got 'NON_STD 44;
/note="Selenocysteine"'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(human_protein_metadata) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">gsub</span>(<span class="st">" "</span>, <span class="st">"_"</span>, <span class="fu">tolower</span>(<span class="fu">colnames</span>(human_protein_metadata))), <span class="st">".human"</span>)</span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>out.tmp <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(human_protein_metadata)){ <span class="co">#nrow(human_protein_metadata)</span></span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a> out.tmp[i] <span class="ot">&lt;-</span>  <span class="fu">list</span>(<span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">"VARIANT "</span>, <span class="st">""</span>, <span class="fu">unlist</span>(<span class="fu">regmatches</span>(human_protein_metadata<span class="sc">$</span>natural_variant.human[i], <span class="fu">gregexpr</span>(<span class="st">"VARIANT [0-9]*"</span>, human_protein_metadata<span class="sc">$</span>natural_variant.human[i]))))))</span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a>human_protein_metadata<span class="sc">$</span>human_variants <span class="ot">&lt;-</span> out.tmp</span>
<span id="cb86-7"><a href="#cb86-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-8"><a href="#cb86-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-9"><a href="#cb86-9" aria-hidden="true" tabindex="-1"></a>tissue_map <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"normal_tissue.tsv"</span>))</span>
<span id="cb86-10"><a href="#cb86-10" aria-hidden="true" tabindex="-1"></a>organ_map <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"Tissue_Groupings.xlsx"</span>))</span>
<span id="cb86-11"><a href="#cb86-11" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> <span class="fu">merge</span>(tissue_map, organ_map, <span class="at">by.x =</span> <span class="st">"Tissue"</span>, <span class="at">by.y =</span> <span class="st">"Var1"</span>, <span class="at">all.x =</span> T)</span>
<span id="cb86-12"><a href="#cb86-12" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(tissue_organ_map<span class="sc">$</span>Level)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Not detected"       "High"               "Low"               
[4] "Medium"             "Ascending"          "Descending"        
[7] "Not representative" "N/A"               </code></pre>
</div>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Level <span class="sc">!=</span> <span class="st">"Not detected"</span>,]</span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Level <span class="sc">!=</span> <span class="st">"Not representative"</span>,]</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Level <span class="sc">!=</span> <span class="st">"N/A"</span>,]</span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(tissue_organ_map<span class="sc">$</span>Reliability)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Uncertain" "Enhanced"  "Approved"  "Supported"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Reliability <span class="sc">!=</span> <span class="st">"Uncertain"</span>,]</span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Reliability <span class="sc">!=</span> <span class="st">"NA"</span>,]</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>tissue_organ_map_condensed <span class="ot">&lt;-</span> tissue_organ_map <span class="sc">%&gt;%</span></span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Tissue, Gene, <span class="st">`</span><span class="at">Cell type</span><span class="st">`</span>, <span class="st">`</span><span class="at">New Name</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Gene) <span class="sc">%&gt;%</span></span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">organ_system =</span> <span class="fu">gsub</span>(<span class="st">"NA;"</span>, <span class="st">""</span>, <span class="fu">paste</span>(<span class="fu">unique</span>(<span class="st">`</span><span class="at">New Name</span><span class="st">`</span>), <span class="at">collapse =</span> <span class="st">";"</span>)),</span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a>                   <span class="at">tissue =</span> <span class="fu">gsub</span>(<span class="st">"NA;"</span>, <span class="st">""</span>, <span class="fu">paste</span>(<span class="fu">unique</span>(Tissue), <span class="at">collapse =</span> <span class="st">";"</span>)),</span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a>                   <span class="at">cell_type =</span> <span class="fu">gsub</span>(<span class="st">"NA;"</span>, <span class="st">""</span>, <span class="fu">paste</span>(<span class="fu">unique</span>(<span class="st">`</span><span class="at">Cell type</span><span class="st">`</span>), <span class="at">collapse =</span> <span class="st">";"</span>)))</span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-11"><a href="#cb90-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-12"><a href="#cb90-12" aria-hidden="true" tabindex="-1"></a>out_dir <span class="ot">&lt;-</span> <span class="st">"../output/"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="mer-data-annotation" class="level2">
<h2 class="anchored" data-anchor-id="mer-data-annotation">8mer Data Annotation</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="do">## This will only run for files that haven't ran already</span></span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../../epitope_alignment/Results/Complete_Final_Set_8mers_3mis/"</span>)</span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".tsv"</span>, files_to_read)]</span>
<span id="cb91-5"><a href="#cb91-5" aria-hidden="true" tabindex="-1"></a>files_completed <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../../../Analysis/8mer_annotated/"</span>)</span>
<span id="cb91-6"><a href="#cb91-6" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="sc">!</span><span class="fu">gsub</span>(<span class="st">".tsv.gz"</span>, <span class="st">""</span>, files_to_read) <span class="sc">%in%</span> <span class="fu">gsub</span>(<span class="st">".csv.gz"</span>, <span class="st">""</span>, files_completed)]</span>
<span id="cb91-7"><a href="#cb91-7" aria-hidden="true" tabindex="-1"></a>final_cohort_naming <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../../Viral_Proteomes/Final_Cohort"</span>)</span>
<span id="cb91-8"><a href="#cb91-8" aria-hidden="true" tabindex="-1"></a>final_cohort_naming <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">gsub</span>(<span class="st">".fasta.gz"</span>, <span class="st">"_8mer_3mis_hits.tsv.gz"</span>, final_cohort_naming))</span>
<span id="cb91-9"><a href="#cb91-9" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[files_to_read <span class="sc">%in%</span> final_cohort_naming]</span>
<span id="cb91-10"><a href="#cb91-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-11"><a href="#cb91-11" aria-hidden="true" tabindex="-1"></a>files_to_read<span class="ot">&lt;-</span> files_to_read[<span class="fu">order</span>(<span class="fu">file.info</span>(files_to_read)<span class="sc">$</span>size)]</span>
<span id="cb91-12"><a href="#cb91-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-13"><a href="#cb91-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb91-14"><a href="#cb91-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb91-15"><a href="#cb91-15" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">nThread =</span> <span class="dv">12</span>)</span>
<span id="cb91-16"><a href="#cb91-16" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb91-17"><a href="#cb91-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Skipping "</span>, active_file, <span class="st">" due to lack of data."</span>))</span>
<span id="cb91-18"><a href="#cb91-18" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb91-19"><a href="#cb91-19" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb91-20"><a href="#cb91-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-21"><a href="#cb91-21" aria-hidden="true" tabindex="-1"></a> active_data[, uniprot <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(target_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb91-22"><a href="#cb91-22" aria-hidden="true" tabindex="-1"></a> active_data[, viral_id <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(query_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb91-23"><a href="#cb91-23" aria-hidden="true" tabindex="-1"></a> active_data[, mismatches <span class="sc">:</span><span class="er">=</span> <span class="fu">mismatch_calling_fast</span>(query, target), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb91-24"><a href="#cb91-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb91-25"><a href="#cb91-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Data Merging "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb91-26"><a href="#cb91-26" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb91-27"><a href="#cb91-27" aria-hidden="true" tabindex="-1"></a>     <span class="fu">lazy_dt</span>() <span class="sc">%&gt;%</span></span>
<span id="cb91-28"><a href="#cb91-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(mart_retrieved, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="st">"uniprot_gn_id"</span>),</span>
<span id="cb91-29"><a href="#cb91-29" aria-hidden="true" tabindex="-1"></a>              <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb91-30"><a href="#cb91-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(tissue_organ_map_condensed, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span> <span class="ot">=</span> <span class="st">"Gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb91-31"><a href="#cb91-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(human_protein_metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span><span class="st">"entry.human"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb91-32"><a href="#cb91-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(viral_protein_metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"viral_id"</span> <span class="ot">=</span> <span class="st">"entry.viral"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb91-33"><a href="#cb91-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">catalytic_activity.same =</span> catalytic_activity.human <span class="sc">==</span> catalytic_activity.viral) <span class="sc">%&gt;%</span></span>
<span id="cb91-34"><a href="#cb91-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.human =</span> <span class="fu">gsub</span>(<span class="st">";.*"</span>, <span class="st">""</span>, catalytic_activity.human)) <span class="sc">%&gt;%</span></span>
<span id="cb91-35"><a href="#cb91-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.viral =</span> <span class="fu">gsub</span>(<span class="st">";.*"</span>, <span class="st">""</span>, catalytic_activity.viral)) <span class="sc">%&gt;%</span></span>
<span id="cb91-36"><a href="#cb91-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.same =</span> reaction.human <span class="sc">==</span> reaction.viral) <span class="sc">%&gt;%</span></span>
<span id="cb91-37"><a href="#cb91-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.same =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(reaction.same), <span class="cn">FALSE</span>, reaction.same)) <span class="sc">%&gt;%</span></span>
<span id="cb91-38"><a href="#cb91-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">catalytic_activity.same =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(catalytic_activity.same), <span class="cn">FALSE</span>, catalytic_activity.same)) <span class="sc">%&gt;%</span></span>
<span id="cb91-39"><a href="#cb91-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">enzyme_number.same =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb91-40"><a href="#cb91-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.table</span>()</span>
<span id="cb91-41"><a href="#cb91-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb91-42"><a href="#cb91-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"EC Screening "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb91-43"><a href="#cb91-43" aria-hidden="true" tabindex="-1"></a>  active_data[, enzyme_number.same <span class="sc">:</span><span class="er">=</span> <span class="fu">fast_ec_screen</span>(ec_number.human, ec_number.viral), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb91-44"><a href="#cb91-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-45"><a href="#cb91-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Final Cleaning "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb91-46"><a href="#cb91-46" aria-hidden="true" tabindex="-1"></a>  active_data_subset <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span> </span>
<span id="cb91-47"><a href="#cb91-47" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">human_variants =</span> <span class="fu">sapply</span>(human_variants, toString)) <span class="sc">%&gt;%</span></span>
<span id="cb91-48"><a href="#cb91-48" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(query, query_seqname, query_start, target_seqname, target_start,</span>
<span id="cb91-49"><a href="#cb91-49" aria-hidden="true" tabindex="-1"></a>           target, mismatches, uniprot, viral_id, ensembl_gene_id,</span>
<span id="cb91-50"><a href="#cb91-50" aria-hidden="true" tabindex="-1"></a>           entrezgene_id, external_gene_name, uniprot_gn_symbol, chromosome_name,</span>
<span id="cb91-51"><a href="#cb91-51" aria-hidden="true" tabindex="-1"></a>           reviewed.human, entry_name.human, protein_names.human, length.human,</span>
<span id="cb91-52"><a href="#cb91-52" aria-hidden="true" tabindex="-1"></a>           active_site.human, binding_site.human, dna_binding.human,</span>
<span id="cb91-53"><a href="#cb91-53" aria-hidden="true" tabindex="-1"></a>           ec_number.human,  reviewed.viral, entry_name.viral,</span>
<span id="cb91-54"><a href="#cb91-54" aria-hidden="true" tabindex="-1"></a>           protein_names.viral, length.viral,</span>
<span id="cb91-55"><a href="#cb91-55" aria-hidden="true" tabindex="-1"></a>           active_site.viral, binding_site.viral, dna_binding.viral,</span>
<span id="cb91-56"><a href="#cb91-56" aria-hidden="true" tabindex="-1"></a>           ec_number.viral, catalytic_activity.same, reaction.human,</span>
<span id="cb91-57"><a href="#cb91-57" aria-hidden="true" tabindex="-1"></a>           reaction.viral, reaction.same,</span>
<span id="cb91-58"><a href="#cb91-58" aria-hidden="true" tabindex="-1"></a>           enzyme_number.same) <span class="sc">%&gt;%</span></span>
<span id="cb91-59"><a href="#cb91-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb91-60"><a href="#cb91-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Sys.time</span>()</span>
<span id="cb91-61"><a href="#cb91-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fwrite</span>(active_data_subset, <span class="fu">paste0</span>(<span class="st">"../../../Analysis/8mer_annotated/"</span>, <span class="fu">gsub</span>(<span class="st">".tsv"</span>, <span class="st">".csv"</span>, active_file)), <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">compress =</span> <span class="st">"gzip"</span>, <span class="at">nThread =</span> <span class="dv">12</span>)</span>
<span id="cb91-62"><a href="#cb91-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(active_data, active_data_subset)</span>
<span id="cb91-63"><a href="#cb91-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gc</span>()</span>
<span id="cb91-64"><a href="#cb91-64" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="mer-data-load-and-processing" class="level2">
<h2 class="anchored" data-anchor-id="mer-data-load-and-processing">8mer Data Load and Processing</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="do">### This only needs to be run once and will take a while, the script when not full render unless you already have the output in the expected path or if you set message: true</span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(seqinr)</span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringi)</span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(svMisc)</span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb92-6"><a href="#cb92-6" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../../8mer_annotated"</span>,<span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb92-7"><a href="#cb92-7" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".csv.gz"</span>, files_to_read)]</span>
<span id="cb92-8"><a href="#cb92-8" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb92-9"><a href="#cb92-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-10"><a href="#cb92-10" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"Number of Proteins"</span>, <span class="st">"Proteome Length"</span>))</span>
<span id="cb92-11"><a href="#cb92-11" aria-hidden="true" tabindex="-1"></a>out_protein_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb92-12"><a href="#cb92-12" aria-hidden="true" tabindex="-1"></a>out_protein_final_min <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb92-13"><a href="#cb92-13" aria-hidden="true" tabindex="-1"></a>out_protein_final_count <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb92-14"><a href="#cb92-14" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb92-15"><a href="#cb92-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" Pathogen number:"</span>,<span class="fu">match</span>(active_file, files_to_read), <span class="st">"/"</span>, <span class="fu">length</span>(files_to_read)))</span>
<span id="cb92-16"><a href="#cb92-16" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">nThread =</span> <span class="dv">18</span>) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb92-17"><a href="#cb92-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb92-18"><a href="#cb92-18" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb92-19"><a href="#cb92-19" aria-hidden="true" tabindex="-1"></a>    out_final[,<span class="fu">gsub</span>(<span class="st">"_8mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file)] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="cn">NA</span>,<span class="cn">NA</span>)</span>
<span id="cb92-20"><a href="#cb92-20" aria-hidden="true" tabindex="-1"></a>      data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>, <span class="fu">gsub</span>(<span class="st">"../../8mer_annotated/"</span>, <span class="st">""</span>, <span class="fu">gsub</span>(<span class="st">"_8mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file)), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb92-21"><a href="#cb92-21" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb92-22"><a href="#cb92-22" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb92-23"><a href="#cb92-23" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb92-24"><a href="#cb92-24" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb92-25"><a href="#cb92-25" aria-hidden="true" tabindex="-1"></a>      PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb92-26"><a href="#cb92-26" aria-hidden="true" tabindex="-1"></a>      PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb92-27"><a href="#cb92-27" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb92-28"><a href="#cb92-28" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb92-29"><a href="#cb92-29" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb92-30"><a href="#cb92-30" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb92-31"><a href="#cb92-31" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb92-32"><a href="#cb92-32" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb92-33"><a href="#cb92-33" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb92-34"><a href="#cb92-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb92-35"><a href="#cb92-35" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data  <span class="sc">%&gt;%</span></span>
<span id="cb92-36"><a href="#cb92-36" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Remove duplicated pathogen proteins identified in </span></span>
<span id="cb92-37"><a href="#cb92-37" aria-hidden="true" tabindex="-1"></a>    <span class="do">## ../../Scripts/Duplicate_Protein_Screen.R</span></span>
<span id="cb92-38"><a href="#cb92-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb92-39"><a href="#cb92-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb92-40"><a href="#cb92-40" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(query, query_seqname, query_start, mismatches, reaction.human,</span>
<span id="cb92-41"><a href="#cb92-41" aria-hidden="true" tabindex="-1"></a>           reaction.viral, reaction.same, catalytic_activity.same,</span>
<span id="cb92-42"><a href="#cb92-42" aria-hidden="true" tabindex="-1"></a>           ec_number.human, ec_number.viral,</span>
<span id="cb92-43"><a href="#cb92-43" aria-hidden="true" tabindex="-1"></a>           enzyme_number.same, external_gene_name)</span>
<span id="cb92-44"><a href="#cb92-44" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb92-45"><a href="#cb92-45" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>reaction.same[<span class="fu">is.na</span>(active_data<span class="sc">$</span>reaction.same)] <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb92-46"><a href="#cb92-46" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>catalytic_activity.same[<span class="fu">is.na</span>(active_data<span class="sc">$</span>catalytic_activity.same)] <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb92-47"><a href="#cb92-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb92-48"><a href="#cb92-48" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(active_data)</span>
<span id="cb92-49"><a href="#cb92-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-50"><a href="#cb92-50" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb92-51"><a href="#cb92-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(reaction.same <span class="sc">!=</span> <span class="cn">TRUE</span> <span class="sc">&amp;</span> catalytic_activity.same <span class="sc">!=</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb92-52"><a href="#cb92-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(enzyme_number.same <span class="sc">!=</span> <span class="cn">TRUE</span>)  </span>
<span id="cb92-53"><a href="#cb92-53" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb92-54"><a href="#cb92-54" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(active_data<span class="sc">$</span>mismatches))</span>
<span id="cb92-55"><a href="#cb92-55" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(out) <span class="sc">==</span> <span class="dv">0</span>){out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="at">V2 =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))}</span>
<span id="cb92-56"><a href="#cb92-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(out) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>, <span class="fu">gsub</span>(<span class="st">"_8mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb92-57"><a href="#cb92-57" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb92-58"><a href="#cb92-58" aria-hidden="true" tabindex="-1"></a>  out_protein <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb92-59"><a href="#cb92-59" aria-hidden="true" tabindex="-1"></a>     <span class="fu">distinct</span>(query, query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb92-60"><a href="#cb92-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb92-61"><a href="#cb92-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb92-62"><a href="#cb92-62" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb92-63"><a href="#cb92-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb92-64"><a href="#cb92-64" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb92-65"><a href="#cb92-65" aria-hidden="true" tabindex="-1"></a>   out_protein_min <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb92-66"><a href="#cb92-66" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb92-67"><a href="#cb92-67" aria-hidden="true" tabindex="-1"></a>     <span class="fu">distinct</span>(query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb92-68"><a href="#cb92-68" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, query_start) <span class="sc">%&gt;%</span></span>
<span id="cb92-69"><a href="#cb92-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_min</span>(mismatches, <span class="at">n=</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb92-70"><a href="#cb92-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() </span>
<span id="cb92-71"><a href="#cb92-71" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb92-72"><a href="#cb92-72" aria-hidden="true" tabindex="-1"></a>   out <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(out_protein_min<span class="sc">$</span>mismatches))</span>
<span id="cb92-73"><a href="#cb92-73" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(out) <span class="sc">==</span> <span class="dv">0</span>){out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="at">V2 =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))}</span>
<span id="cb92-74"><a href="#cb92-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(out) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>, <span class="fu">gsub</span>(<span class="st">"_8mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb92-75"><a href="#cb92-75" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb92-76"><a href="#cb92-76" aria-hidden="true" tabindex="-1"></a>     out_protein_min <span class="ot">&lt;-</span> out_protein_min <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(query_seqname, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb92-77"><a href="#cb92-77" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb92-78"><a href="#cb92-78" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb92-79"><a href="#cb92-79" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb92-80"><a href="#cb92-80" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb92-81"><a href="#cb92-81" aria-hidden="true" tabindex="-1"></a>     out_protein_count <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb92-82"><a href="#cb92-82" aria-hidden="true" tabindex="-1"></a>       <span class="fu">distinct</span>(query, query_seqname, query_start, mismatches, external_gene_name) <span class="sc">%&gt;%</span></span>
<span id="cb92-83"><a href="#cb92-83" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb92-84"><a href="#cb92-84" aria-hidden="true" tabindex="-1"></a>       <span class="fu">distinct</span>(query, query_seqname, query_start, external_gene_name, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb92-85"><a href="#cb92-85" aria-hidden="true" tabindex="-1"></a>       <span class="fu">filter</span>(<span class="sc">!</span>external_gene_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="cn">NA</span>, <span class="st">""</span>, <span class="st">" "</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb92-86"><a href="#cb92-86" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb92-87"><a href="#cb92-87" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb92-88"><a href="#cb92-88" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb92-89"><a href="#cb92-89" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb92-90"><a href="#cb92-90" aria-hidden="true" tabindex="-1"></a>  <span class="do">############## Count Proteome Length ##############</span></span>
<span id="cb92-91"><a href="#cb92-91" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>,  <span class="fu">gsub</span>(<span class="st">"../../8mer_annotated/"</span>, <span class="st">""</span>, <span class="fu">gsub</span>(<span class="st">"_8mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file)), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb92-92"><a href="#cb92-92" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb92-93"><a href="#cb92-93" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb92-94"><a href="#cb92-94" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb92-95"><a href="#cb92-95" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb92-96"><a href="#cb92-96" aria-hidden="true" tabindex="-1"></a>       PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb92-97"><a href="#cb92-97" aria-hidden="true" tabindex="-1"></a>      PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb92-98"><a href="#cb92-98" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb92-99"><a href="#cb92-99" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb92-100"><a href="#cb92-100" aria-hidden="true" tabindex="-1"></a>  out_final <span class="ot">&lt;-</span> <span class="fu">merge</span>(out_final, out, <span class="at">by =</span> <span class="st">"mismatches"</span>, <span class="at">all.x =</span> T)</span>
<span id="cb92-101"><a href="#cb92-101" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb92-102"><a href="#cb92-102" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb92-103"><a href="#cb92-103" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb92-104"><a href="#cb92-104" aria-hidden="true" tabindex="-1"></a>   out_protein_tmp <span class="ot">&lt;-</span> out_protein <span class="sc">%&gt;%</span></span>
<span id="cb92-105"><a href="#cb92-105" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb92-106"><a href="#cb92-106" aria-hidden="true" tabindex="-1"></a>  out_protein_tmp_min <span class="ot">&lt;-</span> out_protein_min <span class="sc">%&gt;%</span></span>
<span id="cb92-107"><a href="#cb92-107" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb92-108"><a href="#cb92-108" aria-hidden="true" tabindex="-1"></a>  out_protein_tmp_count <span class="ot">&lt;-</span> out_protein_count <span class="sc">%&gt;%</span></span>
<span id="cb92-109"><a href="#cb92-109" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb92-110"><a href="#cb92-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-111"><a href="#cb92-111" aria-hidden="true" tabindex="-1"></a>  out_protein_final <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final, out_protein_tmp)</span>
<span id="cb92-112"><a href="#cb92-112" aria-hidden="true" tabindex="-1"></a>  out_protein_final_min <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final_min, out_protein_tmp_min)</span>
<span id="cb92-113"><a href="#cb92-113" aria-hidden="true" tabindex="-1"></a>  out_protein_final_count <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final_count, out_protein_tmp_count)</span>
<span id="cb92-114"><a href="#cb92-114" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb92-115"><a href="#cb92-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-116"><a href="#cb92-116" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb92-117"><a href="#cb92-117" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_final, <span class="st">"../../Key_Data/8mer_proteome_everything.csv"</span>)</span>
<span id="cb92-118"><a href="#cb92-118" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_protein_final, <span class="st">"../../Key_Data/8mer_proteome_protein_everything.csv"</span>)</span>
<span id="cb92-119"><a href="#cb92-119" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_protein_final_min, <span class="st">"../../Key_Data/8mer_proteome_protein_everything_min.csv"</span>)</span>
<span id="cb92-120"><a href="#cb92-120" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_protein_final_count, <span class="st">"../../Key_Data/8mer_proteome_protein_everything_count.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="mer-data-annotation-1" class="level2">
<h2 class="anchored" data-anchor-id="mer-data-annotation-1">12mer Data Annotation</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="do">## This will only run for files that haven't ran already</span></span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../../epitope_alignment/Results/Complete_Final_Set/"</span>)</span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".tsv"</span>, files_to_read)]</span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a>files_completed <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../../../Analysis/12mer_annotated/"</span>)</span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="sc">!</span><span class="fu">gsub</span>(<span class="st">".tsv"</span>, <span class="st">""</span>, files_to_read) <span class="sc">%in%</span> <span class="fu">gsub</span>(<span class="st">".csv.gz"</span>, <span class="st">""</span>, files_completed)]</span>
<span id="cb93-7"><a href="#cb93-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-8"><a href="#cb93-8" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID, <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb93-9"><a href="#cb93-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-10"><a href="#cb93-10" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">order</span>(<span class="fu">file.info</span>(files_to_read)<span class="sc">$</span>size)]</span>
<span id="cb93-11"><a href="#cb93-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-12"><a href="#cb93-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb93-13"><a href="#cb93-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb93-14"><a href="#cb93-14" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">nThread =</span> <span class="dv">12</span>)</span>
<span id="cb93-15"><a href="#cb93-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb93-16"><a href="#cb93-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Skipping "</span>, active_file, <span class="st">" due to lack of data."</span>))</span>
<span id="cb93-17"><a href="#cb93-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb93-18"><a href="#cb93-18" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb93-19"><a href="#cb93-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-20"><a href="#cb93-20" aria-hidden="true" tabindex="-1"></a> active_data[, uniprot <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(target_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb93-21"><a href="#cb93-21" aria-hidden="true" tabindex="-1"></a> active_data[, viral_id <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(query_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb93-22"><a href="#cb93-22" aria-hidden="true" tabindex="-1"></a> active_data[, mismatches <span class="sc">:</span><span class="er">=</span> <span class="fu">mismatch_calling_fast</span>(query, target), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb93-23"><a href="#cb93-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb93-24"><a href="#cb93-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Data Merging "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb93-25"><a href="#cb93-25" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb93-26"><a href="#cb93-26" aria-hidden="true" tabindex="-1"></a>     <span class="fu">lazy_dt</span>() <span class="sc">%&gt;%</span></span>
<span id="cb93-27"><a href="#cb93-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(mart_retrieved, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="st">"uniprot_gn_id"</span>),</span>
<span id="cb93-28"><a href="#cb93-28" aria-hidden="true" tabindex="-1"></a>              <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb93-29"><a href="#cb93-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(tissue_organ_map_condensed, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span> <span class="ot">=</span> <span class="st">"Gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb93-30"><a href="#cb93-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(human_protein_metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span><span class="st">"entry.human"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb93-31"><a href="#cb93-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(viral_protein_metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"viral_id"</span> <span class="ot">=</span> <span class="st">"entry.viral"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb93-32"><a href="#cb93-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">catalytic_activity.same =</span> catalytic_activity.human <span class="sc">==</span> catalytic_activity.viral) <span class="sc">%&gt;%</span></span>
<span id="cb93-33"><a href="#cb93-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.human =</span> <span class="fu">gsub</span>(<span class="st">";.*"</span>, <span class="st">""</span>, catalytic_activity.human)) <span class="sc">%&gt;%</span></span>
<span id="cb93-34"><a href="#cb93-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.viral =</span> <span class="fu">gsub</span>(<span class="st">";.*"</span>, <span class="st">""</span>, catalytic_activity.viral)) <span class="sc">%&gt;%</span></span>
<span id="cb93-35"><a href="#cb93-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.same =</span> reaction.human <span class="sc">==</span> reaction.viral) <span class="sc">%&gt;%</span></span>
<span id="cb93-36"><a href="#cb93-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.same =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(reaction.same), <span class="cn">FALSE</span>, reaction.same)) <span class="sc">%&gt;%</span></span>
<span id="cb93-37"><a href="#cb93-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">catalytic_activity.same =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(catalytic_activity.same), <span class="cn">FALSE</span>, catalytic_activity.same)) <span class="sc">%&gt;%</span></span>
<span id="cb93-38"><a href="#cb93-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">enzyme_number.same =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb93-39"><a href="#cb93-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.table</span>()</span>
<span id="cb93-40"><a href="#cb93-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb93-41"><a href="#cb93-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"EC Screening "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb93-42"><a href="#cb93-42" aria-hidden="true" tabindex="-1"></a>  active_data[, enzyme_number.same <span class="sc">:</span><span class="er">=</span> <span class="fu">fast_ec_screen</span>(ec_number.human, ec_number.viral), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb93-43"><a href="#cb93-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-44"><a href="#cb93-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Final Cleaning "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb93-45"><a href="#cb93-45" aria-hidden="true" tabindex="-1"></a>  active_data_subset <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span> </span>
<span id="cb93-46"><a href="#cb93-46" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">human_variants =</span> <span class="fu">sapply</span>(human_variants, toString)) <span class="sc">%&gt;%</span></span>
<span id="cb93-47"><a href="#cb93-47" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(query, query_seqname, query_start, target_seqname, target_start,</span>
<span id="cb93-48"><a href="#cb93-48" aria-hidden="true" tabindex="-1"></a>           target, mismatches, uniprot, viral_id, ensembl_gene_id,</span>
<span id="cb93-49"><a href="#cb93-49" aria-hidden="true" tabindex="-1"></a>           entrezgene_id, external_gene_name, uniprot_gn_symbol, chromosome_name,</span>
<span id="cb93-50"><a href="#cb93-50" aria-hidden="true" tabindex="-1"></a>           reviewed.human, entry_name.human, protein_names.human, length.human,</span>
<span id="cb93-51"><a href="#cb93-51" aria-hidden="true" tabindex="-1"></a>           active_site.human, binding_site.human, dna_binding.human,</span>
<span id="cb93-52"><a href="#cb93-52" aria-hidden="true" tabindex="-1"></a>           ec_number.human,  reviewed.viral, entry_name.viral,</span>
<span id="cb93-53"><a href="#cb93-53" aria-hidden="true" tabindex="-1"></a>           protein_names.viral, length.viral,</span>
<span id="cb93-54"><a href="#cb93-54" aria-hidden="true" tabindex="-1"></a>           active_site.viral, binding_site.viral, dna_binding.viral,</span>
<span id="cb93-55"><a href="#cb93-55" aria-hidden="true" tabindex="-1"></a>           ec_number.viral, catalytic_activity.same, reaction.human,</span>
<span id="cb93-56"><a href="#cb93-56" aria-hidden="true" tabindex="-1"></a>           reaction.viral, reaction.same,</span>
<span id="cb93-57"><a href="#cb93-57" aria-hidden="true" tabindex="-1"></a>           enzyme_number.same) <span class="sc">%&gt;%</span></span>
<span id="cb93-58"><a href="#cb93-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb93-59"><a href="#cb93-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Sys.time</span>()</span>
<span id="cb93-60"><a href="#cb93-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fwrite</span>(active_data_subset, <span class="fu">paste0</span>(<span class="st">"../../../Analysis/12mer_annotated/"</span>, <span class="fu">gsub</span>(<span class="st">".tsv"</span>, <span class="st">".csv.gz"</span>, active_file)), <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">compress =</span> <span class="st">"gzip"</span>, <span class="at">nThread =</span> <span class="dv">12</span>)</span>
<span id="cb93-61"><a href="#cb93-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(active_data, active_data_subset)</span>
<span id="cb93-62"><a href="#cb93-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gc</span>()</span>
<span id="cb93-63"><a href="#cb93-63" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="mer-data-load-and-processing-1" class="level2">
<h2 class="anchored" data-anchor-id="mer-data-load-and-processing-1">12mer Data Load and Processing</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="do">### This only needs to be run once and will take a while, the script when not full render unless you already have the output in the expected path or if you set message: true</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(seqinr)</span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringi)</span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(svMisc)</span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../12mer_annotated/"</span>)</span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a>elm_filter <span class="ot">=</span> <span class="cn">TRUE</span>; elm_mode <span class="ot">=</span> <span class="st">"regex_motiff"</span></span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb94-10"><a href="#cb94-10" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".csv.gz"</span>, files_to_read)]</span>
<span id="cb94-11"><a href="#cb94-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-12"><a href="#cb94-12" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb94-13"><a href="#cb94-13" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"Number of Proteins"</span>, <span class="st">"Proteome Length"</span>))</span>
<span id="cb94-14"><a href="#cb94-14" aria-hidden="true" tabindex="-1"></a>out_protein_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb94-15"><a href="#cb94-15" aria-hidden="true" tabindex="-1"></a>out_protein_final_min <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb94-16"><a href="#cb94-16" aria-hidden="true" tabindex="-1"></a>out_protein_final_count <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb94-17"><a href="#cb94-17" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb94-18"><a href="#cb94-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" Pathogen number:"</span>,<span class="fu">match</span>(active_file, files_to_read), <span class="st">"/"</span>, <span class="fu">length</span>(files_to_read)))</span>
<span id="cb94-19"><a href="#cb94-19" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">nThread =</span> <span class="dv">18</span>) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb94-20"><a href="#cb94-20" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb94-21"><a href="#cb94-21" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb94-22"><a href="#cb94-22" aria-hidden="true" tabindex="-1"></a>    out_final[,<span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file)] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="cn">NA</span>,<span class="cn">NA</span>)</span>
<span id="cb94-23"><a href="#cb94-23" aria-hidden="true" tabindex="-1"></a>      data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb94-24"><a href="#cb94-24" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb94-25"><a href="#cb94-25" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb94-26"><a href="#cb94-26" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb94-27"><a href="#cb94-27" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb94-28"><a href="#cb94-28" aria-hidden="true" tabindex="-1"></a>      PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb94-29"><a href="#cb94-29" aria-hidden="true" tabindex="-1"></a>      PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb94-30"><a href="#cb94-30" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb94-31"><a href="#cb94-31" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb94-32"><a href="#cb94-32" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb94-33"><a href="#cb94-33" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb94-34"><a href="#cb94-34" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb94-35"><a href="#cb94-35" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb94-36"><a href="#cb94-36" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb94-37"><a href="#cb94-37" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-38"><a href="#cb94-38" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data  <span class="sc">%&gt;%</span></span>
<span id="cb94-39"><a href="#cb94-39" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Remove duplicated pathogen proteins identified in </span></span>
<span id="cb94-40"><a href="#cb94-40" aria-hidden="true" tabindex="-1"></a>    <span class="do">## ../../Scripts/Duplicate_Protein_Screen.R</span></span>
<span id="cb94-41"><a href="#cb94-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb94-42"><a href="#cb94-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb94-43"><a href="#cb94-43" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(query, query_seqname, query_start, mismatches, reaction.human,</span>
<span id="cb94-44"><a href="#cb94-44" aria-hidden="true" tabindex="-1"></a>           reaction.viral, reaction.same, catalytic_activity.same,</span>
<span id="cb94-45"><a href="#cb94-45" aria-hidden="true" tabindex="-1"></a>           ec_number.human, ec_number.viral,</span>
<span id="cb94-46"><a href="#cb94-46" aria-hidden="true" tabindex="-1"></a>           enzyme_number.same, external_gene_name)</span>
<span id="cb94-47"><a href="#cb94-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-48"><a href="#cb94-48" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>reaction.same[<span class="fu">is.na</span>(active_data<span class="sc">$</span>reaction.same)] <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb94-49"><a href="#cb94-49" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>catalytic_activity.same[<span class="fu">is.na</span>(active_data<span class="sc">$</span>catalytic_activity.same)] <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb94-50"><a href="#cb94-50" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-51"><a href="#cb94-51" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(active_data)</span>
<span id="cb94-52"><a href="#cb94-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-53"><a href="#cb94-53" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb94-54"><a href="#cb94-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(reaction.same <span class="sc">!=</span> <span class="cn">TRUE</span> <span class="sc">&amp;</span> catalytic_activity.same <span class="sc">!=</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb94-55"><a href="#cb94-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(enzyme_number.same <span class="sc">!=</span> <span class="cn">TRUE</span>)  </span>
<span id="cb94-56"><a href="#cb94-56" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb94-57"><a href="#cb94-57" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(active_data<span class="sc">$</span>mismatches))</span>
<span id="cb94-58"><a href="#cb94-58" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(out) <span class="sc">==</span> <span class="dv">0</span>){out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="at">V2 =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))}</span>
<span id="cb94-59"><a href="#cb94-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(out) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb94-60"><a href="#cb94-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-61"><a href="#cb94-61" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(elm_filter){</span>
<span id="cb94-62"><a href="#cb94-62" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(elm_mode <span class="sc">==</span> <span class="st">"regex_motiff"</span>){</span>
<span id="cb94-63"><a href="#cb94-63" aria-hidden="true" tabindex="-1"></a>      <span class="cf">for</span>(motiff_i <span class="cf">in</span> <span class="fu">unique</span>(elm_classes<span class="sc">$</span>Regex)){</span>
<span id="cb94-64"><a href="#cb94-64" aria-hidden="true" tabindex="-1"></a>        active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb94-65"><a href="#cb94-65" aria-hidden="true" tabindex="-1"></a>          <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(motiff_i, <span class="fu">paste0</span>(<span class="st">".*"</span>, query, <span class="st">".*"</span>)))</span>
<span id="cb94-66"><a href="#cb94-66" aria-hidden="true" tabindex="-1"></a>        <span class="do">## browser()</span></span>
<span id="cb94-67"><a href="#cb94-67" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb94-68"><a href="#cb94-68" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (elm_mode <span class="sc">==</span> <span class="st">"human_uniprot"</span>){</span>
<span id="cb94-69"><a href="#cb94-69" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb94-70"><a href="#cb94-70" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb94-71"><a href="#cb94-71" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb94-72"><a href="#cb94-72" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-73"><a href="#cb94-73" aria-hidden="true" tabindex="-1"></a>  out_protein <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb94-74"><a href="#cb94-74" aria-hidden="true" tabindex="-1"></a>     <span class="fu">distinct</span>(query, query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb94-75"><a href="#cb94-75" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb94-76"><a href="#cb94-76" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb94-77"><a href="#cb94-77" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb94-78"><a href="#cb94-78" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb94-79"><a href="#cb94-79" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-80"><a href="#cb94-80" aria-hidden="true" tabindex="-1"></a>   out_protein_min <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb94-81"><a href="#cb94-81" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb94-82"><a href="#cb94-82" aria-hidden="true" tabindex="-1"></a>     <span class="fu">distinct</span>(query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb94-83"><a href="#cb94-83" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, query_start) <span class="sc">%&gt;%</span></span>
<span id="cb94-84"><a href="#cb94-84" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_min</span>(mismatches, <span class="at">n=</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb94-85"><a href="#cb94-85" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() </span>
<span id="cb94-86"><a href="#cb94-86" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb94-87"><a href="#cb94-87" aria-hidden="true" tabindex="-1"></a>   out <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(out_protein_min<span class="sc">$</span>mismatches))</span>
<span id="cb94-88"><a href="#cb94-88" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(out) <span class="sc">==</span> <span class="dv">0</span>){out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="at">V2 =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))}</span>
<span id="cb94-89"><a href="#cb94-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(out) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb94-90"><a href="#cb94-90" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb94-91"><a href="#cb94-91" aria-hidden="true" tabindex="-1"></a>     out_protein_min <span class="ot">&lt;-</span> out_protein_min <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(query_seqname, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb94-92"><a href="#cb94-92" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb94-93"><a href="#cb94-93" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb94-94"><a href="#cb94-94" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb94-95"><a href="#cb94-95" aria-hidden="true" tabindex="-1"></a>     out_protein_count <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb94-96"><a href="#cb94-96" aria-hidden="true" tabindex="-1"></a>       <span class="fu">distinct</span>(query, query_seqname, query_start, mismatches, external_gene_name) <span class="sc">%&gt;%</span></span>
<span id="cb94-97"><a href="#cb94-97" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb94-98"><a href="#cb94-98" aria-hidden="true" tabindex="-1"></a>       <span class="fu">distinct</span>(query, query_seqname, query_start, external_gene_name, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb94-99"><a href="#cb94-99" aria-hidden="true" tabindex="-1"></a>       <span class="fu">filter</span>(<span class="sc">!</span>external_gene_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="cn">NA</span>, <span class="st">""</span>, <span class="st">" "</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb94-100"><a href="#cb94-100" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb94-101"><a href="#cb94-101" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb94-102"><a href="#cb94-102" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb94-103"><a href="#cb94-103" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-104"><a href="#cb94-104" aria-hidden="true" tabindex="-1"></a>  <span class="do">############## Count Proteome Length ##############</span></span>
<span id="cb94-105"><a href="#cb94-105" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb94-106"><a href="#cb94-106" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb94-107"><a href="#cb94-107" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb94-108"><a href="#cb94-108" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb94-109"><a href="#cb94-109" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb94-110"><a href="#cb94-110" aria-hidden="true" tabindex="-1"></a>       PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb94-111"><a href="#cb94-111" aria-hidden="true" tabindex="-1"></a>      PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb94-112"><a href="#cb94-112" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb94-113"><a href="#cb94-113" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-114"><a href="#cb94-114" aria-hidden="true" tabindex="-1"></a>  out_final <span class="ot">&lt;-</span> <span class="fu">merge</span>(out_final, out, <span class="at">by =</span> <span class="st">"mismatches"</span>, <span class="at">all.x =</span> T)</span>
<span id="cb94-115"><a href="#cb94-115" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb94-116"><a href="#cb94-116" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb94-117"><a href="#cb94-117" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-118"><a href="#cb94-118" aria-hidden="true" tabindex="-1"></a>   out_protein_tmp <span class="ot">&lt;-</span> out_protein <span class="sc">%&gt;%</span></span>
<span id="cb94-119"><a href="#cb94-119" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb94-120"><a href="#cb94-120" aria-hidden="true" tabindex="-1"></a>  out_protein_tmp_min <span class="ot">&lt;-</span> out_protein_min <span class="sc">%&gt;%</span></span>
<span id="cb94-121"><a href="#cb94-121" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb94-122"><a href="#cb94-122" aria-hidden="true" tabindex="-1"></a>  out_protein_tmp_count <span class="ot">&lt;-</span> out_protein_count <span class="sc">%&gt;%</span></span>
<span id="cb94-123"><a href="#cb94-123" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb94-124"><a href="#cb94-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-125"><a href="#cb94-125" aria-hidden="true" tabindex="-1"></a>  out_protein_final <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final, out_protein_tmp)</span>
<span id="cb94-126"><a href="#cb94-126" aria-hidden="true" tabindex="-1"></a>  out_protein_final_min <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final_min, out_protein_tmp_min)</span>
<span id="cb94-127"><a href="#cb94-127" aria-hidden="true" tabindex="-1"></a>  out_protein_final_count <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final_count, out_protein_tmp_count)</span>
<span id="cb94-128"><a href="#cb94-128" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb94-129"><a href="#cb94-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-130"><a href="#cb94-130" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb94-131"><a href="#cb94-131" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_final, <span class="st">"../Key_Data/12mer_proteome_everything_elmfilter_regex.csv"</span>)</span>
<span id="cb94-132"><a href="#cb94-132" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_protein_final, <span class="st">"../Key_Data/12mer_proteome_protein_everything_elmfilter_regex.csv"</span>)</span>
<span id="cb94-133"><a href="#cb94-133" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_protein_final_min, <span class="st">"../Key_Data/12mer_proteome_protein_everything_min_elmfilter_regex.csv"</span>)</span>
<span id="cb94-134"><a href="#cb94-134" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_protein_final_count, <span class="st">"../Key_Data/12mer_proteome_protein_everything_count_elmfilter_regex.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="mer-data-annotation-2" class="level2">
<h2 class="anchored" data-anchor-id="mer-data-annotation-2">18mer Data Annotation</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="do">## This will only run for files that haven't ran already</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../../epitope_alignment/Results/Complete_Final_Set_18mer_3mis/"</span>)</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".tsv"</span>, files_to_read)]</span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a>files_completed <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../../../Analysis/18mer_annotated/"</span>)</span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="sc">!</span><span class="fu">gsub</span>(<span class="st">".tsv"</span>, <span class="st">""</span>, files_to_read) <span class="sc">%in%</span> <span class="fu">gsub</span>(<span class="st">".csv.gz"</span>, <span class="st">""</span>, files_completed)]</span>
<span id="cb95-7"><a href="#cb95-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-8"><a href="#cb95-8" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID, <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb95-9"><a href="#cb95-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-10"><a href="#cb95-10" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">order</span>(<span class="fu">file.info</span>(files_to_read)<span class="sc">$</span>size)]</span>
<span id="cb95-11"><a href="#cb95-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-12"><a href="#cb95-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb95-13"><a href="#cb95-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb95-14"><a href="#cb95-14" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">nThread =</span> <span class="dv">12</span>)</span>
<span id="cb95-15"><a href="#cb95-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb95-16"><a href="#cb95-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Skipping "</span>, active_file, <span class="st">" due to lack of data."</span>))</span>
<span id="cb95-17"><a href="#cb95-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fwrite</span>(active_data, <span class="fu">paste0</span>(<span class="st">"../../../Analysis/18mer_annotated/"</span>, <span class="fu">gsub</span>(<span class="st">".tsv"</span>, <span class="st">".csv.gz"</span>, active_file)), <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">compress =</span> <span class="st">"gzip"</span>, <span class="at">nThread =</span> <span class="dv">12</span>)</span>
<span id="cb95-18"><a href="#cb95-18" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb95-19"><a href="#cb95-19" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb95-20"><a href="#cb95-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-21"><a href="#cb95-21" aria-hidden="true" tabindex="-1"></a> active_data[, uniprot <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(target_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb95-22"><a href="#cb95-22" aria-hidden="true" tabindex="-1"></a> active_data[, viral_id <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(query_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb95-23"><a href="#cb95-23" aria-hidden="true" tabindex="-1"></a> active_data[, mismatches <span class="sc">:</span><span class="er">=</span> <span class="fu">mismatch_calling_fast</span>(query, target), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb95-24"><a href="#cb95-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb95-25"><a href="#cb95-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Data Merging "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb95-26"><a href="#cb95-26" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb95-27"><a href="#cb95-27" aria-hidden="true" tabindex="-1"></a>     <span class="fu">lazy_dt</span>() <span class="sc">%&gt;%</span></span>
<span id="cb95-28"><a href="#cb95-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(mart_retrieved, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="st">"uniprot_gn_id"</span>),</span>
<span id="cb95-29"><a href="#cb95-29" aria-hidden="true" tabindex="-1"></a>              <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb95-30"><a href="#cb95-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(tissue_organ_map_condensed, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span> <span class="ot">=</span> <span class="st">"Gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb95-31"><a href="#cb95-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(human_protein_metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span><span class="st">"entry.human"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb95-32"><a href="#cb95-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(viral_protein_metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"viral_id"</span> <span class="ot">=</span> <span class="st">"entry.viral"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb95-33"><a href="#cb95-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">catalytic_activity.same =</span> catalytic_activity.human <span class="sc">==</span> catalytic_activity.viral) <span class="sc">%&gt;%</span></span>
<span id="cb95-34"><a href="#cb95-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.human =</span> <span class="fu">gsub</span>(<span class="st">";.*"</span>, <span class="st">""</span>, catalytic_activity.human)) <span class="sc">%&gt;%</span></span>
<span id="cb95-35"><a href="#cb95-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.viral =</span> <span class="fu">gsub</span>(<span class="st">";.*"</span>, <span class="st">""</span>, catalytic_activity.viral)) <span class="sc">%&gt;%</span></span>
<span id="cb95-36"><a href="#cb95-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.same =</span> reaction.human <span class="sc">==</span> reaction.viral) <span class="sc">%&gt;%</span></span>
<span id="cb95-37"><a href="#cb95-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.same =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(reaction.same), <span class="cn">FALSE</span>, reaction.same)) <span class="sc">%&gt;%</span></span>
<span id="cb95-38"><a href="#cb95-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">catalytic_activity.same =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(catalytic_activity.same), <span class="cn">FALSE</span>, catalytic_activity.same)) <span class="sc">%&gt;%</span></span>
<span id="cb95-39"><a href="#cb95-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">enzyme_number.same =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb95-40"><a href="#cb95-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.table</span>()</span>
<span id="cb95-41"><a href="#cb95-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb95-42"><a href="#cb95-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"EC Screening "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb95-43"><a href="#cb95-43" aria-hidden="true" tabindex="-1"></a>  active_data[, enzyme_number.same <span class="sc">:</span><span class="er">=</span> <span class="fu">fast_ec_screen</span>(ec_number.human, ec_number.viral), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb95-44"><a href="#cb95-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-45"><a href="#cb95-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Final Cleaning "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb95-46"><a href="#cb95-46" aria-hidden="true" tabindex="-1"></a>  active_data_subset <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span> </span>
<span id="cb95-47"><a href="#cb95-47" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">human_variants =</span> <span class="fu">sapply</span>(human_variants, toString)) <span class="sc">%&gt;%</span></span>
<span id="cb95-48"><a href="#cb95-48" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(query, query_seqname, query_start, target_seqname, target_start,</span>
<span id="cb95-49"><a href="#cb95-49" aria-hidden="true" tabindex="-1"></a>           target, mismatches, uniprot, viral_id, ensembl_gene_id,</span>
<span id="cb95-50"><a href="#cb95-50" aria-hidden="true" tabindex="-1"></a>           entrezgene_id, external_gene_name, uniprot_gn_symbol, chromosome_name,</span>
<span id="cb95-51"><a href="#cb95-51" aria-hidden="true" tabindex="-1"></a>           reviewed.human, entry_name.human, protein_names.human, length.human,</span>
<span id="cb95-52"><a href="#cb95-52" aria-hidden="true" tabindex="-1"></a>           active_site.human, binding_site.human, dna_binding.human,</span>
<span id="cb95-53"><a href="#cb95-53" aria-hidden="true" tabindex="-1"></a>           ec_number.human,  reviewed.viral, entry_name.viral,</span>
<span id="cb95-54"><a href="#cb95-54" aria-hidden="true" tabindex="-1"></a>           protein_names.viral, length.viral,</span>
<span id="cb95-55"><a href="#cb95-55" aria-hidden="true" tabindex="-1"></a>           active_site.viral, binding_site.viral, dna_binding.viral,</span>
<span id="cb95-56"><a href="#cb95-56" aria-hidden="true" tabindex="-1"></a>           ec_number.viral, catalytic_activity.same, reaction.human,</span>
<span id="cb95-57"><a href="#cb95-57" aria-hidden="true" tabindex="-1"></a>           reaction.viral, reaction.same,</span>
<span id="cb95-58"><a href="#cb95-58" aria-hidden="true" tabindex="-1"></a>           enzyme_number.same) <span class="sc">%&gt;%</span></span>
<span id="cb95-59"><a href="#cb95-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb95-60"><a href="#cb95-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Sys.time</span>()</span>
<span id="cb95-61"><a href="#cb95-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fwrite</span>(active_data_subset, <span class="fu">paste0</span>(<span class="st">"../../../Analysis/18mer_annotated/"</span>, <span class="fu">gsub</span>(<span class="st">".tsv"</span>, <span class="st">".csv.gz"</span>, active_file)), <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">compress =</span> <span class="st">"gzip"</span>, <span class="at">nThread =</span> <span class="dv">12</span>)</span>
<span id="cb95-62"><a href="#cb95-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(active_data, active_data_subset)</span>
<span id="cb95-63"><a href="#cb95-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gc</span>()</span>
<span id="cb95-64"><a href="#cb95-64" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="mer-data-load-and-processing-2" class="level2">
<h2 class="anchored" data-anchor-id="mer-data-load-and-processing-2">18mer Data Load and Processing</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="do">### This only needs to be run once and will take a while, change eval to true if you want to run</span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(seqinr)</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringi)</span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(svMisc)</span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../18mer_annotated/"</span>)</span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".csv.gz"</span>, files_to_read)]</span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"Number of Proteins"</span>, <span class="st">"Proteome Length"</span>))</span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a>out_protein_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a>out_protein_final_min <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a>out_protein_final_count <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb96-16"><a href="#cb96-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" Pathogen number:"</span>,<span class="fu">match</span>(active_file, files_to_read), <span class="st">"/"</span>, <span class="fu">length</span>(files_to_read)))</span>
<span id="cb96-17"><a href="#cb96-17" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">nThread =</span> <span class="dv">18</span>) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb96-18"><a href="#cb96-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-19"><a href="#cb96-19" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb96-20"><a href="#cb96-20" aria-hidden="true" tabindex="-1"></a>    out_final[,<span class="fu">gsub</span>(<span class="st">"_18mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file)] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="cn">NA</span>,<span class="cn">NA</span>)</span>
<span id="cb96-21"><a href="#cb96-21" aria-hidden="true" tabindex="-1"></a>      data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>, <span class="fu">gsub</span>(<span class="st">"_18mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb96-22"><a href="#cb96-22" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb96-23"><a href="#cb96-23" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb96-24"><a href="#cb96-24" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb96-25"><a href="#cb96-25" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb96-26"><a href="#cb96-26" aria-hidden="true" tabindex="-1"></a>      PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb96-27"><a href="#cb96-27" aria-hidden="true" tabindex="-1"></a>      PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb96-28"><a href="#cb96-28" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb96-29"><a href="#cb96-29" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb96-30"><a href="#cb96-30" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb96-31"><a href="#cb96-31" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb96-32"><a href="#cb96-32" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb96-33"><a href="#cb96-33" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb96-34"><a href="#cb96-34" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb96-35"><a href="#cb96-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-36"><a href="#cb96-36" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data  <span class="sc">%&gt;%</span></span>
<span id="cb96-37"><a href="#cb96-37" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Remove duplicated pathogen proteins identified in </span></span>
<span id="cb96-38"><a href="#cb96-38" aria-hidden="true" tabindex="-1"></a>    <span class="do">## ../../Scripts/Duplicate_Protein_Screen.R</span></span>
<span id="cb96-39"><a href="#cb96-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb96-40"><a href="#cb96-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb96-41"><a href="#cb96-41" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(query, query_seqname, query_start, mismatches, reaction.human,</span>
<span id="cb96-42"><a href="#cb96-42" aria-hidden="true" tabindex="-1"></a>           reaction.viral, reaction.same, catalytic_activity.same,</span>
<span id="cb96-43"><a href="#cb96-43" aria-hidden="true" tabindex="-1"></a>           ec_number.human, ec_number.viral,</span>
<span id="cb96-44"><a href="#cb96-44" aria-hidden="true" tabindex="-1"></a>           enzyme_number.same, external_gene_name)</span>
<span id="cb96-45"><a href="#cb96-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-46"><a href="#cb96-46" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>reaction.same[<span class="fu">is.na</span>(active_data<span class="sc">$</span>reaction.same)] <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb96-47"><a href="#cb96-47" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>catalytic_activity.same[<span class="fu">is.na</span>(active_data<span class="sc">$</span>catalytic_activity.same)] <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb96-48"><a href="#cb96-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-49"><a href="#cb96-49" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(active_data)</span>
<span id="cb96-50"><a href="#cb96-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-51"><a href="#cb96-51" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb96-52"><a href="#cb96-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(reaction.same <span class="sc">!=</span> <span class="cn">TRUE</span> <span class="sc">&amp;</span> catalytic_activity.same <span class="sc">!=</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb96-53"><a href="#cb96-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(enzyme_number.same <span class="sc">!=</span> <span class="cn">TRUE</span>)  </span>
<span id="cb96-54"><a href="#cb96-54" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb96-55"><a href="#cb96-55" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(active_data<span class="sc">$</span>mismatches))</span>
<span id="cb96-56"><a href="#cb96-56" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(out) <span class="sc">==</span> <span class="dv">0</span>){out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="at">V2 =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))}</span>
<span id="cb96-57"><a href="#cb96-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(out) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>, <span class="fu">gsub</span>(<span class="st">"_18mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb96-58"><a href="#cb96-58" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-59"><a href="#cb96-59" aria-hidden="true" tabindex="-1"></a>  out_protein <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb96-60"><a href="#cb96-60" aria-hidden="true" tabindex="-1"></a>     <span class="fu">distinct</span>(query, query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb96-61"><a href="#cb96-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb96-62"><a href="#cb96-62" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb96-63"><a href="#cb96-63" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb96-64"><a href="#cb96-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb96-65"><a href="#cb96-65" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-66"><a href="#cb96-66" aria-hidden="true" tabindex="-1"></a>   out_protein_min <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb96-67"><a href="#cb96-67" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb96-68"><a href="#cb96-68" aria-hidden="true" tabindex="-1"></a>     <span class="fu">distinct</span>(query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb96-69"><a href="#cb96-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, query_start) <span class="sc">%&gt;%</span></span>
<span id="cb96-70"><a href="#cb96-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_min</span>(mismatches, <span class="at">n=</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb96-71"><a href="#cb96-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() </span>
<span id="cb96-72"><a href="#cb96-72" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb96-73"><a href="#cb96-73" aria-hidden="true" tabindex="-1"></a>   out <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(out_protein_min<span class="sc">$</span>mismatches))</span>
<span id="cb96-74"><a href="#cb96-74" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(out) <span class="sc">==</span> <span class="dv">0</span>){out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="at">V2 =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))}</span>
<span id="cb96-75"><a href="#cb96-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(out) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>, <span class="fu">gsub</span>(<span class="st">"_18mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb96-76"><a href="#cb96-76" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb96-77"><a href="#cb96-77" aria-hidden="true" tabindex="-1"></a>     out_protein_min <span class="ot">&lt;-</span> out_protein_min <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(query_seqname, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb96-78"><a href="#cb96-78" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb96-79"><a href="#cb96-79" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb96-80"><a href="#cb96-80" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb96-81"><a href="#cb96-81" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb96-82"><a href="#cb96-82" aria-hidden="true" tabindex="-1"></a>     out_protein_count <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb96-83"><a href="#cb96-83" aria-hidden="true" tabindex="-1"></a>       <span class="fu">distinct</span>(query, query_seqname, query_start, mismatches, external_gene_name) <span class="sc">%&gt;%</span></span>
<span id="cb96-84"><a href="#cb96-84" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb96-85"><a href="#cb96-85" aria-hidden="true" tabindex="-1"></a>       <span class="fu">distinct</span>(query, query_seqname, query_start, external_gene_name, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb96-86"><a href="#cb96-86" aria-hidden="true" tabindex="-1"></a>       <span class="fu">filter</span>(<span class="sc">!</span>external_gene_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="cn">NA</span>, <span class="st">""</span>, <span class="st">" "</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb96-87"><a href="#cb96-87" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb96-88"><a href="#cb96-88" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb96-89"><a href="#cb96-89" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb96-90"><a href="#cb96-90" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-91"><a href="#cb96-91" aria-hidden="true" tabindex="-1"></a>  <span class="do">############## Count Proteome Length ##############</span></span>
<span id="cb96-92"><a href="#cb96-92" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>, <span class="fu">gsub</span>(<span class="st">"_18mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb96-93"><a href="#cb96-93" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb96-94"><a href="#cb96-94" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb96-95"><a href="#cb96-95" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb96-96"><a href="#cb96-96" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb96-97"><a href="#cb96-97" aria-hidden="true" tabindex="-1"></a>       PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb96-98"><a href="#cb96-98" aria-hidden="true" tabindex="-1"></a>      PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb96-99"><a href="#cb96-99" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb96-100"><a href="#cb96-100" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-101"><a href="#cb96-101" aria-hidden="true" tabindex="-1"></a>  out_final <span class="ot">&lt;-</span> <span class="fu">merge</span>(out_final, out, <span class="at">by =</span> <span class="st">"mismatches"</span>, <span class="at">all.x =</span> T)</span>
<span id="cb96-102"><a href="#cb96-102" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb96-103"><a href="#cb96-103" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb96-104"><a href="#cb96-104" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-105"><a href="#cb96-105" aria-hidden="true" tabindex="-1"></a>   out_protein_tmp <span class="ot">&lt;-</span> out_protein <span class="sc">%&gt;%</span></span>
<span id="cb96-106"><a href="#cb96-106" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb96-107"><a href="#cb96-107" aria-hidden="true" tabindex="-1"></a>  out_protein_tmp_min <span class="ot">&lt;-</span> out_protein_min <span class="sc">%&gt;%</span></span>
<span id="cb96-108"><a href="#cb96-108" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb96-109"><a href="#cb96-109" aria-hidden="true" tabindex="-1"></a>  out_protein_tmp_count <span class="ot">&lt;-</span> out_protein_count <span class="sc">%&gt;%</span></span>
<span id="cb96-110"><a href="#cb96-110" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb96-111"><a href="#cb96-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-112"><a href="#cb96-112" aria-hidden="true" tabindex="-1"></a>  out_protein_final <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final, out_protein_tmp)</span>
<span id="cb96-113"><a href="#cb96-113" aria-hidden="true" tabindex="-1"></a>  out_protein_final_min <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final_min, out_protein_tmp_min)</span>
<span id="cb96-114"><a href="#cb96-114" aria-hidden="true" tabindex="-1"></a>  out_protein_final_count <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final_count, out_protein_tmp_count)</span>
<span id="cb96-115"><a href="#cb96-115" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb96-116"><a href="#cb96-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-117"><a href="#cb96-117" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb96-118"><a href="#cb96-118" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_final, <span class="st">"../Key_Data/18mer_proteome_everything.csv"</span>)</span>
<span id="cb96-119"><a href="#cb96-119" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_protein_final, <span class="st">"../Key_Data/18mer_proteome_protein_everything.csv"</span>)</span>
<span id="cb96-120"><a href="#cb96-120" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_protein_final_min, <span class="st">"../Key_Data/18mer_proteome_protein_everything_min.csv"</span>)</span>
<span id="cb96-121"><a href="#cb96-121" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_protein_final_count, <span class="st">"../Key_Data/18mer_proteome_protein_everything_count.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="panel-b" class="level2">
<h2 class="anchored" data-anchor-id="panel-b">Panel B</h2>
<section id="panel-b-12mers" class="level4">
<h4 class="anchored" data-anchor-id="panel-b-12mers">Panel B 12mers</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/12mer_proteome_everything.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>out_protein_final <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/12mer_proteome_protein_everything_count.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final)</span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb97-10"><a href="#cb97-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb97-11"><a href="#cb97-11" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb97-12"><a href="#cb97-12" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb97-13"><a href="#cb97-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb97-14"><a href="#cb97-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb97-15"><a href="#cb97-15" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb97-16"><a href="#cb97-16" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb97-17"><a href="#cb97-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb97-18"><a href="#cb97-18" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb97-19"><a href="#cb97-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb97-20"><a href="#cb97-20" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb97-21"><a href="#cb97-21" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb97-22"><a href="#cb97-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-23"><a href="#cb97-23" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">12-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb97-24"><a href="#cb97-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">12-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb97-25"><a href="#cb97-25" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">12-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb97-26"><a href="#cb97-26" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">12-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb97-27"><a href="#cb97-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-28"><a href="#cb97-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-29"><a href="#cb97-29" aria-hidden="true" tabindex="-1"></a>heatdf <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data_meta[,<span class="fu">grepl</span>(<span class="st">"_corrected"</span>, <span class="fu">colnames</span>(mismatch_data_meta))])</span>
<span id="cb97-30"><a href="#cb97-30" aria-hidden="true" tabindex="-1"></a>heatdf <span class="ot">&lt;-</span> <span class="fu">apply</span>(heatdf, <span class="dv">2</span>, scale)</span>
<span id="cb97-31"><a href="#cb97-31" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heatdf) <span class="ot">&lt;-</span> mismatch_data_meta<span class="sc">$</span>ID</span>
<span id="cb97-32"><a href="#cb97-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-33"><a href="#cb97-33" aria-hidden="true" tabindex="-1"></a>gg2 <span class="ot">&lt;-</span> heatdf <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb97-34"><a href="#cb97-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"proteome_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb97-35"><a href="#cb97-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>proteome_id) <span class="sc">%&gt;%</span></span>
<span id="cb97-36"><a href="#cb97-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"proteome_id"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb97-37"><a href="#cb97-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb97-38"><a href="#cb97-38" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">proteome_id</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb97-39"><a href="#cb97-39" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb97-40"><a href="#cb97-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name))  <span class="sc">%&gt;%</span></span>
<span id="cb97-41"><a href="#cb97-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">""</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb97-42"><a href="#cb97-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb97-43"><a href="#cb97-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(name, label, <span class="at">fill=</span>value)) <span class="sc">+</span> <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb97-44"><a href="#cb97-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb97-45"><a href="#cb97-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Number of Mismatches"</span>) <span class="sc">+</span></span>
<span id="cb97-46"><a href="#cb97-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">colours =</span> <span class="fu">c</span>(<span class="st">"#045375"</span>, <span class="st">"#089099"</span>, <span class="st">"#7CCBA2"</span>,</span>
<span id="cb97-47"><a href="#cb97-47" aria-hidden="true" tabindex="-1"></a>                                   <span class="st">"#FCDE9C"</span>, <span class="st">"#FAB27B"</span>, <span class="st">"#E23333"</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">4</span>,<span class="fl">6.7</span>),</span>
<span id="cb97-48"><a href="#cb97-48" aria-hidden="true" tabindex="-1"></a>                       <span class="at">na.value =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb97-49"><a href="#cb97-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb97-50"><a href="#cb97-50" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb97-51"><a href="#cb97-51" aria-hidden="true" tabindex="-1"></a>        <span class="co">#axis.text.y =element_text(size=7),</span></span>
<span id="cb97-52"><a href="#cb97-52" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">9</span>),</span>
<span id="cb97-53"><a href="#cb97-53" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">fill=</span><span class="cn">NA</span>, <span class="at">size=</span><span class="dv">1</span>))</span>
<span id="cb97-54"><a href="#cb97-54" aria-hidden="true" tabindex="-1"></a>gg2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/panel_b_2-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="panel-b-18mers" class="level4">
<h4 class="anchored" data-anchor-id="panel-b-18mers">Panel B 18mers</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>out_final_18mer <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/18mer_proteome_everything.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>) </span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>out_protein_final_18mer <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/18mer_proteome_protein_everything.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"Proteome ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final_18mer)</span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb98-7"><a href="#cb98-7" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb98-8"><a href="#cb98-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-9"><a href="#cb98-9" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb98-10"><a href="#cb98-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb98-11"><a href="#cb98-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb98-12"><a href="#cb98-12" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb98-13"><a href="#cb98-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb98-14"><a href="#cb98-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb98-15"><a href="#cb98-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb98-16"><a href="#cb98-16" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb98-17"><a href="#cb98-17" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb98-18"><a href="#cb98-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb98-19"><a href="#cb98-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb98-20"><a href="#cb98-20" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb98-21"><a href="#cb98-21" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb98-22"><a href="#cb98-22" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb98-23"><a href="#cb98-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-24"><a href="#cb98-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">18-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb98-25"><a href="#cb98-25" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">18-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb98-26"><a href="#cb98-26" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">18-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb98-27"><a href="#cb98-27" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">18-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb98-28"><a href="#cb98-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-29"><a href="#cb98-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-30"><a href="#cb98-30" aria-hidden="true" tabindex="-1"></a>heatdf <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data_meta[,<span class="fu">grepl</span>(<span class="st">"_corrected"</span>, <span class="fu">colnames</span>(mismatch_data_meta))])</span>
<span id="cb98-31"><a href="#cb98-31" aria-hidden="true" tabindex="-1"></a>heatdf <span class="ot">&lt;-</span> <span class="fu">apply</span>(heatdf, <span class="dv">2</span>, scale)</span>
<span id="cb98-32"><a href="#cb98-32" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heatdf) <span class="ot">&lt;-</span> mismatch_data_meta<span class="sc">$</span>ID</span>
<span id="cb98-33"><a href="#cb98-33" aria-hidden="true" tabindex="-1"></a><span class="co">#heatdf[heatdf==0]&lt;-NA</span></span>
<span id="cb98-34"><a href="#cb98-34" aria-hidden="true" tabindex="-1"></a>gg2_18mer <span class="ot">&lt;-</span> heatdf <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb98-35"><a href="#cb98-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"proteome_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb98-36"><a href="#cb98-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>proteome_id) <span class="sc">%&gt;%</span></span>
<span id="cb98-37"><a href="#cb98-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"proteome_id"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb98-38"><a href="#cb98-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb98-39"><a href="#cb98-39" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">proteome_id</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb98-40"><a href="#cb98-40" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb98-41"><a href="#cb98-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb98-42"><a href="#cb98-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">""</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb98-43"><a href="#cb98-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb98-44"><a href="#cb98-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(name, label, <span class="at">fill=</span>value)) <span class="sc">+</span> <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb98-45"><a href="#cb98-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb98-46"><a href="#cb98-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Number of Mismatches"</span>) <span class="sc">+</span></span>
<span id="cb98-47"><a href="#cb98-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">colours =</span> <span class="fu">c</span>(<span class="st">"#045375"</span>, <span class="st">"#089099"</span>, <span class="st">"#7CCBA2"</span>,</span>
<span id="cb98-48"><a href="#cb98-48" aria-hidden="true" tabindex="-1"></a>                                   <span class="st">"#FCDE9C"</span>, <span class="st">"#FAB27B"</span>, <span class="st">"#E23333"</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">4</span>,<span class="fl">6.7</span>),</span>
<span id="cb98-49"><a href="#cb98-49" aria-hidden="true" tabindex="-1"></a>                       <span class="at">na.value =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb98-50"><a href="#cb98-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb98-51"><a href="#cb98-51" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb98-52"><a href="#cb98-52" aria-hidden="true" tabindex="-1"></a>        <span class="co">#axis.text.y = element_text(size=7),</span></span>
<span id="cb98-53"><a href="#cb98-53" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">9</span>),</span>
<span id="cb98-54"><a href="#cb98-54" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">fill=</span><span class="cn">NA</span>, <span class="at">size=</span><span class="dv">1</span>))</span>
<span id="cb98-55"><a href="#cb98-55" aria-hidden="true" tabindex="-1"></a>gg2_18mer</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/unnamed-chunk-1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="panel-b-8mers" class="level4">
<h4 class="anchored" data-anchor-id="panel-b-8mers">Panel B 8mers</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/8mer_proteome_everything.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final)</span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb99-10"><a href="#cb99-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb99-11"><a href="#cb99-11" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic))<span class="sc">%&gt;%</span></span>
<span id="cb99-12"><a href="#cb99-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb99-13"><a href="#cb99-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb99-14"><a href="#cb99-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb99-15"><a href="#cb99-15" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb99-16"><a href="#cb99-16" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb99-17"><a href="#cb99-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb99-18"><a href="#cb99-18" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb99-19"><a href="#cb99-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb99-20"><a href="#cb99-20" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb99-21"><a href="#cb99-21" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb99-22"><a href="#cb99-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-23"><a href="#cb99-23" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">8-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb99-24"><a href="#cb99-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">8-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb99-25"><a href="#cb99-25" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">8-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb99-26"><a href="#cb99-26" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">8-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb99-27"><a href="#cb99-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-28"><a href="#cb99-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-29"><a href="#cb99-29" aria-hidden="true" tabindex="-1"></a>heatdf <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data_meta[,<span class="fu">grepl</span>(<span class="st">"_corrected"</span>, <span class="fu">colnames</span>(mismatch_data_meta))])</span>
<span id="cb99-30"><a href="#cb99-30" aria-hidden="true" tabindex="-1"></a>heatdf <span class="ot">&lt;-</span> <span class="fu">apply</span>(heatdf, <span class="dv">2</span>, scale)</span>
<span id="cb99-31"><a href="#cb99-31" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heatdf) <span class="ot">&lt;-</span> mismatch_data_meta<span class="sc">$</span>ID</span>
<span id="cb99-32"><a href="#cb99-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-33"><a href="#cb99-33" aria-hidden="true" tabindex="-1"></a>gg2_8mers <span class="ot">&lt;-</span> heatdf <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb99-34"><a href="#cb99-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"proteome_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb99-35"><a href="#cb99-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>proteome_id) <span class="sc">%&gt;%</span></span>
<span id="cb99-36"><a href="#cb99-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"proteome_id"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb99-37"><a href="#cb99-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb99-38"><a href="#cb99-38" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">proteome_id</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb99-39"><a href="#cb99-39" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb99-40"><a href="#cb99-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name))  <span class="sc">%&gt;%</span></span>
<span id="cb99-41"><a href="#cb99-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">""</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb99-42"><a href="#cb99-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb99-43"><a href="#cb99-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(name, label, <span class="at">fill=</span>value)) <span class="sc">+</span> <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb99-44"><a href="#cb99-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb99-45"><a href="#cb99-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Number of Mismatches"</span>) <span class="sc">+</span></span>
<span id="cb99-46"><a href="#cb99-46" aria-hidden="true" tabindex="-1"></a>   <span class="co">#scale_fill_gradient(low = "yellow", high = "purple", na.value = NA)</span></span>
<span id="cb99-47"><a href="#cb99-47" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_fill_gradientn(colours = c("blue", "purple", "yellow", "red")) +</span></span>
<span id="cb99-48"><a href="#cb99-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">colours =</span> <span class="fu">c</span>(<span class="st">"#045375"</span>, <span class="st">"#089099"</span>, <span class="st">"#7CCBA2"</span>,</span>
<span id="cb99-49"><a href="#cb99-49" aria-hidden="true" tabindex="-1"></a>                                   <span class="st">"#FCDE9C"</span>, <span class="st">"#FAB27B"</span>, <span class="st">"#E23333"</span>),</span>
<span id="cb99-50"><a href="#cb99-50" aria-hidden="true" tabindex="-1"></a>                       <span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">4</span>,<span class="fl">6.7</span>), <span class="at">oob=</span>squish,</span>
<span id="cb99-51"><a href="#cb99-51" aria-hidden="true" tabindex="-1"></a>                       <span class="at">na.value =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb99-52"><a href="#cb99-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb99-53"><a href="#cb99-53" aria-hidden="true" tabindex="-1"></a>        <span class="co">#axis.text.y = element_blank(),</span></span>
<span id="cb99-54"><a href="#cb99-54" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">7</span>),</span>
<span id="cb99-55"><a href="#cb99-55" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">9</span>),</span>
<span id="cb99-56"><a href="#cb99-56" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">fill=</span><span class="cn">NA</span>, <span class="at">size=</span><span class="dv">1</span>))</span>
<span id="cb99-57"><a href="#cb99-57" aria-hidden="true" tabindex="-1"></a>gg2_8mers</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="panel-assembly" class="level4">
<h4 class="anchored" data-anchor-id="panel-assembly">Panel Assembly</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>taxa <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Proteome.ID <span class="sc">%in%</span> out_protein_final<span class="sc">$</span><span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(family, order, class, phylum, kingdom) <span class="sc">%&gt;%</span></span>
<span id="cb100-9"><a href="#cb100-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb100-10"><a href="#cb100-10" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Anelloviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span id="cb100-11"><a href="#cb100-11" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Kolmioviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Kolmioviridae"</span></span>
<span id="cb100-12"><a href="#cb100-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-13"><a href="#cb100-13" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">lapply</span>(taxa, as.factor), <span class="at">stringsAsFactors =</span><span class="cn">TRUE</span>)</span>
<span id="cb100-14"><a href="#cb100-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(check) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(taxa)</span>
<span id="cb100-15"><a href="#cb100-15" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> check[<span class="fu">complete.cases</span>(check),]</span>
<span id="cb100-16"><a href="#cb100-16" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">taxo2phylog</span>(<span class="fu">as.taxo</span>(check), <span class="at">root=</span><span class="st">"Root"</span>, <span class="at">abbrev=</span>F)</span>
<span id="cb100-17"><a href="#cb100-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-18"><a href="#cb100-18" aria-hidden="true" tabindex="-1"></a>tax.phy <span class="ot">&lt;-</span> <span class="fu">as.phylo</span>(tmp)</span>
<span id="cb100-19"><a href="#cb100-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-20"><a href="#cb100-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-21"><a href="#cb100-21" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tax.phy, <span class="at">branch.length=</span><span class="st">'none'</span>)</span>
<span id="cb100-22"><a href="#cb100-22" aria-hidden="true" tabindex="-1"></a>gg_tree</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/panel_b_5-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>gg_tree[[<span class="st">"data"</span>]] <span class="ot">&lt;-</span> gg_tree[[<span class="st">"data"</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata <span class="sc">%&gt;%</span></span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a>               <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb101-5"><a href="#cb101-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb101-6"><a href="#cb101-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb101-7"><a href="#cb101-7" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(Taxon.mnemonic, family), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>))</span>
<span id="cb101-8"><a href="#cb101-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-9"><a href="#cb101-9" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span>  gg_tree <span class="sc">+</span> <span class="fu">geom_hilight</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">subset =</span> family <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2),<span class="at">fill =</span> family), <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb101-10"><a href="#cb101-10" aria-hidden="true" tabindex="-1"></a>   <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> Family_palette_v2)</span>
<span id="cb101-11"><a href="#cb101-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-12"><a href="#cb101-12" aria-hidden="true" tabindex="-1"></a>panel_b <span class="ot">&lt;-</span> gg2 <span class="sc">%&gt;%</span> <span class="fu">insert_left</span>(gg2_8mers) <span class="sc">%&gt;%</span></span>
<span id="cb101-13"><a href="#cb101-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">insert_left</span>(gg_tree, <span class="at">width =</span> <span class="fl">0.7</span>) <span class="sc">%&gt;%</span></span>
<span id="cb101-14"><a href="#cb101-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">insert_right</span>(gg2_18mer)</span>
<span id="cb101-15"><a href="#cb101-15" aria-hidden="true" tabindex="-1"></a>panel_b</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/panel_b_5-2.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"Figure1B.pdf"</span>), <span class="at">height =</span> <span class="dv">17</span>, <span class="at">width =</span> <span class="dv">7</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="panel-c" class="level2">
<h2 class="anchored" data-anchor-id="panel-c">Panel C</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a>out_final<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/8mer_proteome_everything.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final)</span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb103-4"><a href="#cb103-4" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb103-5"><a href="#cb103-5" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb103-6"><a href="#cb103-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-7"><a href="#cb103-7" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb103-8"><a href="#cb103-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb103-9"><a href="#cb103-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb103-10"><a href="#cb103-10" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic))<span class="sc">%&gt;%</span></span>
<span id="cb103-11"><a href="#cb103-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb103-12"><a href="#cb103-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb103-13"><a href="#cb103-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb103-14"><a href="#cb103-14" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb103-15"><a href="#cb103-15" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb103-16"><a href="#cb103-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb103-17"><a href="#cb103-17" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb103-18"><a href="#cb103-18" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb103-19"><a href="#cb103-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb103-20"><a href="#cb103-20" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb103-21"><a href="#cb103-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-22"><a href="#cb103-22" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">8-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb103-23"><a href="#cb103-23" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">8-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb103-24"><a href="#cb103-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">8-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb103-25"><a href="#cb103-25" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">8-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb103-26"><a href="#cb103-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-27"><a href="#cb103-27" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(mismatch_data_meta, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"8mers"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb103-28"><a href="#cb103-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-29"><a href="#cb103-29" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb103-30"><a href="#cb103-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"Virus"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb103-31"><a href="#cb103-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb103-32"><a href="#cb103-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>))</span>
<span id="cb103-33"><a href="#cb103-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-34"><a href="#cb103-34" aria-hidden="true" tabindex="-1"></a>wt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Chronic.or.Acute,</span>
<span id="cb103-35"><a href="#cb103-35" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb103-36"><a href="#cb103-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb103-37"><a href="#cb103-37" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb103-38"><a href="#cb103-38" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"8mer_Chronic_vs_Acute_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb103-39"><a href="#cb103-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-40"><a href="#cb103-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-41"><a href="#cb103-41" aria-hidden="true" tabindex="-1"></a>panel_c <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb103-42"><a href="#cb103-42" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(grepl("Virus", Taxonomic.lineage)) %&gt;%</span></span>
<span id="cb103-43"><a href="#cb103-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb103-44"><a href="#cb103-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">"Bacteria"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb103-45"><a href="#cb103-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Chronic.or.Acute =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"Bacteria"</span>, Taxonomic.lineage), <span class="st">"Bacteria"</span>, Chronic.or.Acute)) <span class="sc">%&gt;%</span></span>
<span id="cb103-46"><a href="#cb103-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>, name) <span class="sc">%&gt;%</span></span>
<span id="cb103-47"><a href="#cb103-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(value), <span class="dv">0</span>, value)) <span class="sc">%&gt;%</span></span>
<span id="cb103-48"><a href="#cb103-48" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(value), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb103-49"><a href="#cb103-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb103-50"><a href="#cb103-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb103-51"><a href="#cb103-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">""</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb103-52"><a href="#cb103-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb103-53"><a href="#cb103-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb103-54"><a href="#cb103-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb103-55"><a href="#cb103-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb103-56"><a href="#cb103-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log10_trans</span>(),</span>
<span id="cb103-57"><a href="#cb103-57" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log10"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">10</span><span class="sc">^</span>x, <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb103-58"><a href="#cb103-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb103-59"><a href="#cb103-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>name), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb103-60"><a href="#cb103-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb103-61"><a href="#cb103-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span><span class="fu">c</span>(<span class="st">"#1975EE"</span>, <span class="st">"#E78305"</span>, <span class="st">"black"</span>)) <span class="sc">+</span></span>
<span id="cb103-62"><a href="#cb103-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent of 8mers (%)"</span>) <span class="sc">+</span></span>
<span id="cb103-63"><a href="#cb103-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">size =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb103-64"><a href="#cb103-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="fl">1.7</span>, <span class="dv">40</span>, <span class="dv">40</span>),</span>
<span id="cb103-65"><a href="#cb103-65" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb103-66"><a href="#cb103-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb103-67"><a href="#cb103-67" aria-hidden="true" tabindex="-1"></a>panel_c</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/panel_c-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(panel_c<span class="sc">$</span>data, <span class="fu">paste0</span>(out_dir, <span class="st">"Figure1c_datapoints.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="panel-d" class="level2">
<h2 class="anchored" data-anchor-id="panel-d">Panel D</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final)</span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-6"><a href="#cb105-6" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb105-7"><a href="#cb105-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb105-8"><a href="#cb105-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb105-9"><a href="#cb105-9" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic))<span class="sc">%&gt;%</span></span>
<span id="cb105-10"><a href="#cb105-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb105-11"><a href="#cb105-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb105-12"><a href="#cb105-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb105-13"><a href="#cb105-13" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb105-14"><a href="#cb105-14" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb105-15"><a href="#cb105-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb105-16"><a href="#cb105-16" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb105-17"><a href="#cb105-17" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb105-18"><a href="#cb105-18" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb105-19"><a href="#cb105-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb105-20"><a href="#cb105-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-21"><a href="#cb105-21" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">8-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb105-22"><a href="#cb105-22" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">8-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb105-23"><a href="#cb105-23" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">8-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb105-24"><a href="#cb105-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">8-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb105-25"><a href="#cb105-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-26"><a href="#cb105-26" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb105-27"><a href="#cb105-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb105-28"><a href="#cb105-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb105-29"><a href="#cb105-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb105-30"><a href="#cb105-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Herpesviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Herpesviridae"</span>, <span class="st">"Herpesviridae"</span>, <span class="st">"Non-Herpesviridae"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb105-31"><a href="#cb105-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Poxviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Poxviridae"</span>, <span class="st">"Poxviridae"</span>, <span class="st">"Non-Poxviridae"</span>))</span>
<span id="cb105-32"><a href="#cb105-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-33"><a href="#cb105-33" aria-hidden="true" tabindex="-1"></a><span class="do">## These results are used to manually add stars to the graph below in illustrator</span></span>
<span id="cb105-34"><a href="#cb105-34" aria-hidden="true" tabindex="-1"></a>kt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Family,</span>
<span id="cb105-35"><a href="#cb105-35" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"kruskal.test"</span>)</span>
<span id="cb105-36"><a href="#cb105-36" aria-hidden="true" tabindex="-1"></a>wt.p <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Poxviridae,</span>
<span id="cb105-37"><a href="#cb105-37" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb105-38"><a href="#cb105-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb105-39"><a href="#cb105-39" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb105-40"><a href="#cb105-40" aria-hidden="true" tabindex="-1"></a>wt.h <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Herpesviridae,</span>
<span id="cb105-41"><a href="#cb105-41" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb105-42"><a href="#cb105-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb105-43"><a href="#cb105-43" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb105-44"><a href="#cb105-44" aria-hidden="true" tabindex="-1"></a>wt.hp <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Family <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Herpesviridae"</span>, <span class="st">"Poxviridae"</span>)), <span class="at">formula =</span> value <span class="sc">~</span> Poxviridae,</span>
<span id="cb105-45"><a href="#cb105-45" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb105-46"><a href="#cb105-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb105-47"><a href="#cb105-47" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>)))) <span class="sc">%&gt;%</span></span>
<span id="cb105-48"><a href="#cb105-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group1 =</span> <span class="st">"Herpesviridae"</span>)</span>
<span id="cb105-49"><a href="#cb105-49" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(kt, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"8mer_Viral_Family_Kruskal"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb105-50"><a href="#cb105-50" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.p, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"8mer_Pox_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb105-51"><a href="#cb105-51" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.h, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"8mer_Herpes_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb105-52"><a href="#cb105-52" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.hp, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"8mer_Herpes_vs_Pox_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb105-53"><a href="#cb105-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-54"><a href="#cb105-54" aria-hidden="true" tabindex="-1"></a>panel_d <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb105-55"><a href="#cb105-55" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(Family = ifelse(grepl("Bacteria", Taxonomic.lineage), "Bacteria", Family_Viral_Zone)) %&gt;%</span></span>
<span id="cb105-56"><a href="#cb105-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb105-57"><a href="#cb105-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb105-58"><a href="#cb105-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb105-59"><a href="#cb105-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Family, name) <span class="sc">%&gt;%</span></span>
<span id="cb105-60"><a href="#cb105-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(value), <span class="dv">0</span>, value)) <span class="sc">%&gt;%</span></span>
<span id="cb105-61"><a href="#cb105-61" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(value), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb105-62"><a href="#cb105-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(count <span class="sc">&gt;</span> <span class="dv">4</span>) <span class="sc">%&gt;%</span></span>
<span id="cb105-63"><a href="#cb105-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb105-64"><a href="#cb105-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb105-65"><a href="#cb105-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">""</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb105-66"><a href="#cb105-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb105-67"><a href="#cb105-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb105-68"><a href="#cb105-68" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb105-69"><a href="#cb105-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb105-70"><a href="#cb105-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log10_trans</span>(),</span>
<span id="cb105-71"><a href="#cb105-71" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log10"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">10</span><span class="sc">^</span>x, <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb105-72"><a href="#cb105-72" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb105-73"><a href="#cb105-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>name), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb105-74"><a href="#cb105-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb105-75"><a href="#cb105-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span>Family_palette_v2) <span class="sc">+</span></span>
<span id="cb105-76"><a href="#cb105-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent of 8mers (%)"</span>) <span class="sc">+</span></span>
<span id="cb105-77"><a href="#cb105-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">size =</span> <span class="fl">0.1</span>)  <span class="sc">+</span></span>
<span id="cb105-78"><a href="#cb105-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">0.9</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="fl">3.9</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">40</span>, <span class="dv">40</span>, <span class="dv">13</span>),</span>
<span id="cb105-79"><a href="#cb105-79" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.h<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>,</span>
<span id="cb105-80"><a href="#cb105-80" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Herpesviridae"</span>]) <span class="sc">+</span></span>
<span id="cb105-81"><a href="#cb105-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">1.1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="fl">4.1</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">40</span>, <span class="dv">40</span>, <span class="dv">13</span>),</span>
<span id="cb105-82"><a href="#cb105-82" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.p<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>,</span>
<span id="cb105-83"><a href="#cb105-83" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Poxviridae"</span>])  <span class="sc">+</span></span>
<span id="cb105-84"><a href="#cb105-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb105-85"><a href="#cb105-85" aria-hidden="true" tabindex="-1"></a>panel_d</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/panel_d-1.png" class="img-fluid figure-img" width="384"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(panel_d<span class="sc">$</span>data, <span class="fu">paste0</span>(out_dir, <span class="st">"Figure1d_datapoints.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="panel-e" class="level2">
<h2 class="anchored" data-anchor-id="panel-e">Panel E</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/12mer_proteome_everything.csv"</span>, <span class="at">row.names =</span><span class="dv">1</span>)</span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final)</span>
<span id="cb107-4"><a href="#cb107-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb107-5"><a href="#cb107-5" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb107-6"><a href="#cb107-6" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb107-7"><a href="#cb107-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-8"><a href="#cb107-8" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb107-9"><a href="#cb107-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb107-10"><a href="#cb107-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb107-11"><a href="#cb107-11" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic))<span class="sc">%&gt;%</span></span>
<span id="cb107-12"><a href="#cb107-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb107-13"><a href="#cb107-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb107-14"><a href="#cb107-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb107-15"><a href="#cb107-15" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb107-16"><a href="#cb107-16" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb107-17"><a href="#cb107-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb107-18"><a href="#cb107-18" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb107-19"><a href="#cb107-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb107-20"><a href="#cb107-20" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb107-21"><a href="#cb107-21" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb107-22"><a href="#cb107-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-23"><a href="#cb107-23" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">12-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb107-24"><a href="#cb107-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">12-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb107-25"><a href="#cb107-25" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">12-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb107-26"><a href="#cb107-26" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">12-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb107-27"><a href="#cb107-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-28"><a href="#cb107-28" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(mismatch_data_meta, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"12mers"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb107-29"><a href="#cb107-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-30"><a href="#cb107-30" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb107-31"><a href="#cb107-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"Virus"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb107-32"><a href="#cb107-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb107-33"><a href="#cb107-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>))</span>
<span id="cb107-34"><a href="#cb107-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-35"><a href="#cb107-35" aria-hidden="true" tabindex="-1"></a>wt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Chronic.or.Acute,</span>
<span id="cb107-36"><a href="#cb107-36" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb107-37"><a href="#cb107-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb107-38"><a href="#cb107-38" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb107-39"><a href="#cb107-39" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"12mer_Chronic_vs_Acute_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb107-40"><a href="#cb107-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-41"><a href="#cb107-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-42"><a href="#cb107-42" aria-hidden="true" tabindex="-1"></a>panel_e <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb107-43"><a href="#cb107-43" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(grepl("Virus", Taxonomic.lineage)) %&gt;%</span></span>
<span id="cb107-44"><a href="#cb107-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb107-45"><a href="#cb107-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">"Bacteria"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb107-46"><a href="#cb107-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Chronic.or.Acute =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"Bacteria"</span>, Taxonomic.lineage), <span class="st">"Bacteria"</span>, Chronic.or.Acute)) <span class="sc">%&gt;%</span></span>
<span id="cb107-47"><a href="#cb107-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>, name) <span class="sc">%&gt;%</span></span>
<span id="cb107-48"><a href="#cb107-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(value), <span class="dv">0</span>, value)) <span class="sc">%&gt;%</span></span>
<span id="cb107-49"><a href="#cb107-49" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(value), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb107-50"><a href="#cb107-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb107-51"><a href="#cb107-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb107-52"><a href="#cb107-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">""</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb107-53"><a href="#cb107-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb107-54"><a href="#cb107-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb107-55"><a href="#cb107-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb107-56"><a href="#cb107-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb107-57"><a href="#cb107-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log10_trans</span>(),</span>
<span id="cb107-58"><a href="#cb107-58" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log10"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">10</span><span class="sc">^</span>x, <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb107-59"><a href="#cb107-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb107-60"><a href="#cb107-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>name), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb107-61"><a href="#cb107-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb107-62"><a href="#cb107-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span><span class="fu">c</span>(<span class="st">"#1975EE"</span>, <span class="st">"#E78305"</span>, <span class="st">"black"</span>)) <span class="sc">+</span></span>
<span id="cb107-63"><a href="#cb107-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent of 12mers (%)"</span>) <span class="sc">+</span></span>
<span id="cb107-64"><a href="#cb107-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">size =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb107-65"><a href="#cb107-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="fl">0.07</span>, <span class="fl">0.210</span>, <span class="dv">1</span>, <span class="dv">9</span>),</span>
<span id="cb107-66"><a href="#cb107-66" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb107-67"><a href="#cb107-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.key.height=</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">'cm'</span>),</span>
<span id="cb107-68"><a href="#cb107-68" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.width=</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">'cm'</span>),</span>
<span id="cb107-69"><a href="#cb107-69" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb107-70"><a href="#cb107-70" aria-hidden="true" tabindex="-1"></a>panel_e</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/panel_e-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(panel_e<span class="sc">$</span>data, <span class="fu">paste0</span>(out_dir, <span class="st">"Figure1e_datapoints.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="panel-f" class="level2">
<h2 class="anchored" data-anchor-id="panel-f">Panel F</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final)</span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb109-4"><a href="#cb109-4" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb109-5"><a href="#cb109-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-6"><a href="#cb109-6" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb109-7"><a href="#cb109-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb109-8"><a href="#cb109-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb109-9"><a href="#cb109-9" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic))<span class="sc">%&gt;%</span></span>
<span id="cb109-10"><a href="#cb109-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb109-11"><a href="#cb109-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb109-12"><a href="#cb109-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb109-13"><a href="#cb109-13" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb109-14"><a href="#cb109-14" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb109-15"><a href="#cb109-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb109-16"><a href="#cb109-16" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb109-17"><a href="#cb109-17" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb109-18"><a href="#cb109-18" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb109-19"><a href="#cb109-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb109-20"><a href="#cb109-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-21"><a href="#cb109-21" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">12-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb109-22"><a href="#cb109-22" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">12-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb109-23"><a href="#cb109-23" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">12-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb109-24"><a href="#cb109-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">12-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb109-25"><a href="#cb109-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-26"><a href="#cb109-26" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb109-27"><a href="#cb109-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb109-28"><a href="#cb109-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb109-29"><a href="#cb109-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb109-30"><a href="#cb109-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Herpesviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Herpesviridae"</span>, <span class="st">"Herpesviridae"</span>, <span class="st">"Non-Herpesviridae"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb109-31"><a href="#cb109-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Poxviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Poxviridae"</span>, <span class="st">"Poxviridae"</span>, <span class="st">"Non-Poxviridae"</span>))</span>
<span id="cb109-32"><a href="#cb109-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-33"><a href="#cb109-33" aria-hidden="true" tabindex="-1"></a><span class="do">## These results are used to manually add stars to the graph below in illustrator</span></span>
<span id="cb109-34"><a href="#cb109-34" aria-hidden="true" tabindex="-1"></a>kt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Family,</span>
<span id="cb109-35"><a href="#cb109-35" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"kruskal.test"</span>)</span>
<span id="cb109-36"><a href="#cb109-36" aria-hidden="true" tabindex="-1"></a>wt.p <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Poxviridae,</span>
<span id="cb109-37"><a href="#cb109-37" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb109-38"><a href="#cb109-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb109-39"><a href="#cb109-39" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb109-40"><a href="#cb109-40" aria-hidden="true" tabindex="-1"></a>wt.h <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Herpesviridae,</span>
<span id="cb109-41"><a href="#cb109-41" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb109-42"><a href="#cb109-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb109-43"><a href="#cb109-43" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb109-44"><a href="#cb109-44" aria-hidden="true" tabindex="-1"></a>wt.hp <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Family <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Herpesviridae"</span>, <span class="st">"Poxviridae"</span>)), <span class="at">formula =</span> value <span class="sc">~</span> Poxviridae,</span>
<span id="cb109-45"><a href="#cb109-45" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb109-46"><a href="#cb109-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb109-47"><a href="#cb109-47" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>)))) <span class="sc">%&gt;%</span></span>
<span id="cb109-48"><a href="#cb109-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group1 =</span> <span class="st">"Herpesviridae"</span>)</span>
<span id="cb109-49"><a href="#cb109-49" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(kt, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"12mer_Viral_Family_Kruskal"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb109-50"><a href="#cb109-50" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.p, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"12mer_Pox_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb109-51"><a href="#cb109-51" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.h, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"12mer_Herpes_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb109-52"><a href="#cb109-52" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.hp, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"12mer_Herpes_vs_Pox_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb109-53"><a href="#cb109-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-54"><a href="#cb109-54" aria-hidden="true" tabindex="-1"></a>panel_f <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb109-55"><a href="#cb109-55" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(Family = ifelse(grepl("Bacteria", Taxonomic.lineage), "Bacteria", Family_Viral_Zone)) %&gt;%</span></span>
<span id="cb109-56"><a href="#cb109-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb109-57"><a href="#cb109-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb109-58"><a href="#cb109-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb109-59"><a href="#cb109-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Family, name) <span class="sc">%&gt;%</span></span>
<span id="cb109-60"><a href="#cb109-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(value), <span class="dv">0</span>, value)) <span class="sc">%&gt;%</span></span>
<span id="cb109-61"><a href="#cb109-61" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(value), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb109-62"><a href="#cb109-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Family <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2)) <span class="sc">%&gt;%</span></span>
<span id="cb109-63"><a href="#cb109-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb109-64"><a href="#cb109-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb109-65"><a href="#cb109-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">""</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb109-66"><a href="#cb109-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb109-67"><a href="#cb109-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mean_value =</span> <span class="fu">ifelse</span>(mean_value <span class="sc">==</span> <span class="dv">0</span>, <span class="fl">1E-3</span>, mean_value)) <span class="sc">%&gt;%</span></span>
<span id="cb109-68"><a href="#cb109-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb109-69"><a href="#cb109-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb109-70"><a href="#cb109-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb109-71"><a href="#cb109-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb109-72"><a href="#cb109-72" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>name), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb109-73"><a href="#cb109-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb109-74"><a href="#cb109-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span>Family_palette_v2) <span class="sc">+</span></span>
<span id="cb109-75"><a href="#cb109-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent of 12mers (%)"</span>) <span class="sc">+</span></span>
<span id="cb109-76"><a href="#cb109-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">size =</span> <span class="fl">0.1</span>)  <span class="sc">+</span></span>
<span id="cb109-77"><a href="#cb109-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">0.9</span>,<span class="fl">1.9</span>,<span class="fl">2.9</span>,<span class="fl">3.9</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="fl">0.45</span>, <span class="fl">0.75</span>, <span class="fl">1.7</span>, <span class="dv">13</span>),</span>
<span id="cb109-78"><a href="#cb109-78" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.h<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>,</span>
<span id="cb109-79"><a href="#cb109-79" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Herpesviridae"</span>]) <span class="sc">+</span></span>
<span id="cb109-80"><a href="#cb109-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">1.1</span>,<span class="fl">2.1</span>,<span class="fl">3.1</span>,<span class="fl">4.1</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="fl">0.45</span>, <span class="fl">0.75</span>, <span class="fl">1.7</span>, <span class="dv">13</span>),</span>
<span id="cb109-81"><a href="#cb109-81" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.p<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>,</span>
<span id="cb109-82"><a href="#cb109-82" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Poxviridae"</span>]) <span class="sc">+</span></span>
<span id="cb109-83"><a href="#cb109-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.key.height=</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">'cm'</span>),</span>
<span id="cb109-84"><a href="#cb109-84" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.width=</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">'cm'</span>)) <span class="sc">+</span></span>
<span id="cb109-85"><a href="#cb109-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb109-86"><a href="#cb109-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">1e-3</span>, <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb109-87"><a href="#cb109-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log10_trans</span>(),</span>
<span id="cb109-88"><a href="#cb109-88" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log10"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">10</span><span class="sc">^</span>x, <span class="dv">3</span>)))</span>
<span id="cb109-89"><a href="#cb109-89" aria-hidden="true" tabindex="-1"></a>panel_f</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/panel_f-1.png" class="img-fluid figure-img" width="384"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(panel_f<span class="sc">$</span>data, <span class="fu">paste0</span>(out_dir, <span class="st">"Figure1f_datapoints.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="panel-g" class="level2">
<h2 class="anchored" data-anchor-id="panel-g">Panel G</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>out_final_18mer <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/18mer_proteome_everything.csv"</span>, <span class="at">row.names =</span><span class="dv">1</span>)</span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final_18mer)</span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb111-6"><a href="#cb111-6" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb111-7"><a href="#cb111-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-8"><a href="#cb111-8" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb111-9"><a href="#cb111-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb111-10"><a href="#cb111-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb111-11"><a href="#cb111-11" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic))<span class="sc">%&gt;%</span></span>
<span id="cb111-12"><a href="#cb111-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb111-13"><a href="#cb111-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb111-14"><a href="#cb111-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb111-15"><a href="#cb111-15" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb111-16"><a href="#cb111-16" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb111-17"><a href="#cb111-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb111-18"><a href="#cb111-18" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb111-19"><a href="#cb111-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb111-20"><a href="#cb111-20" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb111-21"><a href="#cb111-21" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb111-22"><a href="#cb111-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-23"><a href="#cb111-23" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">18-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb111-24"><a href="#cb111-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">18-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb111-25"><a href="#cb111-25" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">18-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb111-26"><a href="#cb111-26" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">18-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb111-27"><a href="#cb111-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-28"><a href="#cb111-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-29"><a href="#cb111-29" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(mismatch_data_meta, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"18mers"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb111-30"><a href="#cb111-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-31"><a href="#cb111-31" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb111-32"><a href="#cb111-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"Virus"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb111-33"><a href="#cb111-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb111-34"><a href="#cb111-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>))</span>
<span id="cb111-35"><a href="#cb111-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-36"><a href="#cb111-36" aria-hidden="true" tabindex="-1"></a>wt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Chronic.or.Acute,</span>
<span id="cb111-37"><a href="#cb111-37" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb111-38"><a href="#cb111-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb111-39"><a href="#cb111-39" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb111-40"><a href="#cb111-40" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"18mer_Chronic_vs_Acute_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb111-41"><a href="#cb111-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-42"><a href="#cb111-42" aria-hidden="true" tabindex="-1"></a>panel_g <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb111-43"><a href="#cb111-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"Virus"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb111-44"><a href="#cb111-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb111-45"><a href="#cb111-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb111-46"><a href="#cb111-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>, name) <span class="sc">%&gt;%</span></span>
<span id="cb111-47"><a href="#cb111-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(value), <span class="dv">0</span>, value)) <span class="sc">%&gt;%</span></span>
<span id="cb111-48"><a href="#cb111-48" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(value), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb111-49"><a href="#cb111-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb111-50"><a href="#cb111-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb111-51"><a href="#cb111-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">""</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb111-52"><a href="#cb111-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb111-53"><a href="#cb111-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb111-54"><a href="#cb111-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb111-55"><a href="#cb111-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb111-56"><a href="#cb111-56" aria-hidden="true" tabindex="-1"></a>    <span class="co">#scale_y_continuous(trans = log2_trans(),</span></span>
<span id="cb111-57"><a href="#cb111-57" aria-hidden="true" tabindex="-1"></a>    <span class="co"># breaks = trans_breaks("log2", function(x) round(2^x, 3))) +</span></span>
<span id="cb111-58"><a href="#cb111-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb111-59"><a href="#cb111-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>name), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb111-60"><a href="#cb111-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb111-61"><a href="#cb111-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span><span class="fu">c</span>(<span class="st">"#1975EE"</span>, <span class="st">"#E78305"</span>)) <span class="sc">+</span></span>
<span id="cb111-62"><a href="#cb111-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent of 18mers (%)"</span>) <span class="sc">+</span></span>
<span id="cb111-63"><a href="#cb111-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="fl">0.02</span>, <span class="fl">0.05</span>, <span class="fl">0.085</span>, <span class="fl">0.11</span>),</span>
<span id="cb111-64"><a href="#cb111-64" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb111-65"><a href="#cb111-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb111-66"><a href="#cb111-66" aria-hidden="true" tabindex="-1"></a>panel_g</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/panel_g-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(panel_g<span class="sc">$</span>data, <span class="fu">paste0</span>(out_dir, <span class="st">"Figure1g_datapoints.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="panel-h" class="level2">
<h2 class="anchored" data-anchor-id="panel-h">Panel H</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final_18mer)</span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb113-9"><a href="#cb113-9" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic))<span class="sc">%&gt;%</span></span>
<span id="cb113-10"><a href="#cb113-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb113-11"><a href="#cb113-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb113-12"><a href="#cb113-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb113-13"><a href="#cb113-13" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb113-14"><a href="#cb113-14" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb113-15"><a href="#cb113-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb113-16"><a href="#cb113-16" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb113-17"><a href="#cb113-17" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb113-18"><a href="#cb113-18" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb113-19"><a href="#cb113-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb113-20"><a href="#cb113-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-21"><a href="#cb113-21" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">18-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb113-22"><a href="#cb113-22" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">18-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb113-23"><a href="#cb113-23" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">18-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb113-24"><a href="#cb113-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">18-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb113-25"><a href="#cb113-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-26"><a href="#cb113-26" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb113-27"><a href="#cb113-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb113-28"><a href="#cb113-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb113-29"><a href="#cb113-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb113-30"><a href="#cb113-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Herpesviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Herpesviridae"</span>, <span class="st">"Herpesviridae"</span>, <span class="st">"Non-Herpesviridae"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb113-31"><a href="#cb113-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Poxviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Poxviridae"</span>, <span class="st">"Poxviridae"</span>, <span class="st">"Non-Poxviridae"</span>))</span>
<span id="cb113-32"><a href="#cb113-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-33"><a href="#cb113-33" aria-hidden="true" tabindex="-1"></a><span class="do">## These results are used to manually add stars to the graph below in illustrator</span></span>
<span id="cb113-34"><a href="#cb113-34" aria-hidden="true" tabindex="-1"></a>kt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Family,</span>
<span id="cb113-35"><a href="#cb113-35" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"kruskal.test"</span>)</span>
<span id="cb113-36"><a href="#cb113-36" aria-hidden="true" tabindex="-1"></a>wt.p <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Poxviridae,</span>
<span id="cb113-37"><a href="#cb113-37" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-38"><a href="#cb113-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb113-39"><a href="#cb113-39" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb113-40"><a href="#cb113-40" aria-hidden="true" tabindex="-1"></a>wt.h <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Herpesviridae,</span>
<span id="cb113-41"><a href="#cb113-41" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-42"><a href="#cb113-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb113-43"><a href="#cb113-43" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb113-44"><a href="#cb113-44" aria-hidden="true" tabindex="-1"></a>wt.hp <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Family <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Herpesviridae"</span>, <span class="st">"Poxviridae"</span>)), <span class="at">formula =</span> value <span class="sc">~</span> Poxviridae,</span>
<span id="cb113-45"><a href="#cb113-45" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-46"><a href="#cb113-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb113-47"><a href="#cb113-47" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>)))) <span class="sc">%&gt;%</span></span>
<span id="cb113-48"><a href="#cb113-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group1 =</span> <span class="st">"Herpesviridae"</span>)</span>
<span id="cb113-49"><a href="#cb113-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-50"><a href="#cb113-50" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(kt, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"18mer_Viral_Family_Kruskal"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb113-51"><a href="#cb113-51" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.p, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"18mer_Pox_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb113-52"><a href="#cb113-52" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.h, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"18mer_Herpes_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb113-53"><a href="#cb113-53" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.hp, <span class="fu">paste0</span>(out_dir, <span class="st">"SupplementalData2.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"18mer_Herpes_vs_Pox_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb113-54"><a href="#cb113-54" aria-hidden="true" tabindex="-1"></a>panel_h <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb113-55"><a href="#cb113-55" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(Family = ifelse(grepl("Bacteria", Taxonomic.lineage), "Bacteria", Family_Viral_Zone)) %&gt;%</span></span>
<span id="cb113-56"><a href="#cb113-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb113-57"><a href="#cb113-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb113-58"><a href="#cb113-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb113-59"><a href="#cb113-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Family, name) <span class="sc">%&gt;%</span></span>
<span id="cb113-60"><a href="#cb113-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(value), <span class="dv">0</span>, value)) <span class="sc">%&gt;%</span></span>
<span id="cb113-61"><a href="#cb113-61" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(value), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb113-62"><a href="#cb113-62" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(count &gt; 4) %&gt;%</span></span>
<span id="cb113-63"><a href="#cb113-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb113-64"><a href="#cb113-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb113-65"><a href="#cb113-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">""</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb113-66"><a href="#cb113-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb113-67"><a href="#cb113-67" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Only keep Families that have at least one non-zero point</span></span>
<span id="cb113-68"><a href="#cb113-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Family <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Poxviridae"</span>, <span class="st">"Herpesviridae"</span>, <span class="st">"Coronaviridae"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb113-69"><a href="#cb113-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb113-70"><a href="#cb113-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb113-71"><a href="#cb113-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb113-72"><a href="#cb113-72" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb113-73"><a href="#cb113-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>name), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb113-74"><a href="#cb113-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb113-75"><a href="#cb113-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span>Family_palette_v2) <span class="sc">+</span></span>
<span id="cb113-76"><a href="#cb113-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent of 18mers (%)"</span>) <span class="sc">+</span></span>
<span id="cb113-77"><a href="#cb113-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">0.9</span>,<span class="fl">1.9</span>,<span class="fl">2.9</span>,<span class="fl">3.9</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="fl">0.12</span>, <span class="fl">0.24</span>, <span class="fl">0.42</span>, <span class="fl">0.65</span>),</span>
<span id="cb113-78"><a href="#cb113-78" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.h<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>,</span>
<span id="cb113-79"><a href="#cb113-79" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Herpesviridae"</span>]) <span class="sc">+</span></span>
<span id="cb113-80"><a href="#cb113-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">1.1</span>,<span class="fl">2.1</span>,<span class="fl">3.1</span>,<span class="fl">4.1</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="fl">0.12</span>, <span class="fl">0.24</span>, <span class="fl">0.42</span>, <span class="fl">0.65</span>),</span>
<span id="cb113-81"><a href="#cb113-81" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.p<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>,</span>
<span id="cb113-82"><a href="#cb113-82" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Poxviridae"</span>]) <span class="sc">+</span></span>
<span id="cb113-83"><a href="#cb113-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">inherit.aes =</span> <span class="cn">FALSE</span>, <span class="fu">aes</span>(<span class="at">x=</span><span class="fl">0.85</span>, <span class="at">ymin =</span> <span class="fl">0.018380776</span>, <span class="at">ymax =</span> <span class="fl">0.075685311</span>), <span class="at">color =</span> <span class="st">"#545252"</span>,</span>
<span id="cb113-84"><a href="#cb113-84" aria-hidden="true" tabindex="-1"></a>                <span class="at">size =</span> <span class="fl">0.3</span>, <span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb113-85"><a href="#cb113-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">inherit.aes =</span> <span class="cn">FALSE</span>, <span class="fu">aes</span>(<span class="at">x=</span><span class="fl">1.85</span>, <span class="at">ymin =</span> <span class="fl">0.051403280</span>, <span class="at">ymax =</span> <span class="fl">0.182375884</span>), <span class="at">color =</span> <span class="st">"#545252"</span>,</span>
<span id="cb113-86"><a href="#cb113-86" aria-hidden="true" tabindex="-1"></a>                <span class="at">size =</span> <span class="fl">0.3</span>, <span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb113-87"><a href="#cb113-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">inherit.aes =</span> <span class="cn">FALSE</span>, <span class="fu">aes</span>(<span class="at">x=</span><span class="fl">2.85</span>, <span class="at">ymin =</span> <span class="fl">0.085500719</span>, <span class="at">ymax =</span> <span class="fl">0.345138406</span>), <span class="at">color =</span> <span class="st">"#545252"</span>,</span>
<span id="cb113-88"><a href="#cb113-88" aria-hidden="true" tabindex="-1"></a>                <span class="at">size =</span> <span class="fl">0.3</span>, <span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb113-89"><a href="#cb113-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">inherit.aes =</span> <span class="cn">FALSE</span>, <span class="fu">aes</span>(<span class="at">x=</span><span class="fl">3.85</span>, <span class="at">ymin =</span> <span class="fl">0.145041702</span>, <span class="at">ymax =</span> <span class="fl">0.536979034</span>), <span class="at">color =</span> <span class="st">"#545252"</span>,</span>
<span id="cb113-90"><a href="#cb113-90" aria-hidden="true" tabindex="-1"></a>                <span class="at">size =</span> <span class="fl">0.3</span>, <span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb113-91"><a href="#cb113-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">0.75</span>, <span class="fl">1.8</span>, <span class="fl">2.8</span>, <span class="fl">3.8</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="fl">0.04557349</span>, <span class="fl">0.1136466</span>,</span>
<span id="cb113-92"><a href="#cb113-92" aria-hidden="true" tabindex="-1"></a>                                                   <span class="fl">0.2057032</span>, <span class="fl">0.3237431</span>),</span>
<span id="cb113-93"><a href="#cb113-93" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.hp<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>,</span>
<span id="cb113-94"><a href="#cb113-94" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> <span class="st">"#545252"</span>) <span class="sc">+</span></span>
<span id="cb113-95"><a href="#cb113-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb113-96"><a href="#cb113-96" aria-hidden="true" tabindex="-1"></a>panel_h</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/panel_h-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(panel_h<span class="sc">$</span>data, <span class="fu">paste0</span>(out_dir, <span class="st">"Figure1h_datapoints.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="panel-assembly-1" class="level2">
<h2 class="anchored" data-anchor-id="panel-assembly-1">Panel Assembly</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a>panel_c_h <span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(panel_c, panel_d,panel_e, panel_f,panel_g, panel_h, <span class="at">nrow =</span> <span class="dv">3</span>, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>), <span class="at">align =</span> <span class="st">"hv"</span>,</span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"C"</span>, <span class="st">"D"</span>, <span class="st">"E"</span>, <span class="st">"F"</span>, <span class="st">"G"</span>, <span class="st">"H"</span>),</span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">label.x =</span> <span class="fl">0.1</span>)</span>
<span id="cb115-4"><a href="#cb115-4" aria-hidden="true" tabindex="-1"></a>panel_c_h</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/panel_c_f_assembly_save-1.png" class="img-fluid figure-img" width="480"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"Figure1C-H.pdf"</span>), <span class="at">height =</span> <span class="dv">7</span>, <span class="at">width =</span> <span class="dv">4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="supplemental-figures" class="level2">
<h2 class="anchored" data-anchor-id="supplemental-figures">Supplemental Figures</h2>
</section>
<section id="supplemental-figure-1" class="level2">
<h2 class="anchored" data-anchor-id="supplemental-figure-1">Supplemental Figure 1</h2>
<section id="supplemental-panel-1a---8mer-baltimore-classification" class="level3">
<h3 class="anchored" data-anchor-id="supplemental-panel-1a---8mer-baltimore-classification">Supplemental Panel 1A - 8mer Baltimore Classification</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/8mer_proteome_everything.csv"</span>, <span class="at">row.names =</span><span class="dv">1</span>)</span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final)</span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb117-5"><a href="#cb117-5" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb117-6"><a href="#cb117-6" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb117-7"><a href="#cb117-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-8"><a href="#cb117-8" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb117-9"><a href="#cb117-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb117-10"><a href="#cb117-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb117-11"><a href="#cb117-11" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic))<span class="sc">%&gt;%</span></span>
<span id="cb117-12"><a href="#cb117-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb117-13"><a href="#cb117-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb117-14"><a href="#cb117-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb117-15"><a href="#cb117-15" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb117-16"><a href="#cb117-16" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb117-17"><a href="#cb117-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb117-18"><a href="#cb117-18" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb117-19"><a href="#cb117-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb117-20"><a href="#cb117-20" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb117-21"><a href="#cb117-21" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb117-22"><a href="#cb117-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-23"><a href="#cb117-23" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">8-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb117-24"><a href="#cb117-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">8-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb117-25"><a href="#cb117-25" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">8-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb117-26"><a href="#cb117-26" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">8-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb117-27"><a href="#cb117-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-28"><a href="#cb117-28" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb117-29"><a href="#cb117-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Genome)) <span class="sc">%&gt;%</span></span>
<span id="cb117-30"><a href="#cb117-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>)))</span>
<span id="cb117-31"><a href="#cb117-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-32"><a href="#cb117-32" aria-hidden="true" tabindex="-1"></a><span class="do">## These results are used to manually add stars to the graph below in illustrator</span></span>
<span id="cb117-33"><a href="#cb117-33" aria-hidden="true" tabindex="-1"></a>kt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Genome,</span>
<span id="cb117-34"><a href="#cb117-34" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"kruskal.test"</span>)</span>
<span id="cb117-35"><a href="#cb117-35" aria-hidden="true" tabindex="-1"></a>wt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Genome,</span>
<span id="cb117-36"><a href="#cb117-36" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>)</span>
<span id="cb117-37"><a href="#cb117-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-38"><a href="#cb117-38" aria-hidden="true" tabindex="-1"></a>supp_panel_a <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb117-39"><a href="#cb117-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Genome)) <span class="sc">%&gt;%</span></span>
<span id="cb117-40"><a href="#cb117-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb117-41"><a href="#cb117-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Genome, name) <span class="sc">%&gt;%</span></span>
<span id="cb117-42"><a href="#cb117-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(value), <span class="dv">0</span>, value)) <span class="sc">%&gt;%</span></span>
<span id="cb117-43"><a href="#cb117-43" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(value), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb117-44"><a href="#cb117-44" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(count &gt; 4) %&gt;%</span></span>
<span id="cb117-45"><a href="#cb117-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb117-46"><a href="#cb117-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name))  <span class="sc">%&gt;%</span></span>
<span id="cb117-47"><a href="#cb117-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">""</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb117-48"><a href="#cb117-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb117-49"><a href="#cb117-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Genome</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb117-50"><a href="#cb117-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb117-51"><a href="#cb117-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Genome</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb117-52"><a href="#cb117-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log10_trans</span>(),</span>
<span id="cb117-53"><a href="#cb117-53" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log10"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">10</span><span class="sc">^</span>x, <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb117-54"><a href="#cb117-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb117-55"><a href="#cb117-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>name), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb117-56"><a href="#cb117-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb117-57"><a href="#cb117-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> Baltimore_classification_colors) <span class="sc">+</span></span>
<span id="cb117-58"><a href="#cb117-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent of 8mers (%)"</span>) <span class="sc">+</span></span>
<span id="cb117-59"><a href="#cb117-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">size =</span> <span class="fl">0.1</span>)</span>
<span id="cb117-60"><a href="#cb117-60" aria-hidden="true" tabindex="-1"></a>supp_panel_a</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/supp_panel_a-1.png" class="img-fluid figure-img" width="384"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(supp_panel_a<span class="sc">$</span>data, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1a_datapoints.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="supplemental-panel-1b---12mer-baltimore-classification" class="level3">
<h3 class="anchored" data-anchor-id="supplemental-panel-1b---12mer-baltimore-classification">Supplemental Panel 1B - 12mer Baltimore Classification</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/12mer_proteome_everything.csv"</span>, <span class="at">row.names =</span><span class="dv">1</span>)</span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final)</span>
<span id="cb119-4"><a href="#cb119-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb119-5"><a href="#cb119-5" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb119-6"><a href="#cb119-6" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb119-7"><a href="#cb119-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-8"><a href="#cb119-8" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb119-9"><a href="#cb119-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb119-10"><a href="#cb119-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb119-11"><a href="#cb119-11" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic))<span class="sc">%&gt;%</span></span>
<span id="cb119-12"><a href="#cb119-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb119-13"><a href="#cb119-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb119-14"><a href="#cb119-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb119-15"><a href="#cb119-15" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb119-16"><a href="#cb119-16" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb119-17"><a href="#cb119-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb119-18"><a href="#cb119-18" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb119-19"><a href="#cb119-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb119-20"><a href="#cb119-20" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb119-21"><a href="#cb119-21" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb119-22"><a href="#cb119-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-23"><a href="#cb119-23" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">12-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb119-24"><a href="#cb119-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">12-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb119-25"><a href="#cb119-25" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">12-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb119-26"><a href="#cb119-26" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">12-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb119-27"><a href="#cb119-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-28"><a href="#cb119-28" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb119-29"><a href="#cb119-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Genome)) <span class="sc">%&gt;%</span></span>
<span id="cb119-30"><a href="#cb119-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>)))</span>
<span id="cb119-31"><a href="#cb119-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-32"><a href="#cb119-32" aria-hidden="true" tabindex="-1"></a><span class="do">## These results are used to manually add stars to the graph below in illustrator</span></span>
<span id="cb119-33"><a href="#cb119-33" aria-hidden="true" tabindex="-1"></a>kt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Genome,</span>
<span id="cb119-34"><a href="#cb119-34" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"kruskal.test"</span>)</span>
<span id="cb119-35"><a href="#cb119-35" aria-hidden="true" tabindex="-1"></a>wt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Genome,</span>
<span id="cb119-36"><a href="#cb119-36" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>)</span>
<span id="cb119-37"><a href="#cb119-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-38"><a href="#cb119-38" aria-hidden="true" tabindex="-1"></a>supp_panel_b <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb119-39"><a href="#cb119-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Genome)) <span class="sc">%&gt;%</span></span>
<span id="cb119-40"><a href="#cb119-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb119-41"><a href="#cb119-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Genome, name) <span class="sc">%&gt;%</span></span>
<span id="cb119-42"><a href="#cb119-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(value), <span class="dv">0</span>, value)) <span class="sc">%&gt;%</span></span>
<span id="cb119-43"><a href="#cb119-43" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(value), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb119-44"><a href="#cb119-44" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(count &gt; 4) %&gt;%</span></span>
<span id="cb119-45"><a href="#cb119-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb119-46"><a href="#cb119-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb119-47"><a href="#cb119-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">""</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb119-48"><a href="#cb119-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb119-49"><a href="#cb119-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Genome</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb119-50"><a href="#cb119-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb119-51"><a href="#cb119-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Genome</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb119-52"><a href="#cb119-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log10_trans</span>(),</span>
<span id="cb119-53"><a href="#cb119-53" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log10"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">10</span><span class="sc">^</span>x, <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb119-54"><a href="#cb119-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb119-55"><a href="#cb119-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>name), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb119-56"><a href="#cb119-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb119-57"><a href="#cb119-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> Baltimore_classification_colors) <span class="sc">+</span></span>
<span id="cb119-58"><a href="#cb119-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent of 12mers (%)"</span>) <span class="sc">+</span></span>
<span id="cb119-59"><a href="#cb119-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">size =</span> <span class="fl">0.1</span>) </span>
<span id="cb119-60"><a href="#cb119-60" aria-hidden="true" tabindex="-1"></a>supp_panel_b</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/supp_panel_b-1.png" class="img-fluid figure-img" width="384"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(supp_panel_b<span class="sc">$</span>data, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1b_datapoints.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="supplemental-panel-1c---18mer-baltimore-classification" class="level3">
<h3 class="anchored" data-anchor-id="supplemental-panel-1c---18mer-baltimore-classification">Supplemental Panel 1C - 18mer Baltimore Classification</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a>out_final_18mer <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/18mer_proteome_everything.csv"</span>, <span class="at">row.names =</span><span class="dv">1</span>)</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final_18mer)</span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb121-5"><a href="#cb121-5" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb121-6"><a href="#cb121-6" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb121-7"><a href="#cb121-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-8"><a href="#cb121-8" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb121-9"><a href="#cb121-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb121-10"><a href="#cb121-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb121-11"><a href="#cb121-11" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic))<span class="sc">%&gt;%</span></span>
<span id="cb121-12"><a href="#cb121-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb121-13"><a href="#cb121-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb121-14"><a href="#cb121-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb121-15"><a href="#cb121-15" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb121-16"><a href="#cb121-16" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb121-17"><a href="#cb121-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb121-18"><a href="#cb121-18" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb121-19"><a href="#cb121-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb121-20"><a href="#cb121-20" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb121-21"><a href="#cb121-21" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb121-22"><a href="#cb121-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-23"><a href="#cb121-23" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">18-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb121-24"><a href="#cb121-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">18-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb121-25"><a href="#cb121-25" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">18-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb121-26"><a href="#cb121-26" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>(<span class="dv">18-1</span>)<span class="sc">-</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb121-27"><a href="#cb121-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-28"><a href="#cb121-28" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb121-29"><a href="#cb121-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Genome)) <span class="sc">%&gt;%</span></span>
<span id="cb121-30"><a href="#cb121-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>)))</span>
<span id="cb121-31"><a href="#cb121-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-32"><a href="#cb121-32" aria-hidden="true" tabindex="-1"></a><span class="do">## These results are used to manually add stars to the graph below in illustrator</span></span>
<span id="cb121-33"><a href="#cb121-33" aria-hidden="true" tabindex="-1"></a>kt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Genome,</span>
<span id="cb121-34"><a href="#cb121-34" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"kruskal.test"</span>)</span>
<span id="cb121-35"><a href="#cb121-35" aria-hidden="true" tabindex="-1"></a>wt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Genome,</span>
<span id="cb121-36"><a href="#cb121-36" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>)</span>
<span id="cb121-37"><a href="#cb121-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-38"><a href="#cb121-38" aria-hidden="true" tabindex="-1"></a>supp_panel_c <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb121-39"><a href="#cb121-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Genome)) <span class="sc">%&gt;%</span></span>
<span id="cb121-40"><a href="#cb121-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb121-41"><a href="#cb121-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Genome, name) <span class="sc">%&gt;%</span></span>
<span id="cb121-42"><a href="#cb121-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(value), <span class="dv">0</span>, value)) <span class="sc">%&gt;%</span></span>
<span id="cb121-43"><a href="#cb121-43" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(value), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb121-44"><a href="#cb121-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb121-45"><a href="#cb121-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb121-46"><a href="#cb121-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">""</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb121-47"><a href="#cb121-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb121-48"><a href="#cb121-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Genome</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb121-49"><a href="#cb121-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb121-50"><a href="#cb121-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Genome</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb121-51"><a href="#cb121-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log10_trans</span>(),</span>
<span id="cb121-52"><a href="#cb121-52" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log10"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">10</span><span class="sc">^</span>x, <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb121-53"><a href="#cb121-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb121-54"><a href="#cb121-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>name), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb121-55"><a href="#cb121-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb121-56"><a href="#cb121-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> Baltimore_classification_colors) <span class="sc">+</span></span>
<span id="cb121-57"><a href="#cb121-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent of 18mers (%)"</span>) <span class="sc">+</span></span>
<span id="cb121-58"><a href="#cb121-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">size =</span> <span class="fl">0.1</span>)</span>
<span id="cb121-59"><a href="#cb121-59" aria-hidden="true" tabindex="-1"></a>supp_panel_c</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/supp_panel_c-1.png" class="img-fluid figure-img" width="384"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(supp_panel_c<span class="sc">$</span>data, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1c_datapoints.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="supplemental-figure-1-panel-assembly" class="level3">
<h3 class="anchored" data-anchor-id="supplemental-figure-1-panel-assembly">Supplemental Figure 1 Panel Assembly</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a>panel_supp <span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(supp_panel_a, supp_panel_b, supp_panel_c, <span class="at">nrow =</span> <span class="dv">1</span>, <span class="at">ncol =</span> <span class="dv">3</span>, <span class="at">align =</span> <span class="st">"hv"</span>,<span class="at">common.legend =</span> T,<span class="at">legend =</span> <span class="st">"right"</span>,</span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"C"</span>),</span>
<span id="cb123-3"><a href="#cb123-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">label.x =</span> <span class="fl">0.1</span>)</span>
<span id="cb123-4"><a href="#cb123-4" aria-hidden="true" tabindex="-1"></a>panel_supp</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1.pdf"</span>), <span class="at">height =</span> <span class="fl">2.5</span>, <span class="at">width =</span> <span class="fl">7.25</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Supplemental Figure 1D-I are in the FigureS1_ELMFilter.qmd file also in the Figure 1 folder.</p>
<p>Supplemental Figure 1J is in the FigureS1_Mouse_vs_Human.qmd</p>
</section>
</section>
<section id="supplemental-figure-2" class="level2">
<h2 class="anchored" data-anchor-id="supplemental-figure-2">Supplemental Figure 2</h2>
<section id="supplemental-panel-2a" class="level3">
<h3 class="anchored" data-anchor-id="supplemental-panel-2a">Supplemental Panel 2A</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a>out_protein_final_min <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/8mer_proteome_protein_everything_min.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-3"><a href="#cb125-3" aria-hidden="true" tabindex="-1"></a>number_of_proteins <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Proteome.ID, Protein.count)</span>
<span id="cb125-4"><a href="#cb125-4" aria-hidden="true" tabindex="-1"></a>viral_ids <span class="ot">&lt;-</span> <span class="fu">unique</span>(out_protein_final_min<span class="sc">$</span>Proteome.ID)</span>
<span id="cb125-5"><a href="#cb125-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-6"><a href="#cb125-6" aria-hidden="true" tabindex="-1"></a>out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb125-7"><a href="#cb125-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> viral_ids){</span>
<span id="cb125-8"><a href="#cb125-8" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>)</span>
<span id="cb125-9"><a href="#cb125-9" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> file_name[<span class="fu">grepl</span>(i, file_name)]</span>
<span id="cb125-10"><a href="#cb125-10" aria-hidden="true" tabindex="-1"></a> fasta <span class="ot">&lt;-</span> seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="fu">paste0</span>(<span class="st">"../../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>, file_name), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>, <span class="at">whole.header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb125-11"><a href="#cb125-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-12"><a href="#cb125-12" aria-hidden="true" tabindex="-1"></a>  tmp.df <span class="ot">&lt;-</span> out_protein_final_min <span class="sc">%&gt;%</span></span>
<span id="cb125-13"><a href="#cb125-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">number_of_8mer =</span> PLength <span class="sc">-</span> <span class="dv">7</span>) <span class="sc">%&gt;%</span></span>
<span id="cb125-14"><a href="#cb125-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">freq =</span> count <span class="sc">/</span> number_of_8mer) <span class="sc">%&gt;%</span></span>
<span id="cb125-15"><a href="#cb125-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Proteome.ID <span class="sc">==</span> i) <span class="sc">%&gt;%</span></span>
<span id="cb125-16"><a href="#cb125-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">as.factor</span>(mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb125-17"><a href="#cb125-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">factor</span>(query_seqname, <span class="fu">unique</span>(<span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, <span class="fu">names</span>(fasta))))) <span class="sc">%&gt;%</span></span>
<span id="cb125-18"><a href="#cb125-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">complete</span>(query_seqname, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb125-19"><a href="#cb125-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">freq =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(freq), <span class="dv">0</span>, freq))</span>
<span id="cb125-20"><a href="#cb125-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-21"><a href="#cb125-21" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out, tmp.df)</span>
<span id="cb125-22"><a href="#cb125-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-23"><a href="#cb125-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb125-24"><a href="#cb125-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-25"><a href="#cb125-25" aria-hidden="true" tabindex="-1"></a>ggS2A <span class="ot">&lt;-</span> out <span class="sc">%&gt;%</span></span>
<span id="cb125-26"><a href="#cb125-26" aria-hidden="true" tabindex="-1"></a>       <span class="fu">filter</span>(mismatches <span class="sc">!=</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb125-27"><a href="#cb125-27" aria-hidden="true" tabindex="-1"></a>       <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">factor</span>(mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb125-28"><a href="#cb125-28" aria-hidden="true" tabindex="-1"></a>       <span class="fu">left_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb125-29"><a href="#cb125-29" aria-hidden="true" tabindex="-1"></a>       <span class="fu">filter</span>(Family_Viral_Zone <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2)) <span class="sc">%&gt;%</span></span>
<span id="cb125-30"><a href="#cb125-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>freq, <span class="at">y =</span> ..count..<span class="sc">/</span><span class="fu">sum</span>(..count..), <span class="at">color =</span> Family_Viral_Zone, <span class="at">group =</span> Proteome.ID)) <span class="sc">+</span></span>
<span id="cb125-31"><a href="#cb125-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">stat =</span> <span class="st">'bin'</span>) <span class="sc">+</span></span>
<span id="cb125-32"><a href="#cb125-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb125-33"><a href="#cb125-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">"Relative Density of Proteins"</span>) <span class="sc">+</span></span>
<span id="cb125-34"><a href="#cb125-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Percent of 8mers"</span>) <span class="sc">+</span></span>
<span id="cb125-35"><a href="#cb125-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="at">rows =</span> <span class="st">"mismatches"</span>, <span class="at">scale =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb125-36"><a href="#cb125-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> Family_palette_v2) <span class="sc">+</span></span>
<span id="cb125-37"><a href="#cb125-37" aria-hidden="true" tabindex="-1"></a>    <span class="co">#scale_y_continuous(labels = scales::percent)+</span></span>
<span id="cb125-38"><a href="#cb125-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span>percent)<span class="sc">+</span></span>
<span id="cb125-39"><a href="#cb125-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"8mers All Viral Proteins"</span>)</span>
<span id="cb125-40"><a href="#cb125-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-41"><a href="#cb125-41" aria-hidden="true" tabindex="-1"></a>ggS2A</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"FigureS2A.pdf"</span>), <span class="at">height =</span> <span class="fl">4.5</span>, <span class="at">width =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="supplemental-panel-2b" class="level3">
<h3 class="anchored" data-anchor-id="supplemental-panel-2b">Supplemental Panel 2B</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a>out_protein_final_min <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/12mer_proteome_protein_everything_min.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb127-2"><a href="#cb127-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-3"><a href="#cb127-3" aria-hidden="true" tabindex="-1"></a>number_of_proteins <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Proteome.ID, Protein.count)</span>
<span id="cb127-4"><a href="#cb127-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-5"><a href="#cb127-5" aria-hidden="true" tabindex="-1"></a>viral_ids <span class="ot">&lt;-</span> <span class="fu">unique</span>(out_protein_final_min<span class="sc">$</span>Proteome.ID)</span>
<span id="cb127-6"><a href="#cb127-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-7"><a href="#cb127-7" aria-hidden="true" tabindex="-1"></a>out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb127-8"><a href="#cb127-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> viral_ids){</span>
<span id="cb127-9"><a href="#cb127-9" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>)</span>
<span id="cb127-10"><a href="#cb127-10" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> file_name[<span class="fu">grepl</span>(i, file_name)]</span>
<span id="cb127-11"><a href="#cb127-11" aria-hidden="true" tabindex="-1"></a> fasta <span class="ot">&lt;-</span> seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="fu">paste0</span>(<span class="st">"../../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>, file_name), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>, <span class="at">whole.header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb127-12"><a href="#cb127-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb127-13"><a href="#cb127-13" aria-hidden="true" tabindex="-1"></a>  tmp.df <span class="ot">&lt;-</span> out_protein_final_min <span class="sc">%&gt;%</span></span>
<span id="cb127-14"><a href="#cb127-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">number_of_12mer =</span> PLength <span class="sc">-</span> <span class="dv">11</span>) <span class="sc">%&gt;%</span></span>
<span id="cb127-15"><a href="#cb127-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">freq =</span> count <span class="sc">/</span> number_of_12mer) <span class="sc">%&gt;%</span></span>
<span id="cb127-16"><a href="#cb127-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Proteome.ID <span class="sc">==</span> i) <span class="sc">%&gt;%</span></span>
<span id="cb127-17"><a href="#cb127-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">as.factor</span>(mismatches)) <span class="sc">%&gt;%</span></span>
<span id="cb127-18"><a href="#cb127-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">factor</span>(query_seqname, <span class="fu">unique</span>(<span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, <span class="fu">names</span>(fasta))))) <span class="sc">%&gt;%</span></span>
<span id="cb127-19"><a href="#cb127-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">complete</span>(query_seqname, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb127-20"><a href="#cb127-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">freq =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(freq), <span class="dv">0</span>, freq))</span>
<span id="cb127-21"><a href="#cb127-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb127-22"><a href="#cb127-22" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out, tmp.df)</span>
<span id="cb127-23"><a href="#cb127-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb127-24"><a href="#cb127-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb127-25"><a href="#cb127-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-26"><a href="#cb127-26" aria-hidden="true" tabindex="-1"></a>ggS2B <span class="ot">&lt;-</span> out <span class="sc">%&gt;%</span></span>
<span id="cb127-27"><a href="#cb127-27" aria-hidden="true" tabindex="-1"></a>       <span class="fu">mutate</span>(<span class="at">mismatches =</span> <span class="fu">factor</span>(mismatches, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb127-28"><a href="#cb127-28" aria-hidden="true" tabindex="-1"></a>       <span class="fu">left_join</span>(metadata) <span class="sc">%&gt;%</span></span>
<span id="cb127-29"><a href="#cb127-29" aria-hidden="true" tabindex="-1"></a>       <span class="fu">filter</span>(Family_Viral_Zone <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2)) <span class="sc">%&gt;%</span></span>
<span id="cb127-30"><a href="#cb127-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>freq, <span class="at">y =</span> ..count..<span class="sc">/</span><span class="fu">sum</span>(..count..), <span class="at">color =</span> Family_Viral_Zone, <span class="at">group =</span> Proteome.ID)) <span class="sc">+</span></span>
<span id="cb127-31"><a href="#cb127-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">stat =</span> <span class="st">'bin'</span>) <span class="sc">+</span></span>
<span id="cb127-32"><a href="#cb127-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb127-33"><a href="#cb127-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">"Relative Density of Proteins"</span>) <span class="sc">+</span></span>
<span id="cb127-34"><a href="#cb127-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Percent of 12mers"</span>) <span class="sc">+</span></span>
<span id="cb127-35"><a href="#cb127-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="at">rows =</span> <span class="st">"mismatches"</span>, <span class="at">scale =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb127-36"><a href="#cb127-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> Family_palette_v2) <span class="sc">+</span></span>
<span id="cb127-37"><a href="#cb127-37" aria-hidden="true" tabindex="-1"></a>    <span class="co">#scale_y_continuous(labels = scales::percent)+</span></span>
<span id="cb127-38"><a href="#cb127-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span>percent)<span class="sc">+</span></span>
<span id="cb127-39"><a href="#cb127-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"All viral proteins"</span>)</span>
<span id="cb127-40"><a href="#cb127-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-41"><a href="#cb127-41" aria-hidden="true" tabindex="-1"></a>ggS2B</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"FigureS2B.pdf"</span>), <span class="at">height =</span> <span class="dv">6</span>, <span class="at">width =</span> <span class="dv">5</span>)</span>
<span id="cb128-2"><a href="#cb128-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(ggS2B[[<span class="st">"data"</span>]], <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS2B_datapoints.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="supplemental-panel-2c" class="level3">
<h3 class="anchored" data-anchor-id="supplemental-panel-2c">Supplemental Panel 2C</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(seqinr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'seqinr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:treeio':

    read.fasta</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:biomaRt':

    getSequence</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:plyr':

    count</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    count</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biostrings':

    translate</code></pre>
</div>
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a>AA_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"A"</span> <span class="ot">=</span> <span class="st">"#f3c300"</span>, <span class="st">"C"</span> <span class="ot">=</span> <span class="st">"#875692"</span>, <span class="st">"D"</span> <span class="ot">=</span> <span class="st">"#f38400"</span>, <span class="st">"E"</span> <span class="ot">=</span> <span class="st">"#a1caf1"</span>, <span class="st">"F"</span> <span class="ot">=</span> <span class="st">"#be0032"</span>, <span class="st">"G"</span> <span class="ot">=</span> <span class="st">"#c2b290"</span>,</span>
<span id="cb136-2"><a href="#cb136-2" aria-hidden="true" tabindex="-1"></a>               <span class="st">"H"</span> <span class="ot">=</span> <span class="st">"#008856"</span>, <span class="st">"I"</span> <span class="ot">=</span> <span class="st">"#e68fac"</span>, <span class="st">"K"</span> <span class="ot">=</span> <span class="st">"#0067a5"</span>, <span class="st">"L"</span> <span class="ot">=</span> <span class="st">"#f99379"</span>, <span class="st">"M"</span> <span class="ot">=</span> <span class="st">"#604e97"</span>, <span class="st">"N"</span> <span class="ot">=</span> <span class="st">"#f6a600"</span>, <span class="st">"P"</span> <span class="ot">=</span> <span class="st">"#b3446c"</span>, <span class="st">"Q"</span> <span class="ot">=</span> <span class="st">"#dcd300"</span>,</span>
<span id="cb136-3"><a href="#cb136-3" aria-hidden="true" tabindex="-1"></a>               <span class="st">"R"</span> <span class="ot">=</span> <span class="st">"#882d17"</span>, <span class="st">"S"</span> <span class="ot">=</span> <span class="st">"#8db600"</span>, <span class="st">"T"</span> <span class="ot">=</span> <span class="st">"#654522"</span>, <span class="st">"V"</span> <span class="ot">=</span> <span class="st">"#e25822"</span>, <span class="st">"W"</span> <span class="ot">=</span> <span class="st">"#2b3d26"</span>, <span class="st">"Y"</span> <span class="ot">=</span> <span class="st">"lightgray"</span>, <span class="st">"X"</span> <span class="ot">=</span> <span class="st">"black"</span>, <span class="st">"U"</span> <span class="ot">=</span> <span class="st">"#848283"</span>)</span>
<span id="cb136-4"><a href="#cb136-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-5"><a href="#cb136-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-6"><a href="#cb136-6" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>)</span>
<span id="cb136-7"><a href="#cb136-7" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb136-8"><a href="#cb136-8" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".fasta.gz"</span>, files_to_read)]</span>
<span id="cb136-9"><a href="#cb136-9" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste0</span>(metadata<span class="sc">$</span><span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>],<span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb136-10"><a href="#cb136-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-11"><a href="#cb136-11" aria-hidden="true" tabindex="-1"></a>AA_freq <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb136-12"><a href="#cb136-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb136-13"><a href="#cb136-13" aria-hidden="true" tabindex="-1"></a>  tmp.fasta <span class="ot">&lt;-</span> seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(active_file, <span class="at">seqtype =</span> <span class="st">"AA"</span>,</span>
<span id="cb136-14"><a href="#cb136-14" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">as.string =</span> <span class="cn">FALSE</span>, <span class="at">whole.header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb136-15"><a href="#cb136-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb136-16"><a href="#cb136-16" aria-hidden="true" tabindex="-1"></a>  AA_table <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(<span class="fu">unlist</span>(tmp.fasta))) <span class="sc">%&gt;%</span></span>
<span id="cb136-17"><a href="#cb136-17" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"amino_acid"</span> <span class="ot">=</span> <span class="dv">1</span>, <span class="st">"freq"</span> <span class="ot">=</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb136-18"><a href="#cb136-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">virus =</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">".fasta.gz"</span>, <span class="st">""</span>, active_file)))</span>
<span id="cb136-19"><a href="#cb136-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb136-20"><a href="#cb136-20" aria-hidden="true" tabindex="-1"></a>  AA_freq <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(AA_freq, AA_table)</span>
<span id="cb136-21"><a href="#cb136-21" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb137"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a>taxa <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb137-2"><a href="#cb137-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb137-3"><a href="#cb137-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb137-4"><a href="#cb137-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb137-5"><a href="#cb137-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb137-6"><a href="#cb137-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Proteome.ID <span class="sc">%in%</span> AA_freq<span class="sc">$</span>virus) <span class="sc">%&gt;%</span></span>
<span id="cb137-7"><a href="#cb137-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"Taxon.mnemonic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb137-8"><a href="#cb137-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(family, order, class, phylum, kingdom) <span class="sc">%&gt;%</span></span>
<span id="cb137-9"><a href="#cb137-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb137-10"><a href="#cb137-10" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Anelloviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span id="cb137-11"><a href="#cb137-11" aria-hidden="true" tabindex="-1"></a>taxa[taxa<span class="sc">$</span>family <span class="sc">==</span> <span class="st">"Kolmioviridae"</span>,] <span class="ot">&lt;-</span> <span class="st">"Kolmioviridae"</span></span>
<span id="cb137-12"><a href="#cb137-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-13"><a href="#cb137-13" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">lapply</span>(taxa, as.factor), <span class="at">stringsAsFactors =</span><span class="cn">TRUE</span>)</span>
<span id="cb137-14"><a href="#cb137-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(check) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(taxa)</span>
<span id="cb137-15"><a href="#cb137-15" aria-hidden="true" tabindex="-1"></a>check <span class="ot">&lt;-</span> check[<span class="fu">complete.cases</span>(check),]</span>
<span id="cb137-16"><a href="#cb137-16" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">taxo2phylog</span>(<span class="fu">as.taxo</span>(check), <span class="at">root=</span><span class="st">"Root"</span>, <span class="at">abbrev=</span>F)</span>
<span id="cb137-17"><a href="#cb137-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-18"><a href="#cb137-18" aria-hidden="true" tabindex="-1"></a>tax.phy <span class="ot">&lt;-</span> <span class="fu">as.phylo</span>(tmp)</span>
<span id="cb137-19"><a href="#cb137-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-20"><a href="#cb137-20" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tax.phy, <span class="at">branch.length=</span><span class="st">'none'</span>)</span>
<span id="cb137-21"><a href="#cb137-21" aria-hidden="true" tabindex="-1"></a>gg_tree</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb138"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1" aria-hidden="true" tabindex="-1"></a>gg_tree[[<span class="st">"data"</span>]] <span class="ot">&lt;-</span> gg_tree[[<span class="st">"data"</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb138-2"><a href="#cb138-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata <span class="sc">%&gt;%</span></span>
<span id="cb138-3"><a href="#cb138-3" aria-hidden="true" tabindex="-1"></a>               <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb138-4"><a href="#cb138-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb138-5"><a href="#cb138-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb138-6"><a href="#cb138-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb138-7"><a href="#cb138-7" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(Taxon.mnemonic, family), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Taxon.mnemonic"</span>))</span>
<span id="cb138-8"><a href="#cb138-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-9"><a href="#cb138-9" aria-hidden="true" tabindex="-1"></a>gg_tree <span class="ot">&lt;-</span>  gg_tree <span class="sc">+</span> <span class="fu">geom_hilight</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">subset =</span> family <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2),<span class="at">fill =</span> family), <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb138-10"><a href="#cb138-10" aria-hidden="true" tabindex="-1"></a>   <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> Family_palette_v2)</span>
<span id="cb138-11"><a href="#cb138-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-12"><a href="#cb138-12" aria-hidden="true" tabindex="-1"></a>gg_AA <span class="ot">&lt;-</span> AA_freq <span class="sc">%&gt;%</span></span>
<span id="cb138-13"><a href="#cb138-13" aria-hidden="true" tabindex="-1"></a>   <span class="fu">left_join</span>(metadata <span class="sc">%&gt;%</span></span>
<span id="cb138-14"><a href="#cb138-14" aria-hidden="true" tabindex="-1"></a>               <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(Taxon.mnemonic),</span>
<span id="cb138-15"><a href="#cb138-15" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"Simian virus 5"</span>, Taxon.mnemonic)) <span class="sc">%&gt;%</span></span>
<span id="cb138-16"><a href="#cb138-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb138-17"><a href="#cb138-17" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">`</span><span class="at">Proteome.ID</span><span class="st">`</span>, Taxon.mnemonic)),</span>
<span id="cb138-18"><a href="#cb138-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"virus"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb138-19"><a href="#cb138-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>freq, <span class="at">y=</span>Taxon.mnemonic, <span class="at">fill =</span> amino_acid)) <span class="sc">+</span></span>
<span id="cb138-20"><a href="#cb138-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="fu">position_fill</span>(<span class="at">reverse =</span> <span class="cn">TRUE</span>),</span>
<span id="cb138-21"><a href="#cb138-21" aria-hidden="true" tabindex="-1"></a>           <span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb138-22"><a href="#cb138-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb138-23"><a href="#cb138-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> AA_colors) <span class="sc">+</span></span>
<span id="cb138-24"><a href="#cb138-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Frequency"</span>)</span>
<span id="cb138-25"><a href="#cb138-25" aria-hidden="true" tabindex="-1"></a>gg_AA</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/unnamed-chunk-7-2.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb139"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a>panel_S2C <span class="ot">&lt;-</span> gg_AA <span class="sc">%&gt;%</span> <span class="fu">insert_left</span>(gg_tree, <span class="at">width =</span> <span class="fl">0.2</span>) </span>
<span id="cb139-2"><a href="#cb139-2" aria-hidden="true" tabindex="-1"></a>panel_S2C</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figure1_files/figure-html/unnamed-chunk-7-3.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb140"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"FigureS2C.pdf"</span>), <span class="at">height =</span> <span class="dv">15</span>, <span class="at">width =</span> <span class="dv">8</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb141"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb141-1"><a href="#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(AA_freq <span class="sc">%&gt;%</span></span>
<span id="cb141-2"><a href="#cb141-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(virus) <span class="sc">%&gt;%</span></span>
<span id="cb141-3"><a href="#cb141-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">percent =</span> freq <span class="sc">/</span> <span class="fu">sum</span>(freq) <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb141-4"><a href="#cb141-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> amino_acid, <span class="at">values_from =</span> percent, <span class="at">id_cols =</span> virus) <span class="sc">%&gt;%</span></span>
<span id="cb141-5"><a href="#cb141-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(metadata <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Proteome.ID, Pathogen), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"virus"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span> <span class="fu">relocate</span>(virus, Pathogen), <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS2c_datapoints.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>