<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>FigureS1</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="FigureS1_ELMFilter_files/libs/clipboard/clipboard.min.js"></script>
<script src="FigureS1_ELMFilter_files/libs/quarto-html/quarto.js"></script>
<script src="FigureS1_ELMFilter_files/libs/quarto-html/popper.min.js"></script>
<script src="FigureS1_ELMFilter_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="FigureS1_ELMFilter_files/libs/quarto-html/anchor.min.js"></script>
<link href="FigureS1_ELMFilter_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="FigureS1_ELMFilter_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="FigureS1_ELMFilter_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="FigureS1_ELMFilter_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="FigureS1_ELMFilter_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">FigureS1</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="figure-s1" class="level2">
<h2 class="anchored" data-anchor-id="figure-s1">Figure S1</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/stor/work/Melamed_COVID/Cole/git_clones/MolecularMimicry/Analysis/Figure1/src/"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dtplyr)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plyr)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggbeeswarm)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(wesanderson)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(aplot)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(treeio)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggtree)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(aplot)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggbeeswarm)</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="do">### mart &lt;- useMart('ENSEMBL_MART_ENSEMBL')</span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="do">### mart &lt;- useDataset('hsapiens_gene_ensembl', mart)</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>mart <span class="ot">&lt;-</span> <span class="fu">useEnsembl</span>(<span class="st">"ensembl"</span>, <span class="at">dataset =</span> <span class="st">"hsapiens_gene_ensembl"</span>) <span class="do">## , mirror = "useast"</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>mart_retrieved <span class="ot">&lt;-</span> <span class="fu">getBM</span>(</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">mart =</span> mart, <span class="at">useCache =</span> T,</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">attributes =</span> <span class="fu">c</span>(</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>    <span class="st">'ensembl_gene_id'</span>,</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>    <span class="st">'entrezgene_id'</span>,</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>    <span class="st">'external_gene_name'</span>,</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>    <span class="st">'uniprot_gn_symbol'</span>,</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>    <span class="st">'uniprot_gn_id'</span>,</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>    <span class="st">'chromosome_name'</span>,</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>    <span class="st">'description'</span>),</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">uniqueRows=</span><span class="cn">TRUE</span>)</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>elm_classes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"../../Databases/elm_classes.tsv"</span>, <span class="at">skip =</span> <span class="dv">5</span>)</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>elm_classes_human <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"../../Databases/elm_instances_human.tsv"</span>, <span class="at">skip =</span> <span class="dv">5</span>)</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../../Utilities.R"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>meta_path <span class="ot">&lt;-</span> <span class="st">"../../Metadata/"</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> openxlsx<span class="sc">::</span><span class="fu">read.xlsx</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"Final_Cohort_Metadata_v5.xlsx"</span>), <span class="at">sheet =</span> <span class="dv">1</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>duplicated_proteins <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"duplicated_proteins.csv"</span>))</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>human_duplicated_proteins <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"human_duplicated_or_removed_proteins.csv"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="dv">1</span>)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>proteome_metadata <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"uniprot_proteome_metadata.tsv"</span>))</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>viral_protein_metadata <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"uniprot_viral_protein_metadata.xlsx"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in Z1741 / R1741C26: got 'NON_STD 64;
/note="Selenocysteine"; /evidence="ECO:0000250"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U1978 / R1978C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA1978 / R1978C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AIC64083.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in W1994 / R1994C23: got 'MASS SPECTROMETRY:
[Non-structural protein 8]: Mass=21871; Method=Electrospray;
Evidence={ECO:0000269|PubMed:32083638}; MASS SPECTROMETRY: [Non-structural
protein 9]: Mass=12403; Method=Electrospray;
Evidence={ECO:0000269|PubMed:32083638}; MASS SPECTROMETRY: [Non-structural
protein 10]: Mass=14974; Method=Electrospray;
Evidence={ECO:0000269|PubMed:32083638};'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in W1995 / R1995C23: got 'MASS SPECTROMETRY:
[Non-structural protein 8]: Mass=21871; Method=Electrospray;
Evidence={ECO:0000269|PubMed:32083638}; MASS SPECTROMETRY: [Non-structural
protein 9]: Mass=12403; Method=Electrospray;
Evidence={ECO:0000269|PubMed:32083638}; MASS SPECTROMETRY: [Non-structural
protein 10]: Mass=14974; Method=Electrospray;
Evidence={ECO:0000269|PubMed:32083638};'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2169 / R2169C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2169 / R2169C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54757.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2170 / R2170C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2170 / R2170C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54658.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2177 / R2177C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2177 / R2177C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54660.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2178 / R2178C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2178 / R2178C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54759.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2181 / R2181C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2181 / R2181C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54755.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2207 / R2207C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2207 / R2207C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54739.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2217 / R2217C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2217 / R2217C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54742.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2229 / R2229C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2229 / R2229C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54758.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2230 / R2230C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2230 / R2230C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54659.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2244 / R2244C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2244 / R2244C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54749.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U2246 / R2246C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA2246 / R2246C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:AAC54741.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3051 / R3051C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3051 / R3051C27: got 'NON_TER 723;
/evidence="ECO:0000313|EMBL:CAD53434.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3061 / R3061C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3061 / R3061C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53393.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3062 / R3062C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3062 / R3062C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53389.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3063 / R3063C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3063 / R3063C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53394.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3064 / R3064C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3064 / R3064C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53391.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3065 / R3065C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3065 / R3065C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53390.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3066 / R3066C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3066 / R3066C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53386.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3067 / R3067C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3067 / R3067C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53392.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3068 / R3068C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3068 / R3068C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53388.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3116 / R3116C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3116 / R3116C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAD53466.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U3442 / R3442C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA3442 / R3442C27: got 'NON_TER 49;
/evidence="ECO:0000313|EMBL:AAD25087.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in Z3605 / R3605C26: got 'NON_STD 181;
/note="Selenocysteine"; /evidence="ECO:0000313|EMBL:CAL35617.1"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in Z3677 / R3677C26: got 'NON_STD 17;
/note="Selenocysteine"; /evidence="ECO:0000255"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in U4860 / R4860C21: got 'fragment'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in AA4860 / R4860C27: got 'NON_TER 1;
/evidence="ECO:0000313|EMBL:CAL35657.1"'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(viral_protein_metadata) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">gsub</span>(<span class="st">" "</span>, <span class="st">"_"</span>, <span class="fu">tolower</span>(<span class="fu">colnames</span>(viral_protein_metadata))), <span class="st">".viral"</span>)</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>human_protein_metadata <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"uniprot_protein_metadata.xlsx"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L1053 / R1053C12: got 'NON_STD 96;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in R2045 / R2045C18: got 'BIOPHYSICOCHEMICAL
PROPERTIES: Absorption: Abs(max)=420 nm;'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in R2046 / R2046C18: got 'BIOPHYSICOCHEMICAL
PROPERTIES: Absorption: Abs(max)=560 nm;'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in R2047 / R2047C18: got 'BIOPHYSICOCHEMICAL
PROPERTIES: Absorption: Abs(max)=530 nm {ECO:0000305|PubMed:2937147};'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L2235 / R2235C12: got 'NON_STD 49;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L2996 / R2996C12: got 'NON_STD 40;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L3216 / R3216C12: got 'NON_STD 73;
/note="Selenocysteine"; /evidence="ECO:0000269|PubMed:2229017"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L3884 / R3884C12: got 'NON_STD 73;
/note="Selenocysteine"; /evidence="ECO:0000305|PubMed:8039723,
ECO:0000305|PubMed:9705830"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L4382 / R4382C12: got 'NON_STD 126;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L4386 / R4386C12: got 'NON_STD 59;
/note="Selenocysteine"; NON_STD 300; /note="Selenocysteine"; NON_STD 318;
/note="Selenocysteine"; NON_STD 330; /note="Selenocysteine"; NON_STD 345;
/note="Selenocysteine"; NON_STD 352; /note="Selenocysteine"; NON_STD 367;
/note="Selenocysteine"; NON_STD 369; /note="Selenocysteine"; NON_STD 376;
/note="Selenocysteine"; NON_STD 378; /note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L4724 / R4724C12: got 'NON_STD 170;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L5044 / R5044C12: got 'NON_STD 49;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L6708 / R6708C12: got 'NON_STD 648;
/note="Selenocysteine"; /evidence="ECO:0000269|PubMed:8650234"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L8342 / R8342C12: got 'NON_STD 642;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L9715 / R9715C12: got 'NON_STD 48;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L9929 / R9929C12: got 'NON_STD 133;
/note="Selenocysteine"; NON_STD 266; /note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L10958 / R10958C12: got 'NON_STD 60;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L11117 / R11117C12: got 'NON_STD 188;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L11330 / R11330C12: got 'NON_STD 667;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L11638 / R11638C12: got 'NON_STD 387;
/note="Selenocysteine"; /evidence="ECO:0000269|PubMed:12775843"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L12326 / R12326C12: got 'NON_STD 523;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L12976 / R12976C12: got 'NON_STD 127;
/note="Selenocysteine"; NON_STD 462; /note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L12977 / R12977C12: got 'NON_STD 95;
/note="Selenocysteine"; /evidence="ECO:0000305|PubMed:10608886"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L14255 / R14255C12: got 'NON_STD 92;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L14927 / R14927C12: got 'NON_STD 73;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L14928 / R14928C12: got 'NON_STD 273;
/note="Selenocysteine"; /evidence="ECO:0000305|PubMed:12775843"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L18715 / R18715C12: got 'NON_STD 13;
/note="Selenocysteine"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in M20179 / R20179C13: got 'NON_CONS 336..337;
/evidence="ECO:0000305"'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expecting logical in L33085 / R33085C12: got 'NON_STD 44;
/note="Selenocysteine"'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(human_protein_metadata) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">gsub</span>(<span class="st">" "</span>, <span class="st">"_"</span>, <span class="fu">tolower</span>(<span class="fu">colnames</span>(human_protein_metadata))), <span class="st">".human"</span>)</span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>out.tmp <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(human_protein_metadata)){ <span class="co">#nrow(human_protein_metadata)</span></span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a> out.tmp[i] <span class="ot">&lt;-</span>  <span class="fu">list</span>(<span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">"VARIANT "</span>, <span class="st">""</span>, <span class="fu">unlist</span>(<span class="fu">regmatches</span>(human_protein_metadata<span class="sc">$</span>natural_variant.human[i], <span class="fu">gregexpr</span>(<span class="st">"VARIANT [0-9]*"</span>, human_protein_metadata<span class="sc">$</span>natural_variant.human[i]))))))</span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a>human_protein_metadata<span class="sc">$</span>human_variants <span class="ot">&lt;-</span> out.tmp</span>
<span id="cb86-7"><a href="#cb86-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-8"><a href="#cb86-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-9"><a href="#cb86-9" aria-hidden="true" tabindex="-1"></a>tissue_map <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"normal_tissue.tsv"</span>))</span>
<span id="cb86-10"><a href="#cb86-10" aria-hidden="true" tabindex="-1"></a>organ_map <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="fu">paste0</span>(meta_path, <span class="st">"Tissue_Groupings.xlsx"</span>))</span>
<span id="cb86-11"><a href="#cb86-11" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> <span class="fu">merge</span>(tissue_map, organ_map, <span class="at">by.x =</span> <span class="st">"Tissue"</span>, <span class="at">by.y =</span> <span class="st">"Var1"</span>, <span class="at">all.x =</span> T)</span>
<span id="cb86-12"><a href="#cb86-12" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(tissue_organ_map<span class="sc">$</span>Level)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Not detected"       "High"               "Low"               
[4] "Medium"             "Ascending"          "Descending"        
[7] "Not representative" "N/A"               </code></pre>
</div>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Level <span class="sc">!=</span> <span class="st">"Not detected"</span>,]</span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Level <span class="sc">!=</span> <span class="st">"Not representative"</span>,]</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Level <span class="sc">!=</span> <span class="st">"N/A"</span>,]</span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(tissue_organ_map<span class="sc">$</span>Reliability)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Uncertain" "Enhanced"  "Approved"  "Supported"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Reliability <span class="sc">!=</span> <span class="st">"Uncertain"</span>,]</span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>tissue_organ_map <span class="ot">&lt;-</span> tissue_organ_map[tissue_organ_map<span class="sc">$</span>Reliability <span class="sc">!=</span> <span class="st">"NA"</span>,]</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>tissue_organ_map_condensed <span class="ot">&lt;-</span> tissue_organ_map <span class="sc">%&gt;%</span></span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Tissue, Gene, <span class="st">`</span><span class="at">Cell type</span><span class="st">`</span>, <span class="st">`</span><span class="at">New Name</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Gene) <span class="sc">%&gt;%</span></span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">organ_system =</span> <span class="fu">gsub</span>(<span class="st">"NA;"</span>, <span class="st">""</span>, <span class="fu">paste</span>(<span class="fu">unique</span>(<span class="st">`</span><span class="at">New Name</span><span class="st">`</span>), <span class="at">collapse =</span> <span class="st">";"</span>)),</span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a>                   <span class="at">tissue =</span> <span class="fu">gsub</span>(<span class="st">"NA;"</span>, <span class="st">""</span>, <span class="fu">paste</span>(<span class="fu">unique</span>(Tissue), <span class="at">collapse =</span> <span class="st">";"</span>)),</span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a>                   <span class="at">cell_type =</span> <span class="fu">gsub</span>(<span class="st">"NA;"</span>, <span class="st">""</span>, <span class="fu">paste</span>(<span class="fu">unique</span>(<span class="st">`</span><span class="at">Cell type</span><span class="st">`</span>), <span class="at">collapse =</span> <span class="st">";"</span>)))</span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-11"><a href="#cb90-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-12"><a href="#cb90-12" aria-hidden="true" tabindex="-1"></a>out_dir <span class="ot">&lt;-</span> <span class="st">"../output/"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="elm-location-identification" class="level2">
<h2 class="anchored" data-anchor-id="elm-location-identification">ELM location identification</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="do">## This takes a while to run, the generated file is already on the github (so could be and by default is a skipped chunk)</span></span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>)</span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>elm_instance_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_virus <span class="cf">in</span> files_to_read){</span>
<span id="cb91-5"><a href="#cb91-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"On "</span>, <span class="fu">match</span>(active_virus, files_to_read), <span class="st">" virus out of "</span>, <span class="fu">length</span>(files_to_read)))</span>
<span id="cb91-6"><a href="#cb91-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(active_virus)</span>
<span id="cb91-7"><a href="#cb91-7" aria-hidden="true" tabindex="-1"></a>  active_virus_proteome <span class="ot">&lt;-</span> seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>, active_virus), <span class="at">as.string =</span> T, <span class="at">seqtype =</span> <span class="st">"AA"</span>)</span>
<span id="cb91-8"><a href="#cb91-8" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(active_protein <span class="cf">in</span> active_virus_proteome){</span>
<span id="cb91-9"><a href="#cb91-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(active_elm_class <span class="cf">in</span> elm_classes<span class="sc">$</span>Regex){</span>
<span id="cb91-10"><a href="#cb91-10" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb91-11"><a href="#cb91-11" aria-hidden="true" tabindex="-1"></a>      tmp <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_locate_all</span>(active_protein[<span class="dv">1</span>], active_elm_class)</span>
<span id="cb91-12"><a href="#cb91-12" aria-hidden="true" tabindex="-1"></a>      tmp <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(tmp)</span>
<span id="cb91-13"><a href="#cb91-13" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">nrow</span>(tmp) <span class="sc">==</span> <span class="dv">0</span>){<span class="cf">next</span>} <span class="do">## was originally active_elm_class by accident</span></span>
<span id="cb91-14"><a href="#cb91-14" aria-hidden="true" tabindex="-1"></a>      tmp <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(tmp) <span class="sc">%&gt;%</span></span>
<span id="cb91-15"><a href="#cb91-15" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mutate</span>(<span class="at">viral_protein =</span> <span class="fu">attr</span>(active_protein,<span class="st">"Annot"</span>),</span>
<span id="cb91-16"><a href="#cb91-16" aria-hidden="true" tabindex="-1"></a>               <span class="at">Regex =</span> active_elm_class,</span>
<span id="cb91-17"><a href="#cb91-17" aria-hidden="true" tabindex="-1"></a>               <span class="at">File_name =</span> active_virus,</span>
<span id="cb91-18"><a href="#cb91-18" aria-hidden="true" tabindex="-1"></a>               <span class="at">Proteome.ID =</span> <span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>, <span class="fu">gsub</span>(<span class="st">".fasta.gz"</span>, <span class="st">""</span>, active_virus)))</span>
<span id="cb91-19"><a href="#cb91-19" aria-hidden="true" tabindex="-1"></a>      elm_instance_df <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(elm_instance_df, tmp)</span>
<span id="cb91-20"><a href="#cb91-20" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb91-21"><a href="#cb91-21" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb91-22"><a href="#cb91-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb91-23"><a href="#cb91-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-24"><a href="#cb91-24" aria-hidden="true" tabindex="-1"></a>elm_instance_df <span class="ot">&lt;-</span> elm_instance_df <span class="sc">%&gt;%</span></span>
<span id="cb91-25"><a href="#cb91-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(elm_classes <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(Accession, ELMIdentifier, FunctionalSiteName, Description, Regex, Probability))</span>
<span id="cb91-26"><a href="#cb91-26" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb91-27"><a href="#cb91-27" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(elm_instance_df, <span class="st">"/stor/work/Melamed_COVID/Cole/git_clones/MolecularMimicry/Analysis/Key_Data/elm_labeled_viral_sequences_v2.csv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="mer-data-annotation" class="level2">
<h2 class="anchored" data-anchor-id="mer-data-annotation">8mer Data Annotation</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="do">## This will only run for files that haven't ran already</span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../../epitope_alignment/Results/Complete_Final_Set_8mers_3mis/"</span>)</span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".tsv"</span>, files_to_read)]</span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a>files_completed <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../../../Analysis/8mer_annotated/"</span>)</span>
<span id="cb92-6"><a href="#cb92-6" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="sc">!</span><span class="fu">gsub</span>(<span class="st">".tsv.gz"</span>, <span class="st">""</span>, files_to_read) <span class="sc">%in%</span> <span class="fu">gsub</span>(<span class="st">".csv.gz"</span>, <span class="st">""</span>, files_completed)]</span>
<span id="cb92-7"><a href="#cb92-7" aria-hidden="true" tabindex="-1"></a>final_cohort_naming <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../../Viral_Proteomes/Final_Cohort"</span>)</span>
<span id="cb92-8"><a href="#cb92-8" aria-hidden="true" tabindex="-1"></a>final_cohort_naming <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">gsub</span>(<span class="st">".fasta.gz"</span>, <span class="st">"_8mer_3mis_hits.tsv.gz"</span>, final_cohort_naming))</span>
<span id="cb92-9"><a href="#cb92-9" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[files_to_read <span class="sc">%in%</span> final_cohort_naming]</span>
<span id="cb92-10"><a href="#cb92-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-11"><a href="#cb92-11" aria-hidden="true" tabindex="-1"></a>files_to_read<span class="ot">&lt;-</span> files_to_read[<span class="fu">order</span>(<span class="fu">file.info</span>(files_to_read)<span class="sc">$</span>size)]</span>
<span id="cb92-12"><a href="#cb92-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-13"><a href="#cb92-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb92-14"><a href="#cb92-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb92-15"><a href="#cb92-15" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">nThread =</span> <span class="dv">12</span>)</span>
<span id="cb92-16"><a href="#cb92-16" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb92-17"><a href="#cb92-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Skipping "</span>, active_file, <span class="st">" due to lack of data."</span>))</span>
<span id="cb92-18"><a href="#cb92-18" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb92-19"><a href="#cb92-19" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb92-20"><a href="#cb92-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-21"><a href="#cb92-21" aria-hidden="true" tabindex="-1"></a> active_data[, uniprot <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(target_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb92-22"><a href="#cb92-22" aria-hidden="true" tabindex="-1"></a> active_data[, viral_id <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(query_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb92-23"><a href="#cb92-23" aria-hidden="true" tabindex="-1"></a> active_data[, mismatches <span class="sc">:</span><span class="er">=</span> <span class="fu">mismatch_calling_fast</span>(query, target), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb92-24"><a href="#cb92-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb92-25"><a href="#cb92-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Data Merging "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb92-26"><a href="#cb92-26" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb92-27"><a href="#cb92-27" aria-hidden="true" tabindex="-1"></a>     <span class="fu">lazy_dt</span>() <span class="sc">%&gt;%</span></span>
<span id="cb92-28"><a href="#cb92-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(mart_retrieved, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="st">"uniprot_gn_id"</span>),</span>
<span id="cb92-29"><a href="#cb92-29" aria-hidden="true" tabindex="-1"></a>              <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb92-30"><a href="#cb92-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(tissue_organ_map_condensed, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span> <span class="ot">=</span> <span class="st">"Gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb92-31"><a href="#cb92-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(human_protein_metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span><span class="st">"entry.human"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb92-32"><a href="#cb92-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(viral_protein_metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"viral_id"</span> <span class="ot">=</span> <span class="st">"entry.viral"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb92-33"><a href="#cb92-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">catalytic_activity.same =</span> catalytic_activity.human <span class="sc">==</span> catalytic_activity.viral) <span class="sc">%&gt;%</span></span>
<span id="cb92-34"><a href="#cb92-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.human =</span> <span class="fu">gsub</span>(<span class="st">";.*"</span>, <span class="st">""</span>, catalytic_activity.human)) <span class="sc">%&gt;%</span></span>
<span id="cb92-35"><a href="#cb92-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.viral =</span> <span class="fu">gsub</span>(<span class="st">";.*"</span>, <span class="st">""</span>, catalytic_activity.viral)) <span class="sc">%&gt;%</span></span>
<span id="cb92-36"><a href="#cb92-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.same =</span> reaction.human <span class="sc">==</span> reaction.viral) <span class="sc">%&gt;%</span></span>
<span id="cb92-37"><a href="#cb92-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.same =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(reaction.same), <span class="cn">FALSE</span>, reaction.same)) <span class="sc">%&gt;%</span></span>
<span id="cb92-38"><a href="#cb92-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">catalytic_activity.same =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(catalytic_activity.same), <span class="cn">FALSE</span>, catalytic_activity.same)) <span class="sc">%&gt;%</span></span>
<span id="cb92-39"><a href="#cb92-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">enzyme_number.same =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb92-40"><a href="#cb92-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.table</span>()</span>
<span id="cb92-41"><a href="#cb92-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb92-42"><a href="#cb92-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"EC Screening "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb92-43"><a href="#cb92-43" aria-hidden="true" tabindex="-1"></a>  active_data[, enzyme_number.same <span class="sc">:</span><span class="er">=</span> <span class="fu">fast_ec_screen</span>(ec_number.human, ec_number.viral), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb92-44"><a href="#cb92-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-45"><a href="#cb92-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Final Cleaning "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb92-46"><a href="#cb92-46" aria-hidden="true" tabindex="-1"></a>  active_data_subset <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span> </span>
<span id="cb92-47"><a href="#cb92-47" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">human_variants =</span> <span class="fu">sapply</span>(human_variants, toString)) <span class="sc">%&gt;%</span></span>
<span id="cb92-48"><a href="#cb92-48" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(query, query_seqname, query_start, target_seqname, target_start,</span>
<span id="cb92-49"><a href="#cb92-49" aria-hidden="true" tabindex="-1"></a>           target, mismatches, uniprot, viral_id, ensembl_gene_id,</span>
<span id="cb92-50"><a href="#cb92-50" aria-hidden="true" tabindex="-1"></a>           entrezgene_id, external_gene_name, uniprot_gn_symbol, chromosome_name,</span>
<span id="cb92-51"><a href="#cb92-51" aria-hidden="true" tabindex="-1"></a>           reviewed.human, entry_name.human, protein_names.human, length.human,</span>
<span id="cb92-52"><a href="#cb92-52" aria-hidden="true" tabindex="-1"></a>           active_site.human, binding_site.human, dna_binding.human,</span>
<span id="cb92-53"><a href="#cb92-53" aria-hidden="true" tabindex="-1"></a>           ec_number.human,  reviewed.viral, entry_name.viral,</span>
<span id="cb92-54"><a href="#cb92-54" aria-hidden="true" tabindex="-1"></a>           protein_names.viral, length.viral,</span>
<span id="cb92-55"><a href="#cb92-55" aria-hidden="true" tabindex="-1"></a>           active_site.viral, binding_site.viral, dna_binding.viral,</span>
<span id="cb92-56"><a href="#cb92-56" aria-hidden="true" tabindex="-1"></a>           ec_number.viral, catalytic_activity.same, reaction.human,</span>
<span id="cb92-57"><a href="#cb92-57" aria-hidden="true" tabindex="-1"></a>           reaction.viral, reaction.same,</span>
<span id="cb92-58"><a href="#cb92-58" aria-hidden="true" tabindex="-1"></a>           enzyme_number.same) <span class="sc">%&gt;%</span></span>
<span id="cb92-59"><a href="#cb92-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb92-60"><a href="#cb92-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Sys.time</span>()</span>
<span id="cb92-61"><a href="#cb92-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fwrite</span>(active_data_subset, <span class="fu">paste0</span>(<span class="st">"../../../Analysis/8mer_annotated/"</span>, <span class="fu">gsub</span>(<span class="st">".tsv"</span>, <span class="st">".csv"</span>, active_file)), <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">compress =</span> <span class="st">"gzip"</span>, <span class="at">nThread =</span> <span class="dv">12</span>)</span>
<span id="cb92-62"><a href="#cb92-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(active_data, active_data_subset)</span>
<span id="cb92-63"><a href="#cb92-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gc</span>()</span>
<span id="cb92-64"><a href="#cb92-64" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="mer-data-load-and-processing" class="level2">
<h2 class="anchored" data-anchor-id="mer-data-load-and-processing">8mer Data Load and Processing</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="do">### This only needs to be run once and will take a while, the script when not full render unless you already have the output in the expected path or if you set message: true</span></span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(seqinr)</span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringi)</span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(svMisc)</span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a>mode <span class="ot">=</span> <span class="st">"tight_elm"</span></span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a>kmer_length <span class="ot">=</span> <span class="dv">8</span></span>
<span id="cb93-7"><a href="#cb93-7" aria-hidden="true" tabindex="-1"></a>elm_instance_df <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"../../Key_Data/elm_labeled_viral_sequences_v2.csv.gz"</span>)</span>
<span id="cb93-8"><a href="#cb93-8" aria-hidden="true" tabindex="-1"></a>elm_db_expand <span class="ot">&lt;-</span> elm_instance_df <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(viral_protein, start, end, Regex) <span class="sc">%&gt;%</span></span>
<span id="cb93-9"><a href="#cb93-9" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">mutate</span>(<span class="at">viral_protein =</span> <span class="fu">gsub</span>(<span class="st">"&gt;"</span>, <span class="st">""</span>, viral_protein)) <span class="sc">%&gt;%</span></span>
<span id="cb93-10"><a href="#cb93-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb93-11"><a href="#cb93-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">do</span>(<span class="fu">data.frame</span>(<span class="at">query_seqname =</span> .<span class="sc">$</span>viral_protein, <span class="at">query_start =</span>(.<span class="sc">$</span>start<span class="sc">-</span>(kmer_length<span class="dv">-1</span>))<span class="sc">:</span>.<span class="sc">$</span>end)) <span class="sc">%&gt;%</span></span>
<span id="cb93-12"><a href="#cb93-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">in_elm =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb93-13"><a href="#cb93-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb93-14"><a href="#cb93-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(query_seqname)</span>
<span id="cb93-15"><a href="#cb93-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-16"><a href="#cb93-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-17"><a href="#cb93-17" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb93-18"><a href="#cb93-18" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../../8mer_annotated"</span>,<span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb93-19"><a href="#cb93-19" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".csv.gz"</span>, files_to_read)]</span>
<span id="cb93-20"><a href="#cb93-20" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb93-21"><a href="#cb93-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-22"><a href="#cb93-22" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"Number of Proteins"</span>, <span class="st">"Proteome Length"</span>))</span>
<span id="cb93-23"><a href="#cb93-23" aria-hidden="true" tabindex="-1"></a>out_protein_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb93-24"><a href="#cb93-24" aria-hidden="true" tabindex="-1"></a>out_protein_final_min <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb93-25"><a href="#cb93-25" aria-hidden="true" tabindex="-1"></a>out_protein_final_count <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb93-26"><a href="#cb93-26" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb93-27"><a href="#cb93-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" Pathogen number:"</span>,<span class="fu">match</span>(active_file, files_to_read), <span class="st">"/"</span>, <span class="fu">length</span>(files_to_read)))</span>
<span id="cb93-28"><a href="#cb93-28" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">nThread =</span> <span class="dv">18</span>) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb93-29"><a href="#cb93-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb93-30"><a href="#cb93-30" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb93-31"><a href="#cb93-31" aria-hidden="true" tabindex="-1"></a>    out_final[,<span class="fu">gsub</span>(<span class="st">"_8mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file)] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="cn">NA</span>,<span class="cn">NA</span>)</span>
<span id="cb93-32"><a href="#cb93-32" aria-hidden="true" tabindex="-1"></a>      data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>, <span class="fu">gsub</span>(<span class="st">"../../8mer_annotated/"</span>, <span class="st">""</span>, <span class="fu">gsub</span>(<span class="st">"_8mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file)), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb93-33"><a href="#cb93-33" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb93-34"><a href="#cb93-34" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb93-35"><a href="#cb93-35" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb93-36"><a href="#cb93-36" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb93-37"><a href="#cb93-37" aria-hidden="true" tabindex="-1"></a>      PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb93-38"><a href="#cb93-38" aria-hidden="true" tabindex="-1"></a>      PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb93-39"><a href="#cb93-39" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb93-40"><a href="#cb93-40" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb93-41"><a href="#cb93-41" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb93-42"><a href="#cb93-42" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb93-43"><a href="#cb93-43" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb93-44"><a href="#cb93-44" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb93-45"><a href="#cb93-45" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb93-46"><a href="#cb93-46" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb93-47"><a href="#cb93-47" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data  <span class="sc">%&gt;%</span></span>
<span id="cb93-48"><a href="#cb93-48" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Remove duplicated pathogen proteins identified in </span></span>
<span id="cb93-49"><a href="#cb93-49" aria-hidden="true" tabindex="-1"></a>    <span class="do">## ../../Scripts/Duplicate_Protein_Screen.R</span></span>
<span id="cb93-50"><a href="#cb93-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb93-51"><a href="#cb93-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb93-52"><a href="#cb93-52" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(query, query_seqname, query_start, mismatches, reaction.human,</span>
<span id="cb93-53"><a href="#cb93-53" aria-hidden="true" tabindex="-1"></a>           reaction.viral, reaction.same, catalytic_activity.same,</span>
<span id="cb93-54"><a href="#cb93-54" aria-hidden="true" tabindex="-1"></a>           ec_number.human, ec_number.viral,</span>
<span id="cb93-55"><a href="#cb93-55" aria-hidden="true" tabindex="-1"></a>           enzyme_number.same, external_gene_name)</span>
<span id="cb93-56"><a href="#cb93-56" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb93-57"><a href="#cb93-57" aria-hidden="true" tabindex="-1"></a>   <span class="cf">if</span>(mode <span class="sc">==</span> <span class="st">"tight_elm"</span>){</span>
<span id="cb93-58"><a href="#cb93-58" aria-hidden="true" tabindex="-1"></a>    active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb93-59"><a href="#cb93-59" aria-hidden="true" tabindex="-1"></a>      <span class="fu">anti_join</span>(elm_db_expand, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"query_seqname"</span>, <span class="st">"query_start"</span>))</span>
<span id="cb93-60"><a href="#cb93-60" aria-hidden="true" tabindex="-1"></a>  } </span>
<span id="cb93-61"><a href="#cb93-61" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb93-62"><a href="#cb93-62" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>reaction.same[<span class="fu">is.na</span>(active_data<span class="sc">$</span>reaction.same)] <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb93-63"><a href="#cb93-63" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>catalytic_activity.same[<span class="fu">is.na</span>(active_data<span class="sc">$</span>catalytic_activity.same)] <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb93-64"><a href="#cb93-64" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb93-65"><a href="#cb93-65" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(active_data)</span>
<span id="cb93-66"><a href="#cb93-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-67"><a href="#cb93-67" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb93-68"><a href="#cb93-68" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(reaction.same <span class="sc">!=</span> <span class="cn">TRUE</span> <span class="sc">&amp;</span> catalytic_activity.same <span class="sc">!=</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb93-69"><a href="#cb93-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(enzyme_number.same <span class="sc">!=</span> <span class="cn">TRUE</span>)  </span>
<span id="cb93-70"><a href="#cb93-70" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb93-71"><a href="#cb93-71" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(active_data<span class="sc">$</span>mismatches))</span>
<span id="cb93-72"><a href="#cb93-72" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(out) <span class="sc">==</span> <span class="dv">0</span>){out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="at">V2 =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))}</span>
<span id="cb93-73"><a href="#cb93-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(out) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>, <span class="fu">gsub</span>(<span class="st">"_8mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb93-74"><a href="#cb93-74" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb93-75"><a href="#cb93-75" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb93-76"><a href="#cb93-76" aria-hidden="true" tabindex="-1"></a>  out_protein <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb93-77"><a href="#cb93-77" aria-hidden="true" tabindex="-1"></a>     <span class="fu">distinct</span>(query, query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb93-78"><a href="#cb93-78" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb93-79"><a href="#cb93-79" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb93-80"><a href="#cb93-80" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb93-81"><a href="#cb93-81" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb93-82"><a href="#cb93-82" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb93-83"><a href="#cb93-83" aria-hidden="true" tabindex="-1"></a>   out_protein_min <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb93-84"><a href="#cb93-84" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb93-85"><a href="#cb93-85" aria-hidden="true" tabindex="-1"></a>     <span class="fu">distinct</span>(query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb93-86"><a href="#cb93-86" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, query_start) <span class="sc">%&gt;%</span></span>
<span id="cb93-87"><a href="#cb93-87" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_min</span>(mismatches, <span class="at">n=</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb93-88"><a href="#cb93-88" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() </span>
<span id="cb93-89"><a href="#cb93-89" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb93-90"><a href="#cb93-90" aria-hidden="true" tabindex="-1"></a>   out <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(out_protein_min<span class="sc">$</span>mismatches))</span>
<span id="cb93-91"><a href="#cb93-91" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(out) <span class="sc">==</span> <span class="dv">0</span>){out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="at">V2 =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))}</span>
<span id="cb93-92"><a href="#cb93-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(out) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>, <span class="fu">gsub</span>(<span class="st">"_8mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb93-93"><a href="#cb93-93" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb93-94"><a href="#cb93-94" aria-hidden="true" tabindex="-1"></a>     out_protein_min <span class="ot">&lt;-</span> out_protein_min <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(query_seqname, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb93-95"><a href="#cb93-95" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb93-96"><a href="#cb93-96" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb93-97"><a href="#cb93-97" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb93-98"><a href="#cb93-98" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb93-99"><a href="#cb93-99" aria-hidden="true" tabindex="-1"></a>     out_protein_count <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb93-100"><a href="#cb93-100" aria-hidden="true" tabindex="-1"></a>       <span class="fu">distinct</span>(query, query_seqname, query_start, mismatches, external_gene_name) <span class="sc">%&gt;%</span></span>
<span id="cb93-101"><a href="#cb93-101" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb93-102"><a href="#cb93-102" aria-hidden="true" tabindex="-1"></a>       <span class="fu">distinct</span>(query, query_seqname, query_start, external_gene_name, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb93-103"><a href="#cb93-103" aria-hidden="true" tabindex="-1"></a>       <span class="fu">filter</span>(<span class="sc">!</span>external_gene_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="cn">NA</span>, <span class="st">""</span>, <span class="st">" "</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb93-104"><a href="#cb93-104" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb93-105"><a href="#cb93-105" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb93-106"><a href="#cb93-106" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb93-107"><a href="#cb93-107" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb93-108"><a href="#cb93-108" aria-hidden="true" tabindex="-1"></a>  <span class="do">############## Count Proteome Length ##############</span></span>
<span id="cb93-109"><a href="#cb93-109" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>,  <span class="fu">gsub</span>(<span class="st">"../../8mer_annotated/"</span>, <span class="st">""</span>, <span class="fu">gsub</span>(<span class="st">"_8mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file)), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb93-110"><a href="#cb93-110" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb93-111"><a href="#cb93-111" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb93-112"><a href="#cb93-112" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb93-113"><a href="#cb93-113" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb93-114"><a href="#cb93-114" aria-hidden="true" tabindex="-1"></a>       PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb93-115"><a href="#cb93-115" aria-hidden="true" tabindex="-1"></a>      PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb93-116"><a href="#cb93-116" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb93-117"><a href="#cb93-117" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb93-118"><a href="#cb93-118" aria-hidden="true" tabindex="-1"></a>  out_final <span class="ot">&lt;-</span> <span class="fu">merge</span>(out_final, out, <span class="at">by =</span> <span class="st">"mismatches"</span>, <span class="at">all.x =</span> T)</span>
<span id="cb93-119"><a href="#cb93-119" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb93-120"><a href="#cb93-120" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb93-121"><a href="#cb93-121" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb93-122"><a href="#cb93-122" aria-hidden="true" tabindex="-1"></a>   out_protein_tmp <span class="ot">&lt;-</span> out_protein <span class="sc">%&gt;%</span></span>
<span id="cb93-123"><a href="#cb93-123" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb93-124"><a href="#cb93-124" aria-hidden="true" tabindex="-1"></a>  out_protein_tmp_min <span class="ot">&lt;-</span> out_protein_min <span class="sc">%&gt;%</span></span>
<span id="cb93-125"><a href="#cb93-125" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb93-126"><a href="#cb93-126" aria-hidden="true" tabindex="-1"></a>  out_protein_tmp_count <span class="ot">&lt;-</span> out_protein_count <span class="sc">%&gt;%</span></span>
<span id="cb93-127"><a href="#cb93-127" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb93-128"><a href="#cb93-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-129"><a href="#cb93-129" aria-hidden="true" tabindex="-1"></a>  out_protein_final <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final, out_protein_tmp)</span>
<span id="cb93-130"><a href="#cb93-130" aria-hidden="true" tabindex="-1"></a>  out_protein_final_min <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final_min, out_protein_tmp_min)</span>
<span id="cb93-131"><a href="#cb93-131" aria-hidden="true" tabindex="-1"></a>  out_protein_final_count <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final_count, out_protein_tmp_count)</span>
<span id="cb93-132"><a href="#cb93-132" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb93-133"><a href="#cb93-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-134"><a href="#cb93-134" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb93-135"><a href="#cb93-135" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(out_final, "../../Key_Data/8mer_proteome_everything_elmfilter_regex.csv")</span></span>
<span id="cb93-136"><a href="#cb93-136" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(out_protein_final, "../../Key_Data/8mer_proteome_protein_everything_elmfilter_regex.csv")</span></span>
<span id="cb93-137"><a href="#cb93-137" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(out_protein_final_min, "../../Key_Data/8mer_proteome_protein_everything_min_elmfilter_regex.csv")</span></span>
<span id="cb93-138"><a href="#cb93-138" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(out_protein_final_count, "../../Key_Data/8mer_proteome_protein_everything_count_elmfilter_regex.csv")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="mer-data-annotation-1" class="level2">
<h2 class="anchored" data-anchor-id="mer-data-annotation-1">12mer Data Annotation</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="do">## This will only run for files that haven't ran already</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../../epitope_alignment/Results/Complete_Final_Set/"</span>)</span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".tsv"</span>, files_to_read)]</span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>files_completed <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../../../Analysis/12mer_annotated/"</span>)</span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a><span class="co">#files_to_read &lt;- files_to_read[!gsub(".tsv", "", files_to_read) %in% gsub(".csv.gz", "", files_completed)]</span></span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID, <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-10"><a href="#cb94-10" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">order</span>(<span class="fu">file.info</span>(files_to_read)<span class="sc">$</span>size)]</span>
<span id="cb94-11"><a href="#cb94-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-12"><a href="#cb94-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb94-13"><a href="#cb94-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb94-14"><a href="#cb94-14" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">nThread =</span> <span class="dv">12</span>)</span>
<span id="cb94-15"><a href="#cb94-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb94-16"><a href="#cb94-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Skipping "</span>, active_file, <span class="st">" due to lack of data."</span>))</span>
<span id="cb94-17"><a href="#cb94-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb94-18"><a href="#cb94-18" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb94-19"><a href="#cb94-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-20"><a href="#cb94-20" aria-hidden="true" tabindex="-1"></a> active_data[, uniprot <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(target_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb94-21"><a href="#cb94-21" aria-hidden="true" tabindex="-1"></a> active_data[, viral_id <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(query_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb94-22"><a href="#cb94-22" aria-hidden="true" tabindex="-1"></a> active_data[, mismatches <span class="sc">:</span><span class="er">=</span> <span class="fu">mismatch_calling_fast</span>(query, target), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb94-23"><a href="#cb94-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-24"><a href="#cb94-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Data Merging "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb94-25"><a href="#cb94-25" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb94-26"><a href="#cb94-26" aria-hidden="true" tabindex="-1"></a>     <span class="fu">lazy_dt</span>() <span class="sc">%&gt;%</span></span>
<span id="cb94-27"><a href="#cb94-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(mart_retrieved, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="st">"uniprot_gn_id"</span>),</span>
<span id="cb94-28"><a href="#cb94-28" aria-hidden="true" tabindex="-1"></a>              <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb94-29"><a href="#cb94-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(tissue_organ_map_condensed, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span> <span class="ot">=</span> <span class="st">"Gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb94-30"><a href="#cb94-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(human_protein_metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span><span class="st">"entry.human"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb94-31"><a href="#cb94-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(viral_protein_metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"viral_id"</span> <span class="ot">=</span> <span class="st">"entry.viral"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb94-32"><a href="#cb94-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">catalytic_activity.same =</span> catalytic_activity.human <span class="sc">==</span> catalytic_activity.viral) <span class="sc">%&gt;%</span></span>
<span id="cb94-33"><a href="#cb94-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.human =</span> <span class="fu">gsub</span>(<span class="st">";.*"</span>, <span class="st">""</span>, catalytic_activity.human)) <span class="sc">%&gt;%</span></span>
<span id="cb94-34"><a href="#cb94-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.viral =</span> <span class="fu">gsub</span>(<span class="st">";.*"</span>, <span class="st">""</span>, catalytic_activity.viral)) <span class="sc">%&gt;%</span></span>
<span id="cb94-35"><a href="#cb94-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.same =</span> reaction.human <span class="sc">==</span> reaction.viral) <span class="sc">%&gt;%</span></span>
<span id="cb94-36"><a href="#cb94-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.same =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(reaction.same), <span class="cn">FALSE</span>, reaction.same)) <span class="sc">%&gt;%</span></span>
<span id="cb94-37"><a href="#cb94-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">catalytic_activity.same =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(catalytic_activity.same), <span class="cn">FALSE</span>, catalytic_activity.same)) <span class="sc">%&gt;%</span></span>
<span id="cb94-38"><a href="#cb94-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">enzyme_number.same =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb94-39"><a href="#cb94-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.table</span>()</span>
<span id="cb94-40"><a href="#cb94-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-41"><a href="#cb94-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"EC Screening "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb94-42"><a href="#cb94-42" aria-hidden="true" tabindex="-1"></a>  active_data[, enzyme_number.same <span class="sc">:</span><span class="er">=</span> <span class="fu">fast_ec_screen</span>(ec_number.human, ec_number.viral), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb94-43"><a href="#cb94-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-44"><a href="#cb94-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Final Cleaning "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb94-45"><a href="#cb94-45" aria-hidden="true" tabindex="-1"></a>  active_data_subset <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span> </span>
<span id="cb94-46"><a href="#cb94-46" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">human_variants =</span> <span class="fu">sapply</span>(human_variants, toString)) <span class="sc">%&gt;%</span></span>
<span id="cb94-47"><a href="#cb94-47" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(query, query_seqname, query_start, target_seqname, target_start,</span>
<span id="cb94-48"><a href="#cb94-48" aria-hidden="true" tabindex="-1"></a>           target, mismatches, uniprot, viral_id, ensembl_gene_id,</span>
<span id="cb94-49"><a href="#cb94-49" aria-hidden="true" tabindex="-1"></a>           entrezgene_id, external_gene_name, uniprot_gn_symbol, chromosome_name,</span>
<span id="cb94-50"><a href="#cb94-50" aria-hidden="true" tabindex="-1"></a>           reviewed.human, entry_name.human, protein_names.human, length.human,</span>
<span id="cb94-51"><a href="#cb94-51" aria-hidden="true" tabindex="-1"></a>           active_site.human, binding_site.human, dna_binding.human,</span>
<span id="cb94-52"><a href="#cb94-52" aria-hidden="true" tabindex="-1"></a>           ec_number.human,  reviewed.viral, entry_name.viral,</span>
<span id="cb94-53"><a href="#cb94-53" aria-hidden="true" tabindex="-1"></a>           protein_names.viral, length.viral,</span>
<span id="cb94-54"><a href="#cb94-54" aria-hidden="true" tabindex="-1"></a>           active_site.viral, binding_site.viral, dna_binding.viral,</span>
<span id="cb94-55"><a href="#cb94-55" aria-hidden="true" tabindex="-1"></a>           ec_number.viral, catalytic_activity.same, reaction.human,</span>
<span id="cb94-56"><a href="#cb94-56" aria-hidden="true" tabindex="-1"></a>           reaction.viral, reaction.same,</span>
<span id="cb94-57"><a href="#cb94-57" aria-hidden="true" tabindex="-1"></a>           enzyme_number.same, human_variants) <span class="sc">%&gt;%</span></span>
<span id="cb94-58"><a href="#cb94-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb94-59"><a href="#cb94-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Sys.time</span>()</span>
<span id="cb94-60"><a href="#cb94-60" aria-hidden="true" tabindex="-1"></a>  <span class="do">##fwrite(active_data_subset, paste0("../../../Analysis/12mer_annotated/", gsub(".tsv", ".csv.gz", active_file)), row.names = FALSE, compress = "gzip", nThread = 12)</span></span>
<span id="cb94-61"><a href="#cb94-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(active_data, active_data_subset)</span>
<span id="cb94-62"><a href="#cb94-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gc</span>()</span>
<span id="cb94-63"><a href="#cb94-63" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="mer-data-load-and-processing-1" class="level2">
<h2 class="anchored" data-anchor-id="mer-data-load-and-processing-1">12mer Data Load and Processing</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="do">### This only needs to be run once and will take a while, the script when not full render unless you already have the output in the expected path or if you set message: true</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(seqinr)</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringi)</span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(svMisc)</span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a>elm_instance_df <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"../../Key_Data/elm_labeled_viral_sequences_v2.csv.gz"</span>)</span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../12mer_annotated/"</span>)</span>
<span id="cb95-7"><a href="#cb95-7" aria-hidden="true" tabindex="-1"></a>mode <span class="ot">=</span> <span class="st">"tight_elm"</span></span>
<span id="cb95-8"><a href="#cb95-8" aria-hidden="true" tabindex="-1"></a>kmer_length <span class="ot">=</span> <span class="dv">12</span></span>
<span id="cb95-9"><a href="#cb95-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-10"><a href="#cb95-10" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb95-11"><a href="#cb95-11" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb95-12"><a href="#cb95-12" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".csv.gz"</span>, files_to_read)]</span>
<span id="cb95-13"><a href="#cb95-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-14"><a href="#cb95-14" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb95-15"><a href="#cb95-15" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"Number of Proteins"</span>, <span class="st">"Proteome Length"</span>))</span>
<span id="cb95-16"><a href="#cb95-16" aria-hidden="true" tabindex="-1"></a>out_protein_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb95-17"><a href="#cb95-17" aria-hidden="true" tabindex="-1"></a>out_protein_final_min <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb95-18"><a href="#cb95-18" aria-hidden="true" tabindex="-1"></a>out_protein_final_count <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb95-19"><a href="#cb95-19" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb95-20"><a href="#cb95-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" Pathogen number:"</span>,<span class="fu">match</span>(active_file, files_to_read), <span class="st">"/"</span>, <span class="fu">length</span>(files_to_read)))</span>
<span id="cb95-21"><a href="#cb95-21" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">nThread =</span> <span class="dv">18</span>) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb95-22"><a href="#cb95-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-23"><a href="#cb95-23" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb95-24"><a href="#cb95-24" aria-hidden="true" tabindex="-1"></a>    out_final[,<span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file)] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="cn">NA</span>,<span class="cn">NA</span>)</span>
<span id="cb95-25"><a href="#cb95-25" aria-hidden="true" tabindex="-1"></a>      data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb95-26"><a href="#cb95-26" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb95-27"><a href="#cb95-27" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb95-28"><a href="#cb95-28" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb95-29"><a href="#cb95-29" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb95-30"><a href="#cb95-30" aria-hidden="true" tabindex="-1"></a>      PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb95-31"><a href="#cb95-31" aria-hidden="true" tabindex="-1"></a>      PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb95-32"><a href="#cb95-32" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb95-33"><a href="#cb95-33" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb95-34"><a href="#cb95-34" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb95-35"><a href="#cb95-35" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb95-36"><a href="#cb95-36" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb95-37"><a href="#cb95-37" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb95-38"><a href="#cb95-38" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb95-39"><a href="#cb95-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb95-40"><a href="#cb95-40" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data  <span class="sc">%&gt;%</span></span>
<span id="cb95-41"><a href="#cb95-41" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Remove duplicated pathogen proteins identified in </span></span>
<span id="cb95-42"><a href="#cb95-42" aria-hidden="true" tabindex="-1"></a>    <span class="do">## ../../Scripts/Duplicate_Protein_Screen.R</span></span>
<span id="cb95-43"><a href="#cb95-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb95-44"><a href="#cb95-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb95-45"><a href="#cb95-45" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(query, query_seqname, query_start, mismatches, reaction.human,</span>
<span id="cb95-46"><a href="#cb95-46" aria-hidden="true" tabindex="-1"></a>           reaction.viral, reaction.same, catalytic_activity.same,</span>
<span id="cb95-47"><a href="#cb95-47" aria-hidden="true" tabindex="-1"></a>           ec_number.human, ec_number.viral,</span>
<span id="cb95-48"><a href="#cb95-48" aria-hidden="true" tabindex="-1"></a>           enzyme_number.same, external_gene_name)</span>
<span id="cb95-49"><a href="#cb95-49" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb95-50"><a href="#cb95-50" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>reaction.same[<span class="fu">is.na</span>(active_data<span class="sc">$</span>reaction.same)] <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb95-51"><a href="#cb95-51" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>catalytic_activity.same[<span class="fu">is.na</span>(active_data<span class="sc">$</span>catalytic_activity.same)] <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb95-52"><a href="#cb95-52" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb95-53"><a href="#cb95-53" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(active_data)</span>
<span id="cb95-54"><a href="#cb95-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-55"><a href="#cb95-55" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb95-56"><a href="#cb95-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(reaction.same <span class="sc">!=</span> <span class="cn">TRUE</span> <span class="sc">&amp;</span> catalytic_activity.same <span class="sc">!=</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb95-57"><a href="#cb95-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(enzyme_number.same <span class="sc">!=</span> <span class="cn">TRUE</span>)  </span>
<span id="cb95-58"><a href="#cb95-58" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb95-59"><a href="#cb95-59" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(active_data<span class="sc">$</span>mismatches))</span>
<span id="cb95-60"><a href="#cb95-60" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(out) <span class="sc">==</span> <span class="dv">0</span>){out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="at">V2 =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))}</span>
<span id="cb95-61"><a href="#cb95-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(out) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb95-62"><a href="#cb95-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-63"><a href="#cb95-63" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(mode <span class="sc">==</span> <span class="st">"tight_elm"</span>){</span>
<span id="cb95-64"><a href="#cb95-64" aria-hidden="true" tabindex="-1"></a>    active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb95-65"><a href="#cb95-65" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">query_end =</span> query_start <span class="sc">+</span> (kmer_length<span class="dv">-1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb95-66"><a href="#cb95-66" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(elm_instance_df <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(viral_protein, start, end, Regex) <span class="sc">%&gt;%</span></span>
<span id="cb95-67"><a href="#cb95-67" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">mutate</span>(<span class="at">viral_protein =</span> <span class="fu">gsub</span>(<span class="st">"&gt;"</span>, <span class="st">""</span>, viral_protein)),</span>
<span id="cb95-68"><a href="#cb95-68" aria-hidden="true" tabindex="-1"></a>                <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"query_seqname"</span> <span class="ot">=</span> <span class="st">"viral_protein"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb95-69"><a href="#cb95-69" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">in_elm =</span> ((query_start <span class="sc">&gt;=</span> start <span class="sc">&amp;</span> query_start <span class="sc">&lt;=</span> end) <span class="sc">|</span> (query_end <span class="sc">&gt;=</span> start <span class="sc">&amp;</span> query_end <span class="sc">&lt;=</span> end))) <span class="sc">%&gt;%</span></span>
<span id="cb95-70"><a href="#cb95-70" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(query, query_seqname, query_start, mismatches, external_gene_name) <span class="sc">%&gt;%</span></span>
<span id="cb95-71"><a href="#cb95-71" aria-hidden="true" tabindex="-1"></a>      <span class="fu">slice_max</span>(in_elm) <span class="sc">%&gt;%</span></span>
<span id="cb95-72"><a href="#cb95-72" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(query, query_seqname, query_start, mismatches, external_gene_name, in_elm) <span class="sc">%&gt;%</span></span>
<span id="cb95-73"><a href="#cb95-73" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(<span class="sc">!</span>in_elm)</span>
<span id="cb95-74"><a href="#cb95-74" aria-hidden="true" tabindex="-1"></a>  } </span>
<span id="cb95-75"><a href="#cb95-75" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb95-76"><a href="#cb95-76" aria-hidden="true" tabindex="-1"></a>  out_protein <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb95-77"><a href="#cb95-77" aria-hidden="true" tabindex="-1"></a>     <span class="fu">distinct</span>(query, query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb95-78"><a href="#cb95-78" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb95-79"><a href="#cb95-79" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb95-80"><a href="#cb95-80" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb95-81"><a href="#cb95-81" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb95-82"><a href="#cb95-82" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb95-83"><a href="#cb95-83" aria-hidden="true" tabindex="-1"></a>   out_protein_min <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb95-84"><a href="#cb95-84" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb95-85"><a href="#cb95-85" aria-hidden="true" tabindex="-1"></a>     <span class="fu">distinct</span>(query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb95-86"><a href="#cb95-86" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, query_start) <span class="sc">%&gt;%</span></span>
<span id="cb95-87"><a href="#cb95-87" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_min</span>(mismatches, <span class="at">n=</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb95-88"><a href="#cb95-88" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() </span>
<span id="cb95-89"><a href="#cb95-89" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb95-90"><a href="#cb95-90" aria-hidden="true" tabindex="-1"></a>   out <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(out_protein_min<span class="sc">$</span>mismatches))</span>
<span id="cb95-91"><a href="#cb95-91" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(out) <span class="sc">==</span> <span class="dv">0</span>){out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="at">V2 =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))}</span>
<span id="cb95-92"><a href="#cb95-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(out) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb95-93"><a href="#cb95-93" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb95-94"><a href="#cb95-94" aria-hidden="true" tabindex="-1"></a>     out_protein_min <span class="ot">&lt;-</span> out_protein_min <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(query_seqname, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb95-95"><a href="#cb95-95" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb95-96"><a href="#cb95-96" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb95-97"><a href="#cb95-97" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb95-98"><a href="#cb95-98" aria-hidden="true" tabindex="-1"></a>     out_protein_count <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb95-99"><a href="#cb95-99" aria-hidden="true" tabindex="-1"></a>       <span class="fu">distinct</span>(query, query_seqname, query_start, mismatches, external_gene_name) <span class="sc">%&gt;%</span></span>
<span id="cb95-100"><a href="#cb95-100" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb95-101"><a href="#cb95-101" aria-hidden="true" tabindex="-1"></a>       <span class="fu">distinct</span>(query, query_seqname, query_start, external_gene_name, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb95-102"><a href="#cb95-102" aria-hidden="true" tabindex="-1"></a>       <span class="fu">filter</span>(<span class="sc">!</span>external_gene_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="cn">NA</span>, <span class="st">""</span>, <span class="st">" "</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb95-103"><a href="#cb95-103" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb95-104"><a href="#cb95-104" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb95-105"><a href="#cb95-105" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb95-106"><a href="#cb95-106" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb95-107"><a href="#cb95-107" aria-hidden="true" tabindex="-1"></a>  <span class="do">############## Count Proteome Length ##############</span></span>
<span id="cb95-108"><a href="#cb95-108" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>, <span class="fu">gsub</span>(<span class="st">"_12mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb95-109"><a href="#cb95-109" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb95-110"><a href="#cb95-110" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb95-111"><a href="#cb95-111" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb95-112"><a href="#cb95-112" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb95-113"><a href="#cb95-113" aria-hidden="true" tabindex="-1"></a>       PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb95-114"><a href="#cb95-114" aria-hidden="true" tabindex="-1"></a>      PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb95-115"><a href="#cb95-115" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb95-116"><a href="#cb95-116" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb95-117"><a href="#cb95-117" aria-hidden="true" tabindex="-1"></a>  out_final <span class="ot">&lt;-</span> <span class="fu">merge</span>(out_final, out, <span class="at">by =</span> <span class="st">"mismatches"</span>, <span class="at">all.x =</span> T)</span>
<span id="cb95-118"><a href="#cb95-118" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb95-119"><a href="#cb95-119" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb95-120"><a href="#cb95-120" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb95-121"><a href="#cb95-121" aria-hidden="true" tabindex="-1"></a>   out_protein_tmp <span class="ot">&lt;-</span> out_protein <span class="sc">%&gt;%</span></span>
<span id="cb95-122"><a href="#cb95-122" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb95-123"><a href="#cb95-123" aria-hidden="true" tabindex="-1"></a>  out_protein_tmp_min <span class="ot">&lt;-</span> out_protein_min <span class="sc">%&gt;%</span></span>
<span id="cb95-124"><a href="#cb95-124" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb95-125"><a href="#cb95-125" aria-hidden="true" tabindex="-1"></a>  out_protein_tmp_count <span class="ot">&lt;-</span> out_protein_count <span class="sc">%&gt;%</span></span>
<span id="cb95-126"><a href="#cb95-126" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb95-127"><a href="#cb95-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-128"><a href="#cb95-128" aria-hidden="true" tabindex="-1"></a>  out_protein_final <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final, out_protein_tmp)</span>
<span id="cb95-129"><a href="#cb95-129" aria-hidden="true" tabindex="-1"></a>  out_protein_final_min <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final_min, out_protein_tmp_min)</span>
<span id="cb95-130"><a href="#cb95-130" aria-hidden="true" tabindex="-1"></a>  out_protein_final_count <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final_count, out_protein_tmp_count)</span>
<span id="cb95-131"><a href="#cb95-131" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb95-132"><a href="#cb95-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-133"><a href="#cb95-133" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb95-134"><a href="#cb95-134" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_final, <span class="st">"../Key_Data/12mer_proteome_everything_elmfilter_regex.csv"</span>)</span>
<span id="cb95-135"><a href="#cb95-135" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_protein_final, <span class="st">"../Key_Data/12mer_proteome_protein_everything_elmfilter_regex.csv"</span>)</span>
<span id="cb95-136"><a href="#cb95-136" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_protein_final_min, <span class="st">"../Key_Data/12mer_proteome_protein_everything_min_elmfilter_regex.csv"</span>)</span>
<span id="cb95-137"><a href="#cb95-137" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_protein_final_count, <span class="st">"../Key_Data/12mer_proteome_protein_everything_count_elmfilter_regex.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="mer-data-annotation-2" class="level2">
<h2 class="anchored" data-anchor-id="mer-data-annotation-2">18mer Data Annotation</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="do">## This will only run for files that haven't ran already</span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../../epitope_alignment/Results/Complete_Final_Set_18mer_3mis/"</span>)</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".tsv"</span>, files_to_read)]</span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a>files_completed <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../../../Analysis/18mer_annotated/"</span>)</span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="sc">!</span><span class="fu">gsub</span>(<span class="st">".tsv"</span>, <span class="st">""</span>, files_to_read) <span class="sc">%in%</span> <span class="fu">gsub</span>(<span class="st">".csv.gz"</span>, <span class="st">""</span>, files_completed)]</span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID, <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">order</span>(<span class="fu">file.info</span>(files_to_read)<span class="sc">$</span>size)]</span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">nThread =</span> <span class="dv">12</span>)</span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb96-16"><a href="#cb96-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Skipping "</span>, active_file, <span class="st">" due to lack of data."</span>))</span>
<span id="cb96-17"><a href="#cb96-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fwrite</span>(active_data, <span class="fu">paste0</span>(<span class="st">"../../../Analysis/18mer_annotated/"</span>, <span class="fu">gsub</span>(<span class="st">".tsv"</span>, <span class="st">".csv.gz"</span>, active_file)), <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">compress =</span> <span class="st">"gzip"</span>, <span class="at">nThread =</span> <span class="dv">12</span>)</span>
<span id="cb96-18"><a href="#cb96-18" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb96-19"><a href="#cb96-19" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb96-20"><a href="#cb96-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-21"><a href="#cb96-21" aria-hidden="true" tabindex="-1"></a> active_data[, uniprot <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(target_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb96-22"><a href="#cb96-22" aria-hidden="true" tabindex="-1"></a> active_data[, viral_id <span class="sc">:</span><span class="er">=</span> <span class="fu">id_split_fast</span>(query_seqname), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb96-23"><a href="#cb96-23" aria-hidden="true" tabindex="-1"></a> active_data[, mismatches <span class="sc">:</span><span class="er">=</span> <span class="fu">mismatch_calling_fast</span>(query, target), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb96-24"><a href="#cb96-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-25"><a href="#cb96-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Data Merging "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb96-26"><a href="#cb96-26" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb96-27"><a href="#cb96-27" aria-hidden="true" tabindex="-1"></a>     <span class="fu">lazy_dt</span>() <span class="sc">%&gt;%</span></span>
<span id="cb96-28"><a href="#cb96-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(mart_retrieved, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span> <span class="st">"uniprot_gn_id"</span>),</span>
<span id="cb96-29"><a href="#cb96-29" aria-hidden="true" tabindex="-1"></a>              <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb96-30"><a href="#cb96-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(tissue_organ_map_condensed, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span> <span class="ot">=</span> <span class="st">"Gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb96-31"><a href="#cb96-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(human_protein_metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"uniprot"</span> <span class="ot">=</span><span class="st">"entry.human"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb96-32"><a href="#cb96-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(viral_protein_metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"viral_id"</span> <span class="ot">=</span> <span class="st">"entry.viral"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb96-33"><a href="#cb96-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">catalytic_activity.same =</span> catalytic_activity.human <span class="sc">==</span> catalytic_activity.viral) <span class="sc">%&gt;%</span></span>
<span id="cb96-34"><a href="#cb96-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.human =</span> <span class="fu">gsub</span>(<span class="st">";.*"</span>, <span class="st">""</span>, catalytic_activity.human)) <span class="sc">%&gt;%</span></span>
<span id="cb96-35"><a href="#cb96-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.viral =</span> <span class="fu">gsub</span>(<span class="st">";.*"</span>, <span class="st">""</span>, catalytic_activity.viral)) <span class="sc">%&gt;%</span></span>
<span id="cb96-36"><a href="#cb96-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.same =</span> reaction.human <span class="sc">==</span> reaction.viral) <span class="sc">%&gt;%</span></span>
<span id="cb96-37"><a href="#cb96-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reaction.same =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(reaction.same), <span class="cn">FALSE</span>, reaction.same)) <span class="sc">%&gt;%</span></span>
<span id="cb96-38"><a href="#cb96-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">catalytic_activity.same =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(catalytic_activity.same), <span class="cn">FALSE</span>, catalytic_activity.same)) <span class="sc">%&gt;%</span></span>
<span id="cb96-39"><a href="#cb96-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">enzyme_number.same =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb96-40"><a href="#cb96-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.table</span>()</span>
<span id="cb96-41"><a href="#cb96-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-42"><a href="#cb96-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"EC Screening "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb96-43"><a href="#cb96-43" aria-hidden="true" tabindex="-1"></a>  active_data[, enzyme_number.same <span class="sc">:</span><span class="er">=</span> <span class="fu">fast_ec_screen</span>(ec_number.human, ec_number.viral), by <span class="ot">=</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(active_data))]</span>
<span id="cb96-44"><a href="#cb96-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-45"><a href="#cb96-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Final Cleaning "</span>, <span class="fu">Sys.time</span>()))</span>
<span id="cb96-46"><a href="#cb96-46" aria-hidden="true" tabindex="-1"></a>  active_data_subset <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span> </span>
<span id="cb96-47"><a href="#cb96-47" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">human_variants =</span> <span class="fu">sapply</span>(human_variants, toString)) <span class="sc">%&gt;%</span></span>
<span id="cb96-48"><a href="#cb96-48" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(query, query_seqname, query_start, target_seqname, target_start,</span>
<span id="cb96-49"><a href="#cb96-49" aria-hidden="true" tabindex="-1"></a>           target, mismatches, uniprot, viral_id, ensembl_gene_id,</span>
<span id="cb96-50"><a href="#cb96-50" aria-hidden="true" tabindex="-1"></a>           entrezgene_id, external_gene_name, uniprot_gn_symbol, chromosome_name,</span>
<span id="cb96-51"><a href="#cb96-51" aria-hidden="true" tabindex="-1"></a>           reviewed.human, entry_name.human, protein_names.human, length.human,</span>
<span id="cb96-52"><a href="#cb96-52" aria-hidden="true" tabindex="-1"></a>           active_site.human, binding_site.human, dna_binding.human,</span>
<span id="cb96-53"><a href="#cb96-53" aria-hidden="true" tabindex="-1"></a>           ec_number.human,  reviewed.viral, entry_name.viral,</span>
<span id="cb96-54"><a href="#cb96-54" aria-hidden="true" tabindex="-1"></a>           protein_names.viral, length.viral,</span>
<span id="cb96-55"><a href="#cb96-55" aria-hidden="true" tabindex="-1"></a>           active_site.viral, binding_site.viral, dna_binding.viral,</span>
<span id="cb96-56"><a href="#cb96-56" aria-hidden="true" tabindex="-1"></a>           ec_number.viral, catalytic_activity.same, reaction.human,</span>
<span id="cb96-57"><a href="#cb96-57" aria-hidden="true" tabindex="-1"></a>           reaction.viral, reaction.same,</span>
<span id="cb96-58"><a href="#cb96-58" aria-hidden="true" tabindex="-1"></a>           enzyme_number.same) <span class="sc">%&gt;%</span></span>
<span id="cb96-59"><a href="#cb96-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb96-60"><a href="#cb96-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Sys.time</span>()</span>
<span id="cb96-61"><a href="#cb96-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fwrite</span>(active_data_subset, <span class="fu">paste0</span>(<span class="st">"../../../Analysis/18mer_annotated/"</span>, <span class="fu">gsub</span>(<span class="st">".tsv"</span>, <span class="st">".csv.gz"</span>, active_file)), <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">compress =</span> <span class="st">"gzip"</span>, <span class="at">nThread =</span> <span class="dv">12</span>)</span>
<span id="cb96-62"><a href="#cb96-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(active_data, active_data_subset)</span>
<span id="cb96-63"><a href="#cb96-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gc</span>()</span>
<span id="cb96-64"><a href="#cb96-64" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="mer-data-load-and-processing-2" class="level2">
<h2 class="anchored" data-anchor-id="mer-data-load-and-processing-2">18mer Data Load and Processing</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="do">### This only needs to be run once and will take a while, change eval to true if you want to run</span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(seqinr)</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringi)</span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(svMisc)</span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a>elm_instance_df <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"../../Key_Data/elm_labeled_viral_sequences_v2.csv.gz"</span>)</span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"../../18mer_annotated/"</span>)</span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a>mode <span class="ot">=</span> <span class="st">"tight_elm"</span></span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a>kmer_length <span class="ot">=</span> <span class="dv">18</span></span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-10"><a href="#cb97-10" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb97-11"><a href="#cb97-11" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> <span class="fu">list.files</span>()</span>
<span id="cb97-12"><a href="#cb97-12" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="st">".csv.gz"</span>, files_to_read)]</span>
<span id="cb97-13"><a href="#cb97-13" aria-hidden="true" tabindex="-1"></a>files_to_read <span class="ot">&lt;-</span> files_to_read[<span class="fu">grepl</span>(<span class="fu">paste</span>(metadata<span class="sc">$</span>Proteome.ID[metadata<span class="sc">$</span>superkingdom <span class="sc">==</span> <span class="st">"Viruses"</span>], <span class="at">collapse =</span> <span class="st">"|"</span>), files_to_read)]</span>
<span id="cb97-14"><a href="#cb97-14" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"Number of Proteins"</span>, <span class="st">"Proteome Length"</span>))</span>
<span id="cb97-15"><a href="#cb97-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-16"><a href="#cb97-16" aria-hidden="true" tabindex="-1"></a>out_protein_final <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb97-17"><a href="#cb97-17" aria-hidden="true" tabindex="-1"></a>out_protein_final_min <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb97-18"><a href="#cb97-18" aria-hidden="true" tabindex="-1"></a>out_protein_final_count <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb97-19"><a href="#cb97-19" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(active_file <span class="cf">in</span> files_to_read){</span>
<span id="cb97-20"><a href="#cb97-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Starting "</span>, active_file, <span class="st">" Pathogen number:"</span>,<span class="fu">match</span>(active_file, files_to_read), <span class="st">"/"</span>, <span class="fu">length</span>(files_to_read)))</span>
<span id="cb97-21"><a href="#cb97-21" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(active_file, <span class="at">nThread =</span> <span class="dv">18</span>) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb97-22"><a href="#cb97-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-23"><a href="#cb97-23" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(active_data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb97-24"><a href="#cb97-24" aria-hidden="true" tabindex="-1"></a>    out_final[,<span class="fu">gsub</span>(<span class="st">"_18mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file)] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="cn">NA</span>,<span class="cn">NA</span>)</span>
<span id="cb97-25"><a href="#cb97-25" aria-hidden="true" tabindex="-1"></a>      data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>, <span class="fu">gsub</span>(<span class="st">"_18mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb97-26"><a href="#cb97-26" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb97-27"><a href="#cb97-27" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb97-28"><a href="#cb97-28" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb97-29"><a href="#cb97-29" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb97-30"><a href="#cb97-30" aria-hidden="true" tabindex="-1"></a>      PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb97-31"><a href="#cb97-31" aria-hidden="true" tabindex="-1"></a>      PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb97-32"><a href="#cb97-32" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb97-33"><a href="#cb97-33" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb97-34"><a href="#cb97-34" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb97-35"><a href="#cb97-35" aria-hidden="true" tabindex="-1"></a>      out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb97-36"><a href="#cb97-36" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb97-37"><a href="#cb97-37" aria-hidden="true" tabindex="-1"></a>    <span class="cf">next</span></span>
<span id="cb97-38"><a href="#cb97-38" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb97-39"><a href="#cb97-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb97-40"><a href="#cb97-40" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data  <span class="sc">%&gt;%</span></span>
<span id="cb97-41"><a href="#cb97-41" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Remove duplicated pathogen proteins identified in </span></span>
<span id="cb97-42"><a href="#cb97-42" aria-hidden="true" tabindex="-1"></a>    <span class="do">## ../../Scripts/Duplicate_Protein_Screen.R</span></span>
<span id="cb97-43"><a href="#cb97-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>viral_id <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb97-44"><a href="#cb97-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>uniprot <span class="sc">%in%</span> human_duplicated_proteins<span class="sc">$</span>uniprot) <span class="sc">%&gt;%</span></span>
<span id="cb97-45"><a href="#cb97-45" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(query, query_seqname, query_start, mismatches, reaction.human,</span>
<span id="cb97-46"><a href="#cb97-46" aria-hidden="true" tabindex="-1"></a>           reaction.viral, reaction.same, catalytic_activity.same,</span>
<span id="cb97-47"><a href="#cb97-47" aria-hidden="true" tabindex="-1"></a>           ec_number.human, ec_number.viral,</span>
<span id="cb97-48"><a href="#cb97-48" aria-hidden="true" tabindex="-1"></a>           enzyme_number.same, external_gene_name)</span>
<span id="cb97-49"><a href="#cb97-49" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb97-50"><a href="#cb97-50" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>reaction.same[<span class="fu">is.na</span>(active_data<span class="sc">$</span>reaction.same)] <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb97-51"><a href="#cb97-51" aria-hidden="true" tabindex="-1"></a>  active_data<span class="sc">$</span>catalytic_activity.same[<span class="fu">is.na</span>(active_data<span class="sc">$</span>catalytic_activity.same)] <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb97-52"><a href="#cb97-52" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb97-53"><a href="#cb97-53" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(active_data)</span>
<span id="cb97-54"><a href="#cb97-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-55"><a href="#cb97-55" aria-hidden="true" tabindex="-1"></a>  active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb97-56"><a href="#cb97-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(reaction.same <span class="sc">!=</span> <span class="cn">TRUE</span> <span class="sc">&amp;</span> catalytic_activity.same <span class="sc">!=</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb97-57"><a href="#cb97-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(enzyme_number.same <span class="sc">!=</span> <span class="cn">TRUE</span>)  </span>
<span id="cb97-58"><a href="#cb97-58" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb97-59"><a href="#cb97-59" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(active_data<span class="sc">$</span>mismatches))</span>
<span id="cb97-60"><a href="#cb97-60" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(out) <span class="sc">==</span> <span class="dv">0</span>){out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="at">V2 =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))}</span>
<span id="cb97-61"><a href="#cb97-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(out) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>, <span class="fu">gsub</span>(<span class="st">"_18mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb97-62"><a href="#cb97-62" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb97-63"><a href="#cb97-63" aria-hidden="true" tabindex="-1"></a>   <span class="cf">if</span>(mode <span class="sc">==</span> <span class="st">"tight_elm"</span>){</span>
<span id="cb97-64"><a href="#cb97-64" aria-hidden="true" tabindex="-1"></a>    active_data <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb97-65"><a href="#cb97-65" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">query_end =</span> query_start <span class="sc">+</span> (kmer_length<span class="dv">-1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb97-66"><a href="#cb97-66" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(elm_instance_df <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(viral_protein, start, end, Regex) <span class="sc">%&gt;%</span></span>
<span id="cb97-67"><a href="#cb97-67" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">mutate</span>(<span class="at">viral_protein =</span> <span class="fu">gsub</span>(<span class="st">"&gt;"</span>, <span class="st">""</span>, viral_protein)),</span>
<span id="cb97-68"><a href="#cb97-68" aria-hidden="true" tabindex="-1"></a>                <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"query_seqname"</span> <span class="ot">=</span> <span class="st">"viral_protein"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb97-69"><a href="#cb97-69" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">in_elm =</span> ((query_start <span class="sc">&gt;=</span> start <span class="sc">&amp;</span> query_start <span class="sc">&lt;=</span> end) <span class="sc">|</span> (query_end <span class="sc">&gt;=</span> start <span class="sc">&amp;</span> query_end <span class="sc">&lt;=</span> end))) <span class="sc">%&gt;%</span></span>
<span id="cb97-70"><a href="#cb97-70" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(query, query_seqname, query_start, mismatches, external_gene_name) <span class="sc">%&gt;%</span></span>
<span id="cb97-71"><a href="#cb97-71" aria-hidden="true" tabindex="-1"></a>      <span class="fu">slice_max</span>(in_elm) <span class="sc">%&gt;%</span></span>
<span id="cb97-72"><a href="#cb97-72" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(query, query_seqname, query_start, mismatches, external_gene_name, in_elm) <span class="sc">%&gt;%</span></span>
<span id="cb97-73"><a href="#cb97-73" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(<span class="sc">!</span>in_elm)</span>
<span id="cb97-74"><a href="#cb97-74" aria-hidden="true" tabindex="-1"></a>   }</span>
<span id="cb97-75"><a href="#cb97-75" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb97-76"><a href="#cb97-76" aria-hidden="true" tabindex="-1"></a>  out_protein <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb97-77"><a href="#cb97-77" aria-hidden="true" tabindex="-1"></a>     <span class="fu">distinct</span>(query, query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb97-78"><a href="#cb97-78" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb97-79"><a href="#cb97-79" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb97-80"><a href="#cb97-80" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb97-81"><a href="#cb97-81" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb97-82"><a href="#cb97-82" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb97-83"><a href="#cb97-83" aria-hidden="true" tabindex="-1"></a>   out_protein_min <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb97-84"><a href="#cb97-84" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb97-85"><a href="#cb97-85" aria-hidden="true" tabindex="-1"></a>     <span class="fu">distinct</span>(query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb97-86"><a href="#cb97-86" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, query_start) <span class="sc">%&gt;%</span></span>
<span id="cb97-87"><a href="#cb97-87" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_min</span>(mismatches, <span class="at">n=</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb97-88"><a href="#cb97-88" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() </span>
<span id="cb97-89"><a href="#cb97-89" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb97-90"><a href="#cb97-90" aria-hidden="true" tabindex="-1"></a>   out <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(out_protein_min<span class="sc">$</span>mismatches))</span>
<span id="cb97-91"><a href="#cb97-91" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(out) <span class="sc">==</span> <span class="dv">0</span>){out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mismatches =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="at">V2 =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))}</span>
<span id="cb97-92"><a href="#cb97-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(out) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mismatches"</span>, <span class="fu">gsub</span>(<span class="st">"_18mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file))</span>
<span id="cb97-93"><a href="#cb97-93" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb97-94"><a href="#cb97-94" aria-hidden="true" tabindex="-1"></a>     out_protein_min <span class="ot">&lt;-</span> out_protein_min <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(query_seqname, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb97-95"><a href="#cb97-95" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb97-96"><a href="#cb97-96" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb97-97"><a href="#cb97-97" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb97-98"><a href="#cb97-98" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb97-99"><a href="#cb97-99" aria-hidden="true" tabindex="-1"></a>     out_protein_count <span class="ot">&lt;-</span> active_data <span class="sc">%&gt;%</span></span>
<span id="cb97-100"><a href="#cb97-100" aria-hidden="true" tabindex="-1"></a>       <span class="fu">distinct</span>(query, query_seqname, query_start, mismatches, external_gene_name) <span class="sc">%&gt;%</span></span>
<span id="cb97-101"><a href="#cb97-101" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">query_seqname =</span> <span class="fu">gsub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, query_seqname)) <span class="sc">%&gt;%</span></span>
<span id="cb97-102"><a href="#cb97-102" aria-hidden="true" tabindex="-1"></a>       <span class="fu">distinct</span>(query, query_seqname, query_start, external_gene_name, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb97-103"><a href="#cb97-103" aria-hidden="true" tabindex="-1"></a>       <span class="fu">filter</span>(<span class="sc">!</span>external_gene_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="cn">NA</span>, <span class="st">""</span>, <span class="st">" "</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb97-104"><a href="#cb97-104" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(query_seqname, query_start, mismatches) <span class="sc">%&gt;%</span></span>
<span id="cb97-105"><a href="#cb97-105" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb97-106"><a href="#cb97-106" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Proteome ID</span><span class="st">`</span>  <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>,<span class="fu">gsub</span>(<span class="st">".*_UP"</span>, <span class="st">"UP"</span>,active_file)))</span>
<span id="cb97-107"><a href="#cb97-107" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb97-108"><a href="#cb97-108" aria-hidden="true" tabindex="-1"></a>  <span class="do">############## Count Proteome Length ##############</span></span>
<span id="cb97-109"><a href="#cb97-109" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">try</span>(seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"../../epitope_alignment/Viral_Proteomes/Final_Cohort/"</span>, <span class="fu">gsub</span>(<span class="st">"_18mer_3mis_hits.csv.gz"</span>, <span class="st">""</span>, active_file), <span class="st">".fasta.gz"</span>), <span class="at">seqtype =</span> <span class="st">"AA"</span>, <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">set.attributes =</span> <span class="cn">FALSE</span>), <span class="at">silent =</span> <span class="cn">TRUE</span>)</span>
<span id="cb97-110"><a href="#cb97-110" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(data) <span class="sc">==</span> <span class="st">"try-error"</span>){</span>
<span id="cb97-111"><a href="#cb97-111" aria-hidden="true" tabindex="-1"></a>        PLength <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb97-112"><a href="#cb97-112" aria-hidden="true" tabindex="-1"></a>        PCount <span class="ot">&lt;-</span> <span class="st">"Error"</span></span>
<span id="cb97-113"><a href="#cb97-113" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb97-114"><a href="#cb97-114" aria-hidden="true" tabindex="-1"></a>       PLength <span class="ot">&lt;-</span> <span class="fu">nchar</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb97-115"><a href="#cb97-115" aria-hidden="true" tabindex="-1"></a>      PCount <span class="ot">&lt;-</span> <span class="fu">length</span>(data[<span class="sc">!</span><span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">names</span>(data), <span class="at">split=</span><span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>), <span class="cf">function</span>(x) <span class="fu">paste</span>(x[<span class="dv">2</span>])) <span class="sc">%in%</span> duplicated_proteins<span class="sc">$</span>viral_uniprot])</span>
<span id="cb97-116"><a href="#cb97-116" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb97-117"><a href="#cb97-117" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb97-118"><a href="#cb97-118" aria-hidden="true" tabindex="-1"></a>  out_final <span class="ot">&lt;-</span> <span class="fu">merge</span>(out_final, out, <span class="at">by =</span> <span class="st">"mismatches"</span>, <span class="at">all.x =</span> T)</span>
<span id="cb97-119"><a href="#cb97-119" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">6</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> <span class="fu">sum</span>(PLength)</span>
<span id="cb97-120"><a href="#cb97-120" aria-hidden="true" tabindex="-1"></a>  out_final[<span class="dv">5</span>, <span class="fu">ncol</span>(out_final)] <span class="ot">&lt;-</span> PCount</span>
<span id="cb97-121"><a href="#cb97-121" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb97-122"><a href="#cb97-122" aria-hidden="true" tabindex="-1"></a>   out_protein_tmp <span class="ot">&lt;-</span> out_protein <span class="sc">%&gt;%</span></span>
<span id="cb97-123"><a href="#cb97-123" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb97-124"><a href="#cb97-124" aria-hidden="true" tabindex="-1"></a>  out_protein_tmp_min <span class="ot">&lt;-</span> out_protein_min <span class="sc">%&gt;%</span></span>
<span id="cb97-125"><a href="#cb97-125" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb97-126"><a href="#cb97-126" aria-hidden="true" tabindex="-1"></a>  out_protein_tmp_count <span class="ot">&lt;-</span> out_protein_count <span class="sc">%&gt;%</span></span>
<span id="cb97-127"><a href="#cb97-127" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">as.data.frame</span>(PLength) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">"query_seqname"</span>))</span>
<span id="cb97-128"><a href="#cb97-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-129"><a href="#cb97-129" aria-hidden="true" tabindex="-1"></a>  out_protein_final <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final, out_protein_tmp)</span>
<span id="cb97-130"><a href="#cb97-130" aria-hidden="true" tabindex="-1"></a>  out_protein_final_min <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final_min, out_protein_tmp_min)</span>
<span id="cb97-131"><a href="#cb97-131" aria-hidden="true" tabindex="-1"></a>  out_protein_final_count <span class="ot">&lt;-</span> <span class="fu">rbind.data.frame</span>(out_protein_final_count, out_protein_tmp_count)</span>
<span id="cb97-132"><a href="#cb97-132" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb97-133"><a href="#cb97-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-134"><a href="#cb97-134" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span>
<span id="cb97-135"><a href="#cb97-135" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_final, <span class="st">"../Key_Data/18mer_proteome_everything_elmfilter_regex.csv"</span>)</span>
<span id="cb97-136"><a href="#cb97-136" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_protein_final, <span class="st">"../Key_Data/18mer_proteome_protein_everything_elmfilter_regex.csv"</span>)</span>
<span id="cb97-137"><a href="#cb97-137" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_protein_final_min, <span class="st">"../Key_Data/18mer_proteome_protein_everything_min_elmfilter_regex.csv"</span>)</span>
<span id="cb97-138"><a href="#cb97-138" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out_protein_final_count, <span class="st">"../Key_Data/18mer_proteome_protein_everything_count_elmfilter_regex.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="panel-d" class="level2">
<h2 class="anchored" data-anchor-id="panel-d">Panel D</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>out_final<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/8mer_proteome_everything_elmfilter_regex.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final)</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-7"><a href="#cb98-7" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb98-8"><a href="#cb98-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb98-9"><a href="#cb98-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb98-10"><a href="#cb98-10" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic))<span class="sc">%&gt;%</span></span>
<span id="cb98-11"><a href="#cb98-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb98-12"><a href="#cb98-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb98-13"><a href="#cb98-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb98-14"><a href="#cb98-14" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb98-15"><a href="#cb98-15" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb98-16"><a href="#cb98-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb98-17"><a href="#cb98-17" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb98-18"><a href="#cb98-18" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb98-19"><a href="#cb98-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb98-20"><a href="#cb98-20" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb98-21"><a href="#cb98-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-22"><a href="#cb98-22" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">8-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb98-23"><a href="#cb98-23" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">8-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb98-24"><a href="#cb98-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">8-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb98-25"><a href="#cb98-25" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">8-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb98-26"><a href="#cb98-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-27"><a href="#cb98-27" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(mismatch_data_meta, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"8mers"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb98-28"><a href="#cb98-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-29"><a href="#cb98-29" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb98-30"><a href="#cb98-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"Virus"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb98-31"><a href="#cb98-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb98-32"><a href="#cb98-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>))</span>
<span id="cb98-33"><a href="#cb98-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-34"><a href="#cb98-34" aria-hidden="true" tabindex="-1"></a>wt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Chronic.or.Acute,</span>
<span id="cb98-35"><a href="#cb98-35" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb98-36"><a href="#cb98-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb98-37"><a href="#cb98-37" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb98-38"><a href="#cb98-38" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"8mer_Chronic_vs_Acute_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb98-39"><a href="#cb98-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-40"><a href="#cb98-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-41"><a href="#cb98-41" aria-hidden="true" tabindex="-1"></a>panel_d <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb98-42"><a href="#cb98-42" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(grepl("Virus", Taxonomic.lineage)) %&gt;%</span></span>
<span id="cb98-43"><a href="#cb98-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb98-44"><a href="#cb98-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">"Bacteria"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb98-45"><a href="#cb98-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Chronic.or.Acute =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"Bacteria"</span>, Taxonomic.lineage), <span class="st">"Bacteria"</span>, Chronic.or.Acute)) <span class="sc">%&gt;%</span></span>
<span id="cb98-46"><a href="#cb98-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>, name) <span class="sc">%&gt;%</span></span>
<span id="cb98-47"><a href="#cb98-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(value), <span class="dv">0</span>, value)) <span class="sc">%&gt;%</span></span>
<span id="cb98-48"><a href="#cb98-48" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(value), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb98-49"><a href="#cb98-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb98-50"><a href="#cb98-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb98-51"><a href="#cb98-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">"≤"</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb98-52"><a href="#cb98-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"≤1"</span>, <span class="st">"≤2"</span>, <span class="st">"≤3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb98-53"><a href="#cb98-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb98-54"><a href="#cb98-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb98-55"><a href="#cb98-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb98-56"><a href="#cb98-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log10_trans</span>(),</span>
<span id="cb98-57"><a href="#cb98-57" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log10"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">10</span><span class="sc">^</span>x, <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb98-58"><a href="#cb98-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb98-59"><a href="#cb98-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>name), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb98-60"><a href="#cb98-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb98-61"><a href="#cb98-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span><span class="fu">c</span>(<span class="st">"#1975EE"</span>, <span class="st">"#E78305"</span>, <span class="st">"black"</span>)) <span class="sc">+</span></span>
<span id="cb98-62"><a href="#cb98-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent of 8mers (%)"</span>) <span class="sc">+</span></span>
<span id="cb98-63"><a href="#cb98-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">size =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb98-64"><a href="#cb98-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="fl">0.15</span>, <span class="dv">40</span>, <span class="dv">40</span>, <span class="dv">40</span>),</span>
<span id="cb98-65"><a href="#cb98-65" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb98-66"><a href="#cb98-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb98-67"><a href="#cb98-67" aria-hidden="true" tabindex="-1"></a>panel_d</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FigureS1_ELMFilter_files/figure-html/panel_c-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="panel-e" class="level2">
<h2 class="anchored" data-anchor-id="panel-e">Panel E</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final)</span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic))<span class="sc">%&gt;%</span></span>
<span id="cb99-10"><a href="#cb99-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb99-11"><a href="#cb99-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb99-12"><a href="#cb99-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb99-13"><a href="#cb99-13" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb99-14"><a href="#cb99-14" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb99-15"><a href="#cb99-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb99-16"><a href="#cb99-16" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb99-17"><a href="#cb99-17" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb99-18"><a href="#cb99-18" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb99-19"><a href="#cb99-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb99-20"><a href="#cb99-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-21"><a href="#cb99-21" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">8-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb99-22"><a href="#cb99-22" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">8-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb99-23"><a href="#cb99-23" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">8-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb99-24"><a href="#cb99-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">8-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb99-25"><a href="#cb99-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-26"><a href="#cb99-26" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb99-27"><a href="#cb99-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb99-28"><a href="#cb99-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb99-29"><a href="#cb99-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb99-30"><a href="#cb99-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Herpesviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Herpesviridae"</span>, <span class="st">"Herpesviridae"</span>, <span class="st">"Non-Herpesviridae"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb99-31"><a href="#cb99-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Poxviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Poxviridae"</span>, <span class="st">"Poxviridae"</span>, <span class="st">"Non-Poxviridae"</span>))</span>
<span id="cb99-32"><a href="#cb99-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-33"><a href="#cb99-33" aria-hidden="true" tabindex="-1"></a><span class="do">## These results are used to manually add stars to the graph below in illustrator</span></span>
<span id="cb99-34"><a href="#cb99-34" aria-hidden="true" tabindex="-1"></a>kt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Family,</span>
<span id="cb99-35"><a href="#cb99-35" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"kruskal.test"</span>)</span>
<span id="cb99-36"><a href="#cb99-36" aria-hidden="true" tabindex="-1"></a>wt.p <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Poxviridae,</span>
<span id="cb99-37"><a href="#cb99-37" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb99-38"><a href="#cb99-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb99-39"><a href="#cb99-39" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb99-40"><a href="#cb99-40" aria-hidden="true" tabindex="-1"></a>wt.h <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Herpesviridae,</span>
<span id="cb99-41"><a href="#cb99-41" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb99-42"><a href="#cb99-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb99-43"><a href="#cb99-43" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb99-44"><a href="#cb99-44" aria-hidden="true" tabindex="-1"></a>wt.hp <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Family <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Herpesviridae"</span>, <span class="st">"Poxviridae"</span>)), <span class="at">formula =</span> value <span class="sc">~</span> Poxviridae,</span>
<span id="cb99-45"><a href="#cb99-45" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb99-46"><a href="#cb99-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb99-47"><a href="#cb99-47" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>)))) <span class="sc">%&gt;%</span></span>
<span id="cb99-48"><a href="#cb99-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group1 =</span> <span class="st">"Herpesviridae"</span>)</span>
<span id="cb99-49"><a href="#cb99-49" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(kt, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"8mer_Viral_Family_Kruskal"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb99-50"><a href="#cb99-50" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.p, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"8mer_Pox_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb99-51"><a href="#cb99-51" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.h, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"8mer_Herpes_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb99-52"><a href="#cb99-52" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.hp, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"8mer_Herpes_vs_Pox_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb99-53"><a href="#cb99-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-54"><a href="#cb99-54" aria-hidden="true" tabindex="-1"></a>panel_e <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb99-55"><a href="#cb99-55" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(Family = ifelse(grepl("Bacteria", Taxonomic.lineage), "Bacteria", Family_Viral_Zone)) %&gt;%</span></span>
<span id="cb99-56"><a href="#cb99-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb99-57"><a href="#cb99-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb99-58"><a href="#cb99-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb99-59"><a href="#cb99-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Family, name) <span class="sc">%&gt;%</span></span>
<span id="cb99-60"><a href="#cb99-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(value), <span class="dv">0</span>, value)) <span class="sc">%&gt;%</span></span>
<span id="cb99-61"><a href="#cb99-61" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(value), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb99-62"><a href="#cb99-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(count <span class="sc">&gt;</span> <span class="dv">4</span>) <span class="sc">%&gt;%</span></span>
<span id="cb99-63"><a href="#cb99-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb99-64"><a href="#cb99-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb99-65"><a href="#cb99-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">"≤"</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb99-66"><a href="#cb99-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"≤1"</span>, <span class="st">"≤2"</span>, <span class="st">"≤3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb99-67"><a href="#cb99-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb99-68"><a href="#cb99-68" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb99-69"><a href="#cb99-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb99-70"><a href="#cb99-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log10_trans</span>(),</span>
<span id="cb99-71"><a href="#cb99-71" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log10"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">10</span><span class="sc">^</span>x, <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb99-72"><a href="#cb99-72" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb99-73"><a href="#cb99-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>name), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb99-74"><a href="#cb99-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb99-75"><a href="#cb99-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span>Family_palette_v2) <span class="sc">+</span></span>
<span id="cb99-76"><a href="#cb99-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent of 8mers (%)"</span>) <span class="sc">+</span></span>
<span id="cb99-77"><a href="#cb99-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">size =</span> <span class="fl">0.1</span>)  <span class="sc">+</span></span>
<span id="cb99-78"><a href="#cb99-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">0.9</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="fl">3.9</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">13</span>, <span class="dv">13</span>, <span class="dv">13</span>),</span>
<span id="cb99-79"><a href="#cb99-79" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.h<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>,</span>
<span id="cb99-80"><a href="#cb99-80" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Herpesviridae"</span>]) <span class="sc">+</span></span>
<span id="cb99-81"><a href="#cb99-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">1.1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="fl">4.1</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">13</span>, <span class="dv">13</span>, <span class="dv">13</span>),</span>
<span id="cb99-82"><a href="#cb99-82" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.p<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>,</span>
<span id="cb99-83"><a href="#cb99-83" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Poxviridae"</span>])  <span class="sc">+</span></span>
<span id="cb99-84"><a href="#cb99-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb99-85"><a href="#cb99-85" aria-hidden="true" tabindex="-1"></a>panel_e</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FigureS1_ELMFilter_files/figure-html/panel_d-1.png" class="img-fluid" width="384"></p>
</div>
</div>
</section>
<section id="panel-f" class="level2">
<h2 class="anchored" data-anchor-id="panel-f">Panel F</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="co">#out_final &lt;- read.csv("../../Key_Data/12mer_proteome_everything.csv", row.names =1)</span></span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>out_final <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/12mer_proteome_everything_elmfilter_regex.csv"</span>, <span class="at">row.names =</span><span class="dv">1</span>)</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final)</span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-9"><a href="#cb100-9" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb100-10"><a href="#cb100-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb100-11"><a href="#cb100-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb100-12"><a href="#cb100-12" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic))<span class="sc">%&gt;%</span></span>
<span id="cb100-13"><a href="#cb100-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb100-14"><a href="#cb100-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb100-15"><a href="#cb100-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb100-16"><a href="#cb100-16" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb100-17"><a href="#cb100-17" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb100-18"><a href="#cb100-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb100-19"><a href="#cb100-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb100-20"><a href="#cb100-20" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb100-21"><a href="#cb100-21" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb100-22"><a href="#cb100-22" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb100-23"><a href="#cb100-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-24"><a href="#cb100-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">12-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb100-25"><a href="#cb100-25" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">12-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb100-26"><a href="#cb100-26" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">12-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb100-27"><a href="#cb100-27" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">12-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb100-28"><a href="#cb100-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-29"><a href="#cb100-29" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(mismatch_data_meta, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"12mers"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb100-30"><a href="#cb100-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-31"><a href="#cb100-31" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb100-32"><a href="#cb100-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"Virus"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb100-33"><a href="#cb100-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb100-34"><a href="#cb100-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>))</span>
<span id="cb100-35"><a href="#cb100-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-36"><a href="#cb100-36" aria-hidden="true" tabindex="-1"></a>wt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Chronic.or.Acute,</span>
<span id="cb100-37"><a href="#cb100-37" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb100-38"><a href="#cb100-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb100-39"><a href="#cb100-39" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb100-40"><a href="#cb100-40" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"12mer_Chronic_vs_Acute_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb100-41"><a href="#cb100-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-42"><a href="#cb100-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-43"><a href="#cb100-43" aria-hidden="true" tabindex="-1"></a>panel_f <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb100-44"><a href="#cb100-44" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(grepl("Virus", Taxonomic.lineage)) %&gt;%</span></span>
<span id="cb100-45"><a href="#cb100-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb100-46"><a href="#cb100-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">"Bacteria"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb100-47"><a href="#cb100-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Chronic.or.Acute =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"Bacteria"</span>, Taxonomic.lineage), <span class="st">"Bacteria"</span>, Chronic.or.Acute)) <span class="sc">%&gt;%</span></span>
<span id="cb100-48"><a href="#cb100-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>, name) <span class="sc">%&gt;%</span></span>
<span id="cb100-49"><a href="#cb100-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(value), <span class="dv">0</span>, value)) <span class="sc">%&gt;%</span></span>
<span id="cb100-50"><a href="#cb100-50" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(value), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb100-51"><a href="#cb100-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb100-52"><a href="#cb100-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb100-53"><a href="#cb100-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">"≤"</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb100-54"><a href="#cb100-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"≤1"</span>, <span class="st">"≤2"</span>, <span class="st">"≤3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb100-55"><a href="#cb100-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb100-56"><a href="#cb100-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb100-57"><a href="#cb100-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb100-58"><a href="#cb100-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log10_trans</span>(),</span>
<span id="cb100-59"><a href="#cb100-59" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log10"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">10</span><span class="sc">^</span>x, <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb100-60"><a href="#cb100-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb100-61"><a href="#cb100-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>name), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb100-62"><a href="#cb100-62" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb100-63"><a href="#cb100-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span><span class="fu">c</span>(<span class="st">"#1975EE"</span>, <span class="st">"#E78305"</span>, <span class="st">"black"</span>)) <span class="sc">+</span></span>
<span id="cb100-64"><a href="#cb100-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent of 12mers (%)"</span>) <span class="sc">+</span></span>
<span id="cb100-65"><a href="#cb100-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">size =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb100-66"><a href="#cb100-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="fl">0.07</span>, <span class="fl">0.210</span>, <span class="dv">1</span>, <span class="dv">3</span>),</span>
<span id="cb100-67"><a href="#cb100-67" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb100-68"><a href="#cb100-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.key.height=</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">'cm'</span>),</span>
<span id="cb100-69"><a href="#cb100-69" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.width=</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">'cm'</span>),</span>
<span id="cb100-70"><a href="#cb100-70" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb100-71"><a href="#cb100-71" aria-hidden="true" tabindex="-1"></a>panel_f</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FigureS1_ELMFilter_files/figure-html/panel_e-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="panel-g" class="level2">
<h2 class="anchored" data-anchor-id="panel-g">Panel G</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final)</span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb101-5"><a href="#cb101-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-6"><a href="#cb101-6" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb101-7"><a href="#cb101-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb101-8"><a href="#cb101-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb101-9"><a href="#cb101-9" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic))<span class="sc">%&gt;%</span></span>
<span id="cb101-10"><a href="#cb101-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb101-11"><a href="#cb101-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb101-12"><a href="#cb101-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb101-13"><a href="#cb101-13" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb101-14"><a href="#cb101-14" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb101-15"><a href="#cb101-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb101-16"><a href="#cb101-16" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb101-17"><a href="#cb101-17" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb101-18"><a href="#cb101-18" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb101-19"><a href="#cb101-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb101-20"><a href="#cb101-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-21"><a href="#cb101-21" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">12-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb101-22"><a href="#cb101-22" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">12-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb101-23"><a href="#cb101-23" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">12-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb101-24"><a href="#cb101-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">12-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb101-25"><a href="#cb101-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-26"><a href="#cb101-26" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb101-27"><a href="#cb101-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb101-28"><a href="#cb101-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb101-29"><a href="#cb101-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb101-30"><a href="#cb101-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Herpesviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Herpesviridae"</span>, <span class="st">"Herpesviridae"</span>, <span class="st">"Non-Herpesviridae"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb101-31"><a href="#cb101-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Poxviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Poxviridae"</span>, <span class="st">"Poxviridae"</span>, <span class="st">"Non-Poxviridae"</span>))</span>
<span id="cb101-32"><a href="#cb101-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-33"><a href="#cb101-33" aria-hidden="true" tabindex="-1"></a><span class="do">## These results are used to manually add stars to the graph below in illustrator</span></span>
<span id="cb101-34"><a href="#cb101-34" aria-hidden="true" tabindex="-1"></a>kt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Family,</span>
<span id="cb101-35"><a href="#cb101-35" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"kruskal.test"</span>)</span>
<span id="cb101-36"><a href="#cb101-36" aria-hidden="true" tabindex="-1"></a>wt.p <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Poxviridae,</span>
<span id="cb101-37"><a href="#cb101-37" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb101-38"><a href="#cb101-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb101-39"><a href="#cb101-39" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb101-40"><a href="#cb101-40" aria-hidden="true" tabindex="-1"></a>wt.h <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Herpesviridae,</span>
<span id="cb101-41"><a href="#cb101-41" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb101-42"><a href="#cb101-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb101-43"><a href="#cb101-43" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb101-44"><a href="#cb101-44" aria-hidden="true" tabindex="-1"></a>wt.hp <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Family <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Herpesviridae"</span>, <span class="st">"Poxviridae"</span>)), <span class="at">formula =</span> value <span class="sc">~</span> Poxviridae,</span>
<span id="cb101-45"><a href="#cb101-45" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb101-46"><a href="#cb101-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb101-47"><a href="#cb101-47" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>)))) <span class="sc">%&gt;%</span></span>
<span id="cb101-48"><a href="#cb101-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group1 =</span> <span class="st">"Herpesviridae"</span>)</span>
<span id="cb101-49"><a href="#cb101-49" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(kt, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"12mer_Viral_Family_Kruskal"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb101-50"><a href="#cb101-50" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.p, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"12mer_Pox_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb101-51"><a href="#cb101-51" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.h, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"12mer_Herpes_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb101-52"><a href="#cb101-52" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.hp, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"12mer_Herpes_vs_Pox_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb101-53"><a href="#cb101-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-54"><a href="#cb101-54" aria-hidden="true" tabindex="-1"></a>panel_g <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb101-55"><a href="#cb101-55" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(Family = ifelse(grepl("Bacteria", Taxonomic.lineage), "Bacteria", Family_Viral_Zone)) %&gt;%</span></span>
<span id="cb101-56"><a href="#cb101-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb101-57"><a href="#cb101-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb101-58"><a href="#cb101-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb101-59"><a href="#cb101-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Family, name) <span class="sc">%&gt;%</span></span>
<span id="cb101-60"><a href="#cb101-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(value), <span class="dv">0</span>, value)) <span class="sc">%&gt;%</span></span>
<span id="cb101-61"><a href="#cb101-61" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(value), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb101-62"><a href="#cb101-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Family <span class="sc">%in%</span> <span class="fu">names</span>(Family_palette_v2)) <span class="sc">%&gt;%</span></span>
<span id="cb101-63"><a href="#cb101-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb101-64"><a href="#cb101-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb101-65"><a href="#cb101-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">"≤"</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb101-66"><a href="#cb101-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"≤1"</span>, <span class="st">"≤2"</span>, <span class="st">"≤3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb101-67"><a href="#cb101-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mean_value =</span> <span class="fu">ifelse</span>(mean_value <span class="sc">==</span> <span class="dv">0</span>, <span class="fl">1E-3</span>, mean_value)) <span class="sc">%&gt;%</span></span>
<span id="cb101-68"><a href="#cb101-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb101-69"><a href="#cb101-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb101-70"><a href="#cb101-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb101-71"><a href="#cb101-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb101-72"><a href="#cb101-72" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>name), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb101-73"><a href="#cb101-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb101-74"><a href="#cb101-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span>Family_palette_v2) <span class="sc">+</span></span>
<span id="cb101-75"><a href="#cb101-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent of 12mers (%)"</span>) <span class="sc">+</span></span>
<span id="cb101-76"><a href="#cb101-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">size =</span> <span class="fl">0.1</span>)  <span class="sc">+</span></span>
<span id="cb101-77"><a href="#cb101-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">0.9</span>,<span class="fl">1.9</span>,<span class="fl">2.9</span>,<span class="fl">3.9</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="fl">0.45</span>, <span class="fl">0.75</span>, <span class="fl">1.7</span>, <span class="dv">13</span>),</span>
<span id="cb101-78"><a href="#cb101-78" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.h<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>,</span>
<span id="cb101-79"><a href="#cb101-79" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Herpesviridae"</span>]) <span class="sc">+</span></span>
<span id="cb101-80"><a href="#cb101-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">1.1</span>,<span class="fl">2.1</span>,<span class="fl">3.1</span>,<span class="fl">4.1</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="fl">0.45</span>, <span class="fl">0.75</span>, <span class="fl">1.7</span>, <span class="dv">13</span>),</span>
<span id="cb101-81"><a href="#cb101-81" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.p<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>,</span>
<span id="cb101-82"><a href="#cb101-82" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Poxviridae"</span>]) <span class="sc">+</span></span>
<span id="cb101-83"><a href="#cb101-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.key.height=</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">'cm'</span>),</span>
<span id="cb101-84"><a href="#cb101-84" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.width=</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">'cm'</span>)) <span class="sc">+</span></span>
<span id="cb101-85"><a href="#cb101-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb101-86"><a href="#cb101-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">1e-3</span>, <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb101-87"><a href="#cb101-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">log10_trans</span>(),</span>
<span id="cb101-88"><a href="#cb101-88" aria-hidden="true" tabindex="-1"></a>     <span class="at">breaks =</span> <span class="fu">trans_breaks</span>(<span class="st">"log10"</span>, <span class="cf">function</span>(x) <span class="fu">round</span>(<span class="dv">10</span><span class="sc">^</span>x, <span class="dv">3</span>)))</span>
<span id="cb101-89"><a href="#cb101-89" aria-hidden="true" tabindex="-1"></a>panel_g</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FigureS1_ELMFilter_files/figure-html/panel_f-1.png" class="img-fluid" width="384"></p>
</div>
</div>
</section>
<section id="panel-h" class="level2">
<h2 class="anchored" data-anchor-id="panel-h">Panel H</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>out_final_18mer <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../Key_Data/18mer_proteome_everything_elmfilter_regex.csv"</span>, <span class="at">row.names =</span><span class="dv">1</span>)</span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final_18mer)</span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb102-6"><a href="#cb102-6" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb102-7"><a href="#cb102-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-8"><a href="#cb102-8" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb102-9"><a href="#cb102-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb102-10"><a href="#cb102-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb102-11"><a href="#cb102-11" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic))<span class="sc">%&gt;%</span></span>
<span id="cb102-12"><a href="#cb102-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb102-13"><a href="#cb102-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb102-14"><a href="#cb102-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb102-15"><a href="#cb102-15" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb102-16"><a href="#cb102-16" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb102-17"><a href="#cb102-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb102-18"><a href="#cb102-18" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb102-19"><a href="#cb102-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb102-20"><a href="#cb102-20" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb102-21"><a href="#cb102-21" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb102-22"><a href="#cb102-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-23"><a href="#cb102-23" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">18-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb102-24"><a href="#cb102-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">18-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb102-25"><a href="#cb102-25" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">18-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb102-26"><a href="#cb102-26" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">18-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb102-27"><a href="#cb102-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-28"><a href="#cb102-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-29"><a href="#cb102-29" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(mismatch_data_meta, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"18mers"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb102-30"><a href="#cb102-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-31"><a href="#cb102-31" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb102-32"><a href="#cb102-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"Virus"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb102-33"><a href="#cb102-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb102-34"><a href="#cb102-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>))</span>
<span id="cb102-35"><a href="#cb102-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-36"><a href="#cb102-36" aria-hidden="true" tabindex="-1"></a>wt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Chronic.or.Acute,</span>
<span id="cb102-37"><a href="#cb102-37" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb102-38"><a href="#cb102-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb102-39"><a href="#cb102-39" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb102-40"><a href="#cb102-40" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"18mer_Chronic_vs_Acute_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb102-41"><a href="#cb102-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-42"><a href="#cb102-42" aria-hidden="true" tabindex="-1"></a>panel_h <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb102-43"><a href="#cb102-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"Virus"</span>, Taxonomic.lineage)) <span class="sc">%&gt;%</span></span>
<span id="cb102-44"><a href="#cb102-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb102-45"><a href="#cb102-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Chronic.or.Acute <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Acute"</span>, <span class="st">"Chronic"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb102-46"><a href="#cb102-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>, name) <span class="sc">%&gt;%</span></span>
<span id="cb102-47"><a href="#cb102-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(value), <span class="dv">0</span>, value)) <span class="sc">%&gt;%</span></span>
<span id="cb102-48"><a href="#cb102-48" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(value), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb102-49"><a href="#cb102-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb102-50"><a href="#cb102-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb102-51"><a href="#cb102-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">"≤"</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb102-52"><a href="#cb102-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"≤1"</span>, <span class="st">"≤2"</span>, <span class="st">"≤3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb102-53"><a href="#cb102-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb102-54"><a href="#cb102-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb102-55"><a href="#cb102-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Chronic.or.Acute</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb102-56"><a href="#cb102-56" aria-hidden="true" tabindex="-1"></a>    <span class="co">#scale_y_continuous(trans = log2_trans(),</span></span>
<span id="cb102-57"><a href="#cb102-57" aria-hidden="true" tabindex="-1"></a>    <span class="co"># breaks = trans_breaks("log2", function(x) round(2^x, 3))) +</span></span>
<span id="cb102-58"><a href="#cb102-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb102-59"><a href="#cb102-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>name), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb102-60"><a href="#cb102-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb102-61"><a href="#cb102-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span><span class="fu">c</span>(<span class="st">"#1975EE"</span>, <span class="st">"#E78305"</span>)) <span class="sc">+</span></span>
<span id="cb102-62"><a href="#cb102-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent of 18mers (%)"</span>) <span class="sc">+</span></span>
<span id="cb102-63"><a href="#cb102-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="fl">0.02</span>, <span class="fl">0.05</span>, <span class="fl">0.085</span>, <span class="fl">0.06</span>),</span>
<span id="cb102-64"><a href="#cb102-64" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb102-65"><a href="#cb102-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb102-66"><a href="#cb102-66" aria-hidden="true" tabindex="-1"></a>panel_h</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FigureS1_ELMFilter_files/figure-html/panel_g-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="panel-i" class="level2">
<h2 class="anchored" data-anchor-id="panel-i">Panel I</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">t</span>(out_final_18mer)</span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mismatch_data) <span class="ot">&lt;-</span> mismatch_data[<span class="dv">1</span>,]</span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a>mismatch_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mismatch_data[<span class="sc">-</span><span class="dv">1</span>,])</span>
<span id="cb103-4"><a href="#cb103-4" aria-hidden="true" tabindex="-1"></a>mismatch_data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*UP"</span>, <span class="st">"UP"</span>, <span class="fu">rownames</span>(mismatch_data))</span>
<span id="cb103-5"><a href="#cb103-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-6"><a href="#cb103-6" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta <span class="ot">&lt;-</span> mismatch_data <span class="sc">%&gt;%</span></span>
<span id="cb103-7"><a href="#cb103-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"Proteome.ID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb103-8"><a href="#cb103-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxon.mnemonic =</span> <span class="fu">ifelse</span>(Taxon.mnemonic <span class="sc">==</span> <span class="st">"null"</span>,</span>
<span id="cb103-9"><a href="#cb103-9" aria-hidden="true" tabindex="-1"></a>                                 ID, Taxon.mnemonic))<span class="sc">%&gt;%</span></span>
<span id="cb103-10"><a href="#cb103-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>), as.numeric) <span class="sc">%&gt;%</span></span>
<span id="cb103-11"><a href="#cb103-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb103-12"><a href="#cb103-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="st">`</span><span class="at">3</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb103-13"><a href="#cb103-13" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="st">`</span><span class="at">2</span><span class="st">`</span>,<span class="at">na.rm =</span> T),</span>
<span id="cb103-14"><a href="#cb103-14" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span>,<span class="at">na.rm =</span> T)) <span class="sc">%&gt;%</span></span>
<span id="cb103-15"><a href="#cb103-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb103-16"><a href="#cb103-16" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb103-17"><a href="#cb103-17" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb103-18"><a href="#cb103-18" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb103-19"><a href="#cb103-19" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>[<span class="fu">is.na</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb103-20"><a href="#cb103-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-21"><a href="#cb103-21" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">0</span><span class="st">`</span>) <span class="sc">/</span> (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">18-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb103-22"><a href="#cb103-22" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">18-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb103-23"><a href="#cb103-23" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">18-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb103-24"><a href="#cb103-24" aria-hidden="true" tabindex="-1"></a>mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3_corrected</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>) <span class="sc">/</span>  (<span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Proteome Length</span><span class="st">`</span>) <span class="sc">-</span>((<span class="dv">18-1</span>)<span class="sc">*</span><span class="fu">as.numeric</span>(mismatch_data_meta<span class="sc">$</span><span class="st">`</span><span class="at">Number of Proteins</span><span class="st">`</span>)))<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb103-25"><a href="#cb103-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-26"><a href="#cb103-26" aria-hidden="true" tabindex="-1"></a>testing_data <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb103-27"><a href="#cb103-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb103-28"><a href="#cb103-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb103-29"><a href="#cb103-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb103-30"><a href="#cb103-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Herpesviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Herpesviridae"</span>, <span class="st">"Herpesviridae"</span>, <span class="st">"Non-Herpesviridae"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb103-31"><a href="#cb103-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Poxviridae =</span> <span class="fu">ifelse</span>(Family <span class="sc">==</span> <span class="st">"Poxviridae"</span>, <span class="st">"Poxviridae"</span>, <span class="st">"Non-Poxviridae"</span>))</span>
<span id="cb103-32"><a href="#cb103-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-33"><a href="#cb103-33" aria-hidden="true" tabindex="-1"></a><span class="do">## These results are used to manually add stars to the graph below in illustrator</span></span>
<span id="cb103-34"><a href="#cb103-34" aria-hidden="true" tabindex="-1"></a>kt <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Family,</span>
<span id="cb103-35"><a href="#cb103-35" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"kruskal.test"</span>)</span>
<span id="cb103-36"><a href="#cb103-36" aria-hidden="true" tabindex="-1"></a>wt.p <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Poxviridae,</span>
<span id="cb103-37"><a href="#cb103-37" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb103-38"><a href="#cb103-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb103-39"><a href="#cb103-39" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb103-40"><a href="#cb103-40" aria-hidden="true" tabindex="-1"></a>wt.h <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data, <span class="at">formula =</span> value <span class="sc">~</span> Herpesviridae,</span>
<span id="cb103-41"><a href="#cb103-41" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb103-42"><a href="#cb103-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb103-43"><a href="#cb103-43" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>))))</span>
<span id="cb103-44"><a href="#cb103-44" aria-hidden="true" tabindex="-1"></a>wt.hp <span class="ot">&lt;-</span> <span class="fu">compare_means</span>(<span class="at">data =</span> testing_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Family <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Herpesviridae"</span>, <span class="st">"Poxviridae"</span>)), <span class="at">formula =</span> value <span class="sc">~</span> Poxviridae,</span>
<span id="cb103-45"><a href="#cb103-45" aria-hidden="true" tabindex="-1"></a>              <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"name"</span>), <span class="at">method =</span> <span class="st">"wilcox.test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb103-46"><a href="#cb103-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.adj.star =</span> <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.001</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*</span><span class="sc">\n</span><span class="st">*"</span>, <span class="fu">ifelse</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">"*</span><span class="sc">\n</span><span class="st">*"</span>,</span>
<span id="cb103-47"><a href="#cb103-47" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">ifelse</span>(p.adj<span class="sc">&lt;=</span><span class="fl">0.05</span>, <span class="st">"*"</span>, <span class="cn">NA</span>)))) <span class="sc">%&gt;%</span></span>
<span id="cb103-48"><a href="#cb103-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group1 =</span> <span class="st">"Herpesviridae"</span>)</span>
<span id="cb103-49"><a href="#cb103-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-50"><a href="#cb103-50" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(kt, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"18mer_Viral_Family_Kruskal"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb103-51"><a href="#cb103-51" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.p, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"18mer_Pox_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb103-52"><a href="#cb103-52" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.h, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"18mer_Herpes_vs_Others_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb103-53"><a href="#cb103-53" aria-hidden="true" tabindex="-1"></a>xlsx<span class="sc">::</span><span class="fu">write.xlsx</span>(wt.hp, <span class="fu">paste0</span>(out_dir, <span class="st">"FigureS1_data.xlsx"</span>), <span class="at">sheetName =</span> <span class="st">"18mer_Herpes_vs_Pox_Wilcox"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span>
<span id="cb103-54"><a href="#cb103-54" aria-hidden="true" tabindex="-1"></a>panel_i <span class="ot">&lt;-</span> mismatch_data_meta <span class="sc">%&gt;%</span></span>
<span id="cb103-55"><a href="#cb103-55" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(Family = ifelse(grepl("Bacteria", Taxonomic.lineage), "Bacteria", Family_Viral_Zone)) %&gt;%</span></span>
<span id="cb103-56"><a href="#cb103-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Family_Viral_Zone)) <span class="sc">%&gt;%</span></span>
<span id="cb103-57"><a href="#cb103-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family =</span> Family_Viral_Zone) <span class="sc">%&gt;%</span></span>
<span id="cb103-58"><a href="#cb103-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">paste0</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), (<span class="st">"_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb103-59"><a href="#cb103-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Family, name) <span class="sc">%&gt;%</span></span>
<span id="cb103-60"><a href="#cb103-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(value), <span class="dv">0</span>, value)) <span class="sc">%&gt;%</span></span>
<span id="cb103-61"><a href="#cb103-61" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_value =</span> <span class="fu">mean</span>(value), <span class="at">count =</span> <span class="fu">n</span>(), <span class="at">sd =</span> <span class="fu">sd</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb103-62"><a href="#cb103-62" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(count &gt; 4) %&gt;%</span></span>
<span id="cb103-63"><a href="#cb103-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value_stderr =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb103-64"><a href="#cb103-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">"_corrected"</span>, <span class="st">""</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb103-65"><a href="#cb103-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span><span class="fu">ifelse</span>(name <span class="sc">!=</span> <span class="st">"0"</span>,  <span class="fu">paste0</span>(<span class="st">"≤"</span>, name), name)) <span class="sc">%&gt;%</span></span>
<span id="cb103-66"><a href="#cb103-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"≤1"</span>, <span class="st">"≤2"</span>, <span class="st">"≤3"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb103-67"><a href="#cb103-67" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Only keep Families that have at least one non-zero point</span></span>
<span id="cb103-68"><a href="#cb103-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Family <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Poxviridae"</span>, <span class="st">"Herpesviridae"</span>, <span class="st">"Coronaviridae"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb103-69"><a href="#cb103-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>mean_value, <span class="at">color =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb103-70"><a href="#cb103-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb103-71"><a href="#cb103-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="st">`</span><span class="at">Family</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb103-72"><a href="#cb103-72" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb103-73"><a href="#cb103-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_value<span class="sc">-</span>value_stderr, <span class="at">ymax=</span>mean_value<span class="sc">+</span>value_stderr, <span class="at">position=</span>name), <span class="at">width=</span>.<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb103-74"><a href="#cb103-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Number of mismatches"</span>) <span class="sc">+</span></span>
<span id="cb103-75"><a href="#cb103-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span>Family_palette_v2) <span class="sc">+</span></span>
<span id="cb103-76"><a href="#cb103-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent of 18mers (%)"</span>) <span class="sc">+</span></span>
<span id="cb103-77"><a href="#cb103-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">0.9</span>,<span class="fl">1.9</span>,<span class="fl">2.9</span>,<span class="fl">3.9</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="fl">0.03</span>, <span class="fl">0.05</span>, <span class="fl">0.075</span>, <span class="fl">0.1</span>),</span>
<span id="cb103-78"><a href="#cb103-78" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.h<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>,</span>
<span id="cb103-79"><a href="#cb103-79" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Herpesviridae"</span>]) <span class="sc">+</span></span>
<span id="cb103-80"><a href="#cb103-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x=</span> <span class="fu">c</span>(<span class="fl">1.1</span>,<span class="fl">2.1</span>,<span class="fl">3.1</span>,<span class="fl">4.1</span>), <span class="at">y=</span> <span class="fu">c</span>(<span class="fl">0.03</span>, <span class="fl">0.05</span>, <span class="fl">0.075</span>, <span class="fl">0.1</span>),</span>
<span id="cb103-81"><a href="#cb103-81" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> wt.p<span class="sc">$</span>p.adj.star, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">lineheight =</span> <span class="fl">0.5</span>,</span>
<span id="cb103-82"><a href="#cb103-82" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Family_palette_v2[<span class="st">"Poxviridae"</span>]) <span class="sc">+</span></span>
<span id="cb103-83"><a href="#cb103-83" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_errorbar(inherit.aes = FALSE, aes(x=0.85, ymin = 0.018380776, ymax = 0.075685311), color = "#545252",</span></span>
<span id="cb103-84"><a href="#cb103-84" aria-hidden="true" tabindex="-1"></a>  <span class="co">#               size = 0.3, width = 0.1) +</span></span>
<span id="cb103-85"><a href="#cb103-85" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_errorbar(inherit.aes = FALSE, aes(x=1.85, ymin = 0.051403280, ymax = 0.182375884), color = "#545252",</span></span>
<span id="cb103-86"><a href="#cb103-86" aria-hidden="true" tabindex="-1"></a>  <span class="co">#               size = 0.3, width = 0.1) +</span></span>
<span id="cb103-87"><a href="#cb103-87" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_errorbar(inherit.aes = FALSE, aes(x=2.85, ymin = 0.085500719, ymax = 0.345138406), color = "#545252",</span></span>
<span id="cb103-88"><a href="#cb103-88" aria-hidden="true" tabindex="-1"></a>  <span class="co">#               size = 0.3, width = 0.1) +</span></span>
<span id="cb103-89"><a href="#cb103-89" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_errorbar(inherit.aes = FALSE, aes(x=3.85, ymin = 0.145041702, ymax = 0.536979034), color = "#545252",</span></span>
<span id="cb103-90"><a href="#cb103-90" aria-hidden="true" tabindex="-1"></a>  <span class="co">#               size = 0.3, width = 0.1) +</span></span>
<span id="cb103-91"><a href="#cb103-91" aria-hidden="true" tabindex="-1"></a>  <span class="co"># annotate("text", x= c(0.75, 1.8, 2.8, 3.8), y= c(0.04557349, 0.1136466,</span></span>
<span id="cb103-92"><a href="#cb103-92" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                                                  0.2057032, 0.3237431),</span></span>
<span id="cb103-93"><a href="#cb103-93" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          label = wt.hp$p.adj.star, size = 4, lineheight = 0.5,</span></span>
<span id="cb103-94"><a href="#cb103-94" aria-hidden="true" tabindex="-1"></a>  <span class="co">#          color = "#545252") +</span></span>
<span id="cb103-95"><a href="#cb103-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb103-96"><a href="#cb103-96" aria-hidden="true" tabindex="-1"></a>panel_i</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FigureS1_ELMFilter_files/figure-html/panel_h-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="panel-assembly" class="level2">
<h2 class="anchored" data-anchor-id="panel-assembly">Panel Assembly</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>panel_d_i <span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(panel_d, panel_e,panel_f, panel_g,panel_h, panel_i, <span class="at">nrow =</span> <span class="dv">3</span>, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>), <span class="at">align =</span> <span class="st">"hv"</span>,</span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"D"</span>, <span class="st">"E"</span>, <span class="st">"F"</span>, <span class="st">"G"</span>, <span class="st">"H"</span>, <span class="st">"I"</span>),</span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">label.x =</span> <span class="fl">0.1</span>)</span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a>panel_d_i</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="FigureS1_ELMFilter_files/figure-html/panel_c_f_assembly_save-1.png" class="img-fluid" width="480"></p>
</div>
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out_dir, <span class="st">"Figure1S_D-I.pdf"</span>), <span class="at">height =</span> <span class="dv">7</span>, <span class="at">width =</span> <span class="dv">4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>